
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Using cells selected in the ABC Atlas with Gene Expression data. &#8212; Allen Brain Cell Atlas - Data Access</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=9c3e77be" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/simple_style.css?v=3aaed5dc" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=1ae7504c"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/abc_atlas_selection_example';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)" href="../descriptions/WMB_dataset.html" />
    <link rel="prev" title="Accessing 10x RNA-seq gene expression data" href="general_accessing_10x_snRNASeq_tutorial.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/allen_logo.jpeg" class="logo__image only-light" alt="Allen Brain Cell Atlas - Data Access - Home"/>
    <script>document.write(`<img src="../_static/allen_logo.jpeg" class="logo__image only-dark" alt="Allen Brain Cell Atlas - Data Access - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Allen Brain Cell Atlas - Data Access
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">INTRODUCTION</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="general_accessing_10x_snRNASeq_tutorial.html">Accessing 10x RNA-seq gene expression data</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Using cells selected in the ABC Atlas with Gene Expression data.</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">DATA</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WMB_dataset.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv2.html">Whole mouse brain 10Xv2 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv3.html">Whole mouse brain 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10XMulti.html">Whole mouse brain 10XMulti single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10X.html">Clustering analysis of whole adult mouse brain 10X single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-taxonomy.html">Whole adult mouse brain taxonomy of cell types</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-imputed.html">Imputed MERFISH spatial transcriptomics of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-neighborhoods.html">Cluster groups and neighborhood specific UMAP embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Allen-CCF-2020.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-CCF.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_dataset.html">Aging Mouse dataset (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_10Xv3.html">Zeng Aging Mouse 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_taxonomy.html">Zeng Aging Mouse clustering and mapping to the WMB taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zhuang_dataset.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-1.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-2.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-3.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-3)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-4.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-4)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-dataset.html">Consensus Whole Mouse Brain dataset (Macosko, Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-10X.html">Consensus Whole Mouse Brain single cell/nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-taxonomy.html">Consensus Whole Mouse Brain integrated taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-dataset.html">Developing Mouse - Visual Cortex</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-10X.html">Developing Mouse - Visual Cortex single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-taxonomy.html">Developing Mouse - Visual Cortex taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_dataset.html">Human whole-brain transcriptomic cell type atlas (Kimberly Siletti)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10Xv3.html">Human brain 10Xv3 single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10X-clustering.html">Clustering analysis of whole human brain 10X single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-taxonomy.html">Human brain taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_dataset.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-10X.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-MapMyCells.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain (WHB) and Seattle Alzheimer’s Disease Brain Cell Atlas (SEA-AD) Taxonomies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_dataset.html">Human-Mammalian Brain - Basal Ganglia: Data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-BG-taxonomy-CCN20250428.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq data: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG-Aligned.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Aligned gene expression across species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Individual species gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Spatial-BG.html">Human-Mammalian Brain - Basal Ganglia: spatial transcriptomics cells for each species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Macaque-PatchSeq-BG.html">Human-Mammalian Brain - Basal Ganglia: Macaque Patch-seq cells multimodal data</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">NOTEBOOKS</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cluster_annotation_tutorial.html">10x RNA-seq clustering analysis and annotation (CCN20230722)</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle2.html">10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_1.html">10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2a.html">10x RNA-seq gene expression data (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2b.html">10x RNA-seq gene expression data (part 2b)</a></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle3.html">MERFISH whole brain spatial transcriptomics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_1.html">MERFISH whole brain spatial transcriptomics (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2a.html">MERFISH whole brain spatial transcriptomics (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2b.html">MERFISH whole brain spatial transcriptomics (part 2b)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_imputed_genes_example.html">MERFISH whole brain spatial transcriptomics cells with imputed genes</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_groups_and_embeddings_tutorial.html">Cluster neighborhoods and embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_neighborhood_gallery.html">Cell type neighborhood gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="ccf_and_parcellation_annotation_tutorial.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="merfish_ccf_registration_tutorial.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-notebooks.html">Consensus Mouse notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_clustering_analysis_and_annotation.html">Consensus Whole Mouse Brain #1: Clustering and Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_10X_snRNASeq.html">Consensus Whole Mouse Brain #2: 10X gene expression</a></li>

</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-notebooks.html">Developing Mouse - Visual Cortex notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="dev_mouse_vis_cortex_clustering_analysis_and_annotation.html">Developing Mouse - Visual Cortex 10X scRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="dev_mouse_vis_cortex_10X_scRNASeq.html">Developing Mouse - Visual Cortex: 10X gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_notebooks.html">Aging Mouse Notebooks (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_clustering_analysis_and_annotation.html">Mouse Aging clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_10x_snRNASeq_tutorial.html">Mouse Aging 10X RNA-seq gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle4.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="zhuang_merfish_tutorial.html">MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_notebooks.html">Whole Human Brain 10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="WHB_cluster_annotation_tutorial.html">Whole Human Brain 10x RNA-seq clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_1.html">Whole Human Brain (WHB) 10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_2.html">Whole Human Brain 10x RNA-seq gene expression data (part 2)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_notebooks.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_data_and_metadata.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_siletti_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_seaad_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_gene_expression.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Gene Expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_notebooks.html">Human-Mammalian Brain - Basal Ganglia: Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_clustering_analysis_and_annotation.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_10X_snRNASeq_tutorial.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_slabs_and_taxonomy.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_genes.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_macaque_patchseq.html">Human-Mammalian Brain - Basal Ganglia Patchseq</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">ADDITIONAL RESOURCES</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../docs/README.html">Resource Pages</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../docs/abbreviation_terms/README.html">Abbreviation lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/cluster_annotations/README.html">Cluster annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/gene_lists/README.html">Gene lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/parcellation_annotations/README.html">Parcellation annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/supplemental_metadata_descriptions/README.html">Supplemental Metadata Descriptions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/neuroglancer_hmba/README.html">HMBA Basal Ganglia: Neuroglancer Views</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access/issues/new?title=Issue%20on%20page%20%2Fnotebooks/abc_atlas_selection_example.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/abc_atlas_selection_example.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Using cells selected in the ABC Atlas with Gene Expression data.</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#initializing-required-modules-and-instantiating-the-abcprojectcache">Initializing required modules and instantiating the AbcProjectCache.</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#selecting-data-in-the-abc-atlas">Selecting Data in the ABC Atlas.</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#combining-with-the-cell-metadata">Combining with the cell metadata.</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#using-the-matched-data-in-a-cell-by-gene-matrix">Using the matched data in a cell by gene matrix.</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="using-cells-selected-in-the-abc-atlas-with-gene-expression-data">
<h1>Using cells selected in the ABC Atlas with Gene Expression data.<a class="headerlink" href="#using-cells-selected-in-the-abc-atlas-with-gene-expression-data" title="Link to this heading">#</a></h1>
<p>The <a class="reference external" href="https://portal.brain-map.org/atlases-and-data/bkp/abc-atlas">Allen Brain Cell (ABC) Atlas</a> is a powerfull tool for visualizing the data that are used throughout these notebooks. One particualar way users can interact with the ABC Atlas is by selecting and downloading specific cells from either a part of the taxonomy UMAP or brain region they are interested. This notebook will show how combine these cells selected from the ABC Atlas with gene expression data and metadata in these notebooks.</p>
<p>Note that the examples in this notebook can only be used with the Whole Mouse Brain (WMB) MERFISH and 10X data, the Zhuang et al. MERFISH data (1-4), and the Whole Human Brain (WHB) 10X data. The Seattle Alzheimer’s Disease (SEA-AD) dataset is not currently available through this repository (Data and documentation for SEA-AD are <a class="reference external" href="https://portal.brain-map.org/explore/seattle-alzheimers-disease/seattle-alzheimers-disease-brain-cell-atlas-download?edit&amp;amp;language=en">available here</a>).</p>
<p>For this notebook, the user should:</p>
<ul class="simple">
<li><p>Have some familiarity with using the ABC Atlas. If you are unsure of any steps used to manipulate data in the atlas visualization, you can refer to the user guide <a class="reference external" href="https://community.brain-map.org/t/abc-atlas-user-guide/3179">here</a>.</p></li>
<li><p>Be using manifest versions equal to or greater than 20241115. Instructions on listing and changing versions can be found in the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.ipynb">Getting Started notebook</a>.</p></li>
<li><p>Be connected to the internet.</p></li>
<li><p>Have run through the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html">Getting Started</a> and other notebooks in this repo.</p></li>
</ul>
<section id="initializing-required-modules-and-instantiating-the-abcprojectcache">
<h2>Initializing required modules and instantiating the AbcProjectCache.<a class="headerlink" href="#initializing-required-modules-and-instantiating-the-abcprojectcache" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">anndata</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">PIL</span><span class="w"> </span><span class="kn">import</span> <span class="n">Image</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">abc_atlas_access.abc_atlas_cache.abc_project_cache</span><span class="w"> </span><span class="kn">import</span> <span class="n">AbcProjectCache</span>
</pre></div>
</div>
</div>
</div>
<p>We will interact with the data using the <strong>AbcProjectCache</strong>. This cache object tracks which data has been downloaded and serves the path to the requsted data on disk. For metadata, the cache can also directly serve a up a Pandas Dataframe. See the <code class="docutils literal notranslate"><span class="pre">getting_started</span></code> notebook for more details on using the cache including installing it if it has not already been.</p>
<p><strong>Change the download_base variable to where you have downloaded the data in your system.</strong></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">download_base</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">&#39;../../data/abc_atlas&#39;</span><span class="p">)</span>
<span class="n">abc_cache</span> <span class="o">=</span> <span class="n">AbcProjectCache</span><span class="o">.</span><span class="n">from_cache_dir</span><span class="p">(</span><span class="n">download_base</span><span class="p">)</span>

<span class="n">abc_cache</span><span class="o">.</span><span class="n">current_manifest</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;releases/20250531/manifest.json&#39;
</pre></div>
</div>
</div>
</div>
</section>
<section id="selecting-data-in-the-abc-atlas">
<h2>Selecting Data in the ABC Atlas.<a class="headerlink" href="#selecting-data-in-the-abc-atlas" title="Link to this heading">#</a></h2>
<p>For this example we’ll be looking at the WMB MERFISH dataset within ABC Atlas. We’ll filter the data by a specific subtype and region and then select cells from a specific MERFISH slice. You can refer to <a class="reference external" href="https://knowledge.brain-map.org/abcatlas#AQEBSzlKTjIzUDI0S1FDR0s5VTc1QQACSFNZWlBaVzE2NjlVODIxQldZUAADBwFGUzAwRFhWMFQ5UjFYOUZKNFFFAAIAAAFRWTVTOEtNTzVITEpVRjBQMDBLAAIBMDIyIEw1IEVUIENUWCBHbHV0AAABMTVCSzQ3RENJT0YxU0xMVVc5UAACAAABQ0JHQzBVMzBWVjlKUFI2MFRKVQACAAABNzNHVlREWERFR0UyN00yWEpNVAACAAABSTlMTlAwTzFSTjhLNFNOMkdXWQACAVZJU3BtAAABVEZRRkxORVAzVjIyMlk4OEM2NAACAAAEAQAChexuHYN%2B0KIDg7ChtYNCd2oEMk5RVElFN1RBTVA4UFFBSE80UAAFga%2BmSoHprA6BoIFHgJLV6AYABwAABQAGAQECRlMwMERYVjBUOVIxWDlGSjRRRQADfgAAAAQAAAhWRk9GWVBGUUdSS1VEUVVaM0ZGAAlMVkRCSkFXOEJJNVlTUzFRVUJHAAoACwFUTE9LV0NMOTVSVTAzRDlQRVRHAAI3M0dWVERYREVHRTI3TTJYSk1UAAMBBAEAAiMwMDAwMDAAA8gBAAUBAQIjMDAwMDAwAAPIAQAAAAIBAA%3D%3D">this linked view</a> of the atlas to see the cells we will be selecting.</p>
<p>The cells we select come from  the <code class="docutils literal notranslate"><span class="pre">L5</span> <span class="pre">ET</span> <span class="pre">CTX</span> <span class="pre">Glut</span></code> subclass, we’ll also filter on only cells in the VISpm sub-region of the Isocortex. This initial selection leaves 325 cells that overlap with our filters. We’ll use the select tool (dashed box in the upper corner in the linked ABC Atlas view) to select the cells in section 31 of our MERFISH data. Refer to the image below to see our selection and where these selected cells reside (in addition to the above link).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">image</span> <span class="o">=</span> <span class="n">Image</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="s1">&#39;data/abc_atlas_selection_merfish.png&#39;</span><span class="p">)</span>
<span class="n">image</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/ed0714dabfbd7377e9434196c9676d2a229fbcdd69c06e30dde78409368e0d79.png" src="../_images/ed0714dabfbd7377e9434196c9676d2a229fbcdd69c06e30dde78409368e0d79.png" />
</div>
</div>
<p>This gives us 94 cells in our final selection. We download the file by clicking the download arrow on the far right just above our listed, selected cells. This specific selection is provided in the file <a class="reference external" href="https://github.com/AllenInstitute/abc_atlas_access/blob/9c9cd714f9f0265a1b9b2d1cd4875c274d36aec2/notebooks/data/ABC_Atlas_Class_01%20IT-ET%20Glut_cells_2024_10_23_16_42.csv"><code class="docutils literal notranslate"><span class="pre">ABC_Atlas_Class_01</span> <span class="pre">IT-ET</span> <span class="pre">Glut_cells_2024_10_23_16_42.csv</span></code></a> that is packaged in this repo with the notebooks.</p>
</section>
<section id="combining-with-the-cell-metadata">
<h2>Combining with the cell metadata.<a class="headerlink" href="#combining-with-the-cell-metadata" title="Link to this heading">#</a></h2>
<p>Now that we have our subset of cells selected from the ABC Atlas, we can combine them with the cell metadata provided by the <code class="docutils literal notranslate"><span class="pre">AbcProjectCache</span></code> object.</p>
<p>The files <code class="docutils literal notranslate"><span class="pre">cell_metadata</span></code> in the projects/directories MERFISH-C57BL6J-638850, WHB-10Xv3, WMB-10X, Zhuang-ABCA-[1-4] contain a column <code class="docutils literal notranslate"><span class="pre">abc_sample_id</span></code> that allows for merging of data downloaded from an ABC Atlas visualization. Note that only the specific files called <code class="docutils literal notranslate"><span class="pre">cell_metadata</span></code> contain this id. Other derived metadata such as <code class="docutils literal notranslate"><span class="pre">cell_metadata_with_cluster_annotation</span></code> do not contain this id though it can be easily merged in.</p>
<p>Here we’ll just display a few columns of the cell_metadata table. For specifics on the MERFISH metadata and gene expression data used in this example, see the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/notebook_subtitle3.html">notebooks related this MERFISH dataset</a></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;MERFISH-C57BL6J-638850&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">},</span>
<span class="p">)</span>
<span class="n">cell_metadata</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()[[</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">,</span> <span class="s1">&#39;abc_sample_id&#39;</span><span class="p">]]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>abc_sample_id</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1019171907102340387-1</th>
      <td>C57BL6J-638850.37</td>
      <td>c9881423-76a7-4835-ba8b-7942fd384b6b</td>
    </tr>
    <tr>
      <th>1104095349101460194-1</th>
      <td>C57BL6J-638850.26</td>
      <td>aa815488-6487-4e47-8a5e-d82ac9933bc6</td>
    </tr>
    <tr>
      <th>1017092617101450577</th>
      <td>C57BL6J-638850.25</td>
      <td>91ef7a85-8e3e-4410-8ee2-785788df3ebe</td>
    </tr>
    <tr>
      <th>1018093344101130233</th>
      <td>C57BL6J-638850.13</td>
      <td>18991e17-fbd3-4ba0-9c60-1281f56ac520</td>
    </tr>
    <tr>
      <th>1019171912201610094</th>
      <td>C57BL6J-638850.27</td>
      <td>5e155936-e40d-4c6b-8971-e7fb0079274b</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Now we load our selected cells. Note that we rename the <code class="docutils literal notranslate"><span class="pre">Sample</span> <span class="pre">Id</span></code> column to match the one in our cell metadata.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Change path location below to either your own downloaded cells</span>
<span class="c1"># or the location of the csv from this repo (abc_atlas_access/notebooks/ABC_Atlas_Class_01 IT-ET Glut_cells_2024_10_23_16_42.csv)</span>
<span class="n">downloaded_cells_path</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">&#39;data/ABC_Atlas_Class_01 IT-ET Glut_cells_2024_10_23_16_42.csv&#39;</span><span class="p">)</span>

<span class="n">abc_atlas_selection</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">downloaded_cells_path</span><span class="p">)</span>
<span class="n">abc_atlas_selection</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span>
    <span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;Sample Id&#39;</span><span class="p">:</span> <span class="s1">&#39;abc_sample_id&#39;</span><span class="p">},</span>
    <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">abc_atlas_selection</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;abc_sample_id&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">abc_atlas_selection</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Supertype</th>
      <th>Section Label</th>
      <th>Neurotransmitter Type</th>
      <th>Anatomical Division</th>
      <th>Cluster</th>
      <th>Class</th>
      <th>Anatomical Structure</th>
      <th>Subclass</th>
      <th>Anatomical Substructure</th>
    </tr>
    <tr>
      <th>abc_sample_id</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>e23109f2-b70b-48a0-b29f-f4cfd24025ac</th>
      <td>0090 L5 ET CTX Glut_1</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0352 L5 ET CTX Glut_1</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
    <tr>
      <th>d42c7e04-95f7-4134-93e8-66acc1a52aa3</th>
      <td>0091 L5 ET CTX Glut_2</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0359 L5 ET CTX Glut_2</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
    <tr>
      <th>cfc13cd0-adf7-41c4-a849-f1472ea31567</th>
      <td>0091 L5 ET CTX Glut_2</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0359 L5 ET CTX Glut_2</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
    <tr>
      <th>e98716b6-15c2-4f54-8717-589b0da6d246</th>
      <td>0090 L5 ET CTX Glut_1</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0352 L5 ET CTX Glut_1</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
    <tr>
      <th>a28ebb39-2849-49ae-80e4-ee135fce1890</th>
      <td>0092 L5 ET CTX Glut_3</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0369 L5 ET CTX Glut_3</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Now we can use pandas functionality to inner join together to two tables, leaving only the subset of cells, 94, that we selected in the ABC Atlas visualization.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">selected_cells</span> <span class="o">=</span> <span class="n">cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">abc_atlas_selection</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;abc_sample_id&#39;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">selected_cells</span><span class="p">))</span>
<span class="n">selected_cells</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>94
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>cluster_alias</th>
      <th>average_correlation_score</th>
      <th>feature_matrix_label</th>
      <th>donor_label</th>
      <th>donor_genotype</th>
      <th>donor_sex</th>
      <th>x</th>
      <th>y</th>
      <th>z</th>
      <th>abc_sample_id</th>
      <th>Supertype</th>
      <th>Section Label</th>
      <th>Neurotransmitter Type</th>
      <th>Anatomical Division</th>
      <th>Cluster</th>
      <th>Class</th>
      <th>Anatomical Structure</th>
      <th>Subclass</th>
      <th>Anatomical Substructure</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1018093345102260085</th>
      <td>C57BL6J-638850.31</td>
      <td>384</td>
      <td>0.651028</td>
      <td>C57BL6J-638850</td>
      <td>C57BL6J-638850</td>
      <td>wt/wt</td>
      <td>M</td>
      <td>4.087788</td>
      <td>3.208372</td>
      <td>5.4</td>
      <td>21c31354-1979-4b93-b6f2-acaefaa22842</td>
      <td>0091 L5 ET CTX Glut_2</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0360 L5 ET CTX Glut_2</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
    <tr>
      <th>1018093345101290596</th>
      <td>C57BL6J-638850.31</td>
      <td>384</td>
      <td>0.737842</td>
      <td>C57BL6J-638850</td>
      <td>C57BL6J-638850</td>
      <td>wt/wt</td>
      <td>M</td>
      <td>6.753508</td>
      <td>3.045020</td>
      <td>5.4</td>
      <td>484abbf0-56d4-4622-add2-24920cd3cd05</td>
      <td>0091 L5 ET CTX Glut_2</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0360 L5 ET CTX Glut_2</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
    <tr>
      <th>1018093345102270314</th>
      <td>C57BL6J-638850.31</td>
      <td>393</td>
      <td>0.685347</td>
      <td>C57BL6J-638850</td>
      <td>C57BL6J-638850</td>
      <td>wt/wt</td>
      <td>M</td>
      <td>3.936859</td>
      <td>3.295320</td>
      <td>5.4</td>
      <td>147ac3d0-b2f2-4e1c-bd00-e1b740b242bc</td>
      <td>0091 L5 ET CTX Glut_2</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0364 L5 ET CTX Glut_2</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
    <tr>
      <th>1018093345102260156</th>
      <td>C57BL6J-638850.31</td>
      <td>383</td>
      <td>0.808770</td>
      <td>C57BL6J-638850</td>
      <td>C57BL6J-638850</td>
      <td>wt/wt</td>
      <td>M</td>
      <td>4.042362</td>
      <td>3.188116</td>
      <td>5.4</td>
      <td>eaf8a6aa-2ccd-474b-ac40-5bcbc705e2eb</td>
      <td>0091 L5 ET CTX Glut_2</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0359 L5 ET CTX Glut_2</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
    <tr>
      <th>1018093345101290474</th>
      <td>C57BL6J-638850.31</td>
      <td>383</td>
      <td>0.536743</td>
      <td>C57BL6J-638850</td>
      <td>C57BL6J-638850</td>
      <td>wt/wt</td>
      <td>M</td>
      <td>6.833381</td>
      <td>2.998880</td>
      <td>5.4</td>
      <td>e4874d7b-6754-40ff-8792-c0b7627f2bce</td>
      <td>0091 L5 ET CTX Glut_2</td>
      <td>C57BL6J-638850.31</td>
      <td>Glut</td>
      <td>Isocortex</td>
      <td>0359 L5 ET CTX Glut_2</td>
      <td>01 IT-ET Glut</td>
      <td>VISpm</td>
      <td>022 L5 ET CTX Glut</td>
      <td>VISpm5</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Let’s do a quick plot to show that we can grossly reproduces the view of MERFISH section 31.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_section</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">cc</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="n">val</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="n">fig_width</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span> <span class="n">fig_height</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span> <span class="n">cmap</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="n">fig</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">fig</span> <span class="ow">is</span> <span class="kc">None</span> <span class="ow">or</span> <span class="n">ax</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
        <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">val</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">cc</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">cc</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">11</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">11</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s1">&#39;equal&#39;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pred</span> <span class="o">=</span> <span class="p">(</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;C57BL6J-638850.31&#39;</span><span class="p">)</span>
<span class="n">section</span> <span class="o">=</span> <span class="n">cell_metadata</span><span class="p">[</span><span class="n">pred</span><span class="p">]</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">8</span><span class="p">)</span>
<span class="c1"># Plot all cells as grey.</span>
<span class="n">fig_all</span><span class="p">,</span> <span class="n">ax_all</span> <span class="o">=</span> <span class="n">plot_section</span><span class="p">(</span><span class="n">section</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span> <span class="n">section</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span> <span class="n">cc</span><span class="o">=</span><span class="s1">&#39;#D3D3D3&#39;</span><span class="p">,</span> <span class="n">fig</span><span class="o">=</span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
<span class="c1"># Plot our selected cells as red</span>
<span class="n">fig_sel</span><span class="p">,</span> <span class="n">ax_sel</span> <span class="o">=</span> <span class="n">plot_section</span><span class="p">(</span><span class="n">selected_cells</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span> <span class="n">selected_cells</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span> <span class="n">cc</span><span class="o">=</span><span class="s1">&#39;red&#39;</span><span class="p">,</span> <span class="n">fig</span><span class="o">=</span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/bb61ad168b24c512724a4c9d50fc9fc172c2449c332376f43c3d039fe3c2dc76.png" src="../_images/bb61ad168b24c512724a4c9d50fc9fc172c2449c332376f43c3d039fe3c2dc76.png" />
</div>
</div>
</section>
<section id="using-the-matched-data-in-a-cell-by-gene-matrix">
<h2>Using the matched data in a cell by gene matrix.<a class="headerlink" href="#using-the-matched-data-in-a-cell-by-gene-matrix" title="Link to this heading">#</a></h2>
<p>Now that we have have our matched metadata we can simply use the index of our final combined Pandas DataFrame with our cell by gene data. Since we are working with data from an individual MERFISH slice, we can download that directly. If you are doing this with the 10X data, you should use the <code class="docutils literal notranslate"><span class="pre">get_gene_data</span></code> function tutorialized in the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/general_accessing_10x_snRNASeq_tutorial.html"><code class="docutils literal notranslate"><span class="pre">general_accessing_10x_snRNASeq_tutorial</span></code></a> notebook.</p>
<p>First we need to load our gene list giving a mapping from identifier to symbol.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gene</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;MERFISH-C57BL6J-638850&#39;</span><span class="p">,</span> <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;gene&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;gene_identifier&#39;</span><span class="p">)</span>
<span class="n">gene</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene_symbol</th>
      <th>transcript_identifier</th>
      <th>name</th>
      <th>mapped_ncbi_identifier</th>
    </tr>
    <tr>
      <th>gene_identifier</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>ENSMUSG00000026778</th>
      <td>Prkcq</td>
      <td>ENSMUST00000028118</td>
      <td>protein kinase C, theta</td>
      <td>NCBIGene:18761</td>
    </tr>
    <tr>
      <th>ENSMUSG00000026837</th>
      <td>Col5a1</td>
      <td>ENSMUST00000028280</td>
      <td>collagen, type V, alpha 1</td>
      <td>NCBIGene:12831</td>
    </tr>
    <tr>
      <th>ENSMUSG00000001985</th>
      <td>Grik3</td>
      <td>ENSMUST00000030676</td>
      <td>glutamate receptor, ionotropic, kainate 3</td>
      <td>NCBIGene:14807</td>
    </tr>
    <tr>
      <th>ENSMUSG00000039323</th>
      <td>Igfbp2</td>
      <td>ENSMUST00000047328</td>
      <td>insulin-like growth factor binding protein 2</td>
      <td>NCBIGene:16008</td>
    </tr>
    <tr>
      <th>ENSMUSG00000048387</th>
      <td>Osr1</td>
      <td>ENSMUST00000057021</td>
      <td>odd-skipped related transcription factor 1</td>
      <td>NCBIGene:23967</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Now we’ll load and process the data. Selecting by cell id in our data files (even for large ones) is mostly safe on machines with smaller memory footprints. This is assuming one uses the <code class="docutils literal notranslate"><span class="pre">backed</span></code> option and doesn’t attempt to load too many cells (&lt;1000 say). For larger files, the user will want to be carefulas  most of the anndata files we release cannot be loaded into memory on a smaller scale computer. For an example of doing this, or loading across 10X that has been divided into multiple files, see the <a class="reference internal" href="general_accessing_10x_snRNASeq_tutorial.html"><span class="std std-doc">General Accesing 10X data notebook</span></a></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load the anndata, cell by gene matrix.</span>
<span class="n">file_path</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_file_path</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;MERFISH-C57BL6J-638850-sections&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;C57BL6J-638850.31/log2&#39;</span>
<span class="p">)</span>
<span class="n">adata</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">file_path</span><span class="p">,</span> <span class="n">backed</span><span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">)</span>
<span class="c1"># Create a landing Pandas DataFrame for the data.</span>
<span class="n">selected_cell_gene_data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="n">selected_cells</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                                       <span class="n">columns</span><span class="o">=</span><span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>

<span class="c1"># Find the subset of cells we selected in the full matrix.</span>
<span class="n">mask</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">selected_cells</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="c1"># Fill our DataFrame with the </span>
<span class="n">selected_cell_gene_data</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">selected_cells</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="o">.</span><span class="n">index</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">[</span><span class="n">mask</span><span class="p">]</span><span class="o">.</span><span class="n">toarray</span><span class="p">()</span>
<span class="n">selected_cell_gene_data</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">gene</span><span class="o">.</span><span class="n">gene_symbol</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>C57BL6J-638850.31-log2.h5ad: 100%|██████████| 190M/190M [00:23&lt;00:00, 8.20MMB/s]    
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">selected_cell_gene_data</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>gene_symbol</th>
      <th>Prkcq</th>
      <th>Col5a1</th>
      <th>Grik3</th>
      <th>Igfbp2</th>
      <th>Osr1</th>
      <th>Syt6</th>
      <th>Cntnap3</th>
      <th>Lmo3</th>
      <th>Ntn1</th>
      <th>Otp</th>
      <th>...</th>
      <th>Blank-16</th>
      <th>Blank-32</th>
      <th>Blank-18</th>
      <th>Blank-4</th>
      <th>Blank-28</th>
      <th>Blank-33</th>
      <th>Blank-34</th>
      <th>Blank-45</th>
      <th>Blank-23</th>
      <th>Blank-48</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1018093345102260085</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.900271</td>
      <td>1.847696</td>
      <td>0.0</td>
      <td>0.900271</td>
      <td>0.0</td>
      <td>0.900271</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1018093345101290596</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.498822</td>
      <td>0.685901</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.148877</td>
      <td>1.780236</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1018093345102270314</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.429679</td>
      <td>1.429679</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1018093345102260156</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>2.150413</td>
      <td>1.907521</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.755238</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.755238</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1018093345101290474</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.591793</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.591793</td>
      <td>1.591793</td>
      <td>0.0</td>
      <td>1.591793</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1018093345102270545</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.663865</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.663865</td>
      <td>0.0</td>
      <td>1.116793</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.663865</td>
      <td>0.0</td>
      <td>1.116793</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1018093345102270307</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.264144</td>
      <td>2.094565</td>
      <td>0.0</td>
      <td>0.553146</td>
      <td>0.0</td>
      <td>0.952008</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1018093345102270242</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.782628</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.782628</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1018093345101290541</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.866036</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.403429</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.866036</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1018093345101290626</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>2.860291</td>
      <td>0.0</td>
      <td>1.626302</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.626302</td>
      <td>1.257898</td>
      <td>0.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.761911</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
  </tbody>
</table>
<p>94 rows × 550 columns</p>
</div></div></div>
</div>
<p>This process can be repeated for the all other ABC Atlas visualizations (except SEA-AD which we do not currently released through this package). Here is a list of datasets in the <code class="docutils literal notranslate"><span class="pre">AbcProjectCache</span></code> that contain the <code class="docutils literal notranslate"><span class="pre">abc_sample_id</span></code> with links to their Abc Atlas bisualizations:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://knowledge.brain-map.org/abcatlas#AQEBSzlKTjIzUDI0S1FDR0s5VTc1QQACSFNZWlBaVzE2NjlVODIxQldZUAADAAQBAAKE%2FDwdg14ZMgOFdjNXhRhBHAQyTlFUSUU3VEFNUDhQUUFITzRQAAWBr6ZKgemsDoGggUeAktXoBgAHAAAFAAYBAQJGUzAwRFhWMFQ5UjFYOUZKNFFFAAN%2BAAAABAAACFZGT0ZZUEZRR1JLVURRVVozRkYACUxWREJKQVc4Qkk1WVNTMVFVQkcACgALAVRMT0tXQ0w5NVJVMDNEOVBFVEcAAjczR1ZURFhERUdFMjdNMlhKTVQAAwEEAQACIzAwMDAwMAADyAEABQEBAiMwMDAwMDAAA8gBAAAAAgEA">MERFISH-C57BL6J-638850</a></p></li>
<li><p><a class="reference external" href="https://knowledge.brain-map.org/abcatlas#AQIBOU02NEo3UzBBM1pOSUpBNFZTNgACR1o5NFo4M1JYVVRFQ0hZS1FPNAADAAQBAQJ%2FQEmofIMKAQOFBF5thmxgWAAFAAYBAQJYM0s3SzJFRzZVTVhPTjBUOUJFAAN%2BAAAABAAACFVNWko3WkxGOEc1Rkg1UzlZNFUACUMzUlJWQUsxOEhHNlExSk42WlEACgAAATlNNjRKN1MwQTNaTklKQTRWUzYAAlI1RldVVlRFQTJPVzlOSlFNWTkAAwAEAQECgYCoM34%2F4yADhYQxZ4a8jlIABQAGAQECWDNLN0syRUc2VU1YT04wVDlCRQADfgAAAAQAAAg1VDRDM01DM0xOWldFQkRGVUVaAAlDM1JSVkFLMThIRzZRMUpONlpRAAoAAAIDAA%3D%3D">WHB-10Xv3</a></p></li>
<li><p><a class="reference external" href="https://knowledge.brain-map.org/abcatlas#AQEBQVA4Sk5ONUxZQUJHVk1HS1kxQgACUTFOQ1dXUEc2RlowRE5JWEpCUQADAAQBAQKBEIB4gZSwRAOEjCuShFgYTQAFAAYBAQJGUzAwRFhWMFQ5UjFYOUZKNFFFAAN%2BAAAABAAACEc0STRHRkpYSkI5QVRaM1BUWDEACUxWREJKQVc4Qkk1WVNTMVFVQkcACgALAW5vbmUAAm5vbmUAAwEEAQACIzAwMDAwMAADyAEABQEBAiMwMDAwMDAAA8gBAAAAAgEA">WMB-10X</a></p></li>
<li><p><a class="reference external" href="https://knowledge.brain-map.org/abcatlas#AQQBVTVWOTRFUzRKNzZNWVNMN1FMNwACWkdDQlk2MkowTEZLWjU4SklVRQADAAQBAAKENP98hIBpkgOFsotPhbZEPAAFAAYBAQJGUzAwRFhWMFQ5UjFYOUZKNFFFAAN%2BAAAABAAACE1HQTVMVVRINEVUTTg1OUw1SU0ACTVDMDIwMUpTVkUwNFdZNkRNVkMACgAAAUpaUEg4UjFINjBDNDFOMDJVQjIAAk9BN0oyS0JWTkRMMjIwTkRNSzkAAwAEAQACg2DtGYMWxFIDhORlYYRYGE0ABQAGAQECRlMwMERYVjBUOVIxWDlGSjRRRQADfgAAAAQAAAhCQTVEUDczVVI2WjRCWk5GOThIAAk1QzAyMDFKU1ZFMDRXWTZETVZDAAoAAAE4VEtQQzRCVFRTSE0zSEZXWDFYAAJENDJSNlhLN0I2Uks0VTE4UUE0AAMABAEAAoOWSu%2BDlkrvA4WW4BOFKAcEAAUABgEBAkZTMDBEWFYwVDlSMVg5Rko0UUUAA34AAAAEAAAIMDQwTFRLQzZGWjRORFQyQURZQgAJNUMwMjAxSlNWRTA0V1k2RE1WQwAKAAABWE1GWEdISzVITE40NkJVUTFTUgACMllXWUFBVUpZWVcyN09HMlUwWgADAAQBAAKDWIELggwQ%2FAOErP0ahKC7CgAFAAYBAQJGUzAwRFhWMFQ5UjFYOUZKNFFFAAN%2BAAAABAAACFBYVVNYWExJMlNBNUdOSDRITzYACTVDMDIwMUpTVkUwNFdZNkRNVkMACgAAAggA">Zhuang-ABCA-(1-4)</a></p></li>
</ul>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="general_accessing_10x_snRNASeq_tutorial.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Accessing 10x RNA-seq gene expression data</p>
      </div>
    </a>
    <a class="right-next"
       href="../descriptions/WMB_dataset.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#initializing-required-modules-and-instantiating-the-abcprojectcache">Initializing required modules and instantiating the AbcProjectCache.</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#selecting-data-in-the-abc-atlas">Selecting Data in the ABC Atlas.</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#combining-with-the-cell-metadata">Combining with the cell metadata.</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#using-the-matched-data-in-a-cell-by-gene-matrix">Using the matched data in a cell by gene matrix.</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>