

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy &#8212; Allen Brain Cell Atlas - Data Access</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=5b4479735964841361fd" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=5b4479735964841361fd" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/simple_style.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=5b4479735964841361fd" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd" />
  <script src="../_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=5b4479735964841361fd"></script>

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/asap_pmdbs_siletti_taxonomy';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy" href="asap_pmdbs_seaad_taxonomy.html" />
    <link rel="prev" title="ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview" href="asap_pmdbs_data_and_metadata.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/allen_logo.jpeg" class="logo__image only-light" alt="Allen Brain Cell Atlas - Data Access - Home"/>
    <script>document.write(`<img src="../_static/allen_logo.jpeg" class="logo__image only-dark" alt="Allen Brain Cell Atlas - Data Access - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Allen Brain Cell Atlas - Data Access
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">INTRODUCTION</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="general_accessing_10x_snRNASeq_tutorial.html">Accessing 10x RNA-seq gene expression data</a></li>
<li class="toctree-l1"><a class="reference internal" href="abc_atlas_selection_example.html">Using cells selected in the ABC Atlas with Gene Expression data.</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">DATA</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WMB_dataset.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv2.html">Whole mouse brain 10Xv2 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv3.html">Whole mouse brain 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10XMulti.html">Whole mouse brain 10XMulti single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10X.html">Clustering analysis of whole adult mouse brain 10X single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-taxonomy.html">Whole adult mouse brain taxonomy of cell types</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-imputed.html">Imputed MERFISH spatial transcriptomics of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-neighborhoods.html">Cluster groups and neighborhood specific UMAP embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Allen-CCF-2020.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-CCF.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_dataset.html">Aging Mouse dataset (Hongkui Zeng)</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_10Xv3.html">Zeng Aging Mouse 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_taxonomy.html">Zeng Aging Mouse clustering and mapping to the WMB taxonomy of cell types</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zhuang_dataset.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-3"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-1.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-2.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-3.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-3)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-4.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-4)</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-dataset.html">Consensus Whole Mouse Brain dataset</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-4"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-10X.html">Consensus Whole Mouse Brain single cell/nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-taxonomy.html">Consensus Whole Mouse Brain integrated taxonomy</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_dataset.html">Human whole-brain transcriptomic cell type atlas (Kimberly Siletti)</a><input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-5"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10Xv3.html">Human brain 10Xv3 single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10X-clustering.html">Clustering analysis of whole human brain 10X single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-taxonomy.html">Human brain taxonomy of cell types</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_dataset.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS)</a><input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-6"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-10X.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-MapMyCells.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain (WHB) and Seattle Alzheimer’s Disease Brain Cell Atlas (SEA-AD) Taxonomies</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_dataset.html">Human-Mammalian Brain - Basal Ganglia: Data</a><input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-7"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-BG-taxonomy-CCN20250428.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq data: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG-Aligned.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Aligned gene expression across species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Individual species gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Spatial-BG.html">Human-Mammalian Brain - Basal Ganglia: spatial transcriptomics cells for each species</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">NOTEBOOKS</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><input class="toctree-checkbox" id="toctree-checkbox-8" name="toctree-checkbox-8" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-8"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="cluster_annotation_tutorial.html">10x RNA-seq clustering analysis and annotation (CCN20230722)</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle2.html">10x scRNA-seq gene expression data</a><input class="toctree-checkbox" id="toctree-checkbox-9" name="toctree-checkbox-9" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-9"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_1.html">10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2a.html">10x RNA-seq gene expression data (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2b.html">10x RNA-seq gene expression data (part 2b)</a></li>
</ul>
</li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle3.html">MERFISH whole brain spatial transcriptomics</a><input class="toctree-checkbox" id="toctree-checkbox-10" name="toctree-checkbox-10" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-10"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_1.html">MERFISH whole brain spatial transcriptomics (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2a.html">MERFISH whole brain spatial transcriptomics (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2b.html">MERFISH whole brain spatial transcriptomics (part 2b)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_imputed_genes_example.html">MERFISH whole brain spatial transcriptomics cells with imputed genes</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="cluster_groups_and_embeddings_tutorial.html">Cluster neighborhoods and embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_neighborhood_gallery.html">Cell type neighborhood gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="ccf_and_parcellation_annotation_tutorial.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="merfish_ccf_registration_tutorial.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-notebooks.html">Consensus Mouse notebooks</a><input class="toctree-checkbox" id="toctree-checkbox-11" name="toctree-checkbox-11" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-11"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_clustering_analysis_and_annotation.html">Consensus Whole Mouse Brain #1: Clustering and Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_10X_snRNASeq.html">Consensus Whole Mouse Brain #2: 10X gene expression</a></li>

</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_notebooks.html">Aging Mouse Notebooks (Hongkui Zeng)</a><input class="toctree-checkbox" id="toctree-checkbox-12" name="toctree-checkbox-12" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-12"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_clustering_analysis_and_annotation.html">Mouse Aging clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_10x_snRNASeq_tutorial.html">Mouse Aging 10X RNA-seq gene expression</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle4.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><input class="toctree-checkbox" id="toctree-checkbox-13" name="toctree-checkbox-13" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-13"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="zhuang_merfish_tutorial.html">MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_notebooks.html">Whole Human Brain 10x scRNA-seq gene expression data</a><input class="toctree-checkbox" id="toctree-checkbox-14" name="toctree-checkbox-14" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-14"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="WHB_cluster_annotation_tutorial.html">Whole Human Brain 10x RNA-seq clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_1.html">Whole Human Brain (WHB) 10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_2.html">Whole Human Brain 10x RNA-seq gene expression data (part 2)</a></li>
</ul>
</li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_notebooks.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-15" name="toctree-checkbox-15" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-15"><i class="fa-solid fa-chevron-down"></i></label><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_data_and_metadata.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_seaad_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_gene_expression.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Gene Expression</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_notebooks.html">Human-Mammalian Brain - Basal Ganglia: Notebooks</a><input class="toctree-checkbox" id="toctree-checkbox-16" name="toctree-checkbox-16" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-16"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_clustering_analysis_and_annotation.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_10X_snRNASeq_tutorial.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_slabs_and_taxonomy.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_genes.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">ADDITIONAL RESOURCES</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../docs/README.html">Resource Pages</a><input class="toctree-checkbox" id="toctree-checkbox-17" name="toctree-checkbox-17" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-17"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../docs/abbreviation_terms/README.html">Abbreviation lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/cluster_annotations/README.html">Cluster annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/gene_lists/README.html">Gene lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/parcellation_annotations/README.html">Parcellation annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/supplemental_metadata_descriptions/README.html">Supplemental Metadata Descriptions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/neuroglancer_hmba/README.html">HMBA Basal Ganglia: Neuroglancer Views</a></li>
</ul>
</li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access/issues/new?title=Issue%20on%20page%20%2Fnotebooks/asap_pmdbs_siletti_taxonomy.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/asap_pmdbs_siletti_taxonomy.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#asap-data">ASAP Data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#loading-mapmycells-results">Loading MapMyCells Results</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#x-whole-human-brain-taxonomy-siletti-et-al">10X Whole Human Brain Taxonomy (Siletti et al.)</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#umap-plotting">UMAP Plotting</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#transferred-neurotransmitter-types">Transferred Neurotransmitter Types</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#transferred-whole-human-brain-taxonomy-types">Transferred Whole Human Brain Taxonomy Types</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#assignment-correlation-and-probabilities">Assignment Correlation and Probabilities</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#removing-cells-with-low-quality-mapping">Removing cells with low quality mapping</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mapping-the-siletti-superclusters-on-to-the-leiden-clusters">Mapping the Siletti superclusters on to the Leiden Clusters</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="asap-human-postmortem-derived-brain-sequencing-collection-pmdbs-mapping-to-the-whole-human-brain-taxonomy">
<h1>ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy<a class="headerlink" href="#asap-human-postmortem-derived-brain-sequencing-collection-pmdbs-mapping-to-the-whole-human-brain-taxonomy" title="Permalink to this heading">#</a></h1>
<p>In the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/asap_pmdbs_data_and_metadata.html">previous notebook</a>, we provided an overview of the ASAP PMDBS dataset and associated cell, sample and donor metadata.  In this notebook we demonstrated the use of Allen Institute’s MapMyCells tool to provide cell type insight to the PMDBS dataset.</p>
<p><a class="reference external" href="https://knowledge.brain-map.org/mapmycells/process/">MapMyCells</a>(<a class="reference external" href="https://scicrunch.org/resources/data/record/nlx_144509-1/SCR_024672/resolver?q=MapMyCells&amp;l=MapMyCells&amp;i=rrid:scr_024672">RRID: SCR_024672</a>) transforms cell types from a concept in publications to a tool for public research. Scientists worldwide can discover what cell types their transcriptomics and spatial data corresponds with by comparing their data to massive, high-quality reference datasets using multiple mapping methods. Currently available taxonomies include 10X whole mouse brain taxonomy, 10X whole human brain taxonomy and 10X Human MTG SEA-AD taxonomy. Available algorithms include correlation mapping (flatmap), hierarchical correlation mapping and a deep generative model-based algorithm.</p>
<p>In this notebook, we will map the approximately 3 million cells of PMDBS to the 10X whole human brain taxonomy (Siletti et. al.) using the hierarchical correlation mapping method where assignments are first made at the top level of the taxonomy using mean gene expression profiles of the reference dataset and pre-selected set of marker genes to distinguish cluster from its sibling. For robustness, the process is repeated 100 times, using 50% of the marker genes each time. A cell is assigned the class that receives the plurality of votes, a probability and correlation score. This process is repeated down the taxonomy until the last finest level. In this notebook we walkthrough the output of the MapMyCells results.</p>
<p>You need to be connected to the internet to run this notebook and should have run through the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html">getting started notebook</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">anndata</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">abc_atlas_access.abc_atlas_cache.abc_project_cache</span><span class="w"> </span><span class="kn">import</span> <span class="n">AbcProjectCache</span>
</pre></div>
</div>
</div>
</div>
<p>We will interact with the data using the AbcProjectCache. This cache object tracks which data has been downloaded and serves the path to the requested data on disk. For metadata, the cache can also directly serve up a Pandas DataFrame. See the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html">getting started notebook</a> notebook for more details on using the cache including installing it if it has not already been.</p>
<p>Change the download_base variable to where you have downloaded the data in your system.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">download_base</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">&#39;../../data/abc_atlas&#39;</span><span class="p">)</span>
<span class="n">abc_cache</span> <span class="o">=</span> <span class="n">AbcProjectCache</span><span class="o">.</span><span class="n">from_cache_dir</span><span class="p">(</span><span class="n">download_base</span><span class="p">)</span>

<span class="n">abc_cache</span><span class="o">.</span><span class="n">current_manifest</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;releases/20250531/manifest.json&#39;
</pre></div>
</div>
</div>
</div>
<section id="data-overview">
<h2>Data Overview<a class="headerlink" href="#data-overview" title="Permalink to this heading">#</a></h2>
<section id="asap-data">
<h3>ASAP Data<a class="headerlink" href="#asap-data" title="Permalink to this heading">#</a></h3>
<p>We simplify the ASAP PMDB metadata into a single cell_metadata, csv file and there companion tables describing the donor (including sex, age, race, disease state etc.), sample (including data processing metadata, team information, and dissection or region of interest the sample is from), and value_sets which provides a mapping from unique term name (cell type, region of interest etc.) to a plotting color and order.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">donor</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;donor&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
<span class="n">sample</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;sample&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;sample_label&#39;</span><span class="p">)</span>
<span class="n">value_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;value_sets&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>As with our previous tutorial, we’ll first use a Pandas join to merge the sample data into the cell metadata and then merge the donor information. We’ll use the value_sets as a lookup.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_metadata</span> <span class="o">=</span> <span class="n">cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">sample</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;sample_label&#39;</span><span class="p">)</span>
<span class="n">cell_metadata</span> <span class="o">=</span> <span class="n">cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">donor</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
<span class="n">cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>sample_label</th>
      <th>x</th>
      <th>y</th>
      <th>cluster_label</th>
      <th>cluster_label_order</th>
      <th>cluster_label_color</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>...</th>
      <th>cerad_score_color</th>
      <th>cognitive_status</th>
      <th>cognitive_status_order</th>
      <th>cognitive_status_color</th>
      <th>lewy_body_disease_pathology</th>
      <th>lewy_body_disease_pathology_order</th>
      <th>lewy_body_disease_pathology_color</th>
      <th>thal_phase</th>
      <th>thal_phase_order</th>
      <th>thal_phase_color</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAACCCAAGAAACCAT-1_ASAP_PMBDS_000060_s002_Rep1</th>
      <td>AAACCCAAGAAACCAT-1</td>
      <td>ASAP_PMBDS_000060_s002</td>
      <td>0.016373</td>
      <td>1.403025</td>
      <td>cluster_000</td>
      <td>1</td>
      <td>#171c97</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>cc633260-614c-4e5c-aa18-d3e3a0ef81bb</td>
      <td>...</td>
      <td>#4daf4a</td>
      <td>Unknown</td>
      <td>4</td>
      <td>#e8e8e8</td>
      <td>Absent</td>
      <td>1</td>
      <td>#ffffd9</td>
      <td>Thal 3</td>
      <td>4</td>
      <td>#df65b0</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAACTCA-1_ASAP_PMBDS_000088_s001_Rep1</th>
      <td>AAACCCAAGAAACTCA-1</td>
      <td>ASAP_PMBDS_000088_s001</td>
      <td>10.029368</td>
      <td>1.342043</td>
      <td>cluster_013</td>
      <td>14</td>
      <td>#aa5ed0</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>4b4f4355-6463-4bf2-9544-3f6d493de741</td>
      <td>...</td>
      <td>#4daf4a</td>
      <td>Unknown</td>
      <td>4</td>
      <td>#e8e8e8</td>
      <td>Absent</td>
      <td>1</td>
      <td>#ffffd9</td>
      <td>Thal 2</td>
      <td>3</td>
      <td>#c994c7</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAAGCGA-1_ASAP_PMBDS_000177_s001_rep1</th>
      <td>AAACCCAAGAAAGCGA-1</td>
      <td>ASAP_PMBDS_000177_s001</td>
      <td>-9.057206</td>
      <td>-1.032092</td>
      <td>cluster_006</td>
      <td>7</td>
      <td>#f591a5</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>0ea675f3-4b39-4c16-9518-66ddd9c409a1</td>
      <td>...</td>
      <td>#e41a1c</td>
      <td>Mild Cognitive Impairment</td>
      <td>2</td>
      <td>#377eb8</td>
      <td>Brainstem/Limbic</td>
      <td>5</td>
      <td>#41b6c4</td>
      <td>Thal 3</td>
      <td>4</td>
      <td>#df65b0</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAAGCGA-1_ASAP_PMBDS_000185_s001_rep1</th>
      <td>AAACCCAAGAAAGCGA-1</td>
      <td>ASAP_PMBDS_000185_s001</td>
      <td>-0.685645</td>
      <td>-4.943793</td>
      <td>cluster_001</td>
      <td>2</td>
      <td>#e4a9ba</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>0eb0506b-e76b-4f9a-a3b6-bc6bbccfdd32</td>
      <td>...</td>
      <td>#e41a1c</td>
      <td>Mild Cognitive Impairment</td>
      <td>2</td>
      <td>#377eb8</td>
      <td>Brainstem predominant</td>
      <td>4</td>
      <td>#7fcdbb</td>
      <td>Thal 3</td>
      <td>4</td>
      <td>#df65b0</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAATCCA-1_ASAP_PMBDS_000122_s002_1</th>
      <td>AAACCCAAGAAATCCA-1</td>
      <td>ASAP_PMBDS_000122_s002</td>
      <td>-3.507261</td>
      <td>13.855699</td>
      <td>cluster_002</td>
      <td>3</td>
      <td>#a02226</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>1f67fc2a-7e81-4448-8437-e2c6ef8b4b1b</td>
      <td>...</td>
      <td>#e8e8e8</td>
      <td>Unknown</td>
      <td>4</td>
      <td>#e8e8e8</td>
      <td>Unknown</td>
      <td>10</td>
      <td>#e8e8e8</td>
      <td>Unknown</td>
      <td>8</td>
      <td>#e8e8e8</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 50 columns</p>
</div></div></div>
</div>
</section>
<section id="loading-mapmycells-results">
<h3>Loading MapMyCells Results<a class="headerlink" href="#loading-mapmycells-results" title="Permalink to this heading">#</a></h3>
<p>We load the MapMyCells hierarchical correlation mapping to 10X whole human brain taxonomy results. These are calculated for each cell and can be merged into our cell_metadata table on cell_label index. For more information on MapMyCells see <a class="reference external" href="https://portal.brain-map.org/atlases-and-data/bkp/mapmycells">here</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">siletti_mmc_results</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;mmc_results_siletti_whb&#39;</span><span class="p">,</span>
    <span class="n">comment</span><span class="o">=</span><span class="s1">&#39;#&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">siletti_mmc_results</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>mmc_results_siletti_whb.csv: 100%|██████████| 1.11G/1.11G [01:56&lt;00:00, 9.59MMB/s]   
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>supercluster_label</th>
      <th>supercluster_name</th>
      <th>supercluster_bootstrapping_probability</th>
      <th>supercluster_correlation_coefficient</th>
      <th>supercluster_aggregate_probability</th>
      <th>cluster_label</th>
      <th>cluster_name</th>
      <th>cluster_bootstrapping_probability</th>
      <th>cluster_correlation_coefficient</th>
      <th>cluster_aggregate_probability</th>
      <th>subcluster_label</th>
      <th>subcluster_name</th>
      <th>cluster_alias</th>
      <th>subcluster_bootstrapping_probability</th>
      <th>subcluster_correlation_coefficient</th>
      <th>subcluster_aggregate_probability</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AGCGATTCAGAATGTA-1_ASAP_PMBDS_000105_s001_1</th>
      <td>CS202210140_467</td>
      <td>Oligodendrocyte precursor</td>
      <td>1.00</td>
      <td>0.8196</td>
      <td>1.00</td>
      <td>CS202210140_34</td>
      <td>OPC_33</td>
      <td>1.00</td>
      <td>0.5419</td>
      <td>1.00</td>
      <td>CS202210140_3538</td>
      <td>OPC_33_3044</td>
      <td>3044</td>
      <td>0.75</td>
      <td>0.3870</td>
      <td>0.7500</td>
    </tr>
    <tr>
      <th>GGTGAAGCAGACAATA-1_ASAP_PMBDS_000105_s001_1</th>
      <td>CS202210140_469</td>
      <td>Oligodendrocyte</td>
      <td>1.00</td>
      <td>0.8066</td>
      <td>1.00</td>
      <td>CS202210140_50</td>
      <td>Oligo_49</td>
      <td>0.97</td>
      <td>0.6461</td>
      <td>0.97</td>
      <td>CS202210140_3494</td>
      <td>Oligo_49_3000</td>
      <td>3000</td>
      <td>0.60</td>
      <td>0.3379</td>
      <td>0.5820</td>
    </tr>
    <tr>
      <th>ATTTACCGTTCTAACG-1_ASAP_PMBDS_000105_s001_1</th>
      <td>CS202210140_469</td>
      <td>Oligodendrocyte</td>
      <td>1.00</td>
      <td>0.8252</td>
      <td>1.00</td>
      <td>CS202210140_45</td>
      <td>Oligo_44</td>
      <td>0.97</td>
      <td>0.6821</td>
      <td>0.97</td>
      <td>CS202210140_3496</td>
      <td>Oligo_44_3002</td>
      <td>3002</td>
      <td>1.00</td>
      <td>0.5014</td>
      <td>0.9700</td>
    </tr>
    <tr>
      <th>CTCAAGACACGCTTAA-1_ASAP_PMBDS_000105_s001_1</th>
      <td>CS202210140_470</td>
      <td>Astrocyte</td>
      <td>1.00</td>
      <td>0.7359</td>
      <td>1.00</td>
      <td>CS202210140_63</td>
      <td>Astro_62</td>
      <td>0.94</td>
      <td>0.5680</td>
      <td>0.94</td>
      <td>CS202210140_3618</td>
      <td>Astro_62_3124</td>
      <td>3124</td>
      <td>0.94</td>
      <td>0.3598</td>
      <td>0.8836</td>
    </tr>
    <tr>
      <th>TCGCTTGGTGCGACAA-1_ASAP_PMBDS_000105_s001_1</th>
      <td>CS202210140_467</td>
      <td>Oligodendrocyte precursor</td>
      <td>0.94</td>
      <td>0.7004</td>
      <td>0.94</td>
      <td>CS202210140_34</td>
      <td>OPC_33</td>
      <td>1.00</td>
      <td>0.3904</td>
      <td>0.94</td>
      <td>CS202210140_3539</td>
      <td>OPC_33_3045</td>
      <td>3045</td>
      <td>0.54</td>
      <td>0.2774</td>
      <td>0.5076</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="x-whole-human-brain-taxonomy-siletti-et-al">
<h3>10X Whole Human Brain Taxonomy (Siletti et al.)<a class="headerlink" href="#x-whole-human-brain-taxonomy-siletti-et-al" title="Permalink to this heading">#</a></h3>
<p>The 10X Whole Human Brain (WHB) Taxonomy is a three-level, cell types hierarchical tree (31 superclusters, 461 clusters and 3313 subclusters) derived from a &gt; 3 million cell reference dataset derived from samples across the whole human brain. Additionally, categorical neurotransmitter type annotations were assigned to each cluster based on expression (or lack thereof) of one or more neurotransmitter-associated marker genes. For more information on these data see <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/WHB-10x_snRNASeq_tutorial_part_1.html">here</a>. Here we load information about the taxonomy (terms, colors, hierarchy, levels and order) from the  abc_atlas_access cache object.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">membership</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;WHB-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_to_cluster_annotation_membership&#39;</span>
<span class="p">)</span>
<span class="n">membership_groupby</span> <span class="o">=</span> <span class="n">membership</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">])</span>
<span class="n">membership</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_to_cluster_annotation_membership.csv: 100%|██████████| 1.02M/1.02M [00:00&lt;00:00, 2.68MMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_annotation_term_label</th>
      <th>cluster_annotation_term_set_label</th>
      <th>cluster_alias</th>
      <th>cluster_annotation_term_name</th>
      <th>cluster_annotation_term_set_name</th>
      <th>number_of_cells</th>
      <th>color_hex_triplet</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>CS202210140_494</td>
      <td>CCN202210140_SUBC</td>
      <td>0</td>
      <td>URL_297_0</td>
      <td>subcluster</td>
      <td>34</td>
      <td>#7E807A</td>
    </tr>
    <tr>
      <th>1</th>
      <td>CS202210140_495</td>
      <td>CCN202210140_SUBC</td>
      <td>1</td>
      <td>URL_308_1</td>
      <td>subcluster</td>
      <td>220</td>
      <td>#C54945</td>
    </tr>
    <tr>
      <th>2</th>
      <td>CS202210140_496</td>
      <td>CCN202210140_SUBC</td>
      <td>2</td>
      <td>URL_308_2</td>
      <td>subcluster</td>
      <td>187</td>
      <td>#5232B7</td>
    </tr>
    <tr>
      <th>3</th>
      <td>CS202210140_497</td>
      <td>CCN202210140_SUBC</td>
      <td>3</td>
      <td>URL_308_3</td>
      <td>subcluster</td>
      <td>246</td>
      <td>#31BEBA</td>
    </tr>
    <tr>
      <th>4</th>
      <td>CS202210140_498</td>
      <td>CCN202210140_SUBC</td>
      <td>4</td>
      <td>URL_308_4</td>
      <td>subcluster</td>
      <td>188</td>
      <td>#C8A9BC</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">term_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;WHB-taxonomy&#39;</span><span class="p">,</span> <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term_set&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">cluster_details</span><span class="p">[</span><span class="n">term_sets</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span> <span class="c1"># order columns</span>
<span class="n">cluster_details</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="s1">&#39;Other&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_details</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;supercluster&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster&#39;</span><span class="p">,</span> <span class="s1">&#39;subcluster&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_details</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_annotation_term_set.csv: 100%|██████████| 1.33k/1.33k [00:00&lt;00:00, 12.8kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>cluster_annotation_term_set_name</th>
      <th>subcluster</th>
      <th>cluster</th>
      <th>supercluster</th>
      <th>neurotransmitter</th>
    </tr>
    <tr>
      <th>cluster_alias</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2461</th>
      <td>Amex_153_2461</td>
      <td>Amex_153</td>
      <td>Amygdala excitatory</td>
      <td>VGLUT1 VGLUT2</td>
    </tr>
    <tr>
      <th>2462</th>
      <td>Amex_153_2462</td>
      <td>Amex_153</td>
      <td>Amygdala excitatory</td>
      <td>VGLUT1 VGLUT2</td>
    </tr>
    <tr>
      <th>2463</th>
      <td>Amex_153_2463</td>
      <td>Amex_153</td>
      <td>Amygdala excitatory</td>
      <td>VGLUT1 VGLUT2</td>
    </tr>
    <tr>
      <th>2464</th>
      <td>Amex_153_2464</td>
      <td>Amex_153</td>
      <td>Amygdala excitatory</td>
      <td>VGLUT1 VGLUT2</td>
    </tr>
    <tr>
      <th>2465</th>
      <td>Amex_153_2465</td>
      <td>Amex_153</td>
      <td>Amygdala excitatory</td>
      <td>VGLUT1 VGLUT2</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">cluster_colors</span><span class="p">[</span><span class="n">term_sets</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span>
<span class="n">cluster_colors</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;supercluster&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster&#39;</span><span class="p">,</span> <span class="s1">&#39;subcluster&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_colors</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>cluster_annotation_term_set_name</th>
      <th>subcluster</th>
      <th>cluster</th>
      <th>supercluster</th>
      <th>neurotransmitter</th>
    </tr>
    <tr>
      <th>cluster_alias</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2218</th>
      <td>#374B8A</td>
      <td>#062463</td>
      <td>#003380</td>
      <td>#2BDFD1</td>
    </tr>
    <tr>
      <th>2216</th>
      <td>#4BCBC6</td>
      <td>#062463</td>
      <td>#003380</td>
      <td>#2BDFD1</td>
    </tr>
    <tr>
      <th>2217</th>
      <td>#83C943</td>
      <td>#062463</td>
      <td>#003380</td>
      <td>#2BDFD1</td>
    </tr>
    <tr>
      <th>2219</th>
      <td>#BC4440</td>
      <td>#062463</td>
      <td>#003380</td>
      <td>#2BDFD1</td>
    </tr>
    <tr>
      <th>2220</th>
      <td>#CB7ABA</td>
      <td>#062463</td>
      <td>#003380</td>
      <td>#2BDFD1</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_annotation_term</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span><span class="s1">&#39;WHB-taxonomy&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster_annotation_term&#39;</span><span class="p">)</span>
<span class="n">cluster_annotation_term</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">cluster_annotation_term</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="s1">&#39;Other&#39;</span><span class="p">)</span>
<span class="n">cluster_annotation_term</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;name&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_annotation_term</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_annotation_term.csv: 100%|██████████| 547k/547k [00:00&lt;00:00, 2.14MMB/s] 
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>label</th>
      <th>cluster_annotation_term_set_label</th>
      <th>parent_term_label</th>
      <th>parent_term_set_label</th>
      <th>term_set_order</th>
      <th>term_order</th>
      <th>cluster_annotation_term_set_name</th>
      <th>color_hex_triplet</th>
      <th>number_of_cells</th>
      <th>description</th>
    </tr>
    <tr>
      <th>name</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Upper-layer intratelencephalic</th>
      <td>CS202210140_476</td>
      <td>CCN202210140_SUPC</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>0</td>
      <td>0</td>
      <td>supercluster</td>
      <td>#FEA7BA</td>
      <td>455006</td>
      <td>Upper-layer intratelencephalic</td>
    </tr>
    <tr>
      <th>Deep-layer intratelencephalic</th>
      <td>CS202210140_477</td>
      <td>CCN202210140_SUPC</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>0</td>
      <td>1</td>
      <td>supercluster</td>
      <td>#426600</td>
      <td>228467</td>
      <td>Deep-layer intratelencephalic</td>
    </tr>
    <tr>
      <th>Deep-layer near-projecting</th>
      <td>CS202210140_473</td>
      <td>CCN202210140_SUPC</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>0</td>
      <td>2</td>
      <td>supercluster</td>
      <td>#EE1010</td>
      <td>18856</td>
      <td>Deep-layer near-projecting</td>
    </tr>
    <tr>
      <th>Deep-layer corticothalamic and 6b</th>
      <td>CS202210140_474</td>
      <td>CCN202210140_SUPC</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>0</td>
      <td>3</td>
      <td>supercluster</td>
      <td>#5EF0F1</td>
      <td>78396</td>
      <td>Deep-layer corticothalamic and 6b</td>
    </tr>
    <tr>
      <th>MGE interneuron</th>
      <td>CS202210140_484</td>
      <td>CCN202210140_SUPC</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>0</td>
      <td>4</td>
      <td>supercluster</td>
      <td>#DFFE66</td>
      <td>222434</td>
      <td>MGE interneuron</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Finally, we can merge the cell_metadata with the taxonomy information. We only need to add the Neurotransmitter name/information from our <code class="docutils literal notranslate"><span class="pre">cluster_details</span></code>. We’ll join all colors.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_siletti</span> <span class="o">=</span> <span class="n">cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">siletti_mmc_results</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_siletti&#39;</span><span class="p">)</span>
<span class="n">cell_siletti</span> <span class="o">=</span> <span class="n">cell_siletti</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">cluster_details</span><span class="p">[[</span><span class="s1">&#39;neurotransmitter&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;neurotransmitter&#39;</span><span class="p">:</span> <span class="s1">&#39;neurotransmitter_name&#39;</span><span class="p">}),</span>
    <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span>
<span class="p">)</span>
<span class="n">cell_siletti</span> <span class="o">=</span> <span class="n">cell_siletti</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">cell_siletti</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>sample_label</th>
      <th>x</th>
      <th>y</th>
      <th>cluster_label</th>
      <th>cluster_label_order</th>
      <th>cluster_label_color</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>...</th>
      <th>subcluster_name</th>
      <th>cluster_alias</th>
      <th>subcluster_bootstrapping_probability</th>
      <th>subcluster_correlation_coefficient</th>
      <th>subcluster_aggregate_probability</th>
      <th>neurotransmitter_name</th>
      <th>subcluster</th>
      <th>cluster</th>
      <th>supercluster</th>
      <th>neurotransmitter</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAACCCAAGAAACCAT-1_ASAP_PMBDS_000060_s002_Rep1</th>
      <td>AAACCCAAGAAACCAT-1</td>
      <td>ASAP_PMBDS_000060_s002</td>
      <td>0.016373</td>
      <td>1.403025</td>
      <td>cluster_000</td>
      <td>1</td>
      <td>#171c97</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>cc633260-614c-4e5c-aa18-d3e3a0ef81bb</td>
      <td>...</td>
      <td>Misc_404_2701</td>
      <td>2701</td>
      <td>1.00</td>
      <td>0.4805</td>
      <td>0.990</td>
      <td>VGLUT1</td>
      <td>#767F80</td>
      <td>#B7E52E</td>
      <td>#FEFE00</td>
      <td>#2BDFD1</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAACTCA-1_ASAP_PMBDS_000088_s001_Rep1</th>
      <td>AAACCCAAGAAACTCA-1</td>
      <td>ASAP_PMBDS_000088_s001</td>
      <td>10.029368</td>
      <td>1.342043</td>
      <td>cluster_013</td>
      <td>14</td>
      <td>#aa5ed0</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>4b4f4355-6463-4bf2-9544-3f6d493de741</td>
      <td>...</td>
      <td>DLCT6b_109_2894</td>
      <td>2894</td>
      <td>0.96</td>
      <td>0.4384</td>
      <td>0.768</td>
      <td>VGLUT1</td>
      <td>#C446B4</td>
      <td>#68E5B9</td>
      <td>#5EF0F1</td>
      <td>#2BDFD1</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAAGCGA-1_ASAP_PMBDS_000177_s001_rep1</th>
      <td>AAACCCAAGAAAGCGA-1</td>
      <td>ASAP_PMBDS_000177_s001</td>
      <td>-9.057206</td>
      <td>-1.032092</td>
      <td>cluster_006</td>
      <td>7</td>
      <td>#f591a5</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>0ea675f3-4b39-4c16-9518-66ddd9c409a1</td>
      <td>...</td>
      <td>ULIT_138_2778</td>
      <td>2778</td>
      <td>0.82</td>
      <td>0.4664</td>
      <td>0.820</td>
      <td>VGLUT1 VGLUT2</td>
      <td>#474AC5</td>
      <td>#F5DC97</td>
      <td>#FEA7BA</td>
      <td>#22A5BB</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAAGCGA-1_ASAP_PMBDS_000185_s001_rep1</th>
      <td>AAACCCAAGAAAGCGA-1</td>
      <td>ASAP_PMBDS_000185_s001</td>
      <td>-0.685645</td>
      <td>-4.943793</td>
      <td>cluster_001</td>
      <td>2</td>
      <td>#e4a9ba</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>0eb0506b-e76b-4f9a-a3b6-bc6bbccfdd32</td>
      <td>...</td>
      <td>ULIT_124_2626</td>
      <td>2626</td>
      <td>0.69</td>
      <td>0.4889</td>
      <td>0.276</td>
      <td>VGLUT1 VGLUT2</td>
      <td>#73CDB8</td>
      <td>#F0A4E8</td>
      <td>#FEA7BA</td>
      <td>#22A5BB</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAATCCA-1_ASAP_PMBDS_000122_s002_1</th>
      <td>AAACCCAAGAAATCCA-1</td>
      <td>ASAP_PMBDS_000122_s002</td>
      <td>-3.507261</td>
      <td>13.855699</td>
      <td>cluster_002</td>
      <td>3</td>
      <td>#a02226</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>1f67fc2a-7e81-4448-8437-e2c6ef8b4b1b</td>
      <td>...</td>
      <td>Oligo_47_3023</td>
      <td>3023</td>
      <td>1.00</td>
      <td>0.3952</td>
      <td>0.810</td>
      <td>Other</td>
      <td>#85BE78</td>
      <td>#93B620</td>
      <td>#9CBA19</td>
      <td>#666666</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 71 columns</p>
</div></div></div>
</div>
</section>
</section>
<section id="umap-plotting">
<h2>UMAP Plotting<a class="headerlink" href="#umap-plotting" title="Permalink to this heading">#</a></h2>
<p>We define convenience function to plot the Uniform Manifold Approximation and Projection (UMAP) of the ASAP PMDB data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_umap</span><span class="p">(</span>
    <span class="n">xx</span><span class="p">,</span>
    <span class="n">yy</span><span class="p">,</span>
    <span class="n">cc</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">val</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span>
    <span class="n">cmap</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">labels</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
    <span class="n">sizes</span><span class="o">=</span><span class="kc">None</span>
<span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">sizes</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">sizes</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">scatt</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">val</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="n">sizes</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">cc</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">scatt</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">cc</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="n">sizes</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">labels</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="kn">from</span><span class="w"> </span><span class="nn">matplotlib.patches</span><span class="w"> </span><span class="kn">import</span> <span class="n">Rectangle</span>
        <span class="n">unique_labels</span> <span class="o">=</span> <span class="n">labels</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
        <span class="n">unique_colors</span> <span class="o">=</span> <span class="n">cc</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>

        <span class="k">if</span> <span class="n">term_order_lookup</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">term_order</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">argsort</span><span class="p">(</span><span class="n">term_order_lookup</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">unique_labels</span><span class="p">,</span> <span class="s1">&#39;term_order&#39;</span><span class="p">])</span>
            <span class="n">unique_labels</span> <span class="o">=</span> <span class="n">unique_labels</span><span class="p">[</span><span class="n">term_order</span><span class="p">]</span>
            <span class="n">unique_colors</span> <span class="o">=</span> <span class="n">unique_colors</span><span class="p">[</span><span class="n">term_order</span><span class="p">]</span>
            
        <span class="n">rects</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">color</span> <span class="ow">in</span> <span class="n">unique_colors</span><span class="p">:</span>
            <span class="n">rects</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">),</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fc</span><span class="o">=</span><span class="n">color</span><span class="p">))</span>

        <span class="n">legend</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">unique_labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
        <span class="c1"># ax.add_artist(legend)</span>

    <span class="k">if</span> <span class="n">colorbar</span><span class="p">:</span>
        <span class="n">fig</span><span class="o">.</span><span class="n">colorbar</span><span class="p">(</span><span class="n">scatt</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
    
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<section id="transferred-neurotransmitter-types">
<h3>Transferred Neurotransmitter Types<a class="headerlink" href="#transferred-neurotransmitter-types" title="Permalink to this heading">#</a></h3>
<p>Below are the PMDBS cells colorized by their neurotransmitter types that were inferred from mapping to the WHB taxonomy. Neurotransmitter assignment provides a high-level gestalt of the dataset by dividing the UMAP into glutamergic (shades of blue), GABAergic (reds), their combination (purples) and non-neuronal cells (gray) domains.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;neurotransmitter&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;neurotransmitter_name&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">cluster_annotation_term</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">16</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Neurotransmitter&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/6d75e258e4cf30e79a58157a00cddccd14d2e46f836cb7049e399d9dacf7c5cf.png" src="../_images/6d75e258e4cf30e79a58157a00cddccd14d2e46f836cb7049e399d9dacf7c5cf.png" />
</div>
</div>
</section>
<section id="transferred-whole-human-brain-taxonomy-types">
<h3>Transferred Whole Human Brain Taxonomy Types<a class="headerlink" href="#transferred-whole-human-brain-taxonomy-types" title="Permalink to this heading">#</a></h3>
<p>The next set of plots colorized the PMDBS cells by WHB taxonomy at the supercluster, cluster and subcluster levels. At the supercluster level, we see large swathe of the same colors shows that clusters derived from the scVI integration corresponds well to cell types defined by the WHB taxonomy.</p>
<p>Major glutamergic cortical supercluster annotation from WHB includes intratelencephalic (IT), corticothalamic (CT), and near-projecting (NP) cortical cells, hippocampal types (dente gyrus, CA1-3 and CA4), amygdala excitatory (AMEX) and upper rhombic lip (URL) cells. GABAergic superclusters include CGE (caudal ganglionic eminence) derived, MGE (medial ganglionic eminence) derived, LAMP5-LHX6 and Chandelier (LLC) cells, medium spiny neurons (MSN) and eccentric MSNs. Non-neuronal superclusters include astrocytes, ependymal, microglia, oligodendrocyte, OPCs (oligodendrocyte precursor), vascular and fibroblast cells.</p>
<p>The center of the UMAP shows a region of highly mixed assignments, with lower mapping probabilities (shown later) that could potentially indicate issues with data quality and/or batch effect.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;supercluster&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;supercluster_name&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">cluster_annotation_term</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">16</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Supercluster&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/269831789bb9ea22f475cf1f01b82f0a9a42b0aba6eeb110790bbce8d43cddfd.png" src="../_images/269831789bb9ea22f475cf1f01b82f0a9a42b0aba6eeb110790bbce8d43cddfd.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;cluster&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">16</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Cluster&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/814d48172720caaa3709ab56cbd2234b8e20fd9f09fbe1f9e7d9af5c03e4d4a1.png" src="../_images/814d48172720caaa3709ab56cbd2234b8e20fd9f09fbe1f9e7d9af5c03e4d4a1.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;subcluster&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">16</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;subcluster&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/1cede9a1a6fedcd60260e38b175a14377df0e2a63e77a5a060fb8af5e280a699.png" src="../_images/1cede9a1a6fedcd60260e38b175a14377df0e2a63e77a5a060fb8af5e280a699.png" />
</div>
</div>
</section>
<section id="assignment-correlation-and-probabilities">
<h3>Assignment Correlation and Probabilities<a class="headerlink" href="#assignment-correlation-and-probabilities" title="Permalink to this heading">#</a></h3>
<p>For each cell, at each level of the taxonomy, the MapMyCells results provide the following mapping quality metrics: (1) bootstrapping probability: the fraction of bootstrap iterations that chose the assigned node at that level of the taxonomy, (2) aggregate probability: multiplication of the bootstrapping probability at the current level with its parent levels to provide a total probability and (3) correlation coefficient: Pearson’s correlation coefficient between the gene expression profile of the cell and the gene expression profile of the assigned node across the set of pre-defined marker genes appropriate to that taxonomic node.</p>
<p>The next set of plots colorized PDMBS cells by the boostrapping probability at each level. Here we see that cells in the center part of the UMAP have the lowest probability mapping at every level and corresponding to where we also see highly mixed supercluster assignments.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">val</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;supercluster_bootstrapping_probability&#39;</span><span class="p">],</span>
    <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Supercluster Bootstrap Prob&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/211e621b051eadbe8f46a1a8377cf9a1edb82ed8891901ce93a342333847c578.png" src="../_images/211e621b051eadbe8f46a1a8377cf9a1edb82ed8891901ce93a342333847c578.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">val</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;cluster_bootstrapping_probability&#39;</span><span class="p">],</span>
    <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Cluster Bootstrap Prob&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/71cf72b00ae1298d8bbcf3c1ee83fdf34d37cd3309e5d9ce2062c537ba94d07b.png" src="../_images/71cf72b00ae1298d8bbcf3c1ee83fdf34d37cd3309e5d9ce2062c537ba94d07b.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">val</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;subcluster_bootstrapping_probability&#39;</span><span class="p">],</span>
    <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Subcluster Bootstrap Prob&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/309dbe292c38ce0cac53ce6bb9345b35806de5aa4af0116192ca47c1b3f82877.png" src="../_images/309dbe292c38ce0cac53ce6bb9345b35806de5aa4af0116192ca47c1b3f82877.png" />
</div>
</div>
<p>Each of the above plots shows the probability of being assigned to a given celltype at a level in the taxonomy. This assumes that the cell was correctly assigned to the celltype level above. To calculate the confidence that a given cell is assigned to a type at, say, the leaf level, we need the aggregated probability (the multiplication of the bootstrap probability from parent levels).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">val</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;subcluster_aggregate_probability&#39;</span><span class="p">],</span>
    <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Joint Bootstrap Prob&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/9f41769eea84ff9fc4599e58fff3f11926a3ab218bd543004b6a824f7a9ab64e.png" src="../_images/9f41769eea84ff9fc4599e58fff3f11926a3ab218bd543004b6a824f7a9ab64e.png" />
</div>
</div>
<p>As well as the bootstrapping probability, the correlation between the gene expression of a cell and its assigned cell type also provides vital mapping quality information. Here we colorized the PDMBS cell by the correlation coefficient of gene expression of the cell and its assigned supercluster over the set of pre-defined marker genes for the specific supercluster.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">val</span><span class="o">=</span><span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;supercluster_correlation_coefficient&#39;</span><span class="p">],</span>
    <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Supercluster Correlation Coefficient&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/7f72273c91c47c41fd4d5bb1f1bc4c24c442a006a99b46af5efa341b322f86df.png" src="../_images/7f72273c91c47c41fd4d5bb1f1bc4c24c442a006a99b46af5efa341b322f86df.png" />
</div>
</div>
</section>
<section id="removing-cells-with-low-quality-mapping">
<h3>Removing cells with low quality mapping<a class="headerlink" href="#removing-cells-with-low-quality-mapping" title="Permalink to this heading">#</a></h3>
<p>Now that we’ve seen the probabilities and correlations, one can make cuts on this value to only keep cells that MapMyCells is confident of. Here we make a simple cut based on supercluster probability and correlation both of 50%. This removes roughly 1/3 of the cells. These cuts are not definitive, and users should play with them to achieve a cell mapping quality to their liking.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">supercluster_prob05_mask</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">logical_and</span><span class="p">(</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;supercluster_bootstrapping_probability&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mf">0.5</span><span class="p">,</span>
    <span class="n">cell_siletti</span><span class="p">[</span><span class="s1">&#39;supercluster_correlation_coefficient&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mf">0.5</span>
<span class="p">)</span>
<span class="n">cell_supercluster_high</span> <span class="o">=</span> <span class="n">cell_siletti</span><span class="p">[</span><span class="n">supercluster_prob05_mask</span><span class="p">]</span>
<span class="n">cell_supercluster_low</span> <span class="o">=</span> <span class="n">cell_siletti</span><span class="p">[</span><span class="o">~</span><span class="n">supercluster_prob05_mask</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<p>After making this cut, we removed 758,985 cells with low quality assignments. As seen in the plots below, most of these cells are removed from the central region of the UMAP. The majority of Siletti clusters, the larger fraction of cells were removed from are those that were assigned Splatter cell types.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_supercluster_high</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_supercluster_high</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_supercluster_high</span><span class="p">[</span><span class="s1">&#39;supercluster&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_supercluster_high</span><span class="p">[</span><span class="s1">&#39;supercluster_name&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">cluster_annotation_term</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">16</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;High quality mappings&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/8f1bd6d3e0b0c916b1a9b68bd1248be4dc71ca138f51fbc1c899f988ca9da461.png" src="../_images/8f1bd6d3e0b0c916b1a9b68bd1248be4dc71ca138f51fbc1c899f988ca9da461.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_supercluster_low</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_supercluster_low</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_supercluster_low</span><span class="p">[</span><span class="s1">&#39;supercluster&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_supercluster_low</span><span class="p">[</span><span class="s1">&#39;supercluster_name&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">cluster_annotation_term</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">16</span><span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Low quality mappings&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/c1dc1a0152d7cecdba4c9fc8d3317edbd6e5efae0467f5bac98b905a89cc2c58.png" src="../_images/c1dc1a0152d7cecdba4c9fc8d3317edbd6e5efae0467f5bac98b905a89cc2c58.png" />
</div>
</div>
</section>
</section>
<section id="mapping-the-siletti-superclusters-on-to-the-leiden-clusters">
<h2>Mapping the Siletti superclusters on to the Leiden Clusters<a class="headerlink" href="#mapping-the-siletti-superclusters-on-to-the-leiden-clusters" title="Permalink to this heading">#</a></h2>
<p>Finally, we’ll use the mapping of the cells into the Siletti taxonomy to infer mappings of the Leiden clusters that the ASAP PMDBS data provide. To do this, we’ll compute the fraction of cells that are in a given Leiden cluster that were mapped into a give Siletti supercluster.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Compute the number of cells assigned in a given leiden cluster, Siletti supercluster pair.</span>
<span class="n">group</span> <span class="o">=</span> <span class="n">cell_supercluster_high</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s1">&#39;cluster_label&#39;</span><span class="p">,</span> <span class="s1">&#39;supercluster_name&#39;</span><span class="p">])[[</span><span class="s1">&#39;supercluster&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>

<span class="c1"># Create an empty dataframe to store our results for the fraction of a cluster assigned to a supercluster.</span>
<span class="n">assignment_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span>
    <span class="n">index</span><span class="o">=</span><span class="n">cell_supercluster_high</span><span class="p">[</span><span class="s1">&#39;supercluster_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">(),</span>
    <span class="n">columns</span><span class="o">=</span><span class="n">cell_supercluster_high</span><span class="p">[</span><span class="s1">&#39;cluster_label&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
<span class="p">)</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="mf">0.0</span><span class="p">)</span>

<span class="c1"># Compute the fraction for each leiden cluster, supercluster pair the fraction is the relative to the</span>
<span class="c1"># counts in the leiden supercluster.</span>
<span class="k">for</span> <span class="n">clust_lab</span> <span class="ow">in</span> <span class="n">cell_supercluster_high</span><span class="p">[</span><span class="s1">&#39;cluster_label&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="n">subgroup</span> <span class="o">=</span> <span class="n">group</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">clust_lab</span><span class="p">]</span>
    <span class="n">assignment_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">subgroup</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">clust_lab</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">subgroup</span> <span class="o">/</span> <span class="n">subgroup</span><span class="o">.</span><span class="n">sum</span><span class="p">())</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>

<span class="c1"># Make sure our dataframe rows are in ondological order.</span>
<span class="n">assignment_df</span> <span class="o">=</span> <span class="n">assignment_df</span><span class="o">.</span><span class="n">reindex</span><span class="p">(</span>
    <span class="nb">sorted</span><span class="p">(</span><span class="n">assignment_df</span><span class="o">.</span><span class="n">columns</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span>
<span class="p">)</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cluster_annotation_term</span><span class="p">[</span>
    <span class="n">cluster_annotation_term</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;supercluster&#39;</span>
<span class="p">]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;term_order&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">index</span><span class="p">]</span>

<span class="c1"># For each supertype, retrieve the leiden clusters where that supertype has largest </span>
<span class="c1"># fraction of cells from the leiden cluster assigned to it. Order the columns</span>
<span class="c1"># by supertype decending those maxes.</span>
<span class="n">column_order</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">row_idx</span><span class="p">,</span> <span class="n">supertype</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">assignment_df</span><span class="o">.</span><span class="n">index</span><span class="p">):</span>
    <span class="n">saved_cols</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">col_maxes</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">assignment_df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">max_arg</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">argmax</span><span class="p">(</span><span class="n">assignment_df</span><span class="p">[</span><span class="n">col</span><span class="p">])</span>
        <span class="k">if</span> <span class="n">max_arg</span> <span class="o">==</span> <span class="n">row_idx</span><span class="p">:</span>
            <span class="n">saved_cols</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">col</span><span class="p">)</span>
            <span class="n">col_maxes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">assignment_df</span><span class="p">[</span><span class="n">col</span><span class="p">][</span><span class="n">max_arg</span><span class="p">])</span>
    <span class="n">saved_cols</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">saved_cols</span><span class="p">)</span>
    <span class="n">column_order</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="n">saved_cols</span><span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">argsort</span><span class="p">(</span><span class="n">col_maxes</span><span class="p">)[::</span><span class="o">-</span><span class="mi">1</span><span class="p">]])</span>

<span class="n">assignment_df</span> <span class="o">=</span> <span class="n">assignment_df</span><span class="p">[</span><span class="n">column_order</span><span class="p">]</span>
<span class="n">assignment_df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/var/folders/kc/7glrmt5n67x16yj_tg86t49c0000gp/T/ipykernel_53125/1207378633.py:8: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option(&#39;future.no_silent_downcasting&#39;, True)`
  ).fillna(0.0)
/var/folders/kc/7glrmt5n67x16yj_tg86t49c0000gp/T/ipykernel_53125/1207378633.py:34: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  col_maxes.append(assignment_df[col][max_arg])
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_028</th>
      <th>cluster_001</th>
      <th>cluster_021</th>
      <th>cluster_006</th>
      <th>cluster_015</th>
      <th>cluster_016</th>
      <th>cluster_005</th>
      <th>cluster_011</th>
      <th>cluster_019</th>
      <th>cluster_013</th>
      <th>...</th>
      <th>cluster_025</th>
      <th>cluster_023</th>
      <th>cluster_002</th>
      <th>cluster_022</th>
      <th>cluster_008</th>
      <th>cluster_024</th>
      <th>cluster_003</th>
      <th>cluster_027</th>
      <th>cluster_007</th>
      <th>cluster_010</th>
    </tr>
    <tr>
      <th>name</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Upper-layer intratelencephalic</th>
      <td>0.991699</td>
      <td>0.986873</td>
      <td>0.981704</td>
      <td>0.841231</td>
      <td>0.358504</td>
      <td>0.013095</td>
      <td>0.014055</td>
      <td>0.046075</td>
      <td>0.006404</td>
      <td>0.003158</td>
      <td>...</td>
      <td>0.000339</td>
      <td>0.000882</td>
      <td>0.000044</td>
      <td>0.000507</td>
      <td>0.002780</td>
      <td>0.0</td>
      <td>0.001021</td>
      <td>0.005602</td>
      <td>0.001725</td>
      <td>0.007558</td>
    </tr>
    <tr>
      <th>Deep-layer intratelencephalic</th>
      <td>0.007324</td>
      <td>0.004965</td>
      <td>0.009295</td>
      <td>0.154667</td>
      <td>0.178152</td>
      <td>0.970663</td>
      <td>0.962783</td>
      <td>0.943954</td>
      <td>0.003571</td>
      <td>0.002002</td>
      <td>...</td>
      <td>0.000000</td>
      <td>0.000294</td>
      <td>0.000080</td>
      <td>0.000254</td>
      <td>0.001510</td>
      <td>0.0</td>
      <td>0.000911</td>
      <td>0.000000</td>
      <td>0.000145</td>
      <td>0.001393</td>
    </tr>
    <tr>
      <th>Deep-layer near-projecting</th>
      <td>0.000000</td>
      <td>0.000022</td>
      <td>0.000098</td>
      <td>0.000009</td>
      <td>0.003666</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.957153</td>
      <td>0.001601</td>
      <td>...</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000040</td>
      <td>0.0</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>Deep-layer corticothalamic and 6b</th>
      <td>0.000977</td>
      <td>0.000891</td>
      <td>0.000587</td>
      <td>0.001016</td>
      <td>0.100440</td>
      <td>0.002098</td>
      <td>0.008950</td>
      <td>0.005556</td>
      <td>0.021084</td>
      <td>0.987768</td>
      <td>...</td>
      <td>0.000339</td>
      <td>0.000294</td>
      <td>0.000009</td>
      <td>0.000000</td>
      <td>0.000200</td>
      <td>0.0</td>
      <td>0.000066</td>
      <td>0.000000</td>
      <td>0.000158</td>
      <td>0.000697</td>
    </tr>
    <tr>
      <th>MGE interneuron</th>
      <td>0.000000</td>
      <td>0.000093</td>
      <td>0.000391</td>
      <td>0.000027</td>
      <td>0.032258</td>
      <td>0.000039</td>
      <td>0.000037</td>
      <td>0.000000</td>
      <td>0.001360</td>
      <td>0.000044</td>
      <td>...</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000030</td>
      <td>0.000507</td>
      <td>0.000428</td>
      <td>0.0</td>
      <td>0.000104</td>
      <td>0.000000</td>
      <td>0.000290</td>
      <td>0.001541</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 30 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">heatmap</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">row_labels</span><span class="p">,</span> <span class="n">col_labels</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
            <span class="n">cbar_kw</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">cbarlabel</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">xtick_colors</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    from: https://matplotlib.org/stable/gallery/images_contours_and_fields/image_annotated_heatmap.html</span>

<span class="sd">    Create a heatmap from a numpy array and two lists of labels.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    data</span>
<span class="sd">        A 2D numpy array of shape (M, N).</span>
<span class="sd">    row_labels</span>
<span class="sd">        A list or array of length M with the labels for the rows.</span>
<span class="sd">    col_labels</span>
<span class="sd">        A list or array of length N with the labels for the columns.</span>
<span class="sd">    ax</span>
<span class="sd">        A `matplotlib.axes.Axes` instance to which the heatmap is plotted.  If</span>
<span class="sd">        not provided, use current Axes or create a new one.  Optional.</span>
<span class="sd">    cbar_kw</span>
<span class="sd">        A dictionary with arguments to `matplotlib.Figure.colorbar`.  Optional.</span>
<span class="sd">    cbarlabel</span>
<span class="sd">        The label for the colorbar.  Optional.</span>
<span class="sd">    xtick_colors</span>
<span class="sd">        Set of colors to label the x ticklabels by. Must be the same length as</span>
<span class="sd">        col_labels. Optional.</span>
<span class="sd">    **kwargs</span>
<span class="sd">        All other arguments are forwarded to `imshow`.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">if</span> <span class="n">ax</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">gca</span><span class="p">()</span>

    <span class="k">if</span> <span class="n">cbar_kw</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">cbar_kw</span> <span class="o">=</span> <span class="p">{}</span>

    <span class="c1"># Plot the heatmap</span>
    <span class="n">im</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>

    <span class="c1"># Create colorbar</span>
    <span class="n">cbar</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">figure</span><span class="o">.</span><span class="n">colorbar</span><span class="p">(</span><span class="n">im</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="o">**</span><span class="n">cbar_kw</span><span class="p">)</span>
    <span class="n">cbar</span><span class="o">.</span><span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">cbarlabel</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=-</span><span class="mi">90</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s2">&quot;bottom&quot;</span><span class="p">)</span>

    <span class="c1"># Show all ticks and label them with the respective list entries.</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]),</span> <span class="n">labels</span><span class="o">=</span><span class="n">col_labels</span><span class="p">,</span>
                  <span class="n">rotation</span><span class="o">=-</span><span class="mi">30</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s2">&quot;right&quot;</span><span class="p">,</span> <span class="n">rotation_mode</span><span class="o">=</span><span class="s2">&quot;anchor&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="n">labels</span><span class="o">=</span><span class="n">row_labels</span><span class="p">)</span>


    <span class="c1"># Let the horizontal axes labeling appear on top.</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">top</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">bottom</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                   <span class="n">labeltop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">labelbottom</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

    <span class="c1"># Turn spines off and create white grid.</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[:]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">+</span><span class="mi">1</span><span class="p">)</span><span class="o">-</span><span class="mf">.5</span><span class="p">,</span> <span class="n">minor</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">+</span><span class="mi">1</span><span class="p">)</span><span class="o">-</span><span class="mf">.5</span><span class="p">,</span> <span class="n">minor</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="n">which</span><span class="o">=</span><span class="s2">&quot;minor&quot;</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;w&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;-&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">which</span><span class="o">=</span><span class="s2">&quot;minor&quot;</span><span class="p">,</span> <span class="n">bottom</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">left</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">xtick_colors</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">xticks</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">()</span>

        <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">tick</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">xticks</span><span class="p">):</span>
            <span class="n">tick</span><span class="o">.</span><span class="n">set_color</span><span class="p">(</span><span class="n">xtick_colors</span><span class="p">[</span><span class="n">idx</span><span class="p">])</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">xticks</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">im</span><span class="p">,</span> <span class="n">cbar</span>
</pre></div>
</div>
</div>
</div>
<p>Below we plot the heatmap showing the fraction of cells in a Leiden Cluster assigned to a given Siletti supertype. Note the Leiden cluster names are colored with the colors they were assigned with and shown in the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/asap_pmdbs_data_and_metadata.html">ASAP Data Overview</a> tutorial.</p>
<p>We see that some clusters have clear assignments to Siletti. The ordering we produced above shows a step like structure in the mapping. Horizontal sets show multiple Leiden clusters being assigned to a single Siletti supertype while vertical sets show Leiden clusters that have been split across multiple supertypes that are similar ontologically. We can see that there are also a few Leiden clusters that have ambiguous mappings as well, even after our cut on MapMyCells probability.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span> <span class="mi">12</span><span class="p">)</span>

<span class="n">heatmap</span><span class="p">(</span>
    <span class="n">data</span><span class="o">=</span><span class="n">assignment_df</span><span class="p">,</span>
    <span class="n">row_labels</span><span class="o">=</span><span class="n">assignment_df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">(),</span>
    <span class="n">col_labels</span><span class="o">=</span><span class="n">assignment_df</span><span class="o">.</span><span class="n">columns</span><span class="p">,</span>
    <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">,</span>
    <span class="n">vmin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
    <span class="n">vmax</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
    <span class="n">cbarlabel</span><span class="o">=</span><span class="s2">&quot;Fraction assigned to supertype&quot;</span><span class="p">,</span>
    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
    <span class="n">xtick_colors</span><span class="o">=</span><span class="n">value_sets</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">assignment_df</span><span class="o">.</span><span class="n">columns</span><span class="p">][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/d3e0ac26c4409e5011b6462d8a5e485f3b24a7affed000a2d23acde2f74b7a75.png" src="../_images/d3e0ac26c4409e5011b6462d8a5e485f3b24a7affed000a2d23acde2f74b7a75.png" />
</div>
</div>
<p>In the next notebooks, we explore a similar MapMyCells mapping with the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/asap_pmdbs_seaad_taxonomy.html">SEA-AD taxonomy</a> as well as look at <a class="reference internal" href="asap_pmdbs_gene_expression.html"><span class="doc std std-doc">gene expression data in the UMAP</span></a>.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="asap_pmdbs_data_and_metadata.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</p>
      </div>
    </a>
    <a class="right-next"
       href="asap_pmdbs_seaad_taxonomy.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#asap-data">ASAP Data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#loading-mapmycells-results">Loading MapMyCells Results</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#x-whole-human-brain-taxonomy-siletti-et-al">10X Whole Human Brain Taxonomy (Siletti et al.)</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#umap-plotting">UMAP Plotting</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#transferred-neurotransmitter-types">Transferred Neurotransmitter Types</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#transferred-whole-human-brain-taxonomy-types">Transferred Whole Human Brain Taxonomy Types</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#assignment-correlation-and-probabilities">Assignment Correlation and Probabilities</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#removing-cells-with-low-quality-mapping">Removing cells with low quality mapping</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mapping-the-siletti-superclusters-on-to-the-leiden-clusters">Mapping the Siletti superclusters on to the Leiden Clusters</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=5b4479735964841361fd"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>