
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview &#8212; Allen Brain Cell Atlas - Data Access</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=9c3e77be" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/simple_style.css?v=3aaed5dc" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=1ae7504c"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/asap_pmdbs_data_and_metadata';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy" href="asap_pmdbs_siletti_taxonomy.html" />
    <link rel="prev" title="ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks" href="../descriptions/ASAP-PMDBS_notebooks.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/allen_logo.jpeg" class="logo__image only-light" alt="Allen Brain Cell Atlas - Data Access - Home"/>
    <script>document.write(`<img src="../_static/allen_logo.jpeg" class="logo__image only-dark" alt="Allen Brain Cell Atlas - Data Access - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Allen Brain Cell Atlas - Data Access
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">INTRODUCTION</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="general_accessing_10x_snRNASeq_tutorial.html">Accessing 10x RNA-seq gene expression data</a></li>
<li class="toctree-l1"><a class="reference internal" href="abc_atlas_selection_example.html">Using cells selected in the ABC Atlas with Gene Expression data.</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">DATA</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WMB_dataset.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv2.html">Whole mouse brain 10Xv2 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv3.html">Whole mouse brain 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10XMulti.html">Whole mouse brain 10XMulti single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10X.html">Clustering analysis of whole adult mouse brain 10X single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-taxonomy.html">Whole adult mouse brain taxonomy of cell types</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-imputed.html">Imputed MERFISH spatial transcriptomics of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-neighborhoods.html">Cluster groups and neighborhood specific UMAP embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Allen-CCF-2020.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-CCF.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_dataset.html">Aging Mouse dataset (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_10Xv3.html">Zeng Aging Mouse 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_taxonomy.html">Zeng Aging Mouse clustering and mapping to the WMB taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zhuang_dataset.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-1.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-2.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-3.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-3)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-4.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-4)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-dataset.html">Consensus Whole Mouse Brain dataset (Macosko, Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-10X.html">Consensus Whole Mouse Brain single cell/nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-taxonomy.html">Consensus Whole Mouse Brain integrated taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-dataset.html">Developing Mouse - Visual Cortex</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-10X.html">Developing Mouse - Visual Cortex single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-taxonomy.html">Developing Mouse - Visual Cortex taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_dataset.html">Human whole-brain transcriptomic cell type atlas (Kimberly Siletti)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10Xv3.html">Human brain 10Xv3 single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10X-clustering.html">Clustering analysis of whole human brain 10X single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-taxonomy.html">Human brain taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_dataset.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-10X.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-MapMyCells.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain (WHB) and Seattle Alzheimer’s Disease Brain Cell Atlas (SEA-AD) Taxonomies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_dataset.html">Human-Mammalian Brain - Basal Ganglia: Data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-BG-taxonomy-CCN20250428.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq data: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG-Aligned.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Aligned gene expression across species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Individual species gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Spatial-BG.html">Human-Mammalian Brain - Basal Ganglia: spatial transcriptomics cells for each species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Macaque-PatchSeq-BG.html">Human-Mammalian Brain - Basal Ganglia: Macaque Patch-seq cells multimodal data</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">NOTEBOOKS</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cluster_annotation_tutorial.html">10x RNA-seq clustering analysis and annotation (CCN20230722)</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle2.html">10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_1.html">10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2a.html">10x RNA-seq gene expression data (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2b.html">10x RNA-seq gene expression data (part 2b)</a></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle3.html">MERFISH whole brain spatial transcriptomics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_1.html">MERFISH whole brain spatial transcriptomics (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2a.html">MERFISH whole brain spatial transcriptomics (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2b.html">MERFISH whole brain spatial transcriptomics (part 2b)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_imputed_genes_example.html">MERFISH whole brain spatial transcriptomics cells with imputed genes</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_groups_and_embeddings_tutorial.html">Cluster neighborhoods and embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_neighborhood_gallery.html">Cell type neighborhood gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="ccf_and_parcellation_annotation_tutorial.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="merfish_ccf_registration_tutorial.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-notebooks.html">Consensus Mouse notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_clustering_analysis_and_annotation.html">Consensus Whole Mouse Brain #1: Clustering and Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_10X_snRNASeq.html">Consensus Whole Mouse Brain #2: 10X gene expression</a></li>

</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-notebooks.html">Developing Mouse - Visual Cortex notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="dev_mouse_vis_cortex_clustering_analysis_and_annotation.html">Developing Mouse - Visual Cortex 10X scRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="dev_mouse_vis_cortex_10X_scRNASeq.html">Developing Mouse - Visual Cortex: 10X gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_notebooks.html">Aging Mouse Notebooks (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_clustering_analysis_and_annotation.html">Mouse Aging clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_10x_snRNASeq_tutorial.html">Mouse Aging 10X RNA-seq gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle4.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="zhuang_merfish_tutorial.html">MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_notebooks.html">Whole Human Brain 10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="WHB_cluster_annotation_tutorial.html">Whole Human Brain 10x RNA-seq clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_1.html">Whole Human Brain (WHB) 10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_2.html">Whole Human Brain 10x RNA-seq gene expression data (part 2)</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_notebooks.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_siletti_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_seaad_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_gene_expression.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Gene Expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_notebooks.html">Human-Mammalian Brain - Basal Ganglia: Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_clustering_analysis_and_annotation.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_10X_snRNASeq_tutorial.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_slabs_and_taxonomy.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_genes.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_macaque_patchseq.html">Human-Mammalian Brain - Basal Ganglia Patchseq</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">ADDITIONAL RESOURCES</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../docs/README.html">Resource Pages</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../docs/abbreviation_terms/README.html">Abbreviation lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/cluster_annotations/README.html">Cluster annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/gene_lists/README.html">Gene lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/parcellation_annotations/README.html">Parcellation annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/supplemental_metadata_descriptions/README.html">Supplemental Metadata Descriptions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/neuroglancer_hmba/README.html">HMBA Basal Ganglia: Neuroglancer Views</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access/issues/new?title=Issue%20on%20page%20%2Fnotebooks/asap_pmdbs_data_and_metadata.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/asap_pmdbs_data_and_metadata.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#asap-data">ASAP Data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#umap-plotting">UMAP Plotting</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#data-acquisition-metadata">Data Acquisition Metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#donor-metadata">Donor Metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#clinical-pathology">Clinical  Pathology</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#leiden-scvi-clusters">Leiden scVI Clusters</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="asap-human-postmortem-derived-brain-sequencing-collection-pmdbs-data-overview">
<h1>ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview<a class="headerlink" href="#asap-human-postmortem-derived-brain-sequencing-collection-pmdbs-data-overview" title="Link to this heading">#</a></h1>
<p>Aligning Science Across Parkinson’s (ASAP), The Michael J. Fox Foundation for Parkinson’s Research (MJFF), and the Allen Institute for Brain Science (AIBS) are teaming up to further the mission of the ASAP Collaborative Research Network (CRN) program, to accelerate discoveries in the Parkinson’s disease (PD) and neurodegenerative disease research communities. Together we will annotate, enhance and add knowledge to the growing data catalog in the ASAP CRN Cloud through integration of cell type taxonomies using the Allen Institute’s MapMyCells tool and visualization through the Allen Brain Cell (ABC) Atlas web application. This integration of data and knowledge will allow users to visualize and explore the changes in gene expression of specific, highly resolved brain cell types in the context of a large PD cohort of donors.</p>
<p>This initial collaboration focuses on the Human Postmortem-derived Brain Sequencing Collection (PMDBS), a harmonized repository comprised of single nucleus and PolyA RNA-seq data contributed by five ASAP CRN teams (Hafler, Lee, Jakobsson, Scherzer, Hardy). Sequencing data were uniformly aligned to the GRCh38.p13 reference genome (Gencode V32), quality control was performed and low-quality cells were filtered out. A set of highly variable genes were identified and the scVI workflow resulted in an integrated latent variable representation, 2D UMAP coordinates and a set of 30 clusters. Currently, the repository spans roughly 3 millions cells obtained from 9 brain regions and 211 donors with various pathologies (including healthy control). For more details on this dataset and to access the raw data used in its preparation, please visit the
<a class="reference external" href="https://cloud.parkinsonsroadmap.org/collections/postmortem-derived-brain-sequencing-collection/overview">ASAP CRN Cloud webpage</a>.</p>
<p>In this notebook we assemble the metadata associated with individual cells derived from donor, sample and data integration process information.</p>
<p>You need to be connected to the internet to run this notebook and should have run through the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html">getting started notebook</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">anndata</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">abc_atlas_access.abc_atlas_cache.abc_project_cache</span><span class="w"> </span><span class="kn">import</span> <span class="n">AbcProjectCache</span>
</pre></div>
</div>
</div>
</div>
<p>We will interact with the data using the AbcProjectCache. This cache object tracks which data has been downloaded and serves the path to the requested data on disk. For metadata, the cache can also directly serve up a Pandas DataFrame.  See the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html">getting started notebook</a> notebook for more details on using the cache including installing it if it has not already been.</p>
<p>Change the download_base variable to where you have downloaded the data in your system.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">download_base</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">&#39;../../data/abc_atlas&#39;</span><span class="p">)</span>
<span class="n">abc_cache</span> <span class="o">=</span> <span class="n">AbcProjectCache</span><span class="o">.</span><span class="n">from_cache_dir</span><span class="p">(</span><span class="n">download_base</span><span class="p">)</span>

<span class="n">abc_cache</span><span class="o">.</span><span class="n">current_manifest</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;releases/20250531/manifest.json&#39;
</pre></div>
</div>
</div>
</div>
<section id="data-overview">
<h2>Data Overview<a class="headerlink" href="#data-overview" title="Link to this heading">#</a></h2>
<section id="asap-data">
<h3>ASAP Data<a class="headerlink" href="#asap-data" title="Link to this heading">#</a></h3>
<p>We simplify the ASAP-PMDB metadata into a single cell_metadata.csv file and companion tables describing the donor (including sex, age, race, disease state etc.), sample (including data processing metadata, team information, and region of interest from which the sample was dissected from), and value_sets.csv which provides a mapping from unique terms to their plotting color and term order.</p>
<p>We’ll first load the cell metadata. This provides the UMAP x, y coordinates and clusters provided by ASAP. Refer to the collection documentation for description of the data processing and source datasets.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Number of cells = &quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">cell_metadata</span><span class="p">))</span>
<span class="n">cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_metadata.csv: 100%|██████████| 555M/555M [01:01&lt;00:00, 9.09MMB/s] 
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of cells =  2796736
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>sample_label</th>
      <th>x</th>
      <th>y</th>
      <th>cluster_label</th>
      <th>cluster_label_order</th>
      <th>cluster_label_color</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAACCCAAGAAACCAT-1_ASAP_PMBDS_000060_s002_Rep1</th>
      <td>AAACCCAAGAAACCAT-1</td>
      <td>ASAP_PMBDS_000060_s002</td>
      <td>0.016373</td>
      <td>1.403025</td>
      <td>cluster_000</td>
      <td>1</td>
      <td>#171c97</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>cc633260-614c-4e5c-aa18-d3e3a0ef81bb</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAACTCA-1_ASAP_PMBDS_000088_s001_Rep1</th>
      <td>AAACCCAAGAAACTCA-1</td>
      <td>ASAP_PMBDS_000088_s001</td>
      <td>10.029368</td>
      <td>1.342043</td>
      <td>cluster_013</td>
      <td>14</td>
      <td>#aa5ed0</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>4b4f4355-6463-4bf2-9544-3f6d493de741</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAAGCGA-1_ASAP_PMBDS_000177_s001_rep1</th>
      <td>AAACCCAAGAAAGCGA-1</td>
      <td>ASAP_PMBDS_000177_s001</td>
      <td>-9.057206</td>
      <td>-1.032092</td>
      <td>cluster_006</td>
      <td>7</td>
      <td>#f591a5</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>0ea675f3-4b39-4c16-9518-66ddd9c409a1</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAAGCGA-1_ASAP_PMBDS_000185_s001_rep1</th>
      <td>AAACCCAAGAAAGCGA-1</td>
      <td>ASAP_PMBDS_000185_s001</td>
      <td>-0.685645</td>
      <td>-4.943793</td>
      <td>cluster_001</td>
      <td>2</td>
      <td>#e4a9ba</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>0eb0506b-e76b-4f9a-a3b6-bc6bbccfdd32</td>
    </tr>
    <tr>
      <th>AAACCCAAGAAATCCA-1_ASAP_PMBDS_000122_s002_1</th>
      <td>AAACCCAAGAAATCCA-1</td>
      <td>ASAP_PMBDS_000122_s002</td>
      <td>-3.507261</td>
      <td>13.855699</td>
      <td>cluster_002</td>
      <td>3</td>
      <td>#a02226</td>
      <td>ASAP-PMDBS-10X</td>
      <td>ASAP-PMDBS-10X</td>
      <td>1f67fc2a-7e81-4448-8437-e2c6ef8b4b1b</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We can use pandas groupby function to see how many unique items are associated for each field and list them out if the number of items is small.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">print_column_info</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
    
    <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">grouped</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="n">c</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">c</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
        <span class="n">members</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">grouped</span><span class="p">)</span> <span class="o">&lt;</span> <span class="mi">30</span><span class="p">:</span>
            <span class="n">members</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">grouped</span><span class="o">.</span><span class="n">index</span><span class="p">))</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Number of unique </span><span class="si">%s</span><span class="s2"> = </span><span class="si">%d</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">grouped</span><span class="p">),</span> <span class="n">members</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">print_column_info</span><span class="p">(</span><span class="n">cell_metadata</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of unique cell_barcode = 2040051 
Number of unique sample_label = 383 
Number of unique x = 2735345 
Number of unique y = 2711023 
Number of unique cluster_label = 30 
Number of unique cluster_label_order = 30 
Number of unique cluster_label_color = 30 
Number of unique dataset_label = 1 [&#39;ASAP-PMDBS-10X&#39;]
Number of unique feature_matrix_label = 1 [&#39;ASAP-PMDBS-10X&#39;]
Number of unique abc_sample_id = 2796736 
</pre></div>
</div>
</div>
</div>
<p>Nex we’ll load and join the sample metadata. This contains information about the individual tissue samples that make up the combined dataset. This includes which ASAP Team collected and processed the cells, the technique used to measure the gene expression, and the region of interest in the brain the sample came from. The sample table is a version of the Sample metadata shown <a class="reference external" href="https://cloud.parkinsonsroadmap.org/collections/postmortem-derived-brain-sequencing-collection/overview">here</a> with some post processing to disambiguate the data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sample</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;sample&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;sample_label&#39;</span><span class="p">)</span>
<span class="n">sample</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>sample.csv: 100%|██████████| 50.1k/50.1k [00:00&lt;00:00, 547kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>donor_label</th>
      <th>source_dataset_label</th>
      <th>source_dataset_label_order</th>
      <th>source_dataset_label_color</th>
      <th>technique</th>
      <th>technique_order</th>
      <th>technique_color</th>
      <th>region_of_interest_label</th>
      <th>region_of_interest_label_order</th>
      <th>region_of_interest_label_color</th>
    </tr>
    <tr>
      <th>sample_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>ASAP_PMBDS_000001_s001</th>
      <td>ASAP_PMBDS_000001</td>
      <td>team_lee_sn_rnaseq</td>
      <td>1</td>
      <td>#e41a1c</td>
      <td>v1</td>
      <td>1</td>
      <td>#03738C</td>
      <td>hippocampus</td>
      <td>6</td>
      <td>#bfb5d5</td>
    </tr>
    <tr>
      <th>ASAP_PMBDS_000001_s002</th>
      <td>ASAP_PMBDS_000001</td>
      <td>team_lee_sn_rnaseq</td>
      <td>1</td>
      <td>#e41a1c</td>
      <td>v1</td>
      <td>1</td>
      <td>#03738C</td>
      <td>middle frontal gyrus</td>
      <td>1</td>
      <td>#d4b235</td>
    </tr>
    <tr>
      <th>ASAP_PMBDS_000001_s003</th>
      <td>ASAP_PMBDS_000001</td>
      <td>team_lee_sn_rnaseq</td>
      <td>1</td>
      <td>#e41a1c</td>
      <td>v1</td>
      <td>1</td>
      <td>#03738C</td>
      <td>substantia nigra</td>
      <td>9</td>
      <td>#6ca9bf</td>
    </tr>
    <tr>
      <th>ASAP_PMBDS_000002_s001</th>
      <td>ASAP_PMBDS_000002</td>
      <td>team_lee_sn_rnaseq</td>
      <td>1</td>
      <td>#e41a1c</td>
      <td>v1</td>
      <td>1</td>
      <td>#03738C</td>
      <td>hippocampus</td>
      <td>6</td>
      <td>#bfb5d5</td>
    </tr>
    <tr>
      <th>ASAP_PMBDS_000002_s002</th>
      <td>ASAP_PMBDS_000002</td>
      <td>team_lee_sn_rnaseq</td>
      <td>1</td>
      <td>#e41a1c</td>
      <td>v1</td>
      <td>1</td>
      <td>#03738C</td>
      <td>middle frontal gyrus</td>
      <td>1</td>
      <td>#d4b235</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We join the sample information into the full cell metadata table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_metadata</span> <span class="o">=</span> <span class="n">cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">sample</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;sample_label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Next, we load the donor table. This contains information on the donors (age, sex, race, pathology, etc.) that make up the study and is a processed copy of the Donor table from the same ASAP data description page linked above.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">donor</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;donor&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
<span class="n">donor</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>donor.csv: 100%|██████████| 50.3k/50.3k [00:00&lt;00:00, 623kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>donor_race</th>
      <th>donor_race_order</th>
      <th>donor_race_color</th>
      <th>donor_sex</th>
      <th>donor_sex_order</th>
      <th>donor_sex_color</th>
      <th>primary_diagnosis</th>
      <th>primary_diagnosis_order</th>
      <th>primary_diagnosis_color</th>
      <th>age_at_death</th>
      <th>...</th>
      <th>cerad_score_color</th>
      <th>cognitive_status</th>
      <th>cognitive_status_order</th>
      <th>cognitive_status_color</th>
      <th>lewy_body_disease_pathology</th>
      <th>lewy_body_disease_pathology_order</th>
      <th>lewy_body_disease_pathology_color</th>
      <th>thal_phase</th>
      <th>thal_phase_order</th>
      <th>thal_phase_color</th>
    </tr>
    <tr>
      <th>donor_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>ASAP_PMBDS_000001</th>
      <td>White</td>
      <td>2</td>
      <td>#38CAE8</td>
      <td>Male</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>No PD nor other neurological disorder</td>
      <td>2</td>
      <td>#99d594</td>
      <td>78 - 89 yrs</td>
      <td>...</td>
      <td>#e8e8e8</td>
      <td>Normal</td>
      <td>1</td>
      <td>#4daf4a</td>
      <td>Olfactory bulb only</td>
      <td>2</td>
      <td>#edf8b1</td>
      <td>Unknown</td>
      <td>8</td>
      <td>#e8e8e8</td>
    </tr>
    <tr>
      <th>ASAP_PMBDS_000002</th>
      <td>White</td>
      <td>2</td>
      <td>#38CAE8</td>
      <td>Male</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>Other neurological disorder</td>
      <td>3</td>
      <td>#e6f598</td>
      <td>78 - 89 yrs</td>
      <td>...</td>
      <td>#e8e8e8</td>
      <td>Normal</td>
      <td>1</td>
      <td>#4daf4a</td>
      <td>Brainstem predominant</td>
      <td>4</td>
      <td>#7fcdbb</td>
      <td>Thal 0</td>
      <td>1</td>
      <td>#f1eef6</td>
    </tr>
    <tr>
      <th>ASAP_PMBDS_000003</th>
      <td>White</td>
      <td>2</td>
      <td>#38CAE8</td>
      <td>Male</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>Idiopathic Parkinson's disease</td>
      <td>7</td>
      <td>#d53e4f</td>
      <td>&lt; 65 yrs</td>
      <td>...</td>
      <td>#e8e8e8</td>
      <td>Mild Cognitive Impairment</td>
      <td>2</td>
      <td>#377eb8</td>
      <td>Limbic (transitional)</td>
      <td>7</td>
      <td>#225ea8</td>
      <td>Thal 0</td>
      <td>1</td>
      <td>#f1eef6</td>
    </tr>
    <tr>
      <th>ASAP_PMBDS_000004</th>
      <td>White</td>
      <td>2</td>
      <td>#38CAE8</td>
      <td>Male</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>Idiopathic Parkinson's disease</td>
      <td>7</td>
      <td>#d53e4f</td>
      <td>78 - 89 yrs</td>
      <td>...</td>
      <td>#e8e8e8</td>
      <td>Mild Cognitive Impairment</td>
      <td>2</td>
      <td>#377eb8</td>
      <td>Neocortical</td>
      <td>8</td>
      <td>#253494</td>
      <td>Unknown</td>
      <td>8</td>
      <td>#e8e8e8</td>
    </tr>
    <tr>
      <th>ASAP_PMBDS_000005</th>
      <td>White</td>
      <td>2</td>
      <td>#38CAE8</td>
      <td>Male</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>Idiopathic Parkinson's disease</td>
      <td>7</td>
      <td>#d53e4f</td>
      <td>78 - 89 yrs</td>
      <td>...</td>
      <td>#4daf4a</td>
      <td>Normal</td>
      <td>1</td>
      <td>#4daf4a</td>
      <td>Neocortical</td>
      <td>8</td>
      <td>#253494</td>
      <td>Unknown</td>
      <td>8</td>
      <td>#e8e8e8</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 30 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_metadata</span> <span class="o">=</span> <span class="n">cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">donor</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Let’s print out information on our final table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">print_column_info</span><span class="p">(</span><span class="n">cell_metadata</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of unique cell_barcode = 2040051 
Number of unique sample_label = 383 
Number of unique x = 2735345 
Number of unique y = 2711023 
Number of unique cluster_label = 30 
Number of unique cluster_label_order = 30 
Number of unique cluster_label_color = 30 
Number of unique dataset_label = 1 [&#39;ASAP-PMDBS-10X&#39;]
Number of unique feature_matrix_label = 1 [&#39;ASAP-PMDBS-10X&#39;]
Number of unique abc_sample_id = 2796736 
Number of unique donor_label = 211 
Number of unique source_dataset_label = 5 [&#39;team_hafler_sn_rnaseq_pfc&#39;, &#39;team_hardy_sn_rnaseq&#39;, &#39;team_jakobsson_sn_rnaseq&#39;, &#39;team_lee_sn_rnaseq&#39;, &#39;team_scherzer_sn_rnaseq_mtg&#39;]
Number of unique source_dataset_label_order = 5 [1, 2, 3, 4, 5]
Number of unique source_dataset_label_color = 5 [&#39;#377eb8&#39;, &#39;#4daf4a&#39;, &#39;#984ea3&#39;, &#39;#e41a1c&#39;, &#39;#ff7f00&#39;]
Number of unique technique = 3 [&#39;v1&#39;, &#39;v3.1 - Dual Index&#39;, &#39;v3.1 - Single Index&#39;]
Number of unique technique_order = 3 [1, 2, 3]
Number of unique technique_color = 3 [&#39;#03738C&#39;, &#39;#F27329&#39;, &#39;#F2B591&#39;]
Number of unique region_of_interest_label = 9 [&#39;amygdaloid complex&#39;, &#39;anterior cingulate gyrus&#39;, &#39;hippocampus&#39;, &#39;inferior parietal lobule&#39;, &#39;middle frontal gyrus&#39;, &#39;middle temporal gyrus&#39;, &#39;prefrontal cortex&#39;, &#39;putamen&#39;, &#39;substantia nigra&#39;]
Number of unique region_of_interest_label_order = 9 [1, 2, 3, 4, 5, 6, 7, 8, 9]
Number of unique region_of_interest_label_color = 9 [&#39;#6ca9bf&#39;, &#39;#a8b485&#39;, &#39;#bfb5d5&#39;, &#39;#c9e2b1&#39;, &#39;#d38e32&#39;, &#39;#d4b235&#39;, &#39;#d55c92&#39;, &#39;#eee0a8&#39;, &#39;#efb9be&#39;]
Number of unique donor_race = 3 [&#39;Black or African American&#39;, &#39;Unknown&#39;, &#39;White&#39;]
Number of unique donor_race_order = 3 [1, 2, 3]
Number of unique donor_race_color = 3 [&#39;#38CAE8&#39;, &#39;#8F3A5C&#39;, &#39;#e8e8e8&#39;]
Number of unique donor_sex = 2 [&#39;Female&#39;, &#39;Male&#39;]
Number of unique donor_sex_order = 2 [1, 2]
Number of unique donor_sex_color = 2 [&#39;#565353&#39;, &#39;#ADC4C3&#39;]
Number of unique primary_diagnosis = 7 [&quot;Alzheimer&#39;s disease&quot;, &#39;Healthy Control&#39;, &quot;Idiopathic Parkinson&#39;s disease&quot;, &#39;No PD nor other neurological disorder&#39;, &#39;Other neurological disorder&#39;, &quot;Parkinson&#39;s disease&quot;, &quot;Prodromal motor Parkinson&#39;s disease&quot;]
Number of unique primary_diagnosis_order = 7 [1, 2, 3, 4, 5, 6, 7]
Number of unique primary_diagnosis_color = 7 [&#39;#3288bd&#39;, &#39;#99d594&#39;, &#39;#d53e4f&#39;, &#39;#e6f598&#39;, &#39;#fc8d59&#39;, &#39;#fee08b&#39;, &#39;#ffffbf&#39;]
Number of unique age_at_death = 4 [&#39;65 - 77 yrs&#39;, &#39;78 - 89 yrs&#39;, &#39;90+ yrs&#39;, &#39;&lt; 65 yrs&#39;]
Number of unique age_at_death_order = 4 [1, 2, 3, 4]
Number of unique age_at_death_color = 4 [&#39;#2b8cbe&#39;, &#39;#7bccc4&#39;, &#39;#bae4bc&#39;, &#39;#f0f9e8&#39;]
Number of unique apoe4_status = 7 [&#39;Unknown&#39;, &#39;e2/e2&#39;, &#39;e2/e3&#39;, &#39;e2/e4&#39;, &#39;e3/e3&#39;, &#39;e3/e4&#39;, &#39;e4/e4&#39;]
Number of unique apoe4_status_order = 7 [1, 2, 3, 4, 5, 6, 7]
Number of unique apoe4_status_color = 7 [&#39;#1b9e77&#39;, &#39;#66a61e&#39;, &#39;#7570b3&#39;, &#39;#d95f02&#39;, &#39;#e6ab02&#39;, &#39;#e7298a&#39;, &#39;#e8e8e8&#39;]
Number of unique braak_stage = 7 [&#39;Braak 0&#39;, &#39;Braak I&#39;, &#39;Braak II&#39;, &#39;Braak III&#39;, &#39;Braak IV&#39;, &#39;Braak VI&#39;, &#39;Unknown&#39;]
Number of unique braak_stage_order = 6 [1, 2, 3, 4, 5, 7]
Number of unique braak_stage_color = 7 [&#39;#91003f&#39;, &#39;#c994c7&#39;, &#39;#d4b9da&#39;, &#39;#df65b0&#39;, &#39;#e7298a&#39;, &#39;#e8e8e8&#39;, &#39;#f1eef6&#39;]
Number of unique cerad_score = 4 [&#39;Frequent&#39;, &#39;Moderate&#39;, &#39;Sparse&#39;, &#39;Unknown&#39;]
Number of unique cerad_score_order = 4 [1, 2, 3, 4]
Number of unique cerad_score_color = 4 [&#39;#377eb8&#39;, &#39;#4daf4a&#39;, &#39;#e41a1c&#39;, &#39;#e8e8e8&#39;]
Number of unique cognitive_status = 4 [&#39;Dementia&#39;, &#39;Mild Cognitive Impairment&#39;, &#39;Normal&#39;, &#39;Unknown&#39;]
Number of unique cognitive_status_order = 4 [1, 2, 3, 4]
Number of unique cognitive_status_color = 4 [&#39;#377eb8&#39;, &#39;#4daf4a&#39;, &#39;#e41a1c&#39;, &#39;#e8e8e8&#39;]
Number of unique lewy_body_disease_pathology = 10 [&#39;Absent&#39;, &#39;Amygdala predominant&#39;, &#39;Brainstem predominant&#39;, &#39;Brainstem/Limbic&#39;, &#39;Diffuse, neocortical (brainstem, limbic and neocortical involvement)&#39;, &#39;Limbic (transitional)&#39;, &#39;Limbic predominant&#39;, &#39;Neocortical&#39;, &#39;Olfactory bulb only&#39;, &#39;Unknown&#39;]
Number of unique lewy_body_disease_pathology_order = 10 [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Number of unique lewy_body_disease_pathology_color = 10 [&#39;#081d58&#39;, &#39;#1d91c0&#39;, &#39;#225ea8&#39;, &#39;#253494&#39;, &#39;#41b6c4&#39;, &#39;#7fcdbb&#39;, &#39;#c7e9b4&#39;, &#39;#e8e8e8&#39;, &#39;#edf8b1&#39;, &#39;#ffffd9&#39;]
Number of unique thal_phase = 8 [&#39;Thal 0&#39;, &#39;Thal 1&#39;, &#39;Thal 2&#39;, &#39;Thal 3&#39;, &#39;Thal 4&#39;, &#39;Thal 4/5&#39;, &#39;Thal 5&#39;, &#39;Unknown&#39;]
Number of unique thal_phase_order = 8 [1, 2, 3, 4, 5, 6, 7, 8]
Number of unique thal_phase_color = 8 [&#39;#91003f&#39;, &#39;#c994c7&#39;, &#39;#ce1256&#39;, &#39;#d4b9da&#39;, &#39;#df65b0&#39;, &#39;#e7298a&#39;, &#39;#e8e8e8&#39;, &#39;#f1eef6&#39;]
</pre></div>
</div>
</div>
</div>
<p>Finally, we’ll load the value_sets table. This is mainly a mapping between the different sets of terms (e.g. regions of interest, cell types, age ranges), providing unique colors for each term in a set and an order of the set. We’ll primarily use it as a lookup table to order or different terms in plot legends.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">value_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;ASAP-PMDBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;value_sets&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">value_sets</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>value_sets.csv: 100%|██████████| 4.25k/4.25k [00:00&lt;00:00, 68.6kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>field</th>
      <th>order</th>
      <th>color_hex_triplet</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>v1</th>
      <td>technique</td>
      <td>1</td>
      <td>#03738C</td>
    </tr>
    <tr>
      <th>v3.1 - Dual Index</th>
      <td>technique</td>
      <td>3</td>
      <td>#F27329</td>
    </tr>
    <tr>
      <th>v3.1 - Single Index</th>
      <td>technique</td>
      <td>2</td>
      <td>#F2B591</td>
    </tr>
    <tr>
      <th>cluster_000</th>
      <td>cluster_label</td>
      <td>1</td>
      <td>#171c97</td>
    </tr>
    <tr>
      <th>cluster_001</th>
      <td>cluster_label</td>
      <td>2</td>
      <td>#e4a9ba</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
</section>
<section id="umap-plotting">
<h2>UMAP Plotting<a class="headerlink" href="#umap-plotting" title="Link to this heading">#</a></h2>
<p>We define convenience function to plot the Uniform Manifold Approximation and Projection (UMAP) of the ASAP PMDB data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_umap</span><span class="p">(</span>
    <span class="n">xx</span><span class="p">,</span>
    <span class="n">yy</span><span class="p">,</span>
    <span class="n">cc</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">val</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span>
    <span class="n">cmap</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">labels</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
    <span class="n">sizes</span><span class="o">=</span><span class="kc">None</span>
<span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">sizes</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">sizes</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">scatt</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">val</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="n">sizes</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">cc</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">scatt</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">cc</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="n">sizes</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">labels</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="kn">from</span><span class="w"> </span><span class="nn">matplotlib.patches</span><span class="w"> </span><span class="kn">import</span> <span class="n">Rectangle</span>
        <span class="n">unique_labels</span> <span class="o">=</span> <span class="n">labels</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
        <span class="n">unique_colors</span> <span class="o">=</span> <span class="n">cc</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>

        <span class="k">if</span> <span class="n">term_order_lookup</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">term_order</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">argsort</span><span class="p">(</span><span class="n">term_order_lookup</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">unique_labels</span><span class="p">,</span> <span class="s1">&#39;order&#39;</span><span class="p">])</span>
            <span class="n">unique_labels</span> <span class="o">=</span> <span class="n">unique_labels</span><span class="p">[</span><span class="n">term_order</span><span class="p">]</span>
            <span class="n">unique_colors</span> <span class="o">=</span> <span class="n">unique_colors</span><span class="p">[</span><span class="n">term_order</span><span class="p">]</span>
            
        <span class="n">rects</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">color</span> <span class="ow">in</span> <span class="n">unique_colors</span><span class="p">:</span>
            <span class="n">rects</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">),</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fc</span><span class="o">=</span><span class="n">color</span><span class="p">))</span>

        <span class="n">legend</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">unique_labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="c1"># ax.add_artist(legend)</span>

    <span class="k">if</span> <span class="n">colorbar</span><span class="p">:</span>
        <span class="n">fig</span><span class="o">.</span><span class="n">colorbar</span><span class="p">(</span><span class="n">scatt</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
    
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<section id="data-acquisition-metadata">
<h3>Data Acquisition Metadata<a class="headerlink" href="#data-acquisition-metadata" title="Link to this heading">#</a></h3>
<p>Below we plot various information associated with the collection of the data. This includes which of the five teams collected the data, the 10X chemistry version used, and the region of interest the tissue was extracted from. Note, throughout the presentation of these data, that some values are missing for collections of cells. This is due to the value not being present for these cells and samples in the data. These are marked as <code class="docutils literal notranslate"><span class="pre">Unknown</span></code>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;source_dataset_label_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;source_dataset_label&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Source Dataset&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/5f11a1f5ee6445c398f4017e4732f8fce6ac68139c06a3bd8389b96484ea6853.png" src="../_images/5f11a1f5ee6445c398f4017e4732f8fce6ac68139c06a3bd8389b96484ea6853.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;technique_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;technique&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;10X Chemistry/Technique&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/f943275078aa1547b8be84c6dcc45a396734485a1f2aeea0b117755a7fc5c674.png" src="../_images/f943275078aa1547b8be84c6dcc45a396734485a1f2aeea0b117755a7fc5c674.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;region_of_interest_label_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;region_of_interest_label&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Region of Interest&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/08b732e9d3e70bf6fc499deb7a04eaeb7284271a2bf89d258d5797f0c21a89e3.png" src="../_images/08b732e9d3e70bf6fc499deb7a04eaeb7284271a2bf89d258d5797f0c21a89e3.png" />
</div>
</div>
</section>
<section id="donor-metadata">
<h3>Donor Metadata<a class="headerlink" href="#donor-metadata" title="Link to this heading">#</a></h3>
<p>Below we plot various information on the donor where these samples/cells originated.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;donor_race_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;donor_race&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;donor_race&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Donor Race&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/90016a9ebd63abd1c8f1ec865ffa26fa1a2d47e6c1dbcac1090c3ae4bfceb9af.png" src="../_images/90016a9ebd63abd1c8f1ec865ffa26fa1a2d47e6c1dbcac1090c3ae4bfceb9af.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;donor_sex_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;donor_sex&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;donor_sex&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Donor Sex&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/2bde98c236d3a3311ba247bc415e9a7ac4b0e5dd5edffbe1e287eeb16bbe4164.png" src="../_images/2bde98c236d3a3311ba247bc415e9a7ac4b0e5dd5edffbe1e287eeb16bbe4164.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;age_at_death_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;age_at_death&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;age_at_death&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Donor Age At Death&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/ee800add4b3b252c4f04ceebc7468e2336aa1b33109a8c1ec40d1f6bb204468c.png" src="../_images/ee800add4b3b252c4f04ceebc7468e2336aa1b33109a8c1ec40d1f6bb204468c.png" />
</div>
</div>
</section>
<section id="clinical-pathology">
<h3>Clinical  Pathology<a class="headerlink" href="#clinical-pathology" title="Link to this heading">#</a></h3>
<p>Below are various diagnoses and pathologies identified in the donor including healthy controls.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;primary_diagnosis_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;primary_diagnosis&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;primary_diagnosis&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Primary Diagnosis&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/d2ba2b024b12008d3d7d9d4611f961cd5a2518e366792a9430f76f9526bf5c84.png" src="../_images/d2ba2b024b12008d3d7d9d4611f961cd5a2518e366792a9430f76f9526bf5c84.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;cognitive_status_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;cognitive_status&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;cognitive_status&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Cognitive Status&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/d7f03a2f72dfce089c1d3bf975d2af876746c40483c9bcdc4bbf3e9bbff56851.png" src="../_images/d7f03a2f72dfce089c1d3bf975d2af876746c40483c9bcdc4bbf3e9bbff56851.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;lewy_body_disease_pathology_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;lewy_body_disease_pathology&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;lewy_body_disease_pathology&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Lewy Body Disease Pathology&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/13610cf75896b3166e47cc5034ffb6e7263f9ec18aab3c24e5810ddaab7b2411.png" src="../_images/13610cf75896b3166e47cc5034ffb6e7263f9ec18aab3c24e5810ddaab7b2411.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;apoe4_status_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;apoe4_status&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;apoe4_status&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Apoe4 Status&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/346f4a0f4b4a7db8bfe4c6a015a9757b91c99523973a5f36ed85ef35866f59dd.png" src="../_images/346f4a0f4b4a7db8bfe4c6a015a9757b91c99523973a5f36ed85ef35866f59dd.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;braak_stage_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;braak_stage&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;braak_stage&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Braak Stage&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/63c497e418f9da6ab03144374ee35b138c362e2b4ddedfeafad9fb9d5d82db6c.png" src="../_images/63c497e418f9da6ab03144374ee35b138c362e2b4ddedfeafad9fb9d5d82db6c.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;cerad_score_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;cerad_score&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;cerad_score&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Cerad Score&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/e3939feb6faa74d91f14e0d0716c1419d6e65e206c1b2af7f5f1b43b62483543.png" src="../_images/e3939feb6faa74d91f14e0d0716c1419d6e65e206c1b2af7f5f1b43b62483543.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;thal_phase_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;thal_phase&#39;</span><span class="p">],</span>
    <span class="n">term_order_lookup</span><span class="o">=</span><span class="n">value_sets</span><span class="p">[</span><span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;thal_phase&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Thal Phase&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/3b3368ff297b95f1a243aea4a1ed9eb5cf29e054092e677cbdb4e8e994ab4b3c.png" src="../_images/3b3368ff297b95f1a243aea4a1ed9eb5cf29e054092e677cbdb4e8e994ab4b3c.png" />
</div>
</div>
</section>
<section id="leiden-scvi-clusters">
<h3>Leiden scVI Clusters<a class="headerlink" href="#leiden-scvi-clusters" title="Link to this heading">#</a></h3>
<p>As part of the ASAP harmonization, scVI was used to generate a latent space representation, 2D UMAP coordinates and set of 30 clusters using their implementation of the Leiden algorithm.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_metadata</span><span class="p">[</span><span class="s1">&#39;cluster_label_color&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Cluster Label&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/64952c086dd3af6c4aa109c073395eee1a13c681404520662458c6dfd8362793.png" src="../_images/64952c086dd3af6c4aa109c073395eee1a13c681404520662458c6dfd8362793.png" />
</div>
</div>
<p>In the next <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/asap_pmdbs_siletti_taxonomy.html">notebook</a>, we demonstrate the usage of Allen Institute’s <a class="reference external" href="https://portal.brain-map.org/atlases-and-data/bkp/mapmycells">MapMyCells</a> tool to map cells from the ASAP PMDBS dataset to the whole human brain taxonomy (Siletti et al.) to provide cell type annotation and insights to the Leiden scVI clusters above.</p>
<p>We’ll also explore gene expression in the UMAP in a <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/asap_pmdbs_gene_expression.html">later notebook</a>.</p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="../descriptions/ASAP-PMDBS_notebooks.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks</p>
      </div>
    </a>
    <a class="right-next"
       href="asap_pmdbs_siletti_taxonomy.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#asap-data">ASAP Data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#umap-plotting">UMAP Plotting</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#data-acquisition-metadata">Data Acquisition Metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#donor-metadata">Donor Metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#clinical-pathology">Clinical  Pathology</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#leiden-scvi-clusters">Leiden scVI Clusters</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>