
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Consensus Whole Mouse Brain #1: Clustering and Annotations &#8212; Allen Brain Cell Atlas - Data Access</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=9c3e77be" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/simple_style.css?v=3aaed5dc" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=1ae7504c"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/consensus_mouse_clustering_analysis_and_annotation';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Consensus Whole Mouse Brain #2: 10X gene expression" href="consensus_mouse_10X_snRNASeq.html" />
    <link rel="prev" title="Consensus Mouse notebooks" href="../descriptions/Consensus-WMB-notebooks.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/allen_logo.jpeg" class="logo__image only-light" alt="Allen Brain Cell Atlas - Data Access - Home"/>
    <script>document.write(`<img src="../_static/allen_logo.jpeg" class="logo__image only-dark" alt="Allen Brain Cell Atlas - Data Access - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Allen Brain Cell Atlas - Data Access
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">INTRODUCTION</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="general_accessing_10x_snRNASeq_tutorial.html">Accessing 10x RNA-seq gene expression data</a></li>
<li class="toctree-l1"><a class="reference internal" href="abc_atlas_selection_example.html">Using cells selected in the ABC Atlas with Gene Expression data.</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">DATA</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WMB_dataset.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv2.html">Whole mouse brain 10Xv2 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv3.html">Whole mouse brain 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10XMulti.html">Whole mouse brain 10XMulti single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10X.html">Clustering analysis of whole adult mouse brain 10X single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-taxonomy.html">Whole adult mouse brain taxonomy of cell types</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-imputed.html">Imputed MERFISH spatial transcriptomics of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-neighborhoods.html">Cluster groups and neighborhood specific UMAP embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Allen-CCF-2020.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-CCF.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_dataset.html">Aging Mouse dataset (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_10Xv3.html">Zeng Aging Mouse 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_taxonomy.html">Zeng Aging Mouse clustering and mapping to the WMB taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zhuang_dataset.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-1.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-2.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-3.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-3)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-4.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-4)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-dataset.html">Consensus Whole Mouse Brain dataset (Macosko, Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-10X.html">Consensus Whole Mouse Brain single cell/nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-taxonomy.html">Consensus Whole Mouse Brain integrated taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_dataset.html">Human whole-brain transcriptomic cell type atlas (Kimberly Siletti)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10Xv3.html">Human brain 10Xv3 single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10X-clustering.html">Clustering analysis of whole human brain 10X single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-taxonomy.html">Human brain taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_dataset.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-10X.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-MapMyCells.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain (WHB) and Seattle Alzheimer’s Disease Brain Cell Atlas (SEA-AD) Taxonomies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_dataset.html">Human-Mammalian Brain - Basal Ganglia: Data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-BG-taxonomy-CCN20250428.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq data: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG-Aligned.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Aligned gene expression across species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Individual species gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Spatial-BG.html">Human-Mammalian Brain - Basal Ganglia: spatial transcriptomics cells for each species</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">NOTEBOOKS</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cluster_annotation_tutorial.html">10x RNA-seq clustering analysis and annotation (CCN20230722)</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle2.html">10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_1.html">10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2a.html">10x RNA-seq gene expression data (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2b.html">10x RNA-seq gene expression data (part 2b)</a></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle3.html">MERFISH whole brain spatial transcriptomics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_1.html">MERFISH whole brain spatial transcriptomics (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2a.html">MERFISH whole brain spatial transcriptomics (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2b.html">MERFISH whole brain spatial transcriptomics (part 2b)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_imputed_genes_example.html">MERFISH whole brain spatial transcriptomics cells with imputed genes</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_groups_and_embeddings_tutorial.html">Cluster neighborhoods and embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_neighborhood_gallery.html">Cell type neighborhood gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="ccf_and_parcellation_annotation_tutorial.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="merfish_ccf_registration_tutorial.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-notebooks.html">Consensus Mouse notebooks</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Consensus Whole Mouse Brain #1: Clustering and Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_10X_snRNASeq.html">Consensus Whole Mouse Brain #2: 10X gene expression</a></li>

</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_notebooks.html">Aging Mouse Notebooks (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_clustering_analysis_and_annotation.html">Mouse Aging clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_10x_snRNASeq_tutorial.html">Mouse Aging 10X RNA-seq gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle4.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="zhuang_merfish_tutorial.html">MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_notebooks.html">Whole Human Brain 10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="WHB_cluster_annotation_tutorial.html">Whole Human Brain 10x RNA-seq clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_1.html">Whole Human Brain (WHB) 10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_2.html">Whole Human Brain 10x RNA-seq gene expression data (part 2)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_notebooks.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_data_and_metadata.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_siletti_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_seaad_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_gene_expression.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Gene Expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_notebooks.html">Human-Mammalian Brain - Basal Ganglia: Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_clustering_analysis_and_annotation.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_10X_snRNASeq_tutorial.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_slabs_and_taxonomy.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_genes.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">ADDITIONAL RESOURCES</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../docs/README.html">Resource Pages</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../docs/abbreviation_terms/README.html">Abbreviation lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/cluster_annotations/README.html">Cluster annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/gene_lists/README.html">Gene lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/parcellation_annotations/README.html">Parcellation annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/supplemental_metadata_descriptions/README.html">Supplemental Metadata Descriptions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/neuroglancer_hmba/README.html">HMBA Basal Ganglia: Neuroglancer Views</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access/issues/new?title=Issue%20on%20page%20%2Fnotebooks/consensus_mouse_clustering_analysis_and_annotation.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/consensus_mouse_clustering_analysis_and_annotation.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Consensus Whole Mouse Brain #1: Clustering and Annotations</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#library-and-donor-metadata">Library and Donor metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#adding-color-and-feature-order">Adding color and feature order</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#umap-spatial-embedding">UMAP spatial embedding</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxonomy-information">Taxonomy Information</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-taxonomy">Plotting the taxonomy</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#neighborhood-umaps">Neighborhood UMAPS</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#aggregating-cluster-and-cells-counts">Aggregating cluster and cells counts.</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualizing-the-taxonomy">Visualizing the  taxonomy</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="consensus-whole-mouse-brain-1-clustering-and-annotations">
<h1>Consensus Whole Mouse Brain #1: Clustering and Annotations<a class="headerlink" href="#consensus-whole-mouse-brain-1-clustering-and-annotations" title="Link to this heading">#</a></h1>
<p>The Consensus, Whole Mouse Brain (WMB) integrated taxonomy is built upon two publicly released transcriptionally defined WMB taxonomies: one derived from single-cell RNA sequencing (scRNA-seq) data and the other from single-nucleus RNA sequencing (snRNA-seq) data. The <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/WMB-taxonomy.html">Allen Institute for Brain Science (AIBS) taxonomy</a>, based on over 4 million scRNA-seq profiles, defines 5,322 clusters organized hierarchically into 34 classes, 338 subclasses, and 1,201 supertypes. The set of data products for this release can be found <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/WMB_dataset.html">here</a>. In parallel, the Broad Institute taxonomy, constructed from 4.4 million snRNA-seq profiles, defines 16 classes, 223 metaclusters, and 5,030 clusters. Integrating these two large-scale taxonomies into a unified framework represents a natural and impactful next step, enabling a consensus view of cell types across the entire mouse brain and benefiting the broader neuroscience community.</p>
<p>To generate this consensus taxonomy, we applied the AIBS Quality Control (QC) and post-integration QC pipelines, retaining 7,651,713 cells and nuclei. Integration of scRNA-seq and snRNA-seq data was performed using scVI, with subsampling by original clusters to mitigate sampling imbalance across cell types and brain regions, followed by projection of all remaining cells into a shared latent space. The same iterative clustering strategy used in the AIBS taxonomy was applied in a hierarchical manner—globally, at nine neighborhood levels, and across eight finer group levels. The resulting comprehensive taxonomy comprises a hierarchically arranged set of cell types with 9 neighborhoods, 43 classes, 414 subclasses, 1,386 supertypes, and 6,721 clusters. A detailed cell type annotation table accompanies the taxonomy, including hierarchical membership, anatomical localization, and neurotransmitter identity. All associated metadata is publicly available as an AWS Public Dataset hosted on Amazon S3 and through the Allen Brain Cell Atlass Access (abc_atlas_access) package.</p>
<p>Below we explore this taxonomy and combine it with cell and other metadata, visualizing the data in a 2d spatial projection and summary statistics.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%</span><span class="k">matplotlib</span> inline

<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">typing</span><span class="w"> </span><span class="kn">import</span> <span class="n">Tuple</span><span class="p">,</span> <span class="n">Optional</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">abc_atlas_access.abc_atlas_cache.abc_project_cache</span><span class="w"> </span><span class="kn">import</span> <span class="n">AbcProjectCache</span>
</pre></div>
</div>
</div>
</div>
<p>We will interact with the data using the <strong>AbcProjectCache</strong>. This cache object downloads data requested by the user, tracks which files have already been downloaded to your local system, and serves the path to the requested data on disk. For metadata, the cache can also directly serve up a Pandas DataFrame. See the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html"><code class="docutils literal notranslate"><span class="pre">getting_started</span></code></a> notebook for more details on using the cache including installing the package.</p>
<p><strong>Change the download_base variable to where you would like to download the data in your system.</strong></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">download_base</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">&#39;../../data/abc_atlas&#39;</span><span class="p">)</span>
<span class="n">abc_cache</span> <span class="o">=</span> <span class="n">AbcProjectCache</span><span class="o">.</span><span class="n">from_cache_dir</span><span class="p">(</span>
    <span class="n">download_base</span>
<span class="p">)</span>

<span class="n">abc_cache</span><span class="o">.</span><span class="n">current_manifest</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;releases/20251031/manifest.json&#39;
</pre></div>
</div>
</div>
</div>
<section id="data-overview">
<h2>Data overview<a class="headerlink" href="#data-overview" title="Link to this heading">#</a></h2>
<p>Below we list the files available with the Consensus Mouse data release. There are three main packages: Consensus-WMB-AIBS-10X, Consensus-WMB-Macosko-10X, and Consensus-WMB-integrated-taxonomy.</p>
<p>For the Consensus-WMB-AIBS-10X there are no new expression matrices, all gene expression is the same as the previous WMB release and we reuse those files, see the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/consensus_mouse_10X_snRNASeq.html">gene expression tutorial</a> for more details on using the gene expression data. This release provides updated cell metadata. While there is significant overlap between the previous WMB AIBS release and this Consensus WMB release, there are a fraction of cells in this metadata that are not in the previous release and vice versa.</p>
<p>Below we list the available metadata files for WMB-AIBS in the consensus release. The expression matrices are part of the previously released <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/WMB-10Xv2.html">WMB-10Xv2</a> and <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/WMB-10Xv3.html">WMB-10Xv3</a> packages.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Consensus-WMB-AIBS-10X: metadata </span><span class="se">\n\t</span><span class="s2">&quot;</span><span class="p">,</span>
      <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_metadata_files</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-AIBS-10X&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Consensus-WMB-AIBS-10X: metadata 
	 [&#39;cell_metadata&#39;, &#39;donor&#39;, &#39;example_gene_expression&#39;, &#39;library&#39;, &#39;value_sets&#39;]
</pre></div>
</div>
</div>
</div>
<p>Next we list the available expression matrix and metadata files in the WMB-Macosko directories. The expression matrices are divided similarlly to those in the WMB-AIBS release, that is by coarse brain region. The metadata are similarly structured to those in the WMB-AIBS portion of the data listed above.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Consensus-WMB-Macosko-10X: gene expression data (h5ad)</span><span class="se">\n\t</span><span class="s2">&quot;</span><span class="p">,</span>
      <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_expression_matrix_files</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-Macosko-10X&#39;</span><span class="p">))</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Consensus-WMB-Macosko-10X: gene expression data (h5ad)</span><span class="se">\n\t</span><span class="s2">&quot;</span><span class="p">,</span>
      <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_metadata_files</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-Macosko-10X&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Consensus-WMB-Macosko-10X: gene expression data (h5ad)
	 [&#39;Macosko-10X-CB/log2&#39;, &#39;Macosko-10X-CB/raw&#39;, &#39;Macosko-10X-HPF/log2&#39;, &#39;Macosko-10X-HPF/raw&#39;, &#39;Macosko-10X-HY/log2&#39;, &#39;Macosko-10X-HY/raw&#39;, &#39;Macosko-10X-Isocortex/log2&#39;, &#39;Macosko-10X-Isocortex/raw&#39;, &#39;Macosko-10X-MB/log2&#39;, &#39;Macosko-10X-MB/raw&#39;, &#39;Macosko-10X-MY-Pons-BS/log2&#39;, &#39;Macosko-10X-MY-Pons-BS/raw&#39;, &#39;Macosko-10X-OLF/log2&#39;, &#39;Macosko-10X-OLF/raw&#39;, &#39;Macosko-10X-PAL/log2&#39;, &#39;Macosko-10X-PAL/raw&#39;, &#39;Macosko-10X-STR/log2&#39;, &#39;Macosko-10X-STR/raw&#39;, &#39;Macosko-10X-TH/log2&#39;, &#39;Macosko-10X-TH/raw&#39;]
Consensus-WMB-Macosko-10X: gene expression data (h5ad)
	 [&#39;cell_metadata&#39;, &#39;donor&#39;, &#39;example_gene_expression&#39;, &#39;gene&#39;, &#39;library&#39;, &#39;value_sets&#39;]
</pre></div>
</div>
</div>
</div>
<p>Finally, we list the metadata files that make up the consensus taxonomy. This data includes 2d projections for all cells as well as the cells in each neighborhood. We’ll show how to join the taxonomy with the cell metadata files listed above later in this notebook.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Consensus-WMB-integrated-taxonomy: metadata (csv)</span><span class="se">\n\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_metadata_files</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Consensus-WMB-integrated-taxonomy: metadata (csv)
	 [&#39;HY-EA-Glut-GABA_cell_2d_embedding_coordinates&#39;, &#39;MB-GABA_cell_2d_embedding_coordinates&#39;, &#39;MB-Glut-Dopa-Sero_cell_2d_embedding_coordinates&#39;, &#39;NN-IMN_cell_2d_embedding_coordinates&#39;, &#39;P-MY-CB-GABA_cell_2d_embedding_coordinates&#39;, &#39;P-MY-CB-Glut_cell_2d_embedding_coordinates&#39;, &#39;Pallium-Glut_cell_2d_embedding_coordinates&#39;, &#39;Subpallium-GABA_cell_2d_embedding_coordinates&#39;, &#39;TH-EPI-Glut_cell_2d_embedding_coordinates&#39;, &#39;cell_2d_embedding_coordinates&#39;, &#39;cell_to_cluster_membership&#39;, &#39;cluster&#39;, &#39;cluster_annotation_term&#39;, &#39;cluster_annotation_term_set&#39;, &#39;cluster_to_cluster_annotation_membership&#39;]
</pre></div>
</div>
</div>
</div>
<section id="cell-metadata">
<h3>Cell metadata<a class="headerlink" href="#cell-metadata" title="Link to this heading">#</a></h3>
<p>Below we load the metadata for each cell in both the WMB-Macosko and WMB-AIBS portion of the data. These contain base information of the cell’s ID, its barcode and barcoded_cell_sample (if available), the library the cell comes from and two columns defining which h5ad file a given cell’s gene expression is located.</p>
<p>Below we load both the WMB-Macosko and WMB-AIBS cell data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macosko_cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-Macosko-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;cell_label&#39;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Number of cells = &quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">macosko_cell_metadata</span><span class="p">))</span>
<span class="n">macosko_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_metadata.csv: 100%|██████████| 506M/506M [01:29&lt;00:00, 5.64MMB/s]    
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of cells =  3736281
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_ACTTCCGGTGGTCCCA</th>
      <td>ACTTCCGGTGGTCCCA</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
    </tr>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_AACCTTTGTTAAGTCC</th>
      <td>AACCTTTGTTAAGTCC</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
    </tr>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_TTTCCTCTCACCGGTG</th>
      <td>TTTCCTCTCACCGGTG</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
    </tr>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_CACACAACATCATCCC</th>
      <td>CACACAACATCATCCC</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
    </tr>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_ACTATCTCAGTTAAAG</th>
      <td>ACTATCTCAGTTAAAG</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">aibs_cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-AIBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;cell_label&#39;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Number of cells = &quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">aibs_cell_metadata</span><span class="p">))</span>
<span class="n">aibs_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_metadata.csv: 100%|██████████| 375M/375M [01:05&lt;00:00, 5.76MMB/s]    
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of cells =  3915432
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>GCGAGAAGTTAAGGGC-410_B05</th>
      <td>GCGAGAAGTTAAGGGC</td>
      <td>410_B05</td>
      <td>L8TX_201030_01_C12</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
    </tr>
    <tr>
      <th>AATGGCTCAGCTCCTT-411_B06</th>
      <td>AATGGCTCAGCTCCTT</td>
      <td>411_B06</td>
      <td>L8TX_201029_01_E10</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
    </tr>
    <tr>
      <th>AACACACGTTGCTTGA-410_B05</th>
      <td>AACACACGTTGCTTGA</td>
      <td>410_B05</td>
      <td>L8TX_201030_01_C12</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
    </tr>
    <tr>
      <th>CACAGATAGAGGCGGA-410_A05</th>
      <td>CACAGATAGAGGCGGA</td>
      <td>410_A05</td>
      <td>L8TX_201029_01_A10</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
    </tr>
    <tr>
      <th>GATCGTATCGAATCCA-411_B06</th>
      <td>GATCGTATCGAATCCA</td>
      <td>411_B06</td>
      <td>L8TX_201029_01_E10</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We can use pandas <code class="docutils literal notranslate"><span class="pre">groupby</span></code> function to see how many unique items are associated for each field and list them out if the number of unique items is small.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">print_column_info</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
    
    <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">grouped</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="n">c</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">c</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
        <span class="n">members</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">grouped</span><span class="p">)</span> <span class="o">&lt;</span> <span class="mi">30</span><span class="p">:</span>
            <span class="n">members</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">grouped</span><span class="o">.</span><span class="n">index</span><span class="p">))</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Number of unique </span><span class="si">%s</span><span class="s2"> = </span><span class="si">%d</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">grouped</span><span class="p">),</span> <span class="n">members</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">print_column_info</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">aibs_cell_metadata</span><span class="p">,</span> <span class="n">macosko_cell_metadata</span><span class="p">]))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of unique cell_barcode = 3580247 
Number of unique barcoded_cell_sample_label = 781 
Number of unique library_label = 1434 
Number of unique dataset_label = 3 [&#39;Consensus-WMB-Macosko-10X&#39;, &#39;WMB-10Xv2&#39;, &#39;WMB-10Xv3&#39;]
Number of unique feature_matrix_label = 33 
</pre></div>
</div>
</div>
</div>
</section>
<section id="library-and-donor-metadata">
<h3>Library and Donor metadata<a class="headerlink" href="#library-and-donor-metadata" title="Link to this heading">#</a></h3>
<p>Next we load metadata associated with each dataset’s libraries and donors.</p>
<p>Below we load the library metadata. The primary information we’ll be using from these tables are the anatomical region the sample originated from and the id of the donor the library came from.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macosko_library</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-Macosko-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;library&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;library_label&#39;</span><span class="p">)</span>
<span class="n">macosko_library</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>library.csv: 100%|██████████| 27.7k/27.7k [00:00&lt;00:00, 297kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>region_of_interest_acronym</th>
      <th>anatomical_division_label</th>
      <th>donor_label</th>
    </tr>
    <tr>
      <th>library_label</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>pBICCNsMMrACAiF019d210630A1</th>
      <td>ACA</td>
      <td>Isocortex</td>
      <td>F019</td>
    </tr>
    <tr>
      <th>pBICCNsMMrACAiF019d210630A2</th>
      <td>ACA</td>
      <td>Isocortex</td>
      <td>F019</td>
    </tr>
    <tr>
      <th>pBICCNsMMrACAiF019d210630A3</th>
      <td>ACA</td>
      <td>Isocortex</td>
      <td>F019</td>
    </tr>
    <tr>
      <th>pBICCNsMMrACAiF019d210630A4</th>
      <td>ACA</td>
      <td>Isocortex</td>
      <td>F019</td>
    </tr>
    <tr>
      <th>pBICCNsMMrACAiF019d210630A5</th>
      <td>ACA</td>
      <td>Isocortex</td>
      <td>F019</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">aibs_library</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-AIBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;library&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;library_label&#39;</span><span class="p">)</span>
<span class="n">aibs_library</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>library.csv: 100%|██████████| 59.0k/59.0k [00:00&lt;00:00, 484kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>library_method</th>
      <th>alignment_id</th>
      <th>region_of_interest_acronym</th>
      <th>anatomical_division_label</th>
      <th>donor_label</th>
    </tr>
    <tr>
      <th>library_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>L8TX_171026_01_A04</th>
      <td>10Xv2</td>
      <td>1186619234</td>
      <td>MOp</td>
      <td>Isocortex</td>
      <td>Snap25-IRES2-Cre;Ai14-352353</td>
    </tr>
    <tr>
      <th>L8TX_171026_01_A05</th>
      <td>10Xv2</td>
      <td>1178482616</td>
      <td>MOp</td>
      <td>Isocortex</td>
      <td>Snap25-IRES2-Cre;Ai14-352356</td>
    </tr>
    <tr>
      <th>L8TX_171026_01_B04</th>
      <td>10Xv2</td>
      <td>1178483191</td>
      <td>MOp</td>
      <td>Isocortex</td>
      <td>Snap25-IRES2-Cre;Ai14-352353</td>
    </tr>
    <tr>
      <th>L8TX_171026_01_B05</th>
      <td>10Xv2</td>
      <td>1178482921</td>
      <td>MOp</td>
      <td>Isocortex</td>
      <td>Snap25-IRES2-Cre;Ai14-352356</td>
    </tr>
    <tr>
      <th>L8TX_171026_01_C05</th>
      <td>10Xv2</td>
      <td>1186619314</td>
      <td>MOp</td>
      <td>Isocortex</td>
      <td>Snap25-IRES2-Cre;Ai14-352357</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Finally, we’ll load the donor metadata, this provides a formalized column of where the sample originated (Macosko or AIBS) and the sex of the donor. For the WMB-AIBS data, we have additional information on the age of the donor at death.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macosko_donor</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-Macosko-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;donor&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
<span class="n">macosko_donor</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>donor.csv: 100%|██████████| 1.29k/1.29k [00:00&lt;00:00, 16.0kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>donor_sex</th>
      <th>origin_dataset</th>
    </tr>
    <tr>
      <th>donor_label</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1F1</th>
      <td>Female</td>
      <td>WMB-Macosko</td>
    </tr>
    <tr>
      <th>1F3</th>
      <td>Female</td>
      <td>WMB-Macosko</td>
    </tr>
    <tr>
      <th>1F5</th>
      <td>Female</td>
      <td>WMB-Macosko</td>
    </tr>
    <tr>
      <th>1F6</th>
      <td>Female</td>
      <td>WMB-Macosko</td>
    </tr>
    <tr>
      <th>1M1</th>
      <td>Male</td>
      <td>WMB-Macosko</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">aibs_donor</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-AIBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;donor&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
<span class="n">aibs_donor</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>donor.csv: 100%|██████████| 16.6k/16.6k [00:00&lt;00:00, 204kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>donor_sex</th>
      <th>donor_age</th>
      <th>origin_dataset</th>
    </tr>
    <tr>
      <th>donor_label</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Gad2-IRES-Cre;Ai14-529270</th>
      <td>Male</td>
      <td>61 days</td>
      <td>WMB-AIBS</td>
    </tr>
    <tr>
      <th>Gad2-IRES-Cre;Ai14-529271</th>
      <td>Male</td>
      <td>66 days</td>
      <td>WMB-AIBS</td>
    </tr>
    <tr>
      <th>Gad2-IRES-Cre;Ai14-529272</th>
      <td>Female</td>
      <td>62 days</td>
      <td>WMB-AIBS</td>
    </tr>
    <tr>
      <th>Gad2-IRES-Cre;Ai14-529273</th>
      <td>Female</td>
      <td>66 days</td>
      <td>WMB-AIBS</td>
    </tr>
    <tr>
      <th>Gad2-IRES-Cre;Ai14-558836</th>
      <td>Male</td>
      <td>56 days</td>
      <td>WMB-AIBS</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Now that we’ve loaded the additional metadata, we’ll join them into the cell metadata tables on the library and donor label.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macosko_cell_extended</span> <span class="o">=</span> <span class="n">macosko_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">macosko_library</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;library_label&#39;</span><span class="p">)</span>
<span class="n">macosko_cell_extended</span> <span class="o">=</span> <span class="n">macosko_cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">macosko_donor</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
<span class="n">aibs_cell_extended</span> <span class="o">=</span> <span class="n">aibs_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">aibs_library</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;library_label&#39;</span><span class="p">)</span>
<span class="n">aibs_cell_extended</span> <span class="o">=</span> <span class="n">aibs_cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">aibs_donor</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Below we compute statistics using pandas groupby funcationality to count the number of cells in either of the two datasets, AIBS and Macosko. The we show the breakdown of cell count by anatomical region.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">aibs_cell_extended</span><span class="p">,</span> <span class="n">macosko_cell_extended</span><span class="p">])</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;origin_dataset&#39;</span><span class="p">)[[</span><span class="s1">&#39;cell_barcode&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
    </tr>
    <tr>
      <th>origin_dataset</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>WMB-AIBS</th>
      <td>3915432</td>
    </tr>
    <tr>
      <th>WMB-Macosko</th>
      <td>3736281</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">aibs_cell_extended</span><span class="p">,</span> <span class="n">macosko_cell_extended</span><span class="p">])</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;anatomical_division_label&#39;</span><span class="p">)[[</span><span class="s1">&#39;cell_barcode&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
    </tr>
    <tr>
      <th>anatomical_division_label</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CB</th>
      <td>645731</td>
    </tr>
    <tr>
      <th>CTXsp</th>
      <td>119786</td>
    </tr>
    <tr>
      <th>HPF</th>
      <td>704031</td>
    </tr>
    <tr>
      <th>HY</th>
      <td>347613</td>
    </tr>
    <tr>
      <th>Isocortex</th>
      <td>2125008</td>
    </tr>
    <tr>
      <th>MB</th>
      <td>974404</td>
    </tr>
    <tr>
      <th>MY-Pons-BS</th>
      <td>981716</td>
    </tr>
    <tr>
      <th>OLF</th>
      <td>433855</td>
    </tr>
    <tr>
      <th>PAL</th>
      <td>311432</td>
    </tr>
    <tr>
      <th>STR</th>
      <td>421355</td>
    </tr>
    <tr>
      <th>TH</th>
      <td>586782</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="adding-color-and-feature-order">
<h3>Adding color and feature order<a class="headerlink" href="#adding-color-and-feature-order" title="Link to this heading">#</a></h3>
<p>In anticipation of plotting these cells and their metadata, we’ll load a lookup table that maps values in each of our loaded tables to color, ontological ordering, and (if available) external identifiers that represent these data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">value_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-AIBS-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;value_sets&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">value_sets</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>value_sets.csv: 100%|██████████| 2.47k/2.47k [00:00&lt;00:00, 39.8kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>field</th>
      <th>table</th>
      <th>description</th>
      <th>order</th>
      <th>external_identifier</th>
      <th>parent_label</th>
      <th>color_hex_triplet</th>
      <th>comment</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Female</th>
      <td>donor_sex</td>
      <td>donor</td>
      <td>Female</td>
      <td>1</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#565353</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Male</th>
      <td>donor_sex</td>
      <td>donor</td>
      <td>Male</td>
      <td>2</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#ADC4C3</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>WMB-AIBS</th>
      <td>origin_dataset</td>
      <td>donor</td>
      <td>Allen Institute for Brain Science, Whole Mouse...</td>
      <td>1</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#1f77b4</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>WMB-Macosko</th>
      <td>origin_dataset</td>
      <td>donor</td>
      <td>Broad Institute, Macosko lab Whole Mouse Brain...</td>
      <td>2</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#ff7f0e</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>Isocortex</th>
      <td>anatomical_division_label</td>
      <td>library</td>
      <td>Isocortex</td>
      <td>1</td>
      <td>MBA:315</td>
      <td>NaN</td>
      <td>#70FF71</td>
      <td>division, ID and parent from CCF-2020</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>The convenience function below, extracts the color and order information and adds it to our DataFrames.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">extract_value_set</span><span class="p">(</span><span class="n">cell_metadata_df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">input_value_set</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">input_value_set_label</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Add color and order columns to the cell metadata dataframe based on the input</span>
<span class="sd">    value set.</span>

<span class="sd">    Columns are added as {input_value_set_label}_color and {input_value_set_label}_order.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    cell_metadata_df : pd.DataFrame</span>
<span class="sd">        DataFrame containing cell metadata.</span>
<span class="sd">    input_value_set : pd.DataFrame</span>
<span class="sd">        DataFrame containing the value set information.</span>
<span class="sd">    input_value_set_label : str</span>
<span class="sd">        The the column name to extract color and order information for. will be added to the cell metadata.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">cell_metadata_df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">input_value_set_label</span><span class="si">}</span><span class="s1">_color&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_value_set</span><span class="p">[</span>
        <span class="n">input_value_set</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">input_value_set_label</span>
    <span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cell_metadata_df</span><span class="p">[</span><span class="n">input_value_set_label</span><span class="p">]][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">cell_metadata_df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">input_value_set_label</span><span class="si">}</span><span class="s1">_order&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_value_set</span><span class="p">[</span>
        <span class="n">input_value_set</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">input_value_set_label</span>
    <span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cell_metadata_df</span><span class="p">[</span><span class="n">input_value_set_label</span><span class="p">]][</span><span class="s1">&#39;order&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Add region of interest color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">macosko_cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;origin_dataset&#39;</span><span class="p">)</span>
<span class="c1"># Add species common name color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">macosko_cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;donor_sex&#39;</span><span class="p">)</span>
<span class="c1"># Add species scientific name color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">macosko_cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;anatomical_division_label&#39;</span><span class="p">)</span>
<span class="n">macosko_cell_extended</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>region_of_interest_acronym</th>
      <th>anatomical_division_label</th>
      <th>donor_label</th>
      <th>donor_sex</th>
      <th>origin_dataset</th>
      <th>origin_dataset_color</th>
      <th>origin_dataset_order</th>
      <th>donor_sex_color</th>
      <th>donor_sex_order</th>
      <th>anatomical_division_label_color</th>
      <th>anatomical_division_label_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_ACTTCCGGTGGTCCCA</th>
      <td>ACTTCCGGTGGTCCCA</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
      <td>BS</td>
      <td>MY-Pons-BS</td>
      <td>M007</td>
      <td>Male</td>
      <td>WMB-Macosko</td>
      <td>#ff7f0e</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>2</td>
      <td>#FF9BCD</td>
      <td>12</td>
    </tr>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_AACCTTTGTTAAGTCC</th>
      <td>AACCTTTGTTAAGTCC</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
      <td>BS</td>
      <td>MY-Pons-BS</td>
      <td>M007</td>
      <td>Male</td>
      <td>WMB-Macosko</td>
      <td>#ff7f0e</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>2</td>
      <td>#FF9BCD</td>
      <td>12</td>
    </tr>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_TTTCCTCTCACCGGTG</th>
      <td>TTTCCTCTCACCGGTG</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
      <td>BS</td>
      <td>MY-Pons-BS</td>
      <td>M007</td>
      <td>Male</td>
      <td>WMB-Macosko</td>
      <td>#ff7f0e</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>2</td>
      <td>#FF9BCD</td>
      <td>12</td>
    </tr>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_CACACAACATCATCCC</th>
      <td>CACACAACATCATCCC</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
      <td>BS</td>
      <td>MY-Pons-BS</td>
      <td>M007</td>
      <td>Male</td>
      <td>WMB-Macosko</td>
      <td>#ff7f0e</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>2</td>
      <td>#FF9BCD</td>
      <td>12</td>
    </tr>
    <tr>
      <th>pBICCNsMMrBSL1aiM007d190529_ACTATCTCAGTTAAAG</th>
      <td>ACTATCTCAGTTAAAG</td>
      <td>NaN</td>
      <td>pBICCNsMMrBSL1aiM007d190529</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-MY-Pons-BS</td>
      <td>BS</td>
      <td>MY-Pons-BS</td>
      <td>M007</td>
      <td>Male</td>
      <td>WMB-Macosko</td>
      <td>#ff7f0e</td>
      <td>2</td>
      <td>#ADC4C3</td>
      <td>2</td>
      <td>#FF9BCD</td>
      <td>12</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Add region of interest color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">aibs_cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;origin_dataset&#39;</span><span class="p">)</span>
<span class="c1"># Add species common name color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">aibs_cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;donor_sex&#39;</span><span class="p">)</span>
<span class="c1"># Add species scientific name color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">aibs_cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;anatomical_division_label&#39;</span><span class="p">)</span>
<span class="n">aibs_cell_extended</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>library_method</th>
      <th>alignment_id</th>
      <th>region_of_interest_acronym</th>
      <th>anatomical_division_label</th>
      <th>donor_label</th>
      <th>donor_sex</th>
      <th>donor_age</th>
      <th>origin_dataset</th>
      <th>origin_dataset_color</th>
      <th>origin_dataset_order</th>
      <th>donor_sex_color</th>
      <th>donor_sex_order</th>
      <th>anatomical_division_label_color</th>
      <th>anatomical_division_label_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>GCGAGAAGTTAAGGGC-410_B05</th>
      <td>GCGAGAAGTTAAGGGC</td>
      <td>410_B05</td>
      <td>L8TX_201030_01_C12</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
      <td>10Xv3</td>
      <td>1177903638</td>
      <td>RHP</td>
      <td>HPF</td>
      <td>Snap25-IRES2-Cre;Ai14-550850</td>
      <td>Female</td>
      <td>53 days</td>
      <td>WMB-AIBS</td>
      <td>#1f77b4</td>
      <td>1</td>
      <td>#565353</td>
      <td>1</td>
      <td>#7ED04B</td>
      <td>6</td>
    </tr>
    <tr>
      <th>AATGGCTCAGCTCCTT-411_B06</th>
      <td>AATGGCTCAGCTCCTT</td>
      <td>411_B06</td>
      <td>L8TX_201029_01_E10</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
      <td>10Xv3</td>
      <td>1177903464</td>
      <td>RHP</td>
      <td>HPF</td>
      <td>Snap25-IRES2-Cre;Ai14-550851</td>
      <td>Female</td>
      <td>53 days</td>
      <td>WMB-AIBS</td>
      <td>#1f77b4</td>
      <td>1</td>
      <td>#565353</td>
      <td>1</td>
      <td>#7ED04B</td>
      <td>6</td>
    </tr>
    <tr>
      <th>AACACACGTTGCTTGA-410_B05</th>
      <td>AACACACGTTGCTTGA</td>
      <td>410_B05</td>
      <td>L8TX_201030_01_C12</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
      <td>10Xv3</td>
      <td>1177903638</td>
      <td>RHP</td>
      <td>HPF</td>
      <td>Snap25-IRES2-Cre;Ai14-550850</td>
      <td>Female</td>
      <td>53 days</td>
      <td>WMB-AIBS</td>
      <td>#1f77b4</td>
      <td>1</td>
      <td>#565353</td>
      <td>1</td>
      <td>#7ED04B</td>
      <td>6</td>
    </tr>
    <tr>
      <th>CACAGATAGAGGCGGA-410_A05</th>
      <td>CACAGATAGAGGCGGA</td>
      <td>410_A05</td>
      <td>L8TX_201029_01_A10</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
      <td>10Xv3</td>
      <td>1177903446</td>
      <td>RHP</td>
      <td>HPF</td>
      <td>Snap25-IRES2-Cre;Ai14-550850</td>
      <td>Female</td>
      <td>53 days</td>
      <td>WMB-AIBS</td>
      <td>#1f77b4</td>
      <td>1</td>
      <td>#565353</td>
      <td>1</td>
      <td>#7ED04B</td>
      <td>6</td>
    </tr>
    <tr>
      <th>GATCGTATCGAATCCA-411_B06</th>
      <td>GATCGTATCGAATCCA</td>
      <td>411_B06</td>
      <td>L8TX_201029_01_E10</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-HPF</td>
      <td>10Xv3</td>
      <td>1177903464</td>
      <td>RHP</td>
      <td>HPF</td>
      <td>Snap25-IRES2-Cre;Ai14-550851</td>
      <td>Female</td>
      <td>53 days</td>
      <td>WMB-AIBS</td>
      <td>#1f77b4</td>
      <td>1</td>
      <td>#565353</td>
      <td>1</td>
      <td>#7ED04B</td>
      <td>6</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="umap-spatial-embedding">
<h3>UMAP spatial embedding<a class="headerlink" href="#umap-spatial-embedding" title="Link to this heading">#</a></h3>
<p>Now that we have metadata with color information, we can utilize the available Uniform Mapping Approximation and Projection (UMAP) available for this consensus mouse release to visualize the information.</p>
<p>Below we load the projection and join it into a combined set of WMB-AIBS and WMB-Macosko, cell metadata.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_2d_embedding_coordinates</span> <span class="o">=</span> <span class="n">value_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_2d_embedding_coordinates&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">cell_2d_embedding_coordinates</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_2d_embedding_coordinates.csv: 100%|██████████| 490M/490M [01:27&lt;00:00, 5.57MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>x</th>
      <th>y</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>GCGAGAAGTTAAGGGC-410_B05</th>
      <td>16.037980</td>
      <td>3.101109</td>
    </tr>
    <tr>
      <th>AATGGCTCAGCTCCTT-411_B06</th>
      <td>15.951514</td>
      <td>3.144049</td>
    </tr>
    <tr>
      <th>AACACACGTTGCTTGA-410_B05</th>
      <td>15.900673</td>
      <td>3.124507</td>
    </tr>
    <tr>
      <th>CACAGATAGAGGCGGA-410_A05</th>
      <td>16.062553</td>
      <td>3.185574</td>
    </tr>
    <tr>
      <th>GATCGTATCGAATCCA-411_B06</th>
      <td>15.971468</td>
      <td>3.124298</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">aibs_cell_extended</span><span class="p">,</span> <span class="n">macosko_cell_extended</span><span class="p">])</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cell_2d_embedding_coordinates</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">sample</span><span class="p">(</span><span class="n">frac</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="k">del</span> <span class="n">cell_2d_embedding_coordinates</span>
</pre></div>
</div>
</div>
</div>
<p>We define a small helper function <em>plot_umap</em> to visualize the cells on the UMAP. In the examples below we will plot associated cell information colorized by dissection donor species, sex, and region of interest.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_umap</span><span class="p">(</span>
    <span class="n">xx</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">,</span>
    <span class="n">yy</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">,</span>
    <span class="n">cc</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">val</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
    <span class="n">cmap</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Colormap</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">labels</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">term_orders</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
    <span class="n">sizes</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">fig</span><span class="p">:</span> <span class="n">plt</span><span class="o">.</span><span class="n">Figure</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">ax</span><span class="p">:</span> <span class="n">plt</span><span class="o">.</span><span class="n">Axes</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
 <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Tuple</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Figure</span><span class="p">,</span> <span class="n">plt</span><span class="o">.</span><span class="n">Axes</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Plot a scatter plot of the UMAP coordinates.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    xx : array-like</span>
<span class="sd">        x-coordinates of the points to plot.</span>
<span class="sd">    yy : array-like</span>
<span class="sd">        y-coordinates of the points to plot.</span>
<span class="sd">    cc : array-like, optional</span>
<span class="sd">        colors of the points to plot. If None, the points will be colored by the values in `val`.</span>
<span class="sd">    val : array-like, optional</span>
<span class="sd">        values of the points to plot. If None, the points will be colored by the values in `cc`.</span>
<span class="sd">    fig_width : float, optional</span>
<span class="sd">        width of the figure in inches. Default is 8.</span>
<span class="sd">    fig_height : float, optional</span>
<span class="sd">        height of the figure in inches. Default is 8.</span>
<span class="sd">    cmap : str, optional</span>
<span class="sd">        colormap to use for coloring the points. If None, the points will be colored by the values in `cc`.</span>
<span class="sd">    labels : array-like, optional</span>
<span class="sd">        labels for the points to plot. If None, no labels will be added to the plot.</span>
<span class="sd">    term_orders : array-like, optional</span>
<span class="sd">        order of the labels for the legend. If None, the labels will be ordered by their appearance in `labels`.</span>
<span class="sd">    colorbar : bool, optional</span>
<span class="sd">        whether to add a colorbar to the plot. Default is False.</span>
<span class="sd">    sizes : array-like, optional</span>
<span class="sd">        sizes of the points to plot. If None, all points will have the same size.</span>
<span class="sd">    fig : matplotlib.figure.Figure, optional</span>
<span class="sd">        figure to plot on. If None, a new figure will be created with 1 subplot.</span>
<span class="sd">    ax : matplotlib.axes.Axes, optional</span>
<span class="sd">        axes to plot on. If None, a new figure will be created with 1 subplot.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">sizes</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">sizes</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="k">if</span> <span class="n">ax</span> <span class="ow">is</span> <span class="kc">None</span> <span class="ow">or</span> <span class="n">fig</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
        <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">scatt</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">val</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="n">sizes</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">cc</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">scatt</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">cc</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="n">sizes</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">labels</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="kn">from</span><span class="w"> </span><span class="nn">matplotlib.patches</span><span class="w"> </span><span class="kn">import</span> <span class="n">Rectangle</span>
        <span class="n">unique_label_colors</span> <span class="o">=</span> <span class="p">(</span><span class="n">labels</span> <span class="o">+</span> <span class="s1">&#39;,&#39;</span> <span class="o">+</span> <span class="n">cc</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
        <span class="n">unique_labels</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">label_color</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">label_color</span> <span class="ow">in</span> <span class="n">unique_label_colors</span><span class="p">])</span>
        <span class="n">unique_colors</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">label_color</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">label_color</span> <span class="ow">in</span> <span class="n">unique_label_colors</span><span class="p">])</span>

        <span class="k">if</span> <span class="n">term_orders</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">unique_order</span> <span class="o">=</span> <span class="n">term_orders</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
            <span class="n">term_order</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">argsort</span><span class="p">(</span><span class="n">unique_order</span><span class="p">)</span>
            <span class="n">unique_labels</span> <span class="o">=</span> <span class="n">unique_labels</span><span class="p">[</span><span class="n">term_order</span><span class="p">]</span>
            <span class="n">unique_colors</span> <span class="o">=</span> <span class="n">unique_colors</span><span class="p">[</span><span class="n">term_order</span><span class="p">]</span>
            
        <span class="n">rects</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">color</span> <span class="ow">in</span> <span class="n">unique_colors</span><span class="p">:</span>
            <span class="n">rects</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">),</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fc</span><span class="o">=</span><span class="n">color</span><span class="p">))</span>

        <span class="n">legend</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">unique_labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="c1"># ax.add_artist(legend)</span>
    
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>

    <span class="k">if</span> <span class="n">colorbar</span><span class="p">:</span>
        <span class="n">fig</span><span class="o">.</span><span class="n">colorbar</span><span class="p">(</span><span class="n">scatt</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
    
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<p>Below we visualize the location of cells colored by origin_dataset, donor_sex, and the anatomical region the cells belong to.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Select every 10th cell for plotting</span>
<span class="n">sub_selected</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[::</span><span class="mi">10</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;origin_dataset_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;origin_dataset&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;origin_dataset_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Origin Dataset&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/089d193f9dbdb1f0a778b627a8c6160879b7ebad2f6d9d05a36d8994ab3f528c.png" src="../_images/089d193f9dbdb1f0a778b627a8c6160879b7ebad2f6d9d05a36d8994ab3f528c.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;donor_sex_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;donor_sex&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;donor_sex_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Donor Sex&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/1cece868be42921a1329debebf4483f7c3a825389bbad24394417db2ebc27d2f.png" src="../_images/1cece868be42921a1329debebf4483f7c3a825389bbad24394417db2ebc27d2f.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;anatomical_division_label_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;anatomical_division_label&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;anatomical_division_label_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Anatomical Region&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/08bd6558ba1003f7dda670987c28728eeb3747a7dec0156656f9bb496f9afeb4.png" src="../_images/08bd6558ba1003f7dda670987c28728eeb3747a7dec0156656f9bb496f9afeb4.png" />
</div>
</div>
</section>
</section>
<section id="taxonomy-information">
<h2>Taxonomy Information<a class="headerlink" href="#taxonomy-information" title="Link to this heading">#</a></h2>
<p>The final set of metadata we load into our extended cell metadata file maps the cells into their assigned cluster in the taxonomy. We additionally load metadata for the clusters and compute useful information, such as the number of cells in each taxon at each level of the taxonomy.</p>
<p>First, we load information associated with each Cluster in the taxonomy. This includes a useful alias value for each cluster as well as the number of cells in each cluster.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span><span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">cluster</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster.csv: 100%|██████████| 203k/203k [00:00&lt;00:00, 865kMB/s]  
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_alias</th>
      <th>number_of_cells</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20251031_CLUS_0001</th>
      <td>2721</td>
      <td>16355</td>
    </tr>
    <tr>
      <th>CS20251031_CLUS_0002</th>
      <td>16574</td>
      <td>1519</td>
    </tr>
    <tr>
      <th>CS20251031_CLUS_0003</th>
      <td>1736</td>
      <td>307</td>
    </tr>
    <tr>
      <th>CS20251031_CLUS_0004</th>
      <td>1737</td>
      <td>825</td>
    </tr>
    <tr>
      <th>CS20251031_CLUS_0005</th>
      <td>1743</td>
      <td>4671</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Next, we load the table that describes the levels in the taxonomy from Neighborhood at the highest to Cluster at the lowest level.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_annotation_term_set</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term_set&#39;</span><span class="p">,</span>
    <span class="n">skip_hash_check</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">cluster_annotation_term_set</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_annotation_term_set.csv: 100%|██████████| 388/388 [00:00&lt;00:00, 5.77kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>description</th>
      <th>order</th>
      <th>parent_term_set_label</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CCN20251031_NEUR</th>
      <td>neurotransmitter</td>
      <td>neurotransmitter</td>
      <td>0</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CCN20251031_LEVEL_0</th>
      <td>neighborhood</td>
      <td>neighborhood</td>
      <td>1</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CCN20251031_LEVEL_1</th>
      <td>class</td>
      <td>class</td>
      <td>2</td>
      <td>CCN20251031_LEVEL_0</td>
    </tr>
    <tr>
      <th>CCN20251031_LEVEL_2</th>
      <td>subclass</td>
      <td>subclass</td>
      <td>3</td>
      <td>CCN20251031_LEVEL_1</td>
    </tr>
    <tr>
      <th>CCN20251031_LEVEL_3</th>
      <td>supertype</td>
      <td>supertype</td>
      <td>4</td>
      <td>CCN20251031_LEVEL_2</td>
    </tr>
    <tr>
      <th>CCN20251031_LEVEL_4</th>
      <td>cluster</td>
      <td>cluster</td>
      <td>5</td>
      <td>CCN20251031_LEVEL_3</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We load the annotation information defining all the taxons at all levels in the taxonomy. This also includes the term order and color associated with the taxon which we will use to plot later.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_annotation_term</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span><span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster_annotation_term&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">cluster_annotation_term</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_annotation_term.csv: 100%|██████████| 1.38M/1.38M [00:00&lt;00:00, 4.37MMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>cluster_annotation_term_set_label</th>
      <th>cluster_annotation_term_set_name</th>
      <th>color_hex_triplet</th>
      <th>term_order</th>
      <th>term_set_order</th>
      <th>parent_term_label</th>
      <th>parent_term_name</th>
      <th>parent_term_set_label</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20251031_NEUR_0004</th>
      <td>Chol</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#73E785</td>
      <td>4</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0012</th>
      <td>Chol-Dopa</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#B8EC68</td>
      <td>12</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0008</th>
      <td>Dopa</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#fcf04b</td>
      <td>8</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0002</th>
      <td>GABA</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#FF3358</td>
      <td>2</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0006</th>
      <td>GABA-Chol</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#000080</td>
      <td>6</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Finally, we load the cluster to cluster annotation membership table. Each row in this table is a mapping between a cluster and taxon in the taxonomy, including the clusters themselves. We’ll use this table in a <code class="docutils literal notranslate"><span class="pre">groupby</span></code>s to allow us to count up the number of clusters at each taxonomy level and sum the number of cells in each taxon in the taxonomy a all levels.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_to_cluster_annotation_membership</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_to_cluster_annotation_membership&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">cluster_to_cluster_annotation_membership</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_to_cluster_annotation_membership.csv: 100%|██████████| 3.07M/3.07M [00:00&lt;00:00, 5.10MMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_annotation_term_set_name</th>
      <th>cluster_annotation_term_name</th>
      <th>cluster_alias</th>
      <th>cluster_annotation_term_set_label</th>
    </tr>
    <tr>
      <th>cluster_annotation_term_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20251031_NEUR_0001</th>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>2721</td>
      <td>CCN20251031_NEUR</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0001</th>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>16574</td>
      <td>CCN20251031_NEUR</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0001</th>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>1736</td>
      <td>CCN20251031_NEUR</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0001</th>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>1737</td>
      <td>CCN20251031_NEUR</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0001</th>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>1743</td>
      <td>CCN20251031_NEUR</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">membership_with_cluster_info</span> <span class="o">=</span> <span class="n">cluster_to_cluster_annotation_membership</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">cluster</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)[[</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">]],</span>
    <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span>
<span class="p">)</span>
<span class="n">membership_with_cluster_info</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_annotation_term</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_anno_term&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">membership_groupby</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span>
<span class="p">)</span>
<span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_annotation_term_label</th>
      <th>cluster_annotation_term_set_name</th>
      <th>cluster_annotation_term_name</th>
      <th>cluster_alias</th>
      <th>cluster_annotation_term_set_label</th>
      <th>number_of_cells</th>
      <th>name</th>
      <th>cluster_annotation_term_set_label_anno_term</th>
      <th>cluster_annotation_term_set_name_anno_term</th>
      <th>color_hex_triplet</th>
      <th>term_order</th>
      <th>term_set_order</th>
      <th>parent_term_label</th>
      <th>parent_term_name</th>
      <th>parent_term_set_label</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>CS20251031_NEUR_0001</td>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>2721</td>
      <td>CCN20251031_NEUR</td>
      <td>16355</td>
      <td>Glut</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#2B93DF</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>CS20251031_NEUR_0001</td>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>16574</td>
      <td>CCN20251031_NEUR</td>
      <td>1519</td>
      <td>Glut</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#2B93DF</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>CS20251031_NEUR_0001</td>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>1736</td>
      <td>CCN20251031_NEUR</td>
      <td>307</td>
      <td>Glut</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#2B93DF</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>CS20251031_NEUR_0001</td>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>1737</td>
      <td>CCN20251031_NEUR</td>
      <td>825</td>
      <td>Glut</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#2B93DF</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>CS20251031_NEUR_0001</td>
      <td>neurotransmitter</td>
      <td>Glut</td>
      <td>1743</td>
      <td>CCN20251031_NEUR</td>
      <td>4671</td>
      <td>Glut</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#2B93DF</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>From the membership table, we create three tables via a <code class="docutils literal notranslate"><span class="pre">groupby</span></code>. First the alias of each cluster and its parents.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># term_sets = abc_cache.get_metadata_dataframe(directory=&#39;WHB-taxonomy&#39;, file_name=&#39;cluster_annotation_term_set&#39;).set_index(&#39;label&#39;)</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">cluster_details</span><span class="p">[</span><span class="n">cluster_annotation_term_set</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span> <span class="c1"># order columns</span>
<span class="n">cluster_details</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;neighborhood&#39;</span><span class="p">,</span> <span class="s1">&#39;class&#39;</span><span class="p">,</span> <span class="s1">&#39;subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;supertype&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_details</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>cluster_annotation_term_set_name</th>
      <th>neurotransmitter</th>
      <th>neighborhood</th>
      <th>class</th>
      <th>subclass</th>
      <th>supertype</th>
      <th>cluster</th>
    </tr>
    <tr>
      <th>cluster_alias</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>6562</th>
      <td>GABA</td>
      <td>HY-EA-Glut-GABA</td>
      <td>011 CNU-HYa GABA</td>
      <td>090 MEA-BST_Lhx6:Nfib_Gaba</td>
      <td>0376 MEA-BST_Lhx6:Nfib_Gaba 1</td>
      <td>1543 MEA-BST_Lhx6:Nfib_Gaba 1</td>
    </tr>
    <tr>
      <th>6567</th>
      <td>GABA</td>
      <td>HY-EA-Glut-GABA</td>
      <td>011 CNU-HYa GABA</td>
      <td>090 MEA-BST_Lhx6:Nfib_Gaba</td>
      <td>0376 MEA-BST_Lhx6:Nfib_Gaba 1</td>
      <td>1544 MEA-BST_Lhx6:Nfib_Gaba 1</td>
    </tr>
    <tr>
      <th>6576</th>
      <td>GABA</td>
      <td>HY-EA-Glut-GABA</td>
      <td>011 CNU-HYa GABA</td>
      <td>090 MEA-BST_Lhx6:Nfib_Gaba</td>
      <td>0376 MEA-BST_Lhx6:Nfib_Gaba 1</td>
      <td>1545 MEA-BST_Lhx6:Nfib_Gaba 1</td>
    </tr>
    <tr>
      <th>6578</th>
      <td>GABA</td>
      <td>HY-EA-Glut-GABA</td>
      <td>011 CNU-HYa GABA</td>
      <td>090 MEA-BST_Lhx6:Nfib_Gaba</td>
      <td>0376 MEA-BST_Lhx6:Nfib_Gaba 1</td>
      <td>1546 MEA-BST_Lhx6:Nfib_Gaba 1</td>
    </tr>
    <tr>
      <th>6579</th>
      <td>GABA</td>
      <td>HY-EA-Glut-GABA</td>
      <td>011 CNU-HYa GABA</td>
      <td>090 MEA-BST_Lhx6:Nfib_Gaba</td>
      <td>0376 MEA-BST_Lhx6:Nfib_Gaba 1</td>
      <td>1547 MEA-BST_Lhx6:Nfib_Gaba 1</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Next the plotting order of each of the clusters and their parents.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_order</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;term_order&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_order</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;neighborhood&#39;</span><span class="p">,</span> <span class="s1">&#39;class&#39;</span><span class="p">,</span> <span class="s1">&#39;subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;supertype&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_order</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>cluster_annotation_term_set_name</th>
      <th>class</th>
      <th>cluster</th>
      <th>neighborhood</th>
      <th>neurotransmitter</th>
      <th>subclass</th>
      <th>supertype</th>
    </tr>
    <tr>
      <th>cluster_alias</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2721</th>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>16574</th>
      <td>1</td>
      <td>2</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1736</th>
      <td>1</td>
      <td>3</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <th>1737</th>
      <td>1</td>
      <td>4</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <th>1743</th>
      <td>1</td>
      <td>5</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>2</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Finally, the colors we will use to plot for each of the unique taxons at all levels.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">cluster_colors</span><span class="p">[</span><span class="n">cluster_annotation_term_set</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span>
<span class="n">cluster_colors</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;neighborhood&#39;</span><span class="p">,</span> <span class="s1">&#39;class&#39;</span><span class="p">,</span> <span class="s1">&#39;subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;supertype&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster&#39;</span><span class="p">],</span>
    <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">cluster_colors</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>cluster_annotation_term_set_name</th>
      <th>neurotransmitter</th>
      <th>neighborhood</th>
      <th>class</th>
      <th>subclass</th>
      <th>supertype</th>
      <th>cluster</th>
    </tr>
    <tr>
      <th>cluster_alias</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>8736</th>
      <td>#2B93DF</td>
      <td>#006200</td>
      <td>#006200</td>
      <td>#002099</td>
      <td>#663D41</td>
      <td>#8C4599</td>
    </tr>
    <tr>
      <th>8734</th>
      <td>#2B93DF</td>
      <td>#006200</td>
      <td>#006200</td>
      <td>#002099</td>
      <td>#99E0FF</td>
      <td>#410F66</td>
    </tr>
    <tr>
      <th>8730</th>
      <td>#2B93DF</td>
      <td>#006200</td>
      <td>#006200</td>
      <td>#002099</td>
      <td>#99E0FF</td>
      <td>#811799</td>
    </tr>
    <tr>
      <th>8732</th>
      <td>#2B93DF</td>
      <td>#006200</td>
      <td>#006200</td>
      <td>#002099</td>
      <td>#99E0FF</td>
      <td>#FFE4E2</td>
    </tr>
    <tr>
      <th>8738</th>
      <td>#2B93DF</td>
      <td>#006200</td>
      <td>#006200</td>
      <td>#002099</td>
      <td>#99E0FF</td>
      <td>#FFF49B</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Next, we bring it all together by loading the mapping of cells to cluster and join into our final metadata table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_to_cluster_membership</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span><span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">,</span> <span class="s1">&#39;cell_to_cluster_membership&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">cell_to_cluster_membership</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_to_cluster_membership.csv: 100%|██████████| 310M/310M [00:50&lt;00:00, 6.12MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_alias</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CAGGTGCAGGCTAGCA-040_C01</th>
      <td>5491</td>
    </tr>
    <tr>
      <th>CGGACGTGTGTGAATA-063_B01</th>
      <td>6268</td>
    </tr>
    <tr>
      <th>GATCCCTTCGTGCACG-107_B01</th>
      <td>6659</td>
    </tr>
    <tr>
      <th>TCACGAACAACTGCGC-026_A01</th>
      <td>6672</td>
    </tr>
    <tr>
      <th>ACACCAAGTCAAACTC-026_B01</th>
      <td>7067</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We merge this table with information from our clusters.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cell_to_cluster_membership</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_cell_to_cluster_membership&#39;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_color&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_order</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_order&#39;</span><span class="p">)</span>

<span class="k">del</span> <span class="n">cell_to_cluster_membership</span>

<span class="n">cell_extended</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>library_method</th>
      <th>alignment_id</th>
      <th>region_of_interest_acronym</th>
      <th>anatomical_division_label</th>
      <th>donor_label</th>
      <th>...</th>
      <th>class_color</th>
      <th>subclass_color</th>
      <th>supertype_color</th>
      <th>cluster_color</th>
      <th>class_order</th>
      <th>cluster_order</th>
      <th>neighborhood_order</th>
      <th>neurotransmitter_order</th>
      <th>subclass_order</th>
      <th>supertype_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>GATTCAGAGCGCCTTG-040_C01</th>
      <td>GATTCAGAGCGCCTTG</td>
      <td>040_C01</td>
      <td>L8TX_180815_01_E08</td>
      <td>WMB-10Xv2</td>
      <td>WMB-10Xv2-TH</td>
      <td>10Xv2</td>
      <td>1178465136</td>
      <td>TH</td>
      <td>TH</td>
      <td>Snap25-IRES2-Cre;Ai14-404124</td>
      <td>...</td>
      <td>#0D47A1</td>
      <td>#076600</td>
      <td>#42CC3D</td>
      <td>#FFE15A</td>
      <td>19</td>
      <td>3024</td>
      <td>5</td>
      <td>1</td>
      <td>180</td>
      <td>681</td>
    </tr>
    <tr>
      <th>pBICCNsMMrAUDiF022d210715A5_AAGGTAAGTCTGTCCT</th>
      <td>AAGGTAAGTCTGTCCT</td>
      <td>NaN</td>
      <td>pBICCNsMMrAUDiF022d210715A5</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-Isocortex</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>AUD</td>
      <td>Isocortex</td>
      <td>F022</td>
      <td>...</td>
      <td>#FA0087</td>
      <td>#3DCCB7</td>
      <td>#3B9900</td>
      <td>#990047</td>
      <td>1</td>
      <td>50</td>
      <td>1</td>
      <td>1</td>
      <td>3</td>
      <td>12</td>
    </tr>
    <tr>
      <th>pBICCNsMMrOLFiF015d201201B1_CCCTTAGCATGTCTAG</th>
      <td>CCCTTAGCATGTCTAG</td>
      <td>NaN</td>
      <td>pBICCNsMMrOLFiF015d201201B1</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-OLF</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>OLF</td>
      <td>OLF</td>
      <td>F015</td>
      <td>...</td>
      <td>#996B2E</td>
      <td>#FF99CF</td>
      <td>#C5FF73</td>
      <td>#995C88</td>
      <td>40</td>
      <td>6619</td>
      <td>9</td>
      <td>25</td>
      <td>402</td>
      <td>1348</td>
    </tr>
    <tr>
      <th>pBICCNsMMrTHT6iM013d210329A1_ACGTAGTGTACTCCCT</th>
      <td>ACGTAGTGTACTCCCT</td>
      <td>NaN</td>
      <td>pBICCNsMMrTHT6iM013d210329A1</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-TH</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TH</td>
      <td>TH</td>
      <td>M013</td>
      <td>...</td>
      <td>#0D47A1</td>
      <td>#076600</td>
      <td>#5C9900</td>
      <td>#410000</td>
      <td>19</td>
      <td>3033</td>
      <td>5</td>
      <td>1</td>
      <td>180</td>
      <td>683</td>
    </tr>
    <tr>
      <th>TCTGTCGAGTGCTACT-442_C02</th>
      <td>TCTGTCGAGTGCTACT</td>
      <td>442_C02</td>
      <td>L8TX_201119_01_F08</td>
      <td>WMB-10Xv3</td>
      <td>WMB-10Xv3-MB</td>
      <td>10Xv3</td>
      <td>1177904103</td>
      <td>MB</td>
      <td>MB</td>
      <td>Snap25-IRES2-Cre;Ai14-553673</td>
      <td>...</td>
      <td>#594a26</td>
      <td>#AD5CCC</td>
      <td>#0F6635</td>
      <td>#FFEE00</td>
      <td>37</td>
      <td>6509</td>
      <td>9</td>
      <td>25</td>
      <td>392</td>
      <td>1305</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 40 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">print_column_info</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of unique cell_barcode = 3578991 
Number of unique barcoded_cell_sample_label = 781 
Number of unique library_label = 1434 
Number of unique dataset_label = 3 [&#39;Consensus-WMB-Macosko-10X&#39;, &#39;WMB-10Xv2&#39;, &#39;WMB-10Xv3&#39;]
Number of unique feature_matrix_label = 33 
Number of unique library_method = 2 [&#39;10Xv2&#39;, &#39;10Xv3&#39;]
Number of unique alignment_id = 781 
Number of unique region_of_interest_acronym = 42 
Number of unique anatomical_division_label = 11 [&#39;CB&#39;, &#39;CTXsp&#39;, &#39;HPF&#39;, &#39;HY&#39;, &#39;Isocortex&#39;, &#39;MB&#39;, &#39;MY-Pons-BS&#39;, &#39;OLF&#39;, &#39;PAL&#39;, &#39;STR&#39;, &#39;TH&#39;]
Number of unique donor_label = 373 
Number of unique donor_sex = 2 [&#39;Female&#39;, &#39;Male&#39;]
Number of unique donor_age = 22 [&#39;51 days&#39;, &#39;52 days&#39;, &#39;53 days&#39;, &#39;54 days&#39;, &#39;55 days&#39;, &#39;56 days&#39;, &#39;57 days&#39;, &#39;58 days&#39;, &#39;59 days&#39;, &#39;60 days&#39;, &#39;61 days&#39;, &#39;62 days&#39;, &#39;63 days&#39;, &#39;64 days&#39;, &#39;65 days&#39;, &#39;66 days&#39;, &#39;67 days&#39;, &#39;68 days&#39;, &#39;69 days&#39;, &#39;70 days&#39;, &#39;71 days&#39;, &#39;unknown&#39;]
Number of unique origin_dataset = 2 [&#39;WMB-AIBS&#39;, &#39;WMB-Macosko&#39;]
Number of unique origin_dataset_color = 2 [&#39;#1f77b4&#39;, &#39;#ff7f0e&#39;]
Number of unique origin_dataset_order = 2 [1, 2]
Number of unique donor_sex_color = 2 [&#39;#565353&#39;, &#39;#ADC4C3&#39;]
Number of unique donor_sex_order = 2 [1, 2]
Number of unique anatomical_division_label_color = 11 [&#39;#70FF71&#39;, &#39;#7ED04B&#39;, &#39;#8599CC&#39;, &#39;#8ADA87&#39;, &#39;#98D6F9&#39;, &#39;#9AD2BD&#39;, &#39;#E64438&#39;, &#39;#F0F080&#39;, &#39;#FF64FF&#39;, &#39;#FF7080&#39;, &#39;#FF9BCD&#39;]
Number of unique anatomical_division_label_order = 11 [1, 4, 6, 7, 8, 9, 12, 13, 14, 15, 18]
Number of unique x = 7505915 
Number of unique y = 7517910 
Number of unique cluster_alias = 6721 
Number of unique neurotransmitter = 22 [&#39;Chol&#39;, &#39;Chol-Dopa&#39;, &#39;Dopa&#39;, &#39;GABA&#39;, &#39;GABA-Chol&#39;, &#39;GABA-Dopa&#39;, &#39;GABA-Glyc&#39;, &#39;GABA-Hist&#39;, &#39;GABA-Sero&#39;, &#39;Glut&#39;, &#39;Glut-Chol&#39;, &#39;Glut-Dopa&#39;, &#39;Glut-GABA&#39;, &#39;Glut-GABA-Chol&#39;, &#39;Glut-GABA-Dopa&#39;, &#39;Glut-GABA-Glyc&#39;, &#39;Glut-GABA-Sero&#39;, &#39;Glut-Glyc&#39;, &#39;Glut-Sero&#39;, &#39;Glyc&#39;, &#39;NN&#39;, &#39;Sero&#39;]
Number of unique neighborhood = 9 [&#39;HY-EA-Glut-GABA&#39;, &#39;MB-GABA&#39;, &#39;MB-Glut-Dopa-Sero&#39;, &#39;NN-IMN&#39;, &#39;P-MY-CB-GABA&#39;, &#39;P-MY-CB-Glut&#39;, &#39;Pallium-Glut&#39;, &#39;Subpallium-GABA&#39;, &#39;TH-EPI-Glut&#39;]
Number of unique class = 43 
Number of unique subclass = 414 
Number of unique supertype = 1386 
Number of unique cluster = 6721 
Number of unique neurotransmitter_color = 22 [&#39;#000080&#39;, &#39;#0000FF&#39;, &#39;#008080&#39;, &#39;#0a9964&#39;, &#39;#1B9E77&#39;, &#39;#2B93DF&#39;, &#39;#377EB8&#39;, &#39;#533691&#39;, &#39;#66636C&#39;, &#39;#73E785&#39;, &#39;#800000&#39;, &#39;#800080&#39;, &#39;#9189FF&#39;, &#39;#A65628&#39;, &#39;#B8EC68&#39;, &#39;#F781BF&#39;, &#39;#FF3358&#39;, &#39;#FF4500&#39;, &#39;#FF7080&#39;, &#39;#fad502&#39;, &#39;#fcf04b&#39;, &#39;#ff7621&#39;]
Number of unique neighborhood_color = 9 [&#39;#006200&#39;, &#39;#0096C7&#39;, &#39;#03045E&#39;, &#39;#1283FF&#39;, &#39;#9EF01A&#39;, &#39;#B199FF&#39;, &#39;#F0A0FF&#39;, &#39;#FA0087&#39;, &#39;#FF6600&#39;]
Number of unique class_color = 43 
Number of unique subclass_color = 414 
Number of unique supertype_color = 1385 
Number of unique cluster_color = 6126 
Number of unique class_order = 43 
Number of unique cluster_order = 6721 
Number of unique neighborhood_order = 9 [1, 2, 3, 4, 5, 6, 7, 8, 9]
Number of unique neurotransmitter_order = 22 [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 17, 18, 19, 20, 21, 22, 23, 24, 25]
Number of unique subclass_order = 414 
Number of unique supertype_order = 1386 
</pre></div>
</div>
</div>
</div>
<section id="plotting-the-taxonomy">
<h3>Plotting the taxonomy<a class="headerlink" href="#plotting-the-taxonomy" title="Link to this heading">#</a></h3>
<p>Now that we have our cells with associated taxonomy information, we’ll plot them into the UMAP we showed previously.</p>
<p>Below we plot the taxonomy mapping of the cells for each level in the taxonomy. We use the labels and their orders to plot them in a legend. We omit the legends the lower levels as the legends become too busy.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sub_selected</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[::</span><span class="mi">10</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;neighborhood_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;neighborhood&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;neighborhood_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;neighborhood&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/dcced7b2478833d806e7b403e629cdcdb3283aa7f54374d83fb6c8c566a2a4fa.png" src="../_images/dcced7b2478833d806e7b403e629cdcdb3283aa7f54374d83fb6c8c566a2a4fa.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;class_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;class&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;class_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">20</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;class&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/33e7f6b712dff6c5cbe6818a8ca613e91ddd13d35208a4f4c0a5fa01d4585b97.png" src="../_images/33e7f6b712dff6c5cbe6818a8ca613e91ddd13d35208a4f4c0a5fa01d4585b97.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;subclass_color&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">18</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">18</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;subclass&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/1aca800b8dbeafdc038e377ee163a428967356ca8b7802e46f821758ea32ce07.png" src="../_images/1aca800b8dbeafdc038e377ee163a428967356ca8b7802e46f821758ea32ce07.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;supertype_color&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">18</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">18</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;supertype&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/647ed25ba4f783e8dc955bdba39d5d482e9c9b3548d277d3a10de4ffec0632dc.png" src="../_images/647ed25ba4f783e8dc955bdba39d5d482e9c9b3548d277d3a10de4ffec0632dc.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;cluster_color&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">18</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">18</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;cluster&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/ba8b0e7bf9e38bf78a0994980b259397c7710307ee12dd68bb65d416135c855a.png" src="../_images/ba8b0e7bf9e38bf78a0994980b259397c7710307ee12dd68bb65d416135c855a.png" />
</div>
</div>
<p>Additionally, we plot by neurotransmitter.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;neurotransmitter_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;neurotransmitter&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">sub_selected</span><span class="p">[</span><span class="s1">&#39;neurotransmitter_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">16</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;neurotransmitter&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/996b4e5ffcea2ac83a86bbeaed1f18b1bad2516ad12dd6d4b32eb22c696c9675.png" src="../_images/996b4e5ffcea2ac83a86bbeaed1f18b1bad2516ad12dd6d4b32eb22c696c9675.png" />
</div>
</div>
<section id="neighborhood-umaps">
<h4>Neighborhood UMAPS<a class="headerlink" href="#neighborhood-umaps" title="Link to this heading">#</a></h4>
<p>The release also provides individual UMAPs for cells in each of the 9 neighborhoods.</p>
<p>We first subselect one of these neighborhoods, ‘Pallium-Glut’. Note that similar masking can for any column/value pair can be done as well. For instance <code class="docutils literal notranslate"><span class="pre">cell_extended[cell_extended['class']</span> <span class="pre">==</span> <span class="pre">'001</span> <span class="pre">L1-ET</span> <span class="pre">Glut']</span></code> will return a DataFrame with only cells in the class 001 L1-ET Glut.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">neighborhood_cells</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;neighborhood&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Pallium-Glut&#39;</span><span class="p">]</span>
<span class="n">neighborhood_cells</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>library_method</th>
      <th>alignment_id</th>
      <th>region_of_interest_acronym</th>
      <th>anatomical_division_label</th>
      <th>donor_label</th>
      <th>...</th>
      <th>class_color</th>
      <th>subclass_color</th>
      <th>supertype_color</th>
      <th>cluster_color</th>
      <th>class_order</th>
      <th>cluster_order</th>
      <th>neighborhood_order</th>
      <th>neurotransmitter_order</th>
      <th>subclass_order</th>
      <th>supertype_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>pBICCNsMMrAUDiF022d210715A5_AAGGTAAGTCTGTCCT</th>
      <td>AAGGTAAGTCTGTCCT</td>
      <td>NaN</td>
      <td>pBICCNsMMrAUDiF022d210715A5</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-Isocortex</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>AUD</td>
      <td>Isocortex</td>
      <td>F022</td>
      <td>...</td>
      <td>#FA0087</td>
      <td>#3DCCB7</td>
      <td>#3B9900</td>
      <td>#990047</td>
      <td>1</td>
      <td>50</td>
      <td>1</td>
      <td>1</td>
      <td>3</td>
      <td>12</td>
    </tr>
    <tr>
      <th>pBICCNsMMrAUDiM015d210707A3_CGTTGGGGTATAATGG</th>
      <td>CGTTGGGGTATAATGG</td>
      <td>NaN</td>
      <td>pBICCNsMMrAUDiM015d210707A3</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-Isocortex</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>AUD</td>
      <td>Isocortex</td>
      <td>M015</td>
      <td>...</td>
      <td>#FA0087</td>
      <td>#E9530F</td>
      <td>#5C6899</td>
      <td>#FFDFA0</td>
      <td>1</td>
      <td>26</td>
      <td>1</td>
      <td>1</td>
      <td>2</td>
      <td>6</td>
    </tr>
    <tr>
      <th>CACTCCACAAGCCGTC-102_A01</th>
      <td>CACTCCACAAGCCGTC</td>
      <td>102_A01</td>
      <td>L8TX_190321_01_C03</td>
      <td>WMB-10Xv2</td>
      <td>WMB-10Xv2-CTXsp</td>
      <td>10Xv2</td>
      <td>1178484037</td>
      <td>CTXsp</td>
      <td>CTXsp</td>
      <td>Snap25-IRES2-Cre;Ai14-449634</td>
      <td>...</td>
      <td>#FA0087</td>
      <td>#BBFF73</td>
      <td>#CC867A</td>
      <td>#FFF7AA</td>
      <td>1</td>
      <td>296</td>
      <td>1</td>
      <td>1</td>
      <td>16</td>
      <td>71</td>
    </tr>
    <tr>
      <th>TCAGATGTCCGCATCT-018_D01</th>
      <td>TCAGATGTCCGCATCT</td>
      <td>018_D01</td>
      <td>L8TX_180406_01_H01</td>
      <td>WMB-10Xv2</td>
      <td>WMB-10Xv2-HPF</td>
      <td>10Xv2</td>
      <td>1178483093</td>
      <td>ENT</td>
      <td>HPF</td>
      <td>Snap25-IRES2-Cre;Ai14-380340</td>
      <td>...</td>
      <td>#FA0087</td>
      <td>#1F46CC</td>
      <td>#990030</td>
      <td>#662E5E</td>
      <td>1</td>
      <td>246</td>
      <td>1</td>
      <td>1</td>
      <td>14</td>
      <td>61</td>
    </tr>
    <tr>
      <th>pBICCNsMMrACAiM016d210628A3_TATATCCCAACCAGAG</th>
      <td>TATATCCCAACCAGAG</td>
      <td>NaN</td>
      <td>pBICCNsMMrACAiM016d210628A3</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-Isocortex</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ACA</td>
      <td>Isocortex</td>
      <td>M016</td>
      <td>...</td>
      <td>#FA0087</td>
      <td>#660F38</td>
      <td>#0F6619</td>
      <td>#5A9917</td>
      <td>1</td>
      <td>62</td>
      <td>1</td>
      <td>1</td>
      <td>4</td>
      <td>15</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 40 columns</p>
</div></div></div>
</div>
<p>Now we load and join in the coordinates of the neighborhood UMAP. Note the inner join and the suffix added to the joined DataFrame as we already have ‘x’ and ‘y’ columns.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">neighborhood_cells</span> <span class="o">=</span> <span class="n">neighborhood_cells</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
        <span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Pallium-Glut_cell_2d_embedding_coordinates&#39;</span>
        <span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">),</span>
    <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">,</span>
    <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_pallium_glut&#39;</span>
<span class="p">)</span>
<span class="n">neighborhood_cells</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Pallium-Glut_cell_2d_embedding_coordinates.csv: 100%|██████████| 138M/138M [00:25&lt;00:00, 5.44MMB/s]   
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>library_method</th>
      <th>alignment_id</th>
      <th>region_of_interest_acronym</th>
      <th>anatomical_division_label</th>
      <th>donor_label</th>
      <th>...</th>
      <th>supertype_color</th>
      <th>cluster_color</th>
      <th>class_order</th>
      <th>cluster_order</th>
      <th>neighborhood_order</th>
      <th>neurotransmitter_order</th>
      <th>subclass_order</th>
      <th>supertype_order</th>
      <th>x_pallium_glut</th>
      <th>y_pallium_glut</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>pBICCNsMMrAUDiF022d210715A5_AAGGTAAGTCTGTCCT</th>
      <td>AAGGTAAGTCTGTCCT</td>
      <td>NaN</td>
      <td>pBICCNsMMrAUDiF022d210715A5</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-Isocortex</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>AUD</td>
      <td>Isocortex</td>
      <td>F022</td>
      <td>...</td>
      <td>#3B9900</td>
      <td>#990047</td>
      <td>1</td>
      <td>50</td>
      <td>1</td>
      <td>1</td>
      <td>3</td>
      <td>12</td>
      <td>12.221766</td>
      <td>12.160112</td>
    </tr>
    <tr>
      <th>pBICCNsMMrAUDiM015d210707A3_CGTTGGGGTATAATGG</th>
      <td>CGTTGGGGTATAATGG</td>
      <td>NaN</td>
      <td>pBICCNsMMrAUDiM015d210707A3</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-Isocortex</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>AUD</td>
      <td>Isocortex</td>
      <td>M015</td>
      <td>...</td>
      <td>#5C6899</td>
      <td>#FFDFA0</td>
      <td>1</td>
      <td>26</td>
      <td>1</td>
      <td>1</td>
      <td>2</td>
      <td>6</td>
      <td>10.600657</td>
      <td>11.083297</td>
    </tr>
    <tr>
      <th>CACTCCACAAGCCGTC-102_A01</th>
      <td>CACTCCACAAGCCGTC</td>
      <td>102_A01</td>
      <td>L8TX_190321_01_C03</td>
      <td>WMB-10Xv2</td>
      <td>WMB-10Xv2-CTXsp</td>
      <td>10Xv2</td>
      <td>1178484037</td>
      <td>CTXsp</td>
      <td>CTXsp</td>
      <td>Snap25-IRES2-Cre;Ai14-449634</td>
      <td>...</td>
      <td>#CC867A</td>
      <td>#FFF7AA</td>
      <td>1</td>
      <td>296</td>
      <td>1</td>
      <td>1</td>
      <td>16</td>
      <td>71</td>
      <td>6.497256</td>
      <td>3.884721</td>
    </tr>
    <tr>
      <th>TCAGATGTCCGCATCT-018_D01</th>
      <td>TCAGATGTCCGCATCT</td>
      <td>018_D01</td>
      <td>L8TX_180406_01_H01</td>
      <td>WMB-10Xv2</td>
      <td>WMB-10Xv2-HPF</td>
      <td>10Xv2</td>
      <td>1178483093</td>
      <td>ENT</td>
      <td>HPF</td>
      <td>Snap25-IRES2-Cre;Ai14-380340</td>
      <td>...</td>
      <td>#990030</td>
      <td>#662E5E</td>
      <td>1</td>
      <td>246</td>
      <td>1</td>
      <td>1</td>
      <td>14</td>
      <td>61</td>
      <td>1.026828</td>
      <td>10.563732</td>
    </tr>
    <tr>
      <th>pBICCNsMMrACAiM016d210628A3_TATATCCCAACCAGAG</th>
      <td>TATATCCCAACCAGAG</td>
      <td>NaN</td>
      <td>pBICCNsMMrACAiM016d210628A3</td>
      <td>Consensus-WMB-Macosko-10X</td>
      <td>Macosko-10X-Isocortex</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ACA</td>
      <td>Isocortex</td>
      <td>M016</td>
      <td>...</td>
      <td>#0F6619</td>
      <td>#5A9917</td>
      <td>1</td>
      <td>62</td>
      <td>1</td>
      <td>1</td>
      <td>4</td>
      <td>15</td>
      <td>14.539476</td>
      <td>11.780524</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 42 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">neighborhood_cells</span><span class="p">[</span><span class="s1">&#39;x_pallium_glut&#39;</span><span class="p">],</span>
    <span class="n">neighborhood_cells</span><span class="p">[</span><span class="s1">&#39;y_pallium_glut&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">neighborhood_cells</span><span class="p">[</span><span class="s1">&#39;subclass_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">neighborhood_cells</span><span class="p">[</span><span class="s1">&#39;subclass&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">neighborhood_cells</span><span class="p">[</span><span class="s1">&#39;subclass_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">18</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">18</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;Subclass in Pallium-Glut Neighborhood&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/3a32be9c2001228093d5d6f57b085e56f12fc3f9f140c588364247ad1d1b1ccf.png" src="../_images/3a32be9c2001228093d5d6f57b085e56f12fc3f9f140c588364247ad1d1b1ccf.png" />
</div>
</div>
<p>Below is a block of code that will plot a given term (e.g. taxonomy class, origin_dataset, donor_sex) in each of the 9 neighborhood UMAPS. Change the value in term_to_plot to any of the other columns we used above to visualize that data in each of the 9 neighborhoods.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">term_to_plot</span> <span class="o">=</span> <span class="s1">&#39;subclass&#39;</span> <span class="c1"># Change to other term (e.g. taxonomy level, anatomical, donor_sex etc.)</span>
<span class="c1"># Loop through all neighborhoods and plot subclass UMAPs ordered by term_order.</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mi">3</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="mi">18</span><span class="p">,</span> <span class="mi">18</span><span class="p">)</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>

<span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">neighborhood</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">cluster_annotation_term</span><span class="p">[</span>
        <span class="n">cluster_annotation_term</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;neighborhood&#39;</span>
        <span class="p">]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;term_order&#39;</span><span class="p">)[</span><span class="s1">&#39;name&#39;</span><span class="p">]):</span>
    <span class="n">neighborhood_cells</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
        <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
            <span class="s1">&#39;Consensus-WMB-integrated-taxonomy&#39;</span><span class="p">,</span>
            <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">neighborhood</span><span class="si">}</span><span class="s1">_cell_2d_embedding_coordinates&#39;</span>
            <span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">),</span>
        <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">,</span>
        <span class="n">rsuffix</span><span class="o">=</span><span class="sa">f</span><span class="s1">&#39;_</span><span class="si">{</span><span class="n">neighborhood</span><span class="si">}</span><span class="s1">&#39;</span>
    <span class="p">)</span>
    <span class="n">plot_umap</span><span class="p">(</span>
        <span class="n">neighborhood_cells</span><span class="p">[</span><span class="s1">&#39;x_&#39;</span> <span class="o">+</span> <span class="n">neighborhood</span><span class="p">],</span>
        <span class="n">neighborhood_cells</span><span class="p">[</span><span class="s1">&#39;y_&#39;</span> <span class="o">+</span> <span class="n">neighborhood</span><span class="p">],</span>
        <span class="n">cc</span><span class="o">=</span><span class="n">neighborhood_cells</span><span class="p">[</span><span class="n">term_to_plot</span> <span class="o">+</span> <span class="s1">&#39;_color&#39;</span><span class="p">],</span>
        <span class="n">fig</span><span class="o">=</span><span class="n">fig</span><span class="p">,</span>
        <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span>
    <span class="p">)</span>
    <span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">neighborhood</span><span class="si">}</span><span class="s2"> Neighborhood&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Subpallium-GABA_cell_2d_embedding_coordinates.csv: 100%|██████████| 53.7M/53.7M [00:10&lt;00:00, 4.94MMB/s]  
HY-EA-Glut-GABA_cell_2d_embedding_coordinates.csv: 100%|██████████| 16.5M/16.5M [00:02&lt;00:00, 6.20MMB/s]
MB-Glut-Dopa-Sero_cell_2d_embedding_coordinates.csv: 100%|██████████| 15.2M/15.2M [00:02&lt;00:00, 5.13MMB/s]
TH-EPI-Glut_cell_2d_embedding_coordinates.csv: 100%|██████████| 15.8M/15.8M [00:03&lt;00:00, 4.44MMB/s]
MB-GABA_cell_2d_embedding_coordinates.csv: 100%|██████████| 8.94M/8.94M [00:01&lt;00:00, 4.94MMB/s]
P-MY-CB-Glut_cell_2d_embedding_coordinates.csv: 100%|██████████| 55.5M/55.5M [00:09&lt;00:00, 5.74MMB/s]  
P-MY-CB-GABA_cell_2d_embedding_coordinates.csv: 100%|██████████| 11.1M/11.1M [00:01&lt;00:00, 5.98MMB/s]
NN-IMN_cell_2d_embedding_coordinates.csv: 100%|██████████| 184M/184M [00:31&lt;00:00, 5.80MMB/s]   
</pre></div>
</div>
<img alt="../_images/2b691b4db553994f92597fc544f5c5de3f91eaacb977bdbcf0d47d2b9703b0ec.png" src="../_images/2b691b4db553994f92597fc544f5c5de3f91eaacb977bdbcf0d47d2b9703b0ec.png" />
</div>
</div>
</section>
</section>
</section>
<section id="aggregating-cluster-and-cells-counts">
<h2>Aggregating cluster and cells counts.<a class="headerlink" href="#aggregating-cluster-and-cells-counts" title="Link to this heading">#</a></h2>
<p>Let’s investigate the taxonomy information a bit more. In this section, we’ll create bar plots showing the number of clusters and cells at each level in the taxonomy.</p>
<p>First, we need to compute the number of clusters that are in each of the cell type taxons above it.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">term_cluster_count</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
        <span class="p">[</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">]</span>
    <span class="p">)[[</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
<span class="n">term_cluster_count</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;number_of_clusters&#39;</span><span class="p">]</span>
<span class="n">term_cluster_count</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>number_of_clusters</th>
    </tr>
    <tr>
      <th>cluster_annotation_term_label</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20251031_CLAS_0001</th>
      <td>518</td>
    </tr>
    <tr>
      <th>CS20251031_CLAS_0002</th>
      <td>98</td>
    </tr>
    <tr>
      <th>CS20251031_CLAS_0003</th>
      <td>24</td>
    </tr>
    <tr>
      <th>CS20251031_CLAS_0004</th>
      <td>25</td>
    </tr>
    <tr>
      <th>CS20251031_CLAS_0005</th>
      <td>167</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Next we sum the cells that are associated for each level in the taxonomy.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">term_cell_count</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">]</span>
<span class="p">)[[</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">term_cell_count</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>number_of_cells</th>
    </tr>
    <tr>
      <th>cluster_annotation_term_label</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20251031_CLAS_0001</th>
      <td>1628174</td>
    </tr>
    <tr>
      <th>CS20251031_CLAS_0002</th>
      <td>386033</td>
    </tr>
    <tr>
      <th>CS20251031_CLAS_0003</th>
      <td>17251</td>
    </tr>
    <tr>
      <th>CS20251031_CLAS_0004</th>
      <td>142462</td>
    </tr>
    <tr>
      <th>CS20251031_CLAS_0005</th>
      <td>218309</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Join counts with the term dataframe</span>
<span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">cluster_annotation_term</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">term_cluster_count</span><span class="p">)</span>
<span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">term_cell_count</span><span class="p">)</span>
<span class="n">term_with_counts</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>cluster_annotation_term_set_label</th>
      <th>cluster_annotation_term_set_name</th>
      <th>color_hex_triplet</th>
      <th>term_order</th>
      <th>term_set_order</th>
      <th>parent_term_label</th>
      <th>parent_term_name</th>
      <th>parent_term_set_label</th>
      <th>number_of_clusters</th>
      <th>number_of_cells</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20251031_NEUR_0004</th>
      <td>Chol</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#73E785</td>
      <td>4</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>25</td>
      <td>3632</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0012</th>
      <td>Chol-Dopa</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#B8EC68</td>
      <td>12</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1</td>
      <td>261</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0008</th>
      <td>Dopa</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#fcf04b</td>
      <td>8</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>24</td>
      <td>7207</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0002</th>
      <td>GABA</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#FF3358</td>
      <td>2</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>2332</td>
      <td>1301636</td>
    </tr>
    <tr>
      <th>CS20251031_NEUR_0006</th>
      <td>GABA-Chol</td>
      <td>CCN20251031_NEUR</td>
      <td>neurotransmitter</td>
      <td>#000080</td>
      <td>6</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>3</td>
      <td>397</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Below we create a function to plot the cluster and cell counts in a bar graph, coloring by the associated taxon level.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">bar_plot_by_level_and_type</span><span class="p">(</span><span class="n">df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">level</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">fig_width</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mf">8.5</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mf">8.5</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Plot the number of cells by the specified level.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    df : pd.DataFrame</span>
<span class="sd">        DataFrame containing cluster annotation terms with counts.</span>
<span class="sd">    level : str</span>
<span class="sd">        The level of the taxonomy to plot (e.g., &#39;Neighborhood&#39;, &#39;Class&#39;, &#39;Subclass&#39;, &#39;Group&#39;, &#39;Cluster&#39;).</span>
<span class="sd">    fig_width : float, optional</span>
<span class="sd">        Width of the figure in inches. Default is 8.5.</span>
<span class="sd">    fig_height : float, optional</span>
<span class="sd">        Height of the figure in inches. Default is 4.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">ctype</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="s1">&#39;clusters&#39;</span><span class="p">,</span> <span class="s1">&#39;cells&#39;</span><span class="p">]):</span>

        <span class="n">pred</span> <span class="o">=</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">level</span><span class="p">)</span>
        <span class="n">sort_order</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">argsort</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">pred</span><span class="p">][</span><span class="s1">&#39;term_order&#39;</span><span class="p">])</span>
        <span class="n">names</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">pred</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">sort_order</span><span class="p">]</span>
        <span class="n">counts</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">pred</span><span class="p">][</span><span class="s1">&#39;number_of_</span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">ctype</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">sort_order</span><span class="p">]</span>
        <span class="n">colors</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">pred</span><span class="p">][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">sort_order</span><span class="p">]</span>
        
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">barh</span><span class="p">(</span><span class="n">names</span><span class="p">[::</span><span class="o">-</span><span class="mi">1</span><span class="p">],</span> <span class="n">counts</span><span class="p">[::</span><span class="o">-</span><span class="mi">1</span><span class="p">],</span> <span class="n">color</span><span class="o">=</span><span class="n">colors</span><span class="p">[::</span><span class="o">-</span><span class="mi">1</span><span class="p">])</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Number of </span><span class="si">%s</span><span class="s1"> by </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">ctype</span><span class="p">,</span><span class="n">level</span><span class="p">))</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Number of </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">ctype</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">ctype</span> <span class="o">==</span> <span class="s1">&#39;cells&#39;</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
        
        <span class="k">if</span> <span class="n">idx</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">([])</span>

    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<p>Now, we plot bar graphs of the number of clusters and cells by taxonomy level. Below we show neighborhood and class, but this comparison can be made for all levels in the taxonomy.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">bar_plot_by_level_and_type</span><span class="p">(</span><span class="n">term_with_counts</span><span class="p">,</span> <span class="s1">&#39;neighborhood&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/c663215a05fb19a0f2aebf3d87081c754515012c06a8db3a27670ae696962813.png" src="../_images/c663215a05fb19a0f2aebf3d87081c754515012c06a8db3a27670ae696962813.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">bar_plot_by_level_and_type</span><span class="p">(</span><span class="n">term_with_counts</span><span class="p">,</span> <span class="s1">&#39;class&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/ca4d2d29a9029857ddfa56f6fcf528396bb7ca0e49714ba1d86bae06f43f6513.png" src="../_images/ca4d2d29a9029857ddfa56f6fcf528396bb7ca0e49714ba1d86bae06f43f6513.png" />
</div>
</div>
<section id="visualizing-the-taxonomy">
<h3>Visualizing the  taxonomy<a class="headerlink" href="#visualizing-the-taxonomy" title="Link to this heading">#</a></h3>
<p>Finally, we create a pie chart for Neighborhood, Class, Subclass, and Supertype. This is plotted in such a way that the inner rings are all children of the parent taxon above. The width’s of the pie colors are given by the number of clusters in taxon.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">levels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;neighborhood&#39;</span><span class="p">,</span> <span class="s1">&#39;class&#39;</span><span class="p">,</span> <span class="s1">&#39;subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;supertype&#39;</span><span class="p">]</span>
<span class="n">df</span> <span class="o">=</span> <span class="p">{}</span>

<span class="c1"># Copy the term order of the parent into each of the level below it.</span>
<span class="n">term_with_counts</span><span class="p">[</span><span class="s1">&#39;parent_order&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;parent_term_label&#39;</span><span class="p">]):</span>
        <span class="k">continue</span>
    <span class="n">term_with_counts</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">idx</span><span class="p">,</span> <span class="s1">&#39;parent_order&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;parent_term_label&#39;</span><span class="p">]][</span><span class="s1">&#39;term_order&#39;</span><span class="p">]</span>

<span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="k">for</span> <span class="n">lvl</span> <span class="ow">in</span> <span class="n">levels</span><span class="p">:</span>
    <span class="n">pred</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">lvl</span>
    <span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">]</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="p">[</span><span class="n">pred</span><span class="p">]</span>
    <span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;parent_order&#39;</span><span class="p">,</span> <span class="s1">&#39;term_order&#39;</span><span class="p">])</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">10</span><span class="p">)</span>
<span class="n">size</span> <span class="o">=</span> <span class="mf">0.15</span>

<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">lvl</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">levels</span><span class="p">):</span>
    
    <span class="k">if</span> <span class="n">lvl</span> <span class="o">==</span> <span class="s1">&#39;neighborhood&#39;</span><span class="p">:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">pie</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;number_of_clusters&#39;</span><span class="p">],</span>
               <span class="n">colors</span><span class="o">=</span><span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">],</span>
               <span class="n">labels</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">],</span>
               <span class="n">rotatelabels</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
               <span class="n">labeldistance</span><span class="o">=</span><span class="mf">1.025</span><span class="p">,</span>
               <span class="n">radius</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
               <span class="n">wedgeprops</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">width</span><span class="o">=</span><span class="n">size</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="kc">None</span><span class="p">),</span>
               <span class="n">startangle</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="k">else</span> <span class="p">:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">pie</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;number_of_clusters&#39;</span><span class="p">],</span>
               <span class="n">colors</span><span class="o">=</span><span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">],</span>
               <span class="n">radius</span><span class="o">=</span><span class="mi">1</span><span class="o">-</span><span class="n">i</span><span class="o">*</span><span class="n">size</span><span class="p">,</span>
               <span class="n">wedgeprops</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">width</span><span class="o">=</span><span class="n">size</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="kc">None</span><span class="p">),</span>
               <span class="n">startangle</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/b14bba18d5dedba0c4c301933ebb8d4038d4f2345404a32fe4885d03d28a9d33.png" src="../_images/b14bba18d5dedba0c4c301933ebb8d4038d4f2345404a32fe4885d03d28a9d33.png" />
</div>
</div>
<p>In the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/consensus_mouse_10X_snRNASeq.html">next notebook</a>, we’ll explore the gene expression data and combine them with the taxonomy and cell level metadata. You can also explore the previously released Whole Mouse Brain (<a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/WMB-10X.html">WMB-10X</a>) through the notebooks linked <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/notebook_subtitle1.html">here</a>.</p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="../descriptions/Consensus-WMB-notebooks.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Consensus Mouse notebooks</p>
      </div>
    </a>
    <a class="right-next"
       href="consensus_mouse_10X_snRNASeq.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Consensus Whole Mouse Brain #2: 10X gene expression</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#library-and-donor-metadata">Library and Donor metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#adding-color-and-feature-order">Adding color and feature order</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#umap-spatial-embedding">UMAP spatial embedding</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxonomy-information">Taxonomy Information</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-taxonomy">Plotting the taxonomy</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#neighborhood-umaps">Neighborhood UMAPS</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#aggregating-cluster-and-cells-counts">Aggregating cluster and cells counts.</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualizing-the-taxonomy">Visualizing the  taxonomy</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>