
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Developing Mouse - Visual Cortex 10X scRNASeq analysis: clustering and annotations &#8212; Allen Brain Cell Atlas - Data Access</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=9c3e77be" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/simple_style.css?v=3aaed5dc" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=1ae7504c"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/dev_mouse_vis_cortex_clustering_analysis_and_annotation';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Developing Mouse - Visual Cortex: 10X gene expression" href="dev_mouse_vis_cortex_10X_scRNASeq.html" />
    <link rel="prev" title="Developing Mouse - Visual Cortex notebooks" href="../descriptions/Dev-Mouse-Vis-Cortex-notebooks.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/allen_logo.jpeg" class="logo__image only-light" alt="Allen Brain Cell Atlas - Data Access - Home"/>
    <script>document.write(`<img src="../_static/allen_logo.jpeg" class="logo__image only-dark" alt="Allen Brain Cell Atlas - Data Access - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Allen Brain Cell Atlas - Data Access
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">INTRODUCTION</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="general_accessing_10x_snRNASeq_tutorial.html">Accessing 10x RNA-seq gene expression data</a></li>
<li class="toctree-l1"><a class="reference internal" href="abc_atlas_selection_example.html">Using cells selected in the ABC Atlas with Gene Expression data.</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">DATA</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WMB_dataset.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv2.html">Whole mouse brain 10Xv2 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv3.html">Whole mouse brain 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10XMulti.html">Whole mouse brain 10XMulti single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10X.html">Clustering analysis of whole adult mouse brain 10X single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-taxonomy.html">Whole adult mouse brain taxonomy of cell types</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-imputed.html">Imputed MERFISH spatial transcriptomics of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-neighborhoods.html">Cluster groups and neighborhood specific UMAP embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Allen-CCF-2020.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-CCF.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_dataset.html">Aging Mouse dataset (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_10Xv3.html">Zeng Aging Mouse 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_taxonomy.html">Zeng Aging Mouse clustering and mapping to the WMB taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zhuang_dataset.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-1.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-2.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-3.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-3)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-4.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-4)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-dataset.html">Consensus Whole Mouse Brain dataset (Macosko, Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-10X.html">Consensus Whole Mouse Brain single cell/nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-taxonomy.html">Consensus Whole Mouse Brain integrated taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-dataset.html">Developing Mouse - Visual Cortex</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-10X.html">Developing Mouse - Visual Cortex single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-taxonomy.html">Developing Mouse - Visual Cortex taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_dataset.html">Human whole-brain transcriptomic cell type atlas (Kimberly Siletti)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10Xv3.html">Human brain 10Xv3 single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10X-clustering.html">Clustering analysis of whole human brain 10X single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-taxonomy.html">Human brain taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_dataset.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-10X.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-MapMyCells.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain (WHB) and Seattle Alzheimer’s Disease Brain Cell Atlas (SEA-AD) Taxonomies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_dataset.html">Human-Mammalian Brain - Basal Ganglia: Data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-BG-taxonomy-CCN20250428.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq data: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG-Aligned.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Aligned gene expression across species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Individual species gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Spatial-BG.html">Human-Mammalian Brain - Basal Ganglia: spatial transcriptomics cells for each species</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">NOTEBOOKS</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cluster_annotation_tutorial.html">10x RNA-seq clustering analysis and annotation (CCN20230722)</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle2.html">10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_1.html">10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2a.html">10x RNA-seq gene expression data (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2b.html">10x RNA-seq gene expression data (part 2b)</a></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle3.html">MERFISH whole brain spatial transcriptomics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_1.html">MERFISH whole brain spatial transcriptomics (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2a.html">MERFISH whole brain spatial transcriptomics (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2b.html">MERFISH whole brain spatial transcriptomics (part 2b)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_imputed_genes_example.html">MERFISH whole brain spatial transcriptomics cells with imputed genes</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_groups_and_embeddings_tutorial.html">Cluster neighborhoods and embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_neighborhood_gallery.html">Cell type neighborhood gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="ccf_and_parcellation_annotation_tutorial.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="merfish_ccf_registration_tutorial.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-notebooks.html">Consensus Mouse notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_clustering_analysis_and_annotation.html">Consensus Whole Mouse Brain #1: Clustering and Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_10X_snRNASeq.html">Consensus Whole Mouse Brain #2: 10X gene expression</a></li>

</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../descriptions/Dev-Mouse-Vis-Cortex-notebooks.html">Developing Mouse - Visual Cortex notebooks</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Developing Mouse - Visual Cortex 10X scRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="dev_mouse_vis_cortex_10X_scRNASeq.html">Developing Mouse - Visual Cortex: 10X gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_notebooks.html">Aging Mouse Notebooks (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_clustering_analysis_and_annotation.html">Mouse Aging clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_10x_snRNASeq_tutorial.html">Mouse Aging 10X RNA-seq gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle4.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="zhuang_merfish_tutorial.html">MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_notebooks.html">Whole Human Brain 10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="WHB_cluster_annotation_tutorial.html">Whole Human Brain 10x RNA-seq clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_1.html">Whole Human Brain (WHB) 10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_2.html">Whole Human Brain 10x RNA-seq gene expression data (part 2)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_notebooks.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_data_and_metadata.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_siletti_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_seaad_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_gene_expression.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Gene Expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_notebooks.html">Human-Mammalian Brain - Basal Ganglia: Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_clustering_analysis_and_annotation.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_10X_snRNASeq_tutorial.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_slabs_and_taxonomy.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_genes.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">ADDITIONAL RESOURCES</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../docs/README.html">Resource Pages</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../docs/abbreviation_terms/README.html">Abbreviation lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/cluster_annotations/README.html">Cluster annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/gene_lists/README.html">Gene lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/parcellation_annotations/README.html">Parcellation annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/supplemental_metadata_descriptions/README.html">Supplemental Metadata Descriptions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/neuroglancer_hmba/README.html">HMBA Basal Ganglia: Neuroglancer Views</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access/issues/new?title=Issue%20on%20page%20%2Fnotebooks/dev_mouse_vis_cortex_clustering_analysis_and_annotation.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/dev_mouse_vis_cortex_clustering_analysis_and_annotation.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Developing Mouse - Visual Cortex 10X scRNASeq analysis: clustering and annotations</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#donor-and-library-metadata">Donor and Library metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#adding-color-and-feature-order">Adding color and feature order</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#umap-spatial-embedding">UMAP spatial embedding</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxonomy-information">Taxonomy Information</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-taxonomy">Plotting the taxonomy</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#aggregating-cluster-and-cells-counts-per-term-per-species">Aggregating cluster and cells counts per term per species.</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cluster-and-subcluster-count-vs-donor-age">Cluster and subcluster count vs donor age.</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualizing-the-developmental-visual-cortex-taxonomy">Visualizing the developmental visual cortex taxonomy</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="developing-mouse-visual-cortex-10x-scrnaseq-analysis-clustering-and-annotations">
<h1>Developing Mouse - Visual Cortex 10X scRNASeq analysis: clustering and annotations<a class="headerlink" href="#developing-mouse-visual-cortex-10x-scrnaseq-analysis-clustering-and-annotations" title="Link to this heading">#</a></h1>
<p>The Developing Mouse - Visual Cortex taxonomy is derived from a single cell transcriptomic dataset containing 568,654 cells from donors ranging in age from embryonic to adult. The cell-type assignment is a four level taxonomy defining 15 classes, 40 subclasses, 148 clusters and 714 subclusters. This cell-type assignment is broadly consistent with those from the previous studies from the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/WMB-taxonomy.html">Allen Institute for Brain Science, Whole Mouse Brain (WMB)</a> and the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/Consensus-WMB-taxonomy.html">Broad Institute whole mouse brain taxonomy</a> at the subclass level while providing finer cell-type and temporal resolutions with additional subcluster annotations.</p>
<p>To generate this developing taxonomy, we applied the Quality Control (QC) and post-integration QC pipelines. Integration and label transfer of scRNA-seq between ages was performed using Seurat/scVI. A detailed cell type annotation table accompanies the taxonomy, including hierarchical membership and anatomical localization.</p>
<p>You need to be connected to the internet to run this notebook or connected to a cache that has the Developing Mouse data downloaded already.</p>
<p>The notebook presented here shows quick visualizations from precomputed metadata in the atlas, exploring both cell, donor, and library metadata as well as the taxonomy. We also plot visualizations for the various data and metadata in a Uniform Manifold Approximation and Projection (UMAP). For examples on accessing the expression matrices, specifically selecting genes from expression matrices, see the  <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/general_accessing_10x_snRNASeq_tutorial.html"><code class="docutils literal notranslate"><span class="pre">general_accessing_10x_snRNASeq_tutorial.ipynb</span></code></a> tutorial/example. In a related tutorial, we also show how to access and use <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/dev_mouse_vis_cortex_10X_scRNASeq.html">Developing Mouse - Visual Cortex gene expression data</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">typing</span><span class="w"> </span><span class="kn">import</span> <span class="n">Tuple</span><span class="p">,</span> <span class="n">Optional</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">abc_atlas_access.abc_atlas_cache.abc_project_cache</span><span class="w"> </span><span class="kn">import</span> <span class="n">AbcProjectCache</span>
</pre></div>
</div>
</div>
</div>
<p>We will interact with the data using the <strong>AbcProjectCache</strong>. This cache object downloads data requested by the user, tracks which files have already been downloaded to your local system, and serves the path to the requested data on disk. For metadata, the cache can also directly serve up a Pandas DataFrame. See the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html"><code class="docutils literal notranslate"><span class="pre">getting_started</span></code></a> notebook for more details on using the cache including installing it if it has not already been.</p>
<p><strong>Change the download_base variable to where you would like to download the data in your system or a location where a cache is already available.</strong></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">download_base</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">&#39;../../data/abc_atlas&#39;</span><span class="p">)</span>
<span class="n">abc_cache</span> <span class="o">=</span> <span class="n">AbcProjectCache</span><span class="o">.</span><span class="n">from_cache_dir</span><span class="p">(</span>
    <span class="n">download_base</span><span class="p">,</span>
<span class="p">)</span>

<span class="n">abc_cache</span><span class="o">.</span><span class="n">current_manifest</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;releases/20260131/manifest.json&#39;
</pre></div>
</div>
</div>
</div>
<section id="data-overview">
<h2>Data overview<a class="headerlink" href="#data-overview" title="Link to this heading">#</a></h2>
<p>We’ll quickly walk through the data we will be using in this notebook. The Developing Mouse 10X dataset is located across two directories listed in the <code class="docutils literal notranslate"><span class="pre">ABCProjectCache</span></code>. These data are the metadata and data for the cells and the metadata defining the taxonomy. We will be using data and metadata from the following directories:</p>
<ul class="simple">
<li><p>Developing-Mouse-Vis-Cortex-10X</p></li>
<li><p>Developing-Mouse-Vis-Cortex-taxonomy</p></li>
</ul>
<p>Below we list the data and metadata in the Developing Mouse dataset.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Developing-Mouse-Vis-Cortex-10X: gene expression data (h5ad)</span><span class="se">\n\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_expression_matrix_files</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-10X&#39;</span><span class="p">))</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Developing-Mouse-Vis-Cortex-10X: metadata (csv)</span><span class="se">\n\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_metadata_files</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-10X&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Developing-Mouse-Vis-Cortex-10X: gene expression data (h5ad)
	 [&#39;Developing-Mouse-Vis-Cortex-10X/log2&#39;, &#39;Developing-Mouse-Vis-Cortex-10X/raw&#39;]
Developing-Mouse-Vis-Cortex-10X: metadata (csv)
	 [&#39;cell_metadata&#39;, &#39;donor&#39;, &#39;example_gene_expression&#39;, &#39;gene&#39;, &#39;library&#39;, &#39;value_sets&#39;]
</pre></div>
</div>
</div>
</div>
<p>We will also use metadata from the Developing-Mouse-Vis-Cortex-taxonomy directory. Below is the list of available files:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Developing-Mouse-Vis-Cortex-taxonomy: metadata (csv)</span><span class="se">\n\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_metadata_files</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-taxonomy&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Developing-Mouse-Vis-Cortex-taxonomy: metadata (csv)
	 [&#39;cell_2d_embedding_coordinates&#39;, &#39;cell_to_cluster_membership&#39;, &#39;cluster&#39;, &#39;cluster_annotation_term&#39;, &#39;cluster_annotation_term_set&#39;, &#39;cluster_to_cluster_annotation_membership&#39;, &#39;de_genes&#39;]
</pre></div>
</div>
</div>
</div>
<section id="cell-metadata">
<h3>Cell metadata<a class="headerlink" href="#cell-metadata" title="Link to this heading">#</a></h3>
<p>Essential cell metadata is stored as a CSV file that we load as a Pandas DataFrame. Each row represents one cell indexed by a <em>cell label</em>. The <em>cell label</em> is the concatenation of barcode and name of the sample. In this context, the sample is the <em>barcoded cell sample</em> that represents a single load into one port of the 10x Chromium. Note that cell barcodes are only unique within a single <em>barcoded cell sample</em> and that the same barcode can be reused.</p>
<p>Each cell is associated with a <em>library label</em>, <em>donor label</em>, <em>alignment_job_id</em>, <em>feature_matrix_label</em> and <em>dataset_label</em> identifying which data package this cell is part of. This metadata file will be combined with other metadata files that ship with this package to add information associated with the donor, UMAP coordinates, taxonomy assignments, and more.</p>
<p>Below, we load the first of the metadata used in this tutorial. This represents the cell metadata for the aligned dataset.</p>
<p>The command we use below both downloads the data if it is not already available in the local cache and loads the data as a Pandas DataFrame. This pattern of loading metadata is repeated throughout the tutorials.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Number of cells = &quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">cell</span><span class="p">))</span>
<span class="n">cell</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_metadata.csv: 100%|██████████| 105M/105M [00:03&lt;00:00, 30.0MMB/s]   
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of cells =  568654
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>alignment_id</th>
      <th>log.qc.score.cr6</th>
      <th>synchronized_age</th>
      <th>donor_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAACCCAAGGATTTAG-898_A02</th>
      <td>AAACCCAAGGATTTAG</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>338.986629</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
    </tr>
    <tr>
      <th>AAACCCACACATTCTT-898_A02</th>
      <td>AAACCCACACATTCTT</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>425.642457</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
    </tr>
    <tr>
      <th>AAACCCACACTTACAG-898_A02</th>
      <td>AAACCCACACTTACAG</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>342.366675</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
    </tr>
    <tr>
      <th>AAACGAATCACTCTTA-898_A02</th>
      <td>AAACGAATCACTCTTA</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>353.204604</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
    </tr>
    <tr>
      <th>AAACGAATCGACCAAT-898_A02</th>
      <td>AAACGAATCGACCAAT</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>355.052147</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We can use pandas <code class="docutils literal notranslate"><span class="pre">groupby</span></code> function to see how many unique items are associated for each field and list them out if the number of unique items is small.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">print_column_info</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
    
    <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">grouped</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="n">c</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">c</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
        <span class="n">members</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">grouped</span><span class="p">)</span> <span class="o">&lt;</span> <span class="mi">30</span><span class="p">:</span>
            <span class="n">members</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">grouped</span><span class="o">.</span><span class="n">index</span><span class="p">))</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Number of unique </span><span class="si">%s</span><span class="s2"> = </span><span class="si">%d</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">grouped</span><span class="p">),</span> <span class="n">members</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">print_column_info</span><span class="p">(</span><span class="n">cell</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of unique cell_barcode = 527176 
Number of unique barcoded_cell_sample_label = 91 
Number of unique library_label = 91 
Number of unique alignment_id = 91 
Number of unique log.qc.score.cr6 = 496566 
Number of unique synchronized_age = 35 
Number of unique donor_label = 53 
Number of unique dataset_label = 1 [&#39;Developing-Mouse-Vis-Cortex-10X&#39;]
Number of unique feature_matrix_label = 1 [&#39;Developing-Mouse-Vis-Cortex-10X&#39;]
</pre></div>
</div>
</div>
</div>
</section>
<section id="donor-and-library-metadata">
<h3>Donor and Library metadata<a class="headerlink" href="#donor-and-library-metadata" title="Link to this heading">#</a></h3>
<p>The first two associated metadata we load are the <code class="docutils literal notranslate"><span class="pre">donor</span></code> and <code class="docutils literal notranslate"><span class="pre">library</span></code> tables. The donor table contains species, sex, and age information. The library table contains information on 10X methods and brain region of interest the tissue was extracted from.</p>
<p>Below we load the donor metadata. Note that we flatten the values of P54 and above to P56 or “Adult” to match the published analysis. The original age values are available in the donor table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">donor</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;donor&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>

<span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">donor</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;donor_age&#39;</span><span class="p">]</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;P54&#39;</span><span class="p">,</span> <span class="s1">&#39;P59&#39;</span><span class="p">,</span> <span class="s1">&#39;P60&#39;</span><span class="p">,</span> <span class="s1">&#39;P61&#39;</span><span class="p">,</span> <span class="s1">&#39;P68&#39;</span><span class="p">]:</span>
        <span class="n">donor</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">idx</span><span class="p">,</span> <span class="s1">&#39;donor_age&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;P56&#39;</span>

<span class="n">donor</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>donor.csv:   0%|          | 0.00/5.08k [00:00&lt;?, ?MB/s]
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>donor.csv: 100%|██████████| 5.08k/5.08k [00:00&lt;00:00, 96.1kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>donor_species</th>
      <th>species_scientific_name</th>
      <th>species_genus</th>
      <th>donor_sex</th>
      <th>donor_age_value</th>
      <th>donor_age_unit</th>
      <th>donor_age_reference_point</th>
      <th>donor_age</th>
      <th>age_bin</th>
    </tr>
    <tr>
      <th>donor_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>C57BL6J-603767</th>
      <td>NCBITaxon:10090</td>
      <td>Mus musculus</td>
      <td>Mouse</td>
      <td>Male</td>
      <td>18.5</td>
      <td>days</td>
      <td>conception</td>
      <td>E18.5</td>
      <td>E17_E18.5</td>
    </tr>
    <tr>
      <th>C57BL6J-620146</th>
      <td>NCBITaxon:10090</td>
      <td>Mus musculus</td>
      <td>Mouse</td>
      <td>Male</td>
      <td>9.0</td>
      <td>days</td>
      <td>birth</td>
      <td>P9</td>
      <td>9</td>
    </tr>
    <tr>
      <th>C57BL6J-628080</th>
      <td>NCBITaxon:10090</td>
      <td>Mus musculus</td>
      <td>Mouse</td>
      <td>Male</td>
      <td>13.5</td>
      <td>days</td>
      <td>conception</td>
      <td>E13.5</td>
      <td>E13.5_E16.5</td>
    </tr>
    <tr>
      <th>C57BL6J-628083</th>
      <td>NCBITaxon:10090</td>
      <td>Mus musculus</td>
      <td>Mouse</td>
      <td>Male</td>
      <td>13.5</td>
      <td>days</td>
      <td>conception</td>
      <td>E13.5</td>
      <td>E13.5_E16.5</td>
    </tr>
    <tr>
      <th>C57BL6J-628825</th>
      <td>NCBITaxon:10090</td>
      <td>Mus musculus</td>
      <td>Mouse</td>
      <td>Female</td>
      <td>18.0</td>
      <td>days</td>
      <td>conception</td>
      <td>E18</td>
      <td>E17_E18.5</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Next we load the library metadata. The information we will primarily use from this table are the region of interest that each library is associated with</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">library</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;library&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;library_label&#39;</span><span class="p">)</span>
<span class="n">library</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>library.csv: 100%|██████████| 11.2k/11.2k [00:00&lt;00:00, 138kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>library_technique</th>
      <th>barcoded_cell_sample_label</th>
      <th>enrichment_population</th>
      <th>cell_specimen_type</th>
      <th>parcellation_term_identifier</th>
      <th>region_of_interest_name</th>
      <th>region_of_interest_label</th>
      <th>donor_label</th>
    </tr>
    <tr>
      <th>library_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>L8TX_200618_01_D07</th>
      <td>10xV3.1;GEXOnly</td>
      <td>265_B02</td>
      <td>No FACS</td>
      <td>Cells</td>
      <td>MBA:385</td>
      <td>Primary visual area</td>
      <td>VISp</td>
      <td>Snap25-IRES2-Cre;Ai14-536764</td>
    </tr>
    <tr>
      <th>L8TX_200618_01_F06</th>
      <td>10xV3.1;GEXOnly</td>
      <td>264_A01</td>
      <td>RFP-positive, Hoechst-positive</td>
      <td>Cells</td>
      <td>MBA:385</td>
      <td>Primary visual area</td>
      <td>VISp</td>
      <td>Snap25-IRES2-Cre;Ai14-536763</td>
    </tr>
    <tr>
      <th>L8TX_200618_01_G05</th>
      <td>10xV3.1;GEXOnly</td>
      <td>264_B01</td>
      <td>RFP-positive, Hoechst-positive</td>
      <td>Cells</td>
      <td>MBA:385</td>
      <td>Primary visual area</td>
      <td>VISp</td>
      <td>Snap25-IRES2-Cre;Ai14-536763</td>
    </tr>
    <tr>
      <th>L8TX_200618_01_H05</th>
      <td>10xV3.1;GEXOnly</td>
      <td>265_A02</td>
      <td>No FACS</td>
      <td>Cells</td>
      <td>MBA:385</td>
      <td>Primary visual area</td>
      <td>VISp</td>
      <td>Snap25-IRES2-Cre;Ai14-536764</td>
    </tr>
    <tr>
      <th>L8TX_200709_01_B03</th>
      <td>10xV3.1;GEXOnly</td>
      <td>282_B02</td>
      <td>No FACS</td>
      <td>Cells</td>
      <td>MBA:385</td>
      <td>Primary visual area</td>
      <td>VISp</td>
      <td>Snap25-IRES2-Cre;Ai14-538743</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We combine the donor and library tables into an extended cell metadata table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">donor</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;donor_label&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">library</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;library_label&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_library_table&#39;</span><span class="p">)</span>

<span class="k">del</span> <span class="n">cell</span>
</pre></div>
</div>
</div>
</div>
<p>We use the <code class="docutils literal notranslate"><span class="pre">groupby</span></code> function to show the number of cells in each region of interest.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;region_of_interest_name&#39;</span><span class="p">)[[</span><span class="s1">&#39;region_of_interest_label&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>region_of_interest_label</th>
    </tr>
    <tr>
      <th>region_of_interest_name</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Cerebrum - Brain stem</th>
      <td>12813</td>
    </tr>
    <tr>
      <th>Primary visual area</th>
      <td>128550</td>
    </tr>
    <tr>
      <th>Visual areas</th>
      <td>387480</td>
    </tr>
    <tr>
      <th>Visual areas - Posterior parietal association areas</th>
      <td>25539</td>
    </tr>
    <tr>
      <th>brain</th>
      <td>14272</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We can use the group by functionality to group the cells by each age.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;age_bin&#39;</span><span class="p">)[[</span><span class="s1">&#39;library_label&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">count</span><span class="p">()</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;library_label&#39;</span><span class="p">:</span> <span class="s1">&#39;number_of_cells&#39;</span><span class="p">})</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span>
    <span class="p">[</span><span class="s1">&#39;E11.5_E12.5&#39;</span><span class="p">,</span> <span class="s1">&#39;E13.5_E16.5&#39;</span><span class="p">,</span> <span class="s1">&#39;E17_E18.5&#39;</span><span class="p">,</span>
     <span class="s1">&#39;0_1&#39;</span><span class="p">,</span> <span class="s1">&#39;2&#39;</span><span class="p">,</span> <span class="s1">&#39;3&#39;</span><span class="p">,</span> <span class="s1">&#39;4&#39;</span><span class="p">,</span> <span class="s1">&#39;5_6&#39;</span><span class="p">,</span> <span class="s1">&#39;7_8&#39;</span><span class="p">,</span> <span class="s1">&#39;9&#39;</span><span class="p">,</span> <span class="s1">&#39;10&#39;</span><span class="p">,</span>
     <span class="s1">&#39;11&#39;</span><span class="p">,</span> <span class="s1">&#39;12_13&#39;</span><span class="p">,</span> <span class="s1">&#39;14_15&#39;</span><span class="p">,</span> <span class="s1">&#39;16&#39;</span><span class="p">,</span> <span class="s1">&#39;17_19&#39;</span><span class="p">,</span> <span class="s1">&#39;20_28&#39;</span><span class="p">,</span> <span class="s1">&#39;54_68&#39;</span><span class="p">]]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>number_of_cells</th>
    </tr>
    <tr>
      <th>age_bin</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>E11.5_E12.5</th>
      <td>17222</td>
    </tr>
    <tr>
      <th>E13.5_E16.5</th>
      <td>19762</td>
    </tr>
    <tr>
      <th>E17_E18.5</th>
      <td>22970</td>
    </tr>
    <tr>
      <th>0_1</th>
      <td>28315</td>
    </tr>
    <tr>
      <th>2</th>
      <td>23329</td>
    </tr>
    <tr>
      <th>3</th>
      <td>17953</td>
    </tr>
    <tr>
      <th>4</th>
      <td>12744</td>
    </tr>
    <tr>
      <th>5_6</th>
      <td>13356</td>
    </tr>
    <tr>
      <th>7_8</th>
      <td>31637</td>
    </tr>
    <tr>
      <th>9</th>
      <td>31973</td>
    </tr>
    <tr>
      <th>10</th>
      <td>12146</td>
    </tr>
    <tr>
      <th>11</th>
      <td>11885</td>
    </tr>
    <tr>
      <th>12_13</th>
      <td>54048</td>
    </tr>
    <tr>
      <th>14_15</th>
      <td>76385</td>
    </tr>
    <tr>
      <th>16</th>
      <td>24198</td>
    </tr>
    <tr>
      <th>17_19</th>
      <td>28877</td>
    </tr>
    <tr>
      <th>20_28</th>
      <td>68990</td>
    </tr>
    <tr>
      <th>54_68</th>
      <td>72864</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="adding-color-and-feature-order">
<h3>Adding color and feature order<a class="headerlink" href="#adding-color-and-feature-order" title="Link to this heading">#</a></h3>
<p>Each major feature in the donor and library table is associated with unique colors and an ordering with the set of values. Below we load the <code class="docutils literal notranslate"><span class="pre">value_sets</span></code> DataFrame which is a mapping from the various value in the donor and species tables to those colors and orderings. We incorporate these values into the cell metadata table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">value_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-10X&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;value_sets&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">value_sets</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>value_sets.csv: 100%|██████████| 4.33k/4.33k [00:00&lt;00:00, 66.6kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>table</th>
      <th>field</th>
      <th>description</th>
      <th>order</th>
      <th>external_identifier</th>
      <th>parent_label</th>
      <th>color_hex_triplet</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Female</th>
      <td>donor</td>
      <td>donor_sex</td>
      <td>Female</td>
      <td>1</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#565353</td>
    </tr>
    <tr>
      <th>Male</th>
      <td>donor</td>
      <td>donor_sex</td>
      <td>Male</td>
      <td>2</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#ADC4C3</td>
    </tr>
    <tr>
      <th>Mouse</th>
      <td>donor</td>
      <td>species_genus</td>
      <td>Mouse</td>
      <td>1</td>
      <td>NCBITaxon:10088</td>
      <td>NaN</td>
      <td>#c941a7</td>
    </tr>
    <tr>
      <th>Mus musculus</th>
      <td>donor</td>
      <td>species_scientific_name</td>
      <td>Mus musculus</td>
      <td>1</td>
      <td>NCBITaxon:10090</td>
      <td>NaN</td>
      <td>#ffa300</td>
    </tr>
    <tr>
      <th>WholeBrain</th>
      <td>library</td>
      <td>region_of_interest_label</td>
      <td>brain</td>
      <td>1</td>
      <td>MBA:997</td>
      <td>NaN</td>
      <td>#bebebe</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>14_15</th>
      <td>donor</td>
      <td>age_bin</td>
      <td>age bin in days</td>
      <td>14</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#2C49AD</td>
    </tr>
    <tr>
      <th>16</th>
      <td>donor</td>
      <td>age_bin</td>
      <td>age bin in days</td>
      <td>15</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#0F0F99</td>
    </tr>
    <tr>
      <th>17_19</th>
      <td>donor</td>
      <td>age_bin</td>
      <td>age bin in days</td>
      <td>16</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#6B49F9</td>
    </tr>
    <tr>
      <th>20_28</th>
      <td>donor</td>
      <td>age_bin</td>
      <td>age bin in days</td>
      <td>17</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#AE43C6</td>
    </tr>
    <tr>
      <th>54_68</th>
      <td>donor</td>
      <td>age_bin</td>
      <td>age bin in days</td>
      <td>18</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>#C30000</td>
    </tr>
  </tbody>
</table>
<p>68 rows × 7 columns</p>
</div></div></div>
</div>
<p>We define a convenience function to add colors for the various values in the data (e.g. unique region of interest or donor sex values).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">extract_value_set</span><span class="p">(</span>
        <span class="n">cell_metadata_df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
        <span class="n">input_value_set</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
        <span class="n">input_value_set_label</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">dataframe_column</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>
    <span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Add color and order columns to the cell metadata dataframe based on the input</span>
<span class="sd">    value set.</span>

<span class="sd">    Columns are added as {input_value_set_label}_color and {input_value_set_label}_order.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    cell_metadata_df : pd.DataFrame</span>
<span class="sd">        DataFrame containing cell metadata.</span>
<span class="sd">    input_value_set : pd.DataFrame</span>
<span class="sd">        DataFrame containing the value set information.</span>
<span class="sd">    input_value_set_label : str</span>
<span class="sd">        The the column name to extract color and order information for. will be added to the cell metadata.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">dataframe_column</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">dataframe_column</span> <span class="o">=</span> <span class="n">input_value_set_label</span>
    <span class="n">cell_metadata_df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">dataframe_column</span><span class="si">}</span><span class="s1">_color&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_value_set</span><span class="p">[</span>
        <span class="n">input_value_set</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">input_value_set_label</span>
    <span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cell_metadata_df</span><span class="p">[</span><span class="n">dataframe_column</span><span class="p">]][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">cell_metadata_df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">dataframe_column</span><span class="si">}</span><span class="s1">_order&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_value_set</span><span class="p">[</span>
        <span class="n">input_value_set</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">input_value_set_label</span>
    <span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cell_metadata_df</span><span class="p">[</span><span class="n">dataframe_column</span><span class="p">]][</span><span class="s1">&#39;order&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
</pre></div>
</div>
</div>
</div>
<p>Use our function to add the relevant color and order columns to our cell_metadata table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Add region of interest color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;region_of_interest_label&#39;</span><span class="p">)</span>
<span class="c1"># Add region of interest color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;synchronized_age&#39;</span><span class="p">,</span> <span class="s1">&#39;donor_age&#39;</span><span class="p">)</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;synchronized_age&#39;</span><span class="p">)</span>
<span class="c1"># Add region of interest color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;age_bin&#39;</span><span class="p">)</span>
<span class="c1"># Add species common name color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;species_genus&#39;</span><span class="p">)</span>
<span class="c1"># Add species scientific name color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;species_scientific_name&#39;</span><span class="p">)</span>
<span class="c1"># Add donor sex color and order</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">,</span> <span class="n">value_sets</span><span class="p">,</span> <span class="s1">&#39;donor_sex&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>alignment_id</th>
      <th>log.qc.score.cr6</th>
      <th>synchronized_age</th>
      <th>donor_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>donor_species</th>
      <th>...</th>
      <th>synchronized_age_color</th>
      <th>synchronized_age_order</th>
      <th>age_bin_color</th>
      <th>age_bin_order</th>
      <th>species_genus_color</th>
      <th>species_genus_order</th>
      <th>species_scientific_name_color</th>
      <th>species_scientific_name_order</th>
      <th>donor_sex_color</th>
      <th>donor_sex_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAACCCAAGGATTTAG-898_A02</th>
      <td>AAACCCAAGGATTTAG</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>338.986629</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>#E6865C</td>
      <td>3</td>
      <td>#D38F38</td>
      <td>2</td>
      <td>#c941a7</td>
      <td>1</td>
      <td>#ffa300</td>
      <td>1</td>
      <td>#ADC4C3</td>
      <td>2</td>
    </tr>
    <tr>
      <th>AAACCCACACATTCTT-898_A02</th>
      <td>AAACCCACACATTCTT</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>425.642457</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>#E6865C</td>
      <td>3</td>
      <td>#D38F38</td>
      <td>2</td>
      <td>#c941a7</td>
      <td>1</td>
      <td>#ffa300</td>
      <td>1</td>
      <td>#ADC4C3</td>
      <td>2</td>
    </tr>
    <tr>
      <th>AAACCCACACTTACAG-898_A02</th>
      <td>AAACCCACACTTACAG</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>342.366675</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>#E6865C</td>
      <td>3</td>
      <td>#D38F38</td>
      <td>2</td>
      <td>#c941a7</td>
      <td>1</td>
      <td>#ffa300</td>
      <td>1</td>
      <td>#ADC4C3</td>
      <td>2</td>
    </tr>
    <tr>
      <th>AAACGAATCACTCTTA-898_A02</th>
      <td>AAACGAATCACTCTTA</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>353.204604</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>#E6865C</td>
      <td>3</td>
      <td>#D38F38</td>
      <td>2</td>
      <td>#c941a7</td>
      <td>1</td>
      <td>#ffa300</td>
      <td>1</td>
      <td>#ADC4C3</td>
      <td>2</td>
    </tr>
    <tr>
      <th>AAACGAATCGACCAAT-898_A02</th>
      <td>AAACGAATCGACCAAT</td>
      <td>898_A02</td>
      <td>L8TX_211028_01_G04</td>
      <td>1157582505</td>
      <td>355.052147</td>
      <td>E13.5</td>
      <td>C67BL6J-606627</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>#E6865C</td>
      <td>3</td>
      <td>#D38F38</td>
      <td>2</td>
      <td>#c941a7</td>
      <td>1</td>
      <td>#ffa300</td>
      <td>1</td>
      <td>#ADC4C3</td>
      <td>2</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 40 columns</p>
</div></div></div>
</div>
</section>
<section id="umap-spatial-embedding">
<h3>UMAP spatial embedding<a class="headerlink" href="#umap-spatial-embedding" title="Link to this heading">#</a></h3>
<p>Now that we’ve merged our donor and library metadata into the main cells data, our next step is to plot these values in the Uniform Manifold Approximation and Projection (UMAP) for cells in the dataset. The UMAP is a dimension reduction technique that can be used for visualizing and exploring large-dimension datasets.</p>
<p>Below we load this 2-D embedding for a sub selection of our cells and merge the x-y coordinates into the extended cell metadata we are creating.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_2d_embedding_coordinates</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_2d_embedding_coordinates&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">cell_2d_embedding_coordinates</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_2d_embedding_coordinates.csv: 100%|██████████| 25.7M/25.7M [00:00&lt;00:00, 31.2MMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>x</th>
      <th>y</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAACCCAAGGATTTAG-898_A02</th>
      <td>17.504883</td>
      <td>4.327579</td>
    </tr>
    <tr>
      <th>AAACCCACACATTCTT-898_A02</th>
      <td>9.755900</td>
      <td>5.404129</td>
    </tr>
    <tr>
      <th>AAACCCACACTTACAG-898_A02</th>
      <td>10.902691</td>
      <td>4.540480</td>
    </tr>
    <tr>
      <th>AAACGAATCACTCTTA-898_A02</th>
      <td>13.066967</td>
      <td>4.577918</td>
    </tr>
    <tr>
      <th>AAACGAATCGACCAAT-898_A02</th>
      <td>18.801212</td>
      <td>-2.231872</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cell_2d_embedding_coordinates</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">sample</span><span class="p">(</span><span class="n">frac</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span> <span class="c1"># shuffle the rows for plotting purposes</span>

<span class="k">del</span> <span class="n">cell_2d_embedding_coordinates</span>
</pre></div>
</div>
</div>
</div>
<p>We define a small helper function <em>plot_umap</em> to visualize the cells on the UMAP. In the examples below we will plot associated cell information colorized by donor age, sex, and region of interest.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_umap</span><span class="p">(</span>
    <span class="n">xx</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">,</span>
    <span class="n">yy</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">,</span>
    <span class="n">cc</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">val</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
    <span class="n">cmap</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Colormap</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">labels</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">term_orders</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">colorbar</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
    <span class="n">sizes</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span> <span class="o">=</span> <span class="kc">None</span>
 <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Tuple</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Figure</span><span class="p">,</span> <span class="n">plt</span><span class="o">.</span><span class="n">Axes</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Plot a scatter plot of the UMAP coordinates.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    xx : array-like</span>
<span class="sd">        x-coordinates of the points to plot.</span>
<span class="sd">    yy : array-like</span>
<span class="sd">        y-coordinates of the points to plot.</span>
<span class="sd">    cc : array-like, optional</span>
<span class="sd">        colors of the points to plot. If None, the points will be colored by the values in `val`.</span>
<span class="sd">    val : array-like, optional</span>
<span class="sd">        values of the points to plot. If None, the points will be colored by the values in `cc`.</span>
<span class="sd">    fig_width : float, optional</span>
<span class="sd">        width of the figure in inches. Default is 8.</span>
<span class="sd">    fig_height : float, optional</span>
<span class="sd">        height of the figure in inches. Default is 8.</span>
<span class="sd">    cmap : str, optional</span>
<span class="sd">        colormap to use for coloring the points. If None, the points will be colored by the values in `cc`.</span>
<span class="sd">    labels : array-like, optional</span>
<span class="sd">        labels for the points to plot. If None, no labels will be added to the plot.</span>
<span class="sd">    term_orders : array-like, optional</span>
<span class="sd">        order of the labels for the legend. If None, the labels will be ordered by their appearance in `labels`.</span>
<span class="sd">    colorbar : bool, optional</span>
<span class="sd">        whether to add a colorbar to the plot. Default is False.</span>
<span class="sd">    sizes : array-like, optional</span>
<span class="sd">        sizes of the points to plot. If None, all points will have the same size.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">sizes</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">sizes</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">scatt</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">val</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="n">sizes</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">cc</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">scatt</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">cc</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="n">sizes</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">labels</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="kn">from</span><span class="w"> </span><span class="nn">matplotlib.patches</span><span class="w"> </span><span class="kn">import</span> <span class="n">Rectangle</span>
        <span class="n">unique_label_colors</span> <span class="o">=</span> <span class="p">(</span><span class="n">labels</span> <span class="o">+</span> <span class="s1">&#39;,&#39;</span> <span class="o">+</span> <span class="n">cc</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
        <span class="n">unique_labels</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">label_color</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">label_color</span> <span class="ow">in</span> <span class="n">unique_label_colors</span><span class="p">])</span>
        <span class="n">unique_colors</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">label_color</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">label_color</span> <span class="ow">in</span> <span class="n">unique_label_colors</span><span class="p">])</span>

        <span class="k">if</span> <span class="n">term_orders</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">unique_order</span> <span class="o">=</span> <span class="n">term_orders</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
            <span class="n">term_order</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">argsort</span><span class="p">(</span><span class="n">unique_order</span><span class="p">)</span>
            <span class="n">unique_labels</span> <span class="o">=</span> <span class="n">unique_labels</span><span class="p">[</span><span class="n">term_order</span><span class="p">]</span>
            <span class="n">unique_colors</span> <span class="o">=</span> <span class="n">unique_colors</span><span class="p">[</span><span class="n">term_order</span><span class="p">]</span>
            
        <span class="n">rects</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">color</span> <span class="ow">in</span> <span class="n">unique_colors</span><span class="p">:</span>
            <span class="n">rects</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">),</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fc</span><span class="o">=</span><span class="n">color</span><span class="p">))</span>

        <span class="n">legend</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">unique_labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
        <span class="c1"># ax.add_artist(legend)</span>

    <span class="k">if</span> <span class="n">colorbar</span><span class="p">:</span>
        <span class="n">fig</span><span class="o">.</span><span class="n">colorbar</span><span class="p">(</span><span class="n">scatt</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
    
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<p>Plot the various donor and library metadata available.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;donor_sex_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;donor_sex&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;donor_sex_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;donor_sex&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/1864e20e45ae4c393bdec86da65744e9c77dd75f89292f5ba834339d22b80441.png" src="../_images/1864e20e45ae4c393bdec86da65744e9c77dd75f89292f5ba834339d22b80441.png" />
</div>
</div>
<p>Below we show the region of interest for the cells in the dataset.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;region_of_interest_label_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;region_of_interest_label&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;region_of_interest_label_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;region_of_interest_label&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/4fa50d4ffa4dbac0b4b95529db2c3bc0522a6a4a5ce9704c95b37710810dd41e.png" src="../_images/4fa50d4ffa4dbac0b4b95529db2c3bc0522a6a4a5ce9704c95b37710810dd41e.png" />
</div>
</div>
<p>The UMAP with donor age.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;donor_age_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;donor_age&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;donor_age_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;donor_age&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/dfc5734600dbc4b2fcfe339a37cb1e0f3827a7ddbe3b5f4ea60669bf46b31f04.png" src="../_images/dfc5734600dbc4b2fcfe339a37cb1e0f3827a7ddbe3b5f4ea60669bf46b31f04.png" />
</div>
</div>
</section>
</section>
<section id="taxonomy-information">
<h2>Taxonomy Information<a class="headerlink" href="#taxonomy-information" title="Link to this heading">#</a></h2>
<p>The final set of metadata we load into our extended cell metadata file maps the cells into their assigned cluster in the taxonomy. We additionally load metadata for the clusters and compute useful information, such as the number of cells in each taxon at each level of the taxonomy.</p>
<p>First, we load information associated with the lowest level in the taxonomy in the taxonomy. This includes a useful alias value for each cluster as well as the number of cells in each subcluster.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">:</span> <span class="s1">&#39;Int64&#39;</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">cluster</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster.csv: 100%|██████████| 20.5k/20.5k [00:00&lt;00:00, 210kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_alias</th>
      <th>number_of_cells</th>
    </tr>
    <tr>
      <th>cluster_annotation_term_label</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20260131_SCLU_0001</th>
      <td>1</td>
      <td>77</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_0002</th>
      <td>2</td>
      <td>126</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_0003</th>
      <td>3</td>
      <td>184</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_0004</th>
      <td>4</td>
      <td>186</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_0005</th>
      <td>5</td>
      <td>25</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Next, we load the table that describes the levels in the taxonomy from class at the highest to subcluster at the lowest level.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_annotation_term_set</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term_set&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span>
<span class="n">cluster_annotation_term_set</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_annotation_term_set.csv: 100%|██████████| 268/268 [00:00&lt;00:00, 2.76kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>cluster_annotation_term_label</th>
      <th>description</th>
      <th>order</th>
      <th>parent_term_set_label</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>class</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>0</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>subclass</td>
      <td>CCN20260131_LEVEL_1</td>
      <td>subclass</td>
      <td>1</td>
      <td>CCN20260131_LEVEL_0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>cluster</td>
      <td>CCN20260131_LEVEL_2</td>
      <td>cluster</td>
      <td>2</td>
      <td>CCN20260131_LEVEL_1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>subcluster</td>
      <td>CCN20260131_LEVEL_3</td>
      <td>subcluster</td>
      <td>3</td>
      <td>CCN20260131_LEVEL_2</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>For the subclusters, we load information on the annotations for each subcluster. This also includes the term order and color information which we will use to plot later. Note the inclusion of the <code class="docutils literal notranslate"><span class="pre">CCN20230722_label</span></code> column. This value points to the equivalent taxon in the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/descriptions/WMB-taxonomy.html">Whole Mouse Brain taxonomy</a> if it exists.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_annotation_term</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">cluster_annotation_term</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_annotation_term.csv: 100%|██████████| 129k/129k [00:00&lt;00:00, 1.10MMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>cluster_annotation_term_set_label</th>
      <th>cluster_annotation_term_set_name</th>
      <th>color_hex_triplet</th>
      <th>term_order</th>
      <th>term_set_order</th>
      <th>parent_term_label</th>
      <th>parent_term_name</th>
      <th>parent_term_set_label</th>
      <th>CCN20230722_label</th>
    </tr>
    <tr>
      <th>cluster_annotation_term_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20260131_CLAS_009</th>
      <td>Astro-Epen</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#594a26</td>
      <td>9</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CS20230722_CLAS_30</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_013</th>
      <td>CNU-MGE GABA</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#450099</td>
      <td>13</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CS20230722_CLAS_08</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_002</th>
      <td>CR Glut</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#919900</td>
      <td>2</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_011</th>
      <td>CTX-CGE GABA</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#CCFF33</td>
      <td>11</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CS20230722_CLAS_06</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_012</th>
      <td>CTX-MGE GABA</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#f954ee</td>
      <td>12</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CS20230722_CLAS_07</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_051</th>
      <td>RG_5</td>
      <td>CCN20260131_LEVEL_3</td>
      <td>subcluster</td>
      <td>#9e9ac8</td>
      <td>51</td>
      <td>3</td>
      <td>CS20260131_CLUS_004</td>
      <td>RG</td>
      <td>CCN20260131_LEVEL_2</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_052</th>
      <td>RG_6</td>
      <td>CCN20260131_LEVEL_3</td>
      <td>subcluster</td>
      <td>#bcbddc</td>
      <td>52</td>
      <td>3</td>
      <td>CS20260131_CLUS_004</td>
      <td>RG</td>
      <td>CCN20260131_LEVEL_2</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_053</th>
      <td>RG_7</td>
      <td>CCN20260131_LEVEL_3</td>
      <td>subcluster</td>
      <td>#dadaeb</td>
      <td>53</td>
      <td>3</td>
      <td>CS20260131_CLUS_004</td>
      <td>RG</td>
      <td>CCN20260131_LEVEL_2</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_054</th>
      <td>RG_8</td>
      <td>CCN20260131_LEVEL_3</td>
      <td>subcluster</td>
      <td>#636363</td>
      <td>54</td>
      <td>3</td>
      <td>CS20260131_CLUS_004</td>
      <td>RG</td>
      <td>CCN20260131_LEVEL_2</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>CS20260131_SCLU_055</th>
      <td>RG_9</td>
      <td>CCN20260131_LEVEL_3</td>
      <td>subcluster</td>
      <td>#969696</td>
      <td>55</td>
      <td>3</td>
      <td>CS20260131_CLUS_004</td>
      <td>RG</td>
      <td>CCN20260131_LEVEL_2</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
<p>917 rows × 10 columns</p>
</div></div></div>
</div>
<p>Finally, we load the cluster to cluster annotation membership table. Each row in this table is a mapping between the subclusters and every level of the taxonomy it belongs to, including itself. We’ll use this table in a <code class="docutils literal notranslate"><span class="pre">groupby</span></code> to allow us to count up the number of clusters at each taxonomy level and sum the number of cells in each taxon in the taxonomy a all levels.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_to_cluster_annotation_membership</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_to_cluster_annotation_membership&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">membership_with_cluster_info</span> <span class="o">=</span> <span class="n">cluster_to_cluster_annotation_membership</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">cluster</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)[[</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">]],</span>
    <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span>
<span class="p">)</span>
<span class="n">membership_with_cluster_info</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_annotation_term</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_anno_term&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">membership_groupby</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span>
<span class="p">)</span>
<span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_to_cluster_annotation_membership.csv:   0%|          | 0.00/189k [00:00&lt;?, ?MB/s]
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cluster_to_cluster_annotation_membership.csv: 100%|██████████| 189k/189k [00:00&lt;00:00, 1.45MMB/s] 
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_annotation_term_label</th>
      <th>cluster_annotation_term_set_name</th>
      <th>cluster_annotation_term_name</th>
      <th>cluster_alias</th>
      <th>cluster_annotation_term_set_label</th>
      <th>number_of_cells</th>
      <th>name</th>
      <th>cluster_annotation_term_set_label_anno_term</th>
      <th>cluster_annotation_term_set_name_anno_term</th>
      <th>color_hex_triplet</th>
      <th>term_order</th>
      <th>term_set_order</th>
      <th>parent_term_label</th>
      <th>parent_term_name</th>
      <th>parent_term_set_label</th>
      <th>CCN20230722_label</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>CS20260131_CLAS_001</td>
      <td>class</td>
      <td>NEC</td>
      <td>1</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>77</td>
      <td>NEC</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#6BAED6</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>CS20260131_CLAS_001</td>
      <td>class</td>
      <td>NEC</td>
      <td>2</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>126</td>
      <td>NEC</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#6BAED6</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>CS20260131_CLAS_001</td>
      <td>class</td>
      <td>NEC</td>
      <td>3</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>184</td>
      <td>NEC</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#6BAED6</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>CS20260131_CLAS_001</td>
      <td>class</td>
      <td>NEC</td>
      <td>4</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>186</td>
      <td>NEC</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#6BAED6</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>CS20260131_CLAS_001</td>
      <td>class</td>
      <td>NEC</td>
      <td>5</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>25</td>
      <td>NEC</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#6BAED6</td>
      <td>1</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>From the membership table, we create three tables via a <code class="docutils literal notranslate"><span class="pre">groupby</span></code>. First the name of each cluster and its parents.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># term_sets = abc_cache.get_metadata_dataframe(directory=&#39;WHB-taxonomy&#39;, file_name=&#39;cluster_annotation_term_set&#39;).set_index(&#39;label&#39;)</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">cluster_details</span><span class="p">[</span><span class="n">cluster_annotation_term_set</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span> <span class="c1"># order columns</span>
<span class="n">cluster_details</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="s1">&#39;Other&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_details</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>cluster_annotation_term_set_name</th>
      <th>class</th>
      <th>subclass</th>
      <th>cluster</th>
      <th>subcluster</th>
    </tr>
    <tr>
      <th>cluster_alias</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC_1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC_2</td>
    </tr>
    <tr>
      <th>3</th>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC_3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC_4</td>
    </tr>
    <tr>
      <th>5</th>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC</td>
      <td>NEC_5</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Next the plotting order of each of the taxons and their parents.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_order</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;term_order&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_order</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span>
    <span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;class&#39;</span><span class="p">:</span> <span class="s1">&#39;class_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;subclass&#39;</span><span class="p">:</span> <span class="s1">&#39;subclass_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;cluster&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;subcluster&#39;</span><span class="p">:</span> <span class="s1">&#39;subcluster_order&#39;</span><span class="p">},</span>
    <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">cluster_order</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>cluster_annotation_term_set_name</th>
      <th>class_order</th>
      <th>cluster_order</th>
      <th>subclass_order</th>
      <th>subcluster_order</th>
    </tr>
    <tr>
      <th>cluster_alias</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>4</td>
    </tr>
    <tr>
      <th>5</th>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>5</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Finally, the colors we will use to plot for each of the unique taxons at all levels.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">cluster_colors</span><span class="p">[</span><span class="n">cluster_annotation_term_set</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span>
<span class="n">cluster_colors</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>cluster_annotation_term_set_name</th>
      <th>class</th>
      <th>subclass</th>
      <th>cluster</th>
      <th>subcluster</th>
    </tr>
    <tr>
      <th>cluster_alias</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>#6BAED6</td>
      <td>#50B4F0</td>
      <td>#9e9ac8</td>
      <td>#6baed6</td>
    </tr>
    <tr>
      <th>2</th>
      <td>#6BAED6</td>
      <td>#50B4F0</td>
      <td>#9e9ac8</td>
      <td>#9ecae1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>#6BAED6</td>
      <td>#50B4F0</td>
      <td>#9e9ac8</td>
      <td>#c6dbef</td>
    </tr>
    <tr>
      <th>4</th>
      <td>#6BAED6</td>
      <td>#50B4F0</td>
      <td>#9e9ac8</td>
      <td>#e6550d</td>
    </tr>
    <tr>
      <th>5</th>
      <td>#6BAED6</td>
      <td>#50B4F0</td>
      <td>#9e9ac8</td>
      <td>#fd8d3c</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Next, we bring it all together by loading the mapping of cells to subcluster and join into our final metadata table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_to_cluster_membership</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;Developing-Mouse-Vis-Cortex-taxonomy&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">cell_to_cluster_membership</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_to_cluster_membership.csv:   0%|          | 0.00/28.5M [00:00&lt;?, ?MB/s]
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_to_cluster_membership.csv: 100%|██████████| 28.5M/28.5M [00:01&lt;00:00, 26.1MMB/s]  
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cluster_alias</th>
      <th>label</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAACCCAAGGATTTAG-898_A02</th>
      <td>50</td>
      <td>CS20260131_SCLU_0050</td>
    </tr>
    <tr>
      <th>AAACCCACACATTCTT-898_A02</th>
      <td>122</td>
      <td>CS20260131_SCLU_0122</td>
    </tr>
    <tr>
      <th>AAACCCACACTTACAG-898_A02</th>
      <td>118</td>
      <td>CS20260131_SCLU_0118</td>
    </tr>
    <tr>
      <th>AAACGAATCACTCTTA-898_A02</th>
      <td>108</td>
      <td>CS20260131_SCLU_0108</td>
    </tr>
    <tr>
      <th>AAACGAATCGACCAAT-898_A02</th>
      <td>690</td>
      <td>CS20260131_SCLU_0690</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We merge this table with information from our taxons.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cell_to_cluster_membership</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_cell_to_cluster_membership&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_color&#39;</span><span class="p">)</span>
<span class="n">cell_extended</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_order</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>

<span class="k">del</span> <span class="n">cell_to_cluster_membership</span>

<span class="n">cell_extended</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>cell_barcode</th>
      <th>barcoded_cell_sample_label</th>
      <th>library_label</th>
      <th>alignment_id</th>
      <th>log.qc.score.cr6</th>
      <th>synchronized_age</th>
      <th>donor_label</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>donor_species</th>
      <th>...</th>
      <th>cluster</th>
      <th>subcluster</th>
      <th>class_color</th>
      <th>subclass_color</th>
      <th>cluster_color</th>
      <th>subcluster_color</th>
      <th>class_order</th>
      <th>cluster_order</th>
      <th>subclass_order</th>
      <th>subcluster_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CCTCCAAAGTAAGAGG-482_A06</th>
      <td>CCTCCAAAGTAAGAGG</td>
      <td>482_A06</td>
      <td>L8TX_210107_01_D09</td>
      <td>1157582376</td>
      <td>241.8102</td>
      <td>P21</td>
      <td>Snap25-IRES2-Cre;Ai14-562459</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>5312_Microglia NN_1</td>
      <td>5312_Microglia NN_1_3</td>
      <td>#825f45</td>
      <td>#CC1F4E</td>
      <td>#a55194</td>
      <td>#dadaeb</td>
      <td>15</td>
      <td>146</td>
      <td>39</td>
      <td>712</td>
    </tr>
    <tr>
      <th>GAGGCAACAGCTTTCC-479_A03</th>
      <td>GAGGCAACAGCTTTCC</td>
      <td>479_A03</td>
      <td>L8TX_210107_01_A09</td>
      <td>1157582337</td>
      <td>431.3020</td>
      <td>P56</td>
      <td>Snap25-IRES2-Cre;Ai14-562811</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>741_Pvalb Gaba_3</td>
      <td>741_Pvalb Gaba_3_8</td>
      <td>#f954ee</td>
      <td>#b13667</td>
      <td>#c7e9c0</td>
      <td>#969696</td>
      <td>12</td>
      <td>106</td>
      <td>31</td>
      <td>617</td>
    </tr>
    <tr>
      <th>ATCGGATCACCCATAA-441_B01</th>
      <td>ATCGGATCACCCATAA</td>
      <td>441_B01</td>
      <td>L8TX_201120_01_H09</td>
      <td>1178468076</td>
      <td>457.6961</td>
      <td>P56</td>
      <td>Snap25-IRES2-Cre;Ai14-553679</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>437_L6 CT CTX Glut_1</td>
      <td>437_L6 CT CTX Glut_1_15</td>
      <td>#61e2a4</td>
      <td>#34661F</td>
      <td>#c7e9c0</td>
      <td>#cedb9c</td>
      <td>6</td>
      <td>13</td>
      <td>11</td>
      <td>204</td>
    </tr>
    <tr>
      <th>AAGAACACATTGCCGG-418_C04</th>
      <td>AAGAACACATTGCCGG</td>
      <td>418_C04</td>
      <td>L8TX_201105_01_C02</td>
      <td>1157582427</td>
      <td>416.2293</td>
      <td>P28</td>
      <td>Snap25-IRES2-Cre;Ai14-555355</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>118_L2/3 IT CTX Glut_4</td>
      <td>118_L2/3 IT CTX Glut_4_2</td>
      <td>#FA0087</td>
      <td>#E9530F</td>
      <td>#de9ed6</td>
      <td>#c7e9c0</td>
      <td>7</td>
      <td>45</td>
      <td>18</td>
      <td>453</td>
    </tr>
    <tr>
      <th>GTGAGCCGTCTACACA-634_A03</th>
      <td>GTGAGCCGTCTACACA</td>
      <td>634_A03</td>
      <td>L8TX_210506_01_D07</td>
      <td>1157582451</td>
      <td>370.9672</td>
      <td>P5</td>
      <td>Snap25-IRES2-Cre;Ai14-582040</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>Developing-Mouse-Vis-Cortex-10X</td>
      <td>NCBITaxon:10090</td>
      <td>...</td>
      <td>439_L6 CT CTX Glut_1</td>
      <td>439_L6 CT CTX Glut_1_6</td>
      <td>#61e2a4</td>
      <td>#34661F</td>
      <td>#756bb1</td>
      <td>#d6616b</td>
      <td>6</td>
      <td>14</td>
      <td>11</td>
      <td>211</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 56 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">print_column_info</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of unique cell_barcode = 527176 
Number of unique barcoded_cell_sample_label = 91 
Number of unique library_label = 91 
Number of unique alignment_id = 91 
Number of unique log.qc.score.cr6 = 496566 
Number of unique synchronized_age = 35 
Number of unique donor_label = 53 
Number of unique dataset_label = 1 [&#39;Developing-Mouse-Vis-Cortex-10X&#39;]
Number of unique feature_matrix_label = 1 [&#39;Developing-Mouse-Vis-Cortex-10X&#39;]
Number of unique donor_species = 1 [&#39;NCBITaxon:10090&#39;]
Number of unique species_scientific_name = 1 [&#39;Mus musculus&#39;]
Number of unique species_genus = 1 [&#39;Mouse&#39;]
Number of unique donor_sex = 2 [&#39;Female&#39;, &#39;Male&#39;]
Number of unique donor_age_value = 38 
Number of unique donor_age_unit = 1 [&#39;days&#39;]
Number of unique donor_age_reference_point = 2 [&#39;birth&#39;, &#39;conception&#39;]
Number of unique donor_age = 35 
Number of unique age_bin = 18 [&#39;0_1&#39;, &#39;10&#39;, &#39;11&#39;, &#39;12_13&#39;, &#39;14_15&#39;, &#39;16&#39;, &#39;17_19&#39;, &#39;2&#39;, &#39;20_28&#39;, &#39;3&#39;, &#39;4&#39;, &#39;54_68&#39;, &#39;5_6&#39;, &#39;7_8&#39;, &#39;9&#39;, &#39;E11.5_E12.5&#39;, &#39;E13.5_E16.5&#39;, &#39;E17_E18.5&#39;]
Number of unique library_technique = 2 [&#39;10xV3.1;GEXOnly&#39;, &#39;10xV3;GEXOnly&#39;]
Number of unique barcoded_cell_sample_label_library_table = 91 
Number of unique enrichment_population = 5 [&#39;Calcein-positive, Hoechst-positive&#39;, &#39;Hoechst-positive&#39;, &#39;No FACS&#39;, &#39;RFP-positive, DAPI-negative&#39;, &#39;RFP-positive, Hoechst-positive&#39;]
Number of unique cell_specimen_type = 1 [&#39;Cells&#39;]
Number of unique parcellation_term_identifier = 5 [&#39;MBA:22|MBA:669&#39;, &#39;MBA:343|MBA:567&#39;, &#39;MBA:385&#39;, &#39;MBA:669&#39;, &#39;MBA:997&#39;]
Number of unique region_of_interest_name = 5 [&#39;Cerebrum - Brain stem&#39;, &#39;Primary visual area&#39;, &#39;Visual areas&#39;, &#39;Visual areas - Posterior parietal association areas&#39;, &#39;brain&#39;]
Number of unique region_of_interest_label = 5 [&#39;CH-BS&#39;, &#39;VIS&#39;, &#39;VIS-PTLp&#39;, &#39;VISp&#39;, &#39;WholeBrain&#39;]
Number of unique donor_label_library_table = 53 
Number of unique region_of_interest_label_color = 5 [&#39;#0059CC&#39;, &#39;#08858C&#39;, &#39;#26AEFF&#39;, &#39;#B0F0FF&#39;, &#39;#bebebe&#39;]
Number of unique region_of_interest_label_order = 5 [1, 3, 4, 5, 6]
Number of unique donor_age_color = 35 
Number of unique donor_age_order = 35 
Number of unique synchronized_age_color = 35 
Number of unique synchronized_age_order = 35 
Number of unique age_bin_color = 18 [&#39;#00CC00&#39;, &#39;#00D4E6&#39;, &#39;#0F0F99&#39;, &#39;#18F2E1&#39;, &#39;#2C49AD&#39;, &#39;#31CCA0&#39;, &#39;#34D916&#39;, &#39;#4EAD73&#39;, &#39;#5A93C6&#39;, &#39;#6B49F9&#39;, &#39;#73BF7F&#39;, &#39;#7FEC3C&#39;, &#39;#82C5D9&#39;, &#39;#AE43C6&#39;, &#39;#BCD233&#39;, &#39;#C30000&#39;, &#39;#D38F38&#39;, &#39;#F99389&#39;]
Number of unique age_bin_order = 18 [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18]
Number of unique species_genus_color = 1 [&#39;#c941a7&#39;]
Number of unique species_genus_order = 1 [1]
Number of unique species_scientific_name_color = 1 [&#39;#ffa300&#39;]
Number of unique species_scientific_name_order = 1 [1]
Number of unique donor_sex_color = 2 [&#39;#565353&#39;, &#39;#ADC4C3&#39;]
Number of unique donor_sex_order = 2 [1, 2]
Number of unique x = 564967 
Number of unique y = 565338 
Number of unique cluster_alias = 714 
Number of unique label = 714 
Number of unique class = 15 [&#39;Astro-Epen&#39;, &#39;CNU-MGE GABA&#39;, &#39;CR Glut&#39;, &#39;CTX-CGE GABA&#39;, &#39;CTX-MGE GABA&#39;, &#39;Glioblast&#39;, &#39;IMN&#39;, &#39;IP&#39;, &#39;IT Glut&#39;, &#39;Immune&#39;, &#39;NEC&#39;, &#39;OPC-Oligo&#39;, &#39;RG&#39;, &#39;Vascular&#39;, &#39;nonIT Glut&#39;]
Number of unique subclass = 40 
Number of unique cluster = 148 
Number of unique subcluster = 714 
Number of unique class_color = 15 [&#39;#03045E&#39;, &#39;#16f2f2&#39;, &#39;#450099&#39;, &#39;#594a26&#39;, &#39;#61e2a4&#39;, &#39;#637939&#39;, &#39;#6BAED6&#39;, &#39;#825f45&#39;, &#39;#858881&#39;, &#39;#919900&#39;, &#39;#CCFF33&#39;, &#39;#CEDB9C&#39;, &#39;#FA0087&#39;, &#39;#FDAE6B&#39;, &#39;#f954ee&#39;]
Number of unique subclass_color = 40 
Number of unique cluster_color = 49 
Number of unique subcluster_color = 40 
Number of unique class_order = 15 [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
Number of unique cluster_order = 148 
Number of unique subclass_order = 40 
Number of unique subcluster_order = 714 
</pre></div>
</div>
</div>
</div>
<section id="plotting-the-taxonomy">
<h3>Plotting the taxonomy<a class="headerlink" href="#plotting-the-taxonomy" title="Link to this heading">#</a></h3>
<p>Now that we have our cells with associated taxonomy information, we’ll plot them into the UMAP we showed previously.</p>
<p>Below we plot the taxonomy mapping of the cells for each level in the taxonomy.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;class_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;class&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;class_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;class&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/b5733afe7f47adc898bde17879d8f704d2002aa25925ec0560f006e2a3e5be88.png" src="../_images/b5733afe7f47adc898bde17879d8f704d2002aa25925ec0560f006e2a3e5be88.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;subclass_color&#39;</span><span class="p">],</span>
    <span class="n">labels</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;subclass&#39;</span><span class="p">],</span>
    <span class="n">term_orders</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;subclass_order&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;subclass&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/f30603d5232aa7e2894d5c4c86f1cc1a8a053e942956eecb1784dc28798f7c79.png" src="../_images/f30603d5232aa7e2894d5c4c86f1cc1a8a053e942956eecb1784dc28798f7c79.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;cluster_color&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">12</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;cluster&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/cc9f4c44b4294a1430f2c30c70bc34e6cae3e041e454e4d6193adc6ee489961c.png" src="../_images/cc9f4c44b4294a1430f2c30c70bc34e6cae3e041e454e4d6193adc6ee489961c.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_umap</span><span class="p">(</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span>
    <span class="n">cc</span><span class="o">=</span><span class="n">cell_extended</span><span class="p">[</span><span class="s1">&#39;subcluster_color&#39;</span><span class="p">],</span>
    <span class="n">fig_width</span><span class="o">=</span><span class="mi">18</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">18</span>
<span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;subcluster&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/06445a29af185e80408229f39db80c279fa79ec57f3886a7f5987dfd6c2af092.png" src="../_images/06445a29af185e80408229f39db80c279fa79ec57f3886a7f5987dfd6c2af092.png" />
</div>
</div>
</section>
<section id="aggregating-cluster-and-cells-counts-per-term-per-species">
<h3>Aggregating cluster and cells counts per term per species.<a class="headerlink" href="#aggregating-cluster-and-cells-counts-per-term-per-species" title="Link to this heading">#</a></h3>
<p>Let’s investigate the taxonomy information a bit more. In this section, we’ll create bar plots showing the number of subclusters and cells at each level in the taxonomy.</p>
<p>First, we need to compute the number of subclusters that are in each of the cell type taxons above it. This is accomplished by a simple groupby in Pandas.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">term_cluster_count</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
        <span class="p">[</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">]</span>
    <span class="p">)[[</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
<span class="n">term_cluster_count</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;number_of_subclusters&#39;</span><span class="p">]</span>
<span class="n">term_cluster_count</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>number_of_subclusters</th>
    </tr>
    <tr>
      <th>cluster_annotation_term_label</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20260131_CLAS_001</th>
      <td>26</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_002</th>
      <td>20</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_003</th>
      <td>30</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_004</th>
      <td>29</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_005</th>
      <td>56</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">term_cell_count</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">]</span>
<span class="p">)[[</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">term_cell_count</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>number_of_cells</th>
    </tr>
    <tr>
      <th>cluster_annotation_term_label</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20260131_CLAS_001</th>
      <td>4605</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_002</th>
      <td>1611</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_003</th>
      <td>5656</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_004</th>
      <td>4957</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_005</th>
      <td>31718</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Join counts with the term dataframe</span>
<span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">cluster_annotation_term</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">term_cluster_count</span><span class="p">)</span>
<span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">term_cell_count</span><span class="p">)</span>
<span class="n">term_with_counts</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>cluster_annotation_term_set_label</th>
      <th>cluster_annotation_term_set_name</th>
      <th>color_hex_triplet</th>
      <th>term_order</th>
      <th>term_set_order</th>
      <th>parent_term_label</th>
      <th>parent_term_name</th>
      <th>parent_term_set_label</th>
      <th>CCN20230722_label</th>
      <th>number_of_subclusters</th>
      <th>number_of_cells</th>
    </tr>
    <tr>
      <th>cluster_annotation_term_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CS20260131_CLAS_009</th>
      <td>Astro-Epen</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#594a26</td>
      <td>9</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CS20230722_CLAS_30</td>
      <td>25</td>
      <td>32116</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_013</th>
      <td>CNU-MGE GABA</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#450099</td>
      <td>13</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CS20230722_CLAS_08</td>
      <td>3</td>
      <td>208</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_002</th>
      <td>CR Glut</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#919900</td>
      <td>2</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20</td>
      <td>1611</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_011</th>
      <td>CTX-CGE GABA</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#CCFF33</td>
      <td>11</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CS20230722_CLAS_06</td>
      <td>52</td>
      <td>17567</td>
    </tr>
    <tr>
      <th>CS20260131_CLAS_012</th>
      <td>CTX-MGE GABA</td>
      <td>CCN20260131_LEVEL_0</td>
      <td>class</td>
      <td>#f954ee</td>
      <td>12</td>
      <td>0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CS20230722_CLAS_07</td>
      <td>91</td>
      <td>38296</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Below we create a function to plot the subcluster and cell counts in a bar graph, coloring by the associated taxon level.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">bar_plot_by_level_and_type</span><span class="p">(</span><span class="n">df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">level</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">fig_width</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mf">8.5</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">4</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Plot the number of cells by the specified level.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    df : pd.DataFrame</span>
<span class="sd">        DataFrame containing cluster annotation terms with counts.</span>
<span class="sd">    level : str</span>
<span class="sd">        The level of the taxonomy to plot (e.g., &#39;Neighborhood&#39;, &#39;Class&#39;, &#39;Subclass&#39;, &#39;Group&#39;, &#39;Cluster&#39;).</span>
<span class="sd">    fig_width : float, optional</span>
<span class="sd">        Width of the figure in inches. Default is 8.5.</span>
<span class="sd">    fig_height : float, optional</span>
<span class="sd">        Height of the figure in inches. Default is 4.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">ctype</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="s1">&#39;subclusters&#39;</span><span class="p">,</span> <span class="s1">&#39;cells&#39;</span><span class="p">]):</span>

        <span class="n">pred</span> <span class="o">=</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">level</span><span class="p">)</span>
        <span class="n">sort_order</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">argsort</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">pred</span><span class="p">][</span><span class="s1">&#39;term_order&#39;</span><span class="p">])</span>
        <span class="n">names</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">pred</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">sort_order</span><span class="p">]</span>
        <span class="n">counts</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">pred</span><span class="p">][</span><span class="s1">&#39;number_of_</span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">ctype</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">sort_order</span><span class="p">]</span>
        <span class="n">colors</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">pred</span><span class="p">][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">sort_order</span><span class="p">]</span>
        
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">barh</span><span class="p">(</span><span class="n">names</span><span class="p">,</span> <span class="n">counts</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">colors</span><span class="p">)</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Number of </span><span class="si">%s</span><span class="s1"> by </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">ctype</span><span class="p">,</span><span class="n">level</span><span class="p">))</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Number of </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">ctype</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">ctype</span> <span class="o">==</span> <span class="s1">&#39;cells&#39;</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
        
        <span class="k">if</span> <span class="n">idx</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">([])</span>

    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<p>Now let’s plot the counts the taxonomy levels class and subclass.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">bar_plot_by_level_and_type</span><span class="p">(</span><span class="n">term_with_counts</span><span class="p">,</span> <span class="s1">&#39;class&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/ac1f3c46fe2f59e6d2bdca71f362d78a0db7a4ee10618b3e28dabcc9f39b0186.png" src="../_images/ac1f3c46fe2f59e6d2bdca71f362d78a0db7a4ee10618b3e28dabcc9f39b0186.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">bar_plot_by_level_and_type</span><span class="p">(</span>
    <span class="n">term_with_counts</span><span class="p">,</span>
    <span class="s1">&#39;subclass&#39;</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="o">=</span><span class="mi">8</span>
<span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/1dcddc8e4447d14a1a368e0a47593e18017cbf6bbc77649c050c5d86076e22c7.png" src="../_images/1dcddc8e4447d14a1a368e0a47593e18017cbf6bbc77649c050c5d86076e22c7.png" />
</div>
</div>
</section>
<section id="cluster-and-subcluster-count-vs-donor-age">
<h3>Cluster and subcluster count vs donor age.<a class="headerlink" href="#cluster-and-subcluster-count-vs-donor-age" title="Link to this heading">#</a></h3>
<p>Below we compute the number of subclusters and clusters for each of the unique values of age in the dataset. We again compute this by Pandas groupbys.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">age_per_subcluster</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
        <span class="p">[</span><span class="s1">&#39;donor_age_order&#39;</span><span class="p">]</span>
    <span class="p">)[[</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span>
<span class="n">age_per_subcluster</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;number_of_subclusters&#39;</span><span class="p">]</span>
<span class="n">age_per_subcluster</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>number_of_subclusters</th>
    </tr>
    <tr>
      <th>donor_age_order</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>65</td>
    </tr>
    <tr>
      <th>2</th>
      <td>68</td>
    </tr>
    <tr>
      <th>3</th>
      <td>150</td>
    </tr>
    <tr>
      <th>4</th>
      <td>132</td>
    </tr>
    <tr>
      <th>6</th>
      <td>144</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">age_per_cluster</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
        <span class="p">[</span><span class="s1">&#39;donor_age_order&#39;</span><span class="p">]</span>
    <span class="p">)[[</span><span class="s1">&#39;cluster&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span>
<span class="n">age_per_cluster</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;number_of_clusters&#39;</span><span class="p">]</span>
<span class="n">age_per_cluster</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>number_of_clusters</th>
    </tr>
    <tr>
      <th>donor_age_order</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>12</td>
    </tr>
    <tr>
      <th>2</th>
      <td>13</td>
    </tr>
    <tr>
      <th>3</th>
      <td>28</td>
    </tr>
    <tr>
      <th>4</th>
      <td>26</td>
    </tr>
    <tr>
      <th>6</th>
      <td>26</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We finally plot the number of (sub)clusters for each unique age, showing that both are increasing as a function of age.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">age_values</span> <span class="o">=</span> <span class="n">value_sets</span><span class="p">[</span>
    <span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;synchronized_age&#39;</span>
<span class="p">]</span>

<span class="n">figure</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">8</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span>
    <span class="nb">range</span><span class="p">(</span><span class="mi">35</span><span class="p">),</span> <span class="n">age_per_subcluster</span><span class="p">[</span><span class="s1">&#39;number_of_subclusters&#39;</span><span class="p">],</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;-&#39;</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;o&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;number of subclusters&#39;</span>
<span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span>
    <span class="nb">range</span><span class="p">(</span><span class="mi">35</span><span class="p">),</span> <span class="n">age_per_cluster</span><span class="p">[</span><span class="s1">&#39;number_of_clusters&#39;</span><span class="p">],</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;-&#39;</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;o&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;number of clusters&#39;</span>
<span class="p">)</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">35</span><span class="p">))</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">age_values</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;order&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">age_per_cluster</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="s1">&#39;label&#39;</span><span class="p">],</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;synchronized_age&#39;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;count&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s2">&quot;Number of clusters and subclusters by synchronized_age&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;matplotlib.legend.Legend at 0x169890590&gt;
</pre></div>
</div>
<img alt="../_images/d73adb88d410fe6b624b9b55c788d6f4dbf4355a805284498effaadb0565d1ef.png" src="../_images/d73adb88d410fe6b624b9b55c788d6f4dbf4355a805284498effaadb0565d1ef.png" />
</div>
</div>
</section>
<section id="visualizing-the-developmental-visual-cortex-taxonomy">
<h3>Visualizing the developmental visual cortex taxonomy<a class="headerlink" href="#visualizing-the-developmental-visual-cortex-taxonomy" title="Link to this heading">#</a></h3>
<p>Term sets: class, subclass, cluster, and subcluster define the Developing Mouse, Visual Cortex taxonomy. We can visualize the taxonomy as a sunburst diagram that shows the single inheritance hierarchy through a series of rings, that are sliced for each annotation term. Each ring corresponds to a level in the hierarchy. We have ordered the rings so that the class level. Rings are divided based on their hierarchical relationship to the parent slice.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">levels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;class&#39;</span><span class="p">,</span> <span class="s1">&#39;subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster&#39;</span><span class="p">,</span> <span class="s1">&#39;subcluster&#39;</span><span class="p">]</span>
<span class="n">df</span> <span class="o">=</span> <span class="p">{}</span>

<span class="c1"># Copy the term order of the parent into each of the level below it.</span>
<span class="k">if</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">name</span> <span class="o">!=</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">:</span>
    <span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">term_with_counts</span><span class="p">[</span><span class="s1">&#39;parent_order&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;parent_term_label&#39;</span><span class="p">]):</span>
        <span class="k">continue</span>
    <span class="n">term_with_counts</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">idx</span><span class="p">,</span> <span class="s1">&#39;parent_order&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;parent_term_label&#39;</span><span class="p">]][</span><span class="s1">&#39;term_order&#39;</span><span class="p">]</span>

<span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="k">for</span> <span class="n">lvl</span> <span class="ow">in</span> <span class="n">levels</span><span class="p">:</span>
    <span class="n">pred</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">lvl</span>
    <span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">]</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="p">[</span><span class="n">pred</span><span class="p">]</span>
    <span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;parent_order&#39;</span><span class="p">,</span> <span class="s1">&#39;term_order&#39;</span><span class="p">])</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">10</span><span class="p">)</span>
<span class="n">size</span> <span class="o">=</span> <span class="mf">0.15</span>

<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">lvl</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">levels</span><span class="p">):</span>
    
    <span class="k">if</span> <span class="n">lvl</span> <span class="o">==</span> <span class="s1">&#39;class&#39;</span><span class="p">:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">pie</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;number_of_subclusters&#39;</span><span class="p">],</span>
               <span class="n">colors</span><span class="o">=</span><span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">],</span>
               <span class="n">labels</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">],</span>
               <span class="n">rotatelabels</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
               <span class="n">labeldistance</span><span class="o">=</span><span class="mf">1.025</span><span class="p">,</span>
               <span class="n">radius</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
               <span class="n">wedgeprops</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">width</span><span class="o">=</span><span class="n">size</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="kc">None</span><span class="p">),</span>
               <span class="n">startangle</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="k">else</span> <span class="p">:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">pie</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;number_of_subclusters&#39;</span><span class="p">],</span>
               <span class="n">colors</span><span class="o">=</span><span class="n">df</span><span class="p">[</span><span class="n">lvl</span><span class="p">][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">],</span>
               <span class="n">radius</span><span class="o">=</span><span class="mi">1</span><span class="o">-</span><span class="n">i</span><span class="o">*</span><span class="n">size</span><span class="p">,</span>
               <span class="n">wedgeprops</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">width</span><span class="o">=</span><span class="n">size</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="kc">None</span><span class="p">),</span>
               <span class="n">startangle</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">term_with_counts</span> <span class="o">=</span> <span class="n">term_with_counts</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/b4c32f3331c68094a2297cce54d4ac05b434d651db16f6913ea22d3a7ce3f52f.png" src="../_images/b4c32f3331c68094a2297cce54d4ac05b434d651db16f6913ea22d3a7ce3f52f.png" />
</div>
</div>
<p>In the next tutorial, we show how to access and use <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/dev_mouse_vis_cortex_10X_scRNASeq_tutorial.html">Deveolping Mouse gene expression data</a>.</p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="../descriptions/Dev-Mouse-Vis-Cortex-notebooks.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Developing Mouse - Visual Cortex notebooks</p>
      </div>
    </a>
    <a class="right-next"
       href="dev_mouse_vis_cortex_10X_scRNASeq.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Developing Mouse - Visual Cortex: 10X gene expression</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#donor-and-library-metadata">Donor and Library metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#adding-color-and-feature-order">Adding color and feature order</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#umap-spatial-embedding">UMAP spatial embedding</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxonomy-information">Taxonomy Information</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-taxonomy">Plotting the taxonomy</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#aggregating-cluster-and-cells-counts-per-term-per-species">Aggregating cluster and cells counts per term per species.</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cluster-and-subcluster-count-vs-donor-age">Cluster and subcluster count vs donor age.</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualizing-the-developmental-visual-cortex-taxonomy">Visualizing the developmental visual cortex taxonomy</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>