
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation &#8212; Allen Brain Cell Atlas - Data Access</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=9c3e77be" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/simple_style.css?v=3aaed5dc" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=1ae7504c"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/hmba_bg_spatial_slabs_and_taxonomy';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression" href="hmba_bg_spatial_genes.html" />
    <link rel="prev" title="Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression" href="hmba_bg_10X_snRNASeq_tutorial.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/allen_logo.jpeg" class="logo__image only-light" alt="Allen Brain Cell Atlas - Data Access - Home"/>
    <script>document.write(`<img src="../_static/allen_logo.jpeg" class="logo__image only-dark" alt="Allen Brain Cell Atlas - Data Access - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Allen Brain Cell Atlas - Data Access
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">INTRODUCTION</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="general_accessing_10x_snRNASeq_tutorial.html">Accessing 10x RNA-seq gene expression data</a></li>
<li class="toctree-l1"><a class="reference internal" href="abc_atlas_selection_example.html">Using cells selected in the ABC Atlas with Gene Expression data.</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">DATA</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WMB_dataset.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv2.html">Whole mouse brain 10Xv2 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv3.html">Whole mouse brain 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10XMulti.html">Whole mouse brain 10XMulti single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10X.html">Clustering analysis of whole adult mouse brain 10X single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-taxonomy.html">Whole adult mouse brain taxonomy of cell types</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-imputed.html">Imputed MERFISH spatial transcriptomics of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-neighborhoods.html">Cluster groups and neighborhood specific UMAP embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Allen-CCF-2020.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-CCF.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_dataset.html">Aging Mouse dataset (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_10Xv3.html">Zeng Aging Mouse 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_taxonomy.html">Zeng Aging Mouse clustering and mapping to the WMB taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zhuang_dataset.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-1.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-2.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-3.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-3)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-4.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-4)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-dataset.html">Consensus Whole Mouse Brain dataset (Macosko, Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-10X.html">Consensus Whole Mouse Brain single cell/nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-taxonomy.html">Consensus Whole Mouse Brain integrated taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_dataset.html">Human whole-brain transcriptomic cell type atlas (Kimberly Siletti)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10Xv3.html">Human brain 10Xv3 single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10X-clustering.html">Clustering analysis of whole human brain 10X single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-taxonomy.html">Human brain taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_dataset.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-10X.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-MapMyCells.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain (WHB) and Seattle Alzheimer’s Disease Brain Cell Atlas (SEA-AD) Taxonomies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_dataset.html">Human-Mammalian Brain - Basal Ganglia: Data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-BG-taxonomy-CCN20250428.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq data: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG-Aligned.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Aligned gene expression across species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Individual species gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Spatial-BG.html">Human-Mammalian Brain - Basal Ganglia: spatial transcriptomics cells for each species</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">NOTEBOOKS</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cluster_annotation_tutorial.html">10x RNA-seq clustering analysis and annotation (CCN20230722)</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle2.html">10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_1.html">10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2a.html">10x RNA-seq gene expression data (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2b.html">10x RNA-seq gene expression data (part 2b)</a></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle3.html">MERFISH whole brain spatial transcriptomics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_1.html">MERFISH whole brain spatial transcriptomics (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2a.html">MERFISH whole brain spatial transcriptomics (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2b.html">MERFISH whole brain spatial transcriptomics (part 2b)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_imputed_genes_example.html">MERFISH whole brain spatial transcriptomics cells with imputed genes</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_groups_and_embeddings_tutorial.html">Cluster neighborhoods and embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_neighborhood_gallery.html">Cell type neighborhood gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="ccf_and_parcellation_annotation_tutorial.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="merfish_ccf_registration_tutorial.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-notebooks.html">Consensus Mouse notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_clustering_analysis_and_annotation.html">Consensus Whole Mouse Brain #1: Clustering and Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_10X_snRNASeq.html">Consensus Whole Mouse Brain #2: 10X gene expression</a></li>

</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_notebooks.html">Aging Mouse Notebooks (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_clustering_analysis_and_annotation.html">Mouse Aging clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_10x_snRNASeq_tutorial.html">Mouse Aging 10X RNA-seq gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle4.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="zhuang_merfish_tutorial.html">MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_notebooks.html">Whole Human Brain 10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="WHB_cluster_annotation_tutorial.html">Whole Human Brain 10x RNA-seq clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_1.html">Whole Human Brain (WHB) 10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_2.html">Whole Human Brain 10x RNA-seq gene expression data (part 2)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_notebooks.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_data_and_metadata.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_siletti_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_seaad_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_gene_expression.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Gene Expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../descriptions/HMBA-BG_notebooks.html">Human-Mammalian Brain - Basal Ganglia: Notebooks</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_clustering_analysis_and_annotation.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_10X_snRNASeq_tutorial.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_genes.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">ADDITIONAL RESOURCES</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../docs/README.html">Resource Pages</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../docs/abbreviation_terms/README.html">Abbreviation lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/cluster_annotations/README.html">Cluster annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/gene_lists/README.html">Gene lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/parcellation_annotations/README.html">Parcellation annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/supplemental_metadata_descriptions/README.html">Supplemental Metadata Descriptions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/neuroglancer_hmba/README.html">HMBA Basal Ganglia: Neuroglancer Views</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access/issues/new?title=Issue%20on%20page%20%2Fnotebooks/hmba_bg_spatial_slabs_and_taxonomy.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/hmba_bg_spatial_slabs_and_taxonomy.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data Overview</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#specimen-metadata">Specimen metadata</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#spatial-slab-plane-coordinates">Spatial, slab-plane coordinates</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#geojson-polygons">Geojson polygons</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#manual-anatomical-annotations">Manual anatomical annotations</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-slabs">Plotting the Slabs</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#displaying-the-manual-anatomical-annotations">Displaying the manual anatomical annotations.</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxonomy-mapping">Taxonomy mapping</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-taxonomy">Plotting the taxonomy</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#selecting-cell-types">Selecting cell types</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="human-mammalian-brain-basal-ganglia-spatial-transcriptomic-slab-coordinates-and-annotation">
<h1>Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation<a class="headerlink" href="#human-mammalian-brain-basal-ganglia-spatial-transcriptomic-slab-coordinates-and-annotation" title="Link to this heading">#</a></h1>
<p>The basal ganglia are a set of subcortical structures critical for motor control, particularly in the context of action selection, motor learning and emotional state, whose coarse functional organization is well-described in the literature. Although single-cell RNA-Seq studies have greatly enhanced our understanding of cellular diversity in the brain, the exact spatial distribution of cell types within the brain has been difficult to determine. Using spatial transcriptomics platforms such as MERSCOPE or Xenium enables spatial profiling of hundreds of genes for each cell, enabling mapping of the cells to the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_clustering_analysis_and_annotation.html">consensus basal ganglia cell type taxonomy</a>.</p>
<p>In this notebook, we walk through the data available in this release and show how they are joined together to enable analyses at the cell level.</p>
<p>You need to be connected to the internet to run this notebook and have run through the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html">getting started notebook</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">geojson</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">typing</span><span class="w"> </span><span class="kn">import</span> <span class="n">List</span><span class="p">,</span> <span class="n">Tuple</span><span class="p">,</span> <span class="n">Optional</span>

<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">abc_atlas_access.abc_atlas_cache.abc_project_cache</span><span class="w"> </span><span class="kn">import</span> <span class="n">AbcProjectCache</span>
</pre></div>
</div>
</div>
</div>
<p>We will interact with the data using the <strong>AbcProjectCache</strong>. This cache object tracks which data has been downloaded and serves the path to the requested data on disk. For metadata, the cache can also directly serve up a Pandas DataFrame. See the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html"><code class="docutils literal notranslate"><span class="pre">getting_started</span></code></a> notebook for more details on using the cache including installing it if it has not already been.</p>
<p><strong>Change the download_base variable to where you would like to download the data in your system.</strong></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">download_base</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">&#39;../../data/abc_atlas&#39;</span><span class="p">)</span>
<span class="n">abc_cache</span> <span class="o">=</span> <span class="n">AbcProjectCache</span><span class="o">.</span><span class="n">from_cache_dir</span><span class="p">(</span>
    <span class="n">download_base</span><span class="p">,</span>
<span class="p">)</span>

<span class="n">abc_cache</span><span class="o">.</span><span class="n">current_manifest</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;releases/20251031/manifest.json&#39;
</pre></div>
</div>
</div>
</div>
<section id="data-overview">
<h2>Data Overview<a class="headerlink" href="#data-overview" title="Link to this heading">#</a></h2>
<p>Below we list the available metadata and expression matrix files available for each species.</p>
<p>The various species in this dataset are divided up by their donor_id and transcriptomic platform. MERSCOPE for Human and Macaque, Xenium for Marmoset.</p>
<p>The mapping between donor to species is:</p>
<ul class="simple">
<li><p>Human: H22.30.001</p></li>
<li><p>Macaque: QM23.50.001</p></li>
<li><p>Marmoset: CJ23.56.004</p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">dataset_name</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span> <span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span> <span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">]:</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Dataset: </span><span class="si">{</span><span class="n">dataset_name</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">metadata_files</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_metadata_files</span><span class="p">(</span><span class="n">dataset_name</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Metadata files: </span><span class="si">{</span><span class="n">metadata_files</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">expression_matrix_files</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">list_expression_matrix_files</span><span class="p">(</span><span class="n">dataset_name</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Expression matrix files: </span><span class="si">{</span><span class="n">expression_matrix_files</span><span class="si">}</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Dataset: HMBA-MERSCOPE-H22.30.001-BG
Metadata files: [&#39;cell_anatomical_annotations&#39;, &#39;cell_metadata&#39;, &#39;cell_supplemental_metadata&#39;, &#39;cell_to_cluster_membership&#39;, &#39;donor&#39;, &#39;example_gene_expression&#39;, &#39;gene&#39;, &#39;mmc_results&#39;, &#39;slab_plane_coordinates&#39;, &#39;specimen_metadata&#39;, &#39;value_sets&#39;, &#39;visualization&#39;]
Expression matrix files: [&#39;MERSCOPE-H22.30.001-BG/log2&#39;, &#39;MERSCOPE-H22.30.001-BG/raw&#39;]

Dataset: HMBA-MERSCOPE-QM23.50.001-BG
Metadata files: [&#39;cell_anatomical_annotations&#39;, &#39;cell_metadata&#39;, &#39;cell_supplemental_metadata&#39;, &#39;cell_to_cluster_membership&#39;, &#39;donor&#39;, &#39;example_gene_expression&#39;, &#39;gene&#39;, &#39;mmc_results&#39;, &#39;slab_plane_coordinates&#39;, &#39;specimen_metadata&#39;, &#39;value_sets&#39;, &#39;visualization&#39;]
Expression matrix files: [&#39;MERSCOPE-QM23.50.001-BG/log2&#39;, &#39;MERSCOPE-QM23.50.001-BG/raw&#39;]

Dataset: HMBA-Xenium-CJ23.56.004-BG
Metadata files: [&#39;cell_anatomical_annotations&#39;, &#39;cell_metadata&#39;, &#39;cell_supplemental_metadata&#39;, &#39;cell_to_cluster_membership&#39;, &#39;donor&#39;, &#39;example_gene_expression&#39;, &#39;gene&#39;, &#39;mmc_results&#39;, &#39;slab_plane_coordinates&#39;, &#39;specimen_metadata&#39;, &#39;value_sets&#39;, &#39;visualization&#39;]
Expression matrix files: [&#39;Xenium-CJ23.56.004-BG/log2&#39;, &#39;Xenium-CJ23.56.004-BG/raw&#39;]
</pre></div>
</div>
</div>
</div>
</section>
<section id="cell-metadata">
<h2>Cell metadata<a class="headerlink" href="#cell-metadata" title="Link to this heading">#</a></h2>
<p>Each species in the HMBA spatial dataset has it’s own h5ad files and associated cell metadata tables. We load each of these in turn starting from Human to Macaque to Marmoset. These metadata contain the basic identifiers for each cell and forms the basis by which we will join all other cell level and associated data. Below we load the cell metadata for each species.</p>
<p>Load the Human cells.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_metadata.csv: 100%|██████████| 996M/996M [02:49&lt;00:00, 5.86MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1324989690_1-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7564.9175</td>
      <td>1985.1162</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>ddc7b0c5-1bb0-4f02-a1e4-ddacac84c9aa</td>
    </tr>
    <tr>
      <th>1324989690_2-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7553.6514</td>
      <td>1986.4812</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>52d3e787-0140-42d2-b7cf-026f398cb597</td>
    </tr>
    <tr>
      <th>1324989690_3-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7687.5170</td>
      <td>1987.5452</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>4af7a7a0-823a-47bd-bbe5-344ddd70cc9e</td>
    </tr>
    <tr>
      <th>1324989690_4-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7574.2090</td>
      <td>1999.2940</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>a4455148-9d49-41b7-8750-00d7a3fb225e</td>
    </tr>
    <tr>
      <th>1324989690_5-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7432.2330</td>
      <td>1999.0550</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>41946bab-3c5a-4890-ae7b-cbffcab06c3e</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Load the Macaque cells.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_metadata.csv: 100%|██████████| 628M/628M [01:55&lt;00:00, 5.43MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1291813781_1-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6328.8564</td>
      <td>271.71730</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1291813781_2-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6313.3560</td>
      <td>297.27830</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1291813781_3-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6323.7240</td>
      <td>297.38340</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1291813781_4-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6370.7470</td>
      <td>377.92710</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1291813781_5-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6375.0240</td>
      <td>386.81006</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Load the Marmoset cells.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_metadata.csv: 100%|██████████| 297M/297M [00:50&lt;00:00, 5.95MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1382042951_abmkfncc-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5923.492676</td>
      <td>6160.654297</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>c8c08d8c-5147-4d03-b6f4-ca7bf4d0b8d1</td>
    </tr>
    <tr>
      <th>1382042951_abmmdgea-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>6137.572266</td>
      <td>6217.075684</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>f10bad2c-9cc7-44bf-bd7e-4e61d28121bd</td>
    </tr>
    <tr>
      <th>1382042951_abmojfce-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5802.674805</td>
      <td>5614.007812</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>080466ff-3dea-44f1-b6e2-7892e8c12c95</td>
    </tr>
    <tr>
      <th>1382042951_abmomnim-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5805.942383</td>
      <td>5599.325684</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>cee3bf8b-a159-42d1-983a-ac149b9edda9</td>
    </tr>
    <tr>
      <th>1382042951_aeiamgga-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>6291.268066</td>
      <td>6068.720215</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>3f24c46e-9ab1-4a54-a6fc-591f752edb48</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<section id="specimen-metadata">
<h3>Specimen metadata<a class="headerlink" href="#specimen-metadata" title="Link to this heading">#</a></h3>
<p>In order to associate each cell to the associated specimen in comes from, we load the specimen table for each species. This provides a mapping for each cell to the specimen/tissue/brain section it originates from. It also allows us to associate the cell to its associated <code class="docutils literal notranslate"><span class="pre">slab_plane</span></code> through the specimen id. We’ll explain more regarding the <code class="docutils literal notranslate"><span class="pre">slab_plane</span></code> when we load the our cells coordinates in the plane.</p>
<p>Below we load the Human specimen data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_specimen</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;specimen_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">)</span>
<span class="n">human_specimen</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>specimen_metadata.csv: 100%|██████████| 21.2k/21.2k [00:00&lt;00:00, 173kMB/s] 
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_barcode</th>
      <th>brain_section_thickness</th>
      <th>brain_section_plane_of_section</th>
      <th>brain_section_ordinal</th>
      <th>brain_section_depth</th>
      <th>section_set_label</th>
      <th>section_set_ordinal</th>
      <th>section_set_base_ordinal</th>
      <th>brain_block_label</th>
      <th>brain_block_number</th>
      <th>brain_slab_label</th>
      <th>brain_slab_ordinal</th>
      <th>brain_slab_thickness</th>
      <th>hemisphere</th>
      <th>brain_division_label</th>
      <th>brain_division_ordinal</th>
      <th>donor_label</th>
      <th>slab_plane_ordinal</th>
      <th>slab_plane_label</th>
      <th>slab_plane_order</th>
    </tr>
    <tr>
      <th>brain_section_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>H22.30.001.BS.3.01.01.02</th>
      <td>1327888285</td>
      <td>14</td>
      <td>transverse (rostral to caudal)</td>
      <td>74</td>
      <td>1036</td>
      <td>H22.30.001.BS.3.01.01</td>
      <td>1</td>
      <td>72</td>
      <td>H22.30.001.BS.3.01</td>
      <td>1</td>
      <td>H22.30.001.BS.3</td>
      <td>2</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.BS</td>
      <td>2</td>
      <td>H22.30.001</td>
      <td>1</td>
      <td>H22.30.001.BS.3.Plane.01</td>
      <td>20201</td>
    </tr>
    <tr>
      <th>H22.30.001.BS.3.01.03.02</th>
      <td>1327888566</td>
      <td>14</td>
      <td>transverse (rostral to caudal)</td>
      <td>109</td>
      <td>1526</td>
      <td>H22.30.001.BS.3.01.03</td>
      <td>3</td>
      <td>107</td>
      <td>H22.30.001.BS.3.01</td>
      <td>1</td>
      <td>H22.30.001.BS.3</td>
      <td>2</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.BS</td>
      <td>2</td>
      <td>H22.30.001</td>
      <td>3</td>
      <td>H22.30.001.BS.3.Plane.03</td>
      <td>20203</td>
    </tr>
    <tr>
      <th>H22.30.001.BS.3.02.01.02</th>
      <td>1327890979</td>
      <td>14</td>
      <td>transverse (rostral to caudal)</td>
      <td>37</td>
      <td>518</td>
      <td>H22.30.001.BS.3.02.01</td>
      <td>1</td>
      <td>35</td>
      <td>H22.30.001.BS.3.02</td>
      <td>2</td>
      <td>H22.30.001.BS.3</td>
      <td>2</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.BS</td>
      <td>2</td>
      <td>H22.30.001</td>
      <td>1</td>
      <td>H22.30.001.BS.3.Plane.01</td>
      <td>20201</td>
    </tr>
    <tr>
      <th>H22.30.001.BS.3.02.03.03</th>
      <td>1327891526</td>
      <td>14</td>
      <td>transverse (rostral to caudal)</td>
      <td>146</td>
      <td>2044</td>
      <td>H22.30.001.BS.3.02.03</td>
      <td>3</td>
      <td>143</td>
      <td>H22.30.001.BS.3.02</td>
      <td>2</td>
      <td>H22.30.001.BS.3</td>
      <td>2</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.BS</td>
      <td>2</td>
      <td>H22.30.001</td>
      <td>3</td>
      <td>H22.30.001.BS.3.Plane.03</td>
      <td>20203</td>
    </tr>
    <tr>
      <th>H22.30.001.CX.13.01.03.02</th>
      <td>1304618482</td>
      <td>10</td>
      <td>coronal (caudal to rostral)</td>
      <td>143</td>
      <td>1430</td>
      <td>H22.30.001.CX.13.01.03</td>
      <td>3</td>
      <td>141</td>
      <td>H22.30.001.CX.13.01</td>
      <td>1</td>
      <td>H22.30.001.CX.13</td>
      <td>13</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.CX</td>
      <td>1</td>
      <td>H22.30.001</td>
      <td>5</td>
      <td>H22.30.001.CX.13.Plane.05</td>
      <td>11305</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>The Macaque specimen table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_specimen</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;specimen_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">)</span>
<span class="n">macaque_specimen</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>specimen_metadata.csv: 100%|██████████| 19.8k/19.8k [00:00&lt;00:00, 224kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_barcode</th>
      <th>brain_section_thickness</th>
      <th>brain_section_plane_of_section</th>
      <th>brain_section_ordinal</th>
      <th>brain_section_depth</th>
      <th>section_set_label</th>
      <th>section_set_ordinal</th>
      <th>section_set_base_ordinal</th>
      <th>brain_block_label</th>
      <th>brain_block_number</th>
      <th>brain_slab_label</th>
      <th>brain_slab_ordinal</th>
      <th>brain_slab_thickness</th>
      <th>hemisphere</th>
      <th>brain_division_label</th>
      <th>brain_division_ordinal</th>
      <th>donor_label</th>
      <th>slab_plane_ordinal</th>
      <th>slab_plane_label</th>
      <th>slab_plane_order</th>
    </tr>
    <tr>
      <th>brain_section_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>QM23.50.001.CX.42.01.03.02</th>
      <td>1291809823</td>
      <td>10</td>
      <td>coronal (caudal to rostral)</td>
      <td>133</td>
      <td>1330</td>
      <td>QM23.50.001.CX.42.01.03</td>
      <td>3</td>
      <td>131</td>
      <td>QM23.50.001.CX.42.01</td>
      <td>1</td>
      <td>QM23.50.001.CX.42</td>
      <td>42</td>
      <td>4.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>5</td>
      <td>QM23.50.001.CX.42.Plane.05</td>
      <td>14205</td>
    </tr>
    <tr>
      <th>QM23.50.001.CX.42.01.05.03</th>
      <td>1291809799</td>
      <td>10</td>
      <td>coronal (caudal to rostral)</td>
      <td>234</td>
      <td>2340</td>
      <td>QM23.50.001.CX.42.01.05</td>
      <td>5</td>
      <td>231</td>
      <td>QM23.50.001.CX.42.01</td>
      <td>1</td>
      <td>QM23.50.001.CX.42</td>
      <td>42</td>
      <td>4.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>3</td>
      <td>QM23.50.001.CX.42.Plane.03</td>
      <td>14203</td>
    </tr>
    <tr>
      <th>QM23.50.001.CX.42.02.03.02</th>
      <td>1291810606</td>
      <td>10</td>
      <td>coronal (caudal to rostral)</td>
      <td>134</td>
      <td>1340</td>
      <td>QM23.50.001.CX.42.02.03</td>
      <td>3</td>
      <td>132</td>
      <td>QM23.50.001.CX.42.02</td>
      <td>2</td>
      <td>QM23.50.001.CX.42</td>
      <td>42</td>
      <td>4.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>5</td>
      <td>QM23.50.001.CX.42.Plane.05</td>
      <td>14205</td>
    </tr>
    <tr>
      <th>QM23.50.001.CX.42.02.05.03</th>
      <td>1291810584</td>
      <td>10</td>
      <td>coronal (caudal to rostral)</td>
      <td>234</td>
      <td>2340</td>
      <td>QM23.50.001.CX.42.02.05</td>
      <td>5</td>
      <td>231</td>
      <td>QM23.50.001.CX.42.02</td>
      <td>2</td>
      <td>QM23.50.001.CX.42</td>
      <td>42</td>
      <td>4.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>3</td>
      <td>QM23.50.001.CX.42.Plane.03</td>
      <td>14203</td>
    </tr>
    <tr>
      <th>QM23.50.001.CX.42.02.07.03</th>
      <td>1291810555</td>
      <td>10</td>
      <td>coronal (caudal to rostral)</td>
      <td>334</td>
      <td>3340</td>
      <td>QM23.50.001.CX.42.02.07</td>
      <td>7</td>
      <td>331</td>
      <td>QM23.50.001.CX.42.02</td>
      <td>2</td>
      <td>QM23.50.001.CX.42</td>
      <td>42</td>
      <td>4.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>1</td>
      <td>QM23.50.001.CX.42.Plane.01</td>
      <td>14201</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>And finally the Marmoset specimen table.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_specimen</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;specimen_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">)</span>
<span class="n">marmoset_specimen</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>specimen_metadata.csv: 100%|██████████| 10.4k/10.4k [00:00&lt;00:00, 136kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_barcode</th>
      <th>brain_section_thickness</th>
      <th>brain_section_plane_of_section</th>
      <th>brain_section_ordinal</th>
      <th>brain_section_depth</th>
      <th>section_set_label</th>
      <th>section_set_ordinal</th>
      <th>section_set_base_ordinal</th>
      <th>brain_block_label</th>
      <th>brain_block_number</th>
      <th>brain_slab_label</th>
      <th>brain_slab_ordinal</th>
      <th>brain_slab_thickness</th>
      <th>hemisphere</th>
      <th>brain_division_label</th>
      <th>brain_division_ordinal</th>
      <th>donor_label</th>
      <th>slab_plane_ordinal</th>
      <th>slab_plane_label</th>
      <th>slab_plane_order</th>
    </tr>
    <tr>
      <th>brain_section_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CJ23.56.004.CX.42.01.06</th>
      <td>1382042951</td>
      <td>10</td>
      <td>coronal (rostral to caudal)</td>
      <td>7</td>
      <td>70</td>
      <td>CJ23.56.004.CX.42.01</td>
      <td>1</td>
      <td>1</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>1</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>14201</td>
    </tr>
    <tr>
      <th>CJ23.56.004.CX.42.02.06</th>
      <td>1382042748</td>
      <td>10</td>
      <td>coronal (rostral to caudal)</td>
      <td>27</td>
      <td>270</td>
      <td>CJ23.56.004.CX.42.02</td>
      <td>2</td>
      <td>21</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>2</td>
      <td>CJ23.56.004.CX.42.Plane.02</td>
      <td>14202</td>
    </tr>
    <tr>
      <th>CJ23.56.004.CX.42.03.07</th>
      <td>1382042618</td>
      <td>10</td>
      <td>coronal (rostral to caudal)</td>
      <td>48</td>
      <td>480</td>
      <td>CJ23.56.004.CX.42.03</td>
      <td>3</td>
      <td>41</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>3</td>
      <td>CJ23.56.004.CX.42.Plane.03</td>
      <td>14203</td>
    </tr>
    <tr>
      <th>CJ23.56.004.CX.42.04.06</th>
      <td>1382042399</td>
      <td>10</td>
      <td>coronal (rostral to caudal)</td>
      <td>67</td>
      <td>670</td>
      <td>CJ23.56.004.CX.42.04</td>
      <td>4</td>
      <td>61</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>4</td>
      <td>CJ23.56.004.CX.42.Plane.04</td>
      <td>14204</td>
    </tr>
    <tr>
      <th>CJ23.56.004.CX.42.05.06</th>
      <td>1382042114</td>
      <td>10</td>
      <td>coronal (rostral to caudal)</td>
      <td>87</td>
      <td>870</td>
      <td>CJ23.56.004.CX.42.05</td>
      <td>5</td>
      <td>81</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>CJ23.56.004.CX.42</td>
      <td>42</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>5</td>
      <td>CJ23.56.004.CX.42.Plane.05</td>
      <td>14205</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Join the specimen tables with the cell metadata for each species.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">macaque_specimen</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_specimen&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">marmoset_specimen</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_specimen&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">human_specimen</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_specimen&#39;</span><span class="p">)</span>

<span class="c1"># clean up memory</span>
<span class="k">del</span> <span class="n">macaque_specimen</span>
<span class="k">del</span> <span class="n">marmoset_specimen</span>
<span class="k">del</span> <span class="n">human_specimen</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="spatial-slab-plane-coordinates">
<h2>Spatial, slab-plane coordinates<a class="headerlink" href="#spatial-slab-plane-coordinates" title="Link to this heading">#</a></h2>
<p>We load the slab-plane coordinates for all of our cells. Each cell comes from a brain section that is a slice of a block cut from a cross-sectional slab of brain. The brain sections are spatially registered with sections from the neighboring blocks within the brain slab. These associated brain sections are referred to a slab-plane. This slab-plane coordinate system in millimeters are the coordinates we will plot all of our cells in this notebook.</p>
<p>Again, we load the coordinates for each of the 3 different species.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_slab_plane_coordinates</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;slab_plane_coordinates&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">human_cell_slab_plane_coordinates</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>slab_plane_coordinates.csv: 100%|██████████| 480M/480M [01:20&lt;00:00, 5.94MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>slab_plane_label</th>
      <th>x_slab_mm</th>
      <th>y_slab_mm</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1324989690_1-20250227</th>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>28.232720</td>
      <td>79.516699</td>
    </tr>
    <tr>
      <th>1324989690_2-20250227</th>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>28.242950</td>
      <td>79.520359</td>
    </tr>
    <tr>
      <th>1324989690_3-20250227</th>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>28.129210</td>
      <td>79.461518</td>
    </tr>
    <tr>
      <th>1324989690_4-20250227</th>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>28.231203</td>
      <td>79.500081</td>
    </tr>
    <tr>
      <th>1324989690_5-20250227</th>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>28.352236</td>
      <td>79.561696</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_cell_slab_plane_coordinates</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;slab_plane_coordinates&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_slab_plane_coordinates</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>slab_plane_coordinates.csv: 100%|██████████| 292M/292M [00:48&lt;00:00, 6.07MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>slab_plane_label</th>
      <th>x_slab_mm</th>
      <th>y_slab_mm</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1291813781_1-20250227</th>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>49.982522</td>
      <td>35.779509</td>
    </tr>
    <tr>
      <th>1291813781_2-20250227</th>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>49.986300</td>
      <td>35.748134</td>
    </tr>
    <tr>
      <th>1291813781_3-20250227</th>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>49.976306</td>
      <td>35.751906</td>
    </tr>
    <tr>
      <th>1291813781_4-20250227</th>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>49.896221</td>
      <td>35.688888</td>
    </tr>
    <tr>
      <th>1291813781_5-20250227</th>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>49.888261</td>
      <td>35.681598</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_cell_slab_plane_coordinates</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;slab_plane_coordinates&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_slab_plane_coordinates</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>slab_plane_coordinates.csv: 100%|██████████| 212M/212M [00:34&lt;00:00, 6.14MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>x_rotated_um</th>
      <th>y_rotated_um</th>
      <th>slab_plane_label</th>
      <th>x_slab_mm</th>
      <th>y_slab_mm</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1382042951_abmkfncc-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>2935.931563</td>
      <td>-7974.969447</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>22.351041</td>
      <td>13.197057</td>
    </tr>
    <tr>
      <th>1382042951_abmmdgea-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>2760.454341</td>
      <td>-8109.955328</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>22.161344</td>
      <td>13.311199</td>
    </tr>
    <tr>
      <th>1382042951_abmojfce-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>2835.431429</td>
      <td>-7424.225350</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>22.313860</td>
      <td>12.638454</td>
    </tr>
    <tr>
      <th>1382042951_abmomnim-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>2826.728643</td>
      <td>-7411.957363</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>22.306610</td>
      <td>12.625276</td>
    </tr>
    <tr>
      <th>1382042951_aeiamgga-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>2561.271672</td>
      <td>-8032.767191</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>21.972238</td>
      <td>13.211848</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>And finally we join each species into their respective cell metadata.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">human_cell_slab_plane_coordinates</span><span class="p">[[</span><span class="s1">&#39;x_slab_mm&#39;</span><span class="p">,</span> <span class="s1">&#39;y_slab_mm&#39;</span><span class="p">]],</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span>
<span class="p">)</span>
<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">macaque_cell_slab_plane_coordinates</span><span class="p">[[</span><span class="s1">&#39;x_slab_mm&#39;</span><span class="p">,</span> <span class="s1">&#39;y_slab_mm&#39;</span><span class="p">]],</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span>
<span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">marmoset_cell_slab_plane_coordinates</span><span class="p">[[</span><span class="s1">&#39;x_slab_mm&#39;</span><span class="p">,</span> <span class="s1">&#39;y_slab_mm&#39;</span><span class="p">]],</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span>
<span class="p">)</span>

<span class="c1"># clean up memory</span>
<span class="k">del</span> <span class="n">human_cell_slab_plane_coordinates</span>
<span class="k">del</span> <span class="n">macaque_cell_slab_plane_coordinates</span>
<span class="k">del</span> <span class="n">marmoset_cell_slab_plane_coordinates</span>
</pre></div>
</div>
</div>
</div>
<section id="geojson-polygons">
<h3>Geojson polygons<a class="headerlink" href="#geojson-polygons" title="Link to this heading">#</a></h3>
<p>The dataset includes polygons outlining select regions of the Basal Ganglia, which can be plotted over slab coordinates. These polygon annotations were manually drawn using a combination of taxonomy cell types, select gene expression from the spatial transcriptomics gene panel, and cell segmentation stains with best attempts made to be consistent across species and planes. Manual annotations include the striatum (STR; encompasses the caudate, putamen, nucleus accumbens, and olfactory tubercle), the globus pallidus (GP) and it’s nested subregions the external (GPe) and internal (GPi) segments, the ventral pallidum (VeP), the substantia nigra (SN), and the subthalamic nucleus (STH). However, please note that these polygons are not derived from registration to the Harmonized Ontology of Mammalian Brain Anatomy (HOMBA). They should instead be used as cellular groupings to orient the user in the anatomical context of the spatial transcriptomics data and subset the data into more manageable, anatomically-related chunks. These polygons are provided in <a class="reference external" href="https://geojson.org/">geojson</a> format, and here we use the associated geojson python package to load the jsons for use in our visualization.</p>
<p>First though, we load the term set for these data. While these geojsons are manually drawn and not derived from the HOMBA, we do use the abbreviations and colors associated with these structures through their DHBA identifier when such an identifier exists. These colorings match those of the previous <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_clustering_analysis_and_annotation.html">HMBA-BG 10X tutorial</a>. We load these colors and identifiers below.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">value_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;value_sets&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">)</span>
<span class="n">pred</span> <span class="o">=</span> <span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span>
<span class="n">parcellation_terms</span> <span class="o">=</span> <span class="n">value_sets</span><span class="p">[</span><span class="n">pred</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">parcellation_terms</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>value_sets.csv: 100%|██████████| 3.79k/3.79k [00:00&lt;00:00, 47.5kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>table</th>
      <th>field</th>
      <th>description</th>
      <th>order</th>
      <th>external_identifier</th>
      <th>parent_label</th>
      <th>color_hex_triplet</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Basal ganglia and adjacent</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>Basal ganglia and adjacent</td>
      <td>1</td>
      <td>basal_ganglia_and_adjacent</td>
      <td>NaN</td>
      <td>#BEBEBE</td>
    </tr>
    <tr>
      <th>STR</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>striatum</td>
      <td>2</td>
      <td>DHBA:10333</td>
      <td>Basal ganglia and adjacent</td>
      <td>#cd86cc</td>
    </tr>
    <tr>
      <th>GP</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>globus pallidus</td>
      <td>3</td>
      <td>DHBA:10342</td>
      <td>Basal ganglia and adjacent</td>
      <td>#7AE361</td>
    </tr>
    <tr>
      <th>GPe</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>external segment of globus pallidus</td>
      <td>4</td>
      <td>DHBA:10343</td>
      <td>Basal ganglia and adjacent</td>
      <td>#53DB33</td>
    </tr>
    <tr>
      <th>GPi</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>internal segment of globus pallidus</td>
      <td>5</td>
      <td>DHBA:10344</td>
      <td>Basal ganglia and adjacent</td>
      <td>#A7DB33</td>
    </tr>
    <tr>
      <th>VeP</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>ventral pallidus</td>
      <td>6</td>
      <td>DHBA:10345</td>
      <td>Basal ganglia and adjacent</td>
      <td>#9b19f5</td>
    </tr>
    <tr>
      <th>STH</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>subthalamic nucleus</td>
      <td>7</td>
      <td>DHBA:10466</td>
      <td>Basal ganglia and adjacent</td>
      <td>#907991</td>
    </tr>
    <tr>
      <th>SN</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>substantia nigra</td>
      <td>8</td>
      <td>DHBA:12251</td>
      <td>Basal ganglia and adjacent</td>
      <td>#89522A</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We define two functions that will be used in our plotting of the geojsons. The first overplots the geojson over the specified plot, handling the different kinds of geojson shapes/polygons. The second loads the data based on the appropriate slab and species provided.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_geo_data</span><span class="p">(</span><span class="n">geo_data</span><span class="p">:</span> <span class="nb">dict</span><span class="p">,</span> <span class="n">ax</span><span class="p">:</span> <span class="n">plt</span><span class="o">.</span><span class="n">Axes</span><span class="p">,</span> <span class="n">terms</span><span class="o">=</span><span class="n">parcellation_terms</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Plot a geojson polygon over the input axis.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    geo_data: dict</span>
<span class="sd">        Geojson formatted dictionary containing Polygons to plot.</span>
<span class="sd">    ax: matplotlib.pyplot.axis</span>
<span class="sd">        Axis object to plot the geojson data over.</span>
<span class="sd">    terms: pandas.DataFrame</span>
<span class="sd">        Term list containing the colors and orders for the parcellation, brain region terms. </span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">geo_data</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="k">return</span> <span class="kc">None</span>
    <span class="c1"># If the geo_data is a FeatureCollection, we need to iterate over its</span>
    <span class="c1"># features and plot each one.</span>
    <span class="k">if</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;FeatureCollection&#39;</span><span class="p">:</span>
        <span class="k">for</span> <span class="n">sub_data</span> <span class="ow">in</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;features&#39;</span><span class="p">]:</span>
            <span class="n">plot_geo_data</span><span class="p">(</span><span class="n">sub_data</span><span class="p">,</span> <span class="n">ax</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Feature&#39;</span><span class="p">:</span>
        <span class="n">geometry_data</span> <span class="o">=</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;geometry&#39;</span><span class="p">]</span>
        <span class="n">name</span> <span class="o">=</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;properties&#39;</span><span class="p">][</span><span class="s1">&#39;label&#39;</span><span class="p">]</span>
        <span class="n">color</span> <span class="o">=</span> <span class="n">terms</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">name</span><span class="p">,</span> <span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span> <span class="c1"># Get the color for this object from the term list.</span>
        <span class="c1"># If the geometry is a GeometryCollection, we need to iterate over its</span>
        <span class="c1"># geometries that make up the complete feature.</span>
        <span class="k">if</span> <span class="n">geometry_data</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;GeometryCollection&#39;</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">geometry</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">geometry_data</span><span class="p">[</span><span class="s1">&#39;geometries&#39;</span><span class="p">]):</span>
                <span class="n">coords</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">geojson</span><span class="o">.</span><span class="n">coords</span><span class="p">(</span><span class="n">geometry</span><span class="p">)))</span>
                <span class="c1"># Only label the first geometry in the collection. Others</span>
                <span class="c1"># will be plotted without a label to avoid duplicate legend entries.</span>
                <span class="k">if</span> <span class="n">idx</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
                    <span class="n">ax</span><span class="o">.</span><span class="n">fill</span><span class="p">(</span><span class="n">coords</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">],</span>
                            <span class="n">coords</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">],</span>
                            <span class="n">edgecolor</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
                            <span class="n">fill</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                            <span class="n">linewidth</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
                            <span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">)</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">ax</span><span class="o">.</span><span class="n">fill</span><span class="p">(</span><span class="n">coords</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">],</span>
                            <span class="n">coords</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">],</span>
                            <span class="n">edgecolor</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
                            <span class="n">fill</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                            <span class="n">linewidth</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">coords</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">geojson</span><span class="o">.</span><span class="n">coords</span><span class="p">(</span><span class="n">geometry_data</span><span class="p">)))</span>
            <span class="k">try</span><span class="p">:</span>
                <span class="n">ax</span><span class="o">.</span><span class="n">fill</span><span class="p">(</span><span class="n">coords</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">],</span>
                        <span class="n">coords</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">],</span>
                        <span class="n">edgecolor</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
                        <span class="n">fill</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">linewidth</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
                        <span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">)</span>
            <span class="k">except</span> <span class="ne">ValueError</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
                <span class="kn">import</span><span class="w"> </span><span class="nn">pdb</span><span class="p">;</span> <span class="n">pdb</span><span class="o">.</span><span class="n">set_trace</span><span class="p">()</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;unknown data&quot;</span><span class="p">,</span> <span class="n">geo_data</span><span class="p">)</span>

<span class="k">def</span><span class="w"> </span><span class="nf">get_polygons</span><span class="p">(</span><span class="n">slab_name</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">species</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">dict</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Load the geojson polygons for the specified slab and species.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    slab_name: str</span>
<span class="sd">        Name of the slab to load.</span>
<span class="sd">    species: str</span>
<span class="sd">        Species of the slab, one of &#39;macaque&#39;, &#39;marmoset&#39;, or &#39;human&#39;.</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    dict</span>
<span class="sd">        Geojson formatted dictionary containing the polygons for the specified slab and species.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">if</span> <span class="n">species</span> <span class="o">==</span> <span class="s1">&#39;macaque&#39;</span><span class="p">:</span>
        <span class="n">base_dir</span> <span class="o">=</span> <span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span>
    <span class="k">elif</span> <span class="n">species</span> <span class="o">==</span> <span class="s1">&#39;marmoset&#39;</span><span class="p">:</span>
        <span class="n">base_dir</span> <span class="o">=</span> <span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span>
    <span class="k">elif</span> <span class="n">species</span> <span class="o">==</span> <span class="s1">&#39;human&#39;</span><span class="p">:</span>
        <span class="n">base_dir</span> <span class="o">=</span> <span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s1">&#39;Invalid species&#39;</span><span class="p">)</span>

    <span class="n">file_path</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_file_path</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="n">base_dir</span><span class="p">,</span> <span class="n">file_name</span><span class="o">=</span><span class="n">slab_name</span><span class="p">)</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">file_path</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="k">return</span> <span class="kc">None</span>

    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">file_path</span><span class="p">)</span> <span class="k">as</span> <span class="n">jfile</span><span class="p">:</span>
        <span class="n">data</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">jfile</span><span class="p">)</span>
    
    <span class="k">return</span> <span class="n">data</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="manual-anatomical-annotations">
<h2>Manual anatomical annotations<a class="headerlink" href="#manual-anatomical-annotations" title="Link to this heading">#</a></h2>
<p>Using the manually drawn anatomical polygons described above, each cell was assigned membership in the annotated anatomical regions. The tables we load below contain boolean columns denoting which anatomical regions the cell could be considered a member of. Note that a given cell can belong to multiple different anatomical annotations as the polygons overlap and can be subsets of each other. For instance, all cells in this release are within the “Basal Ganglia and adjacent” annotation, and most cells in either “GPe” or “GPi” are within the parent annotation “GP”. The final column, <code class="docutils literal notranslate"><span class="pre">parcellation_term_symbol</span></code> is a single value representing a single unique assignment to a structure based on selecting from STH in reverse order of the columns, keeping the first True assignment. For this notebook, we will use only the <code class="docutils literal notranslate"><span class="pre">parcellation_term_symbol</span></code> to plot our data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_anatomical_annotations</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_anatomical_annotations&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">human_cell_anatomical_annotations</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_anatomical_annotations.csv: 100%|██████████| 448M/448M [01:19&lt;00:00, 5.61MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Basal ganglia and adjacent</th>
      <th>STR</th>
      <th>GP</th>
      <th>GPe</th>
      <th>GPi</th>
      <th>VeP</th>
      <th>SN</th>
      <th>STH</th>
      <th>parcellation_term_symbol</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1321115602_1-20250227</th>
      <td>True</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>STR</td>
    </tr>
    <tr>
      <th>1321115602_2-20250227</th>
      <td>True</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>STR</td>
    </tr>
    <tr>
      <th>1321115602_3-20250227</th>
      <td>True</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>STR</td>
    </tr>
    <tr>
      <th>1321115602_4-20250227</th>
      <td>True</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>STR</td>
    </tr>
    <tr>
      <th>1321115602_5-20250227</th>
      <td>True</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>STR</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_cell_anatomical_annotations</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_anatomical_annotations&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_anatomical_annotations</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_anatomical_annotations.csv: 100%|██████████| 275M/275M [00:48&lt;00:00, 5.62MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Basal ganglia and adjacent</th>
      <th>STR</th>
      <th>GP</th>
      <th>GPe</th>
      <th>GPi</th>
      <th>VeP</th>
      <th>SN</th>
      <th>STH</th>
      <th>parcellation_term_symbol</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1291813781_1-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
    <tr>
      <th>1291813781_2-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
    <tr>
      <th>1291813781_3-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
    <tr>
      <th>1291813781_4-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
    <tr>
      <th>1291813781_5-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_cell_anatomical_annotations</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_anatomical_annotations&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_anatomical_annotations</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_anatomical_annotations.csv: 100%|██████████| 116M/116M [00:19&lt;00:00, 5.91MMB/s]   
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Basal ganglia and adjacent</th>
      <th>STR</th>
      <th>GP</th>
      <th>GPe</th>
      <th>GPi</th>
      <th>VeP</th>
      <th>SN</th>
      <th>STH</th>
      <th>parcellation_term_symbol</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1382042951_abmkfncc-1-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
    <tr>
      <th>1382042951_abmmdgea-1-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
    <tr>
      <th>1382042951_abmojfce-1-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
    <tr>
      <th>1382042951_abmomnim-1-20250227</th>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>Basal ganglia and adjacent</td>
    </tr>
    <tr>
      <th>1382042951_aeiamgga-1-20250227</th>
      <td>True</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>STR</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Below, we define a function to extract color and order information from the <code class="docutils literal notranslate"><span class="pre">value_set</span></code> table for each anatomical parcellation term. This will allow us to plot the color of the anatomical region we assigned each cell to using the manually drawn, geojson polygons.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">extract_value_set</span><span class="p">(</span><span class="n">cell_metadata_df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">input_value_set</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">input_value_set_label</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Add color and order columns to the cell metadata dataframe based on the input</span>
<span class="sd">    value set.</span>

<span class="sd">    Columns are added as {input_value_set_label}_color and {input_value_set_label}_order.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    cell_metadata_df : pd.DataFrame</span>
<span class="sd">        DataFrame containing cell metadata.</span>
<span class="sd">    input_value_set : pd.DataFrame</span>
<span class="sd">        DataFrame containing the value set information.</span>
<span class="sd">    input_value_set_label : str</span>
<span class="sd">        The the column name to extract color and order information for. will be added to the cell metadata.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">cell_metadata_df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">input_value_set_label</span><span class="si">}</span><span class="s1">_color&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_value_set</span><span class="p">[</span>
        <span class="n">input_value_set</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">input_value_set_label</span>
    <span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cell_metadata_df</span><span class="p">[</span><span class="n">input_value_set_label</span><span class="p">]][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">cell_metadata_df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">input_value_set_label</span><span class="si">}</span><span class="s1">_order&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_value_set</span><span class="p">[</span>
        <span class="n">input_value_set</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">input_value_set_label</span>
    <span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cell_metadata_df</span><span class="p">[</span><span class="n">input_value_set_label</span><span class="p">]][</span><span class="s1">&#39;order&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
</pre></div>
</div>
</div>
</div>
<p>Merge the manual annotation information into the cell meatadata tables and add in the color and ordering information.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">human_cell_anatomical_annotations</span><span class="p">[[</span><span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">]]</span>
<span class="p">)</span>
<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">macaque_cell_anatomical_annotations</span><span class="p">[[</span><span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">]]</span>
<span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">marmoset_cell_anatomical_annotations</span><span class="p">[[</span><span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">]]</span>
<span class="p">)</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">human_cell_metadata</span><span class="p">,</span> <span class="n">parcellation_terms</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">)</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">macaque_cell_metadata</span><span class="p">,</span> <span class="n">parcellation_terms</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">)</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">marmoset_cell_metadata</span><span class="p">,</span> <span class="n">parcellation_terms</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">)</span>

<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;parcellation_term_symbol_order&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;parcellation_term_symbol_order&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;parcellation_term_symbol_order&#39;</span><span class="p">)</span>

<span class="c1"># clean up memory</span>
<span class="k">del</span> <span class="n">human_cell_anatomical_annotations</span>
<span class="k">del</span> <span class="n">macaque_cell_anatomical_annotations</span>
<span class="k">del</span> <span class="n">marmoset_cell_anatomical_annotations</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="plotting-the-slabs">
<h2>Plotting the Slabs<a class="headerlink" href="#plotting-the-slabs" title="Link to this heading">#</a></h2>
<p>We define a helper function <em>plot_slabs</em> to visualize the cells for a specified set of slabs either by colorized metadata or gene expression. The code also allows for plotting the annotation polygons over the data if requested. This function can be used to plot individual frames as well.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_slabs</span><span class="p">(</span>
    <span class="n">df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
    <span class="n">feature</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">slab_list</span><span class="p">:</span> <span class="n">List</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
    <span class="n">species</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">cmap</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Colormap</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">40</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
    <span class="n">plot_polygons</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
    <span class="n">point_size</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mf">1.0</span><span class="p">,</span>
    <span class="n">fix_bounds</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Tuple</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Figure</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Plot spatial transcriptomics data.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    df: pandas.DataFrame</span>
<span class="sd">        cell metadata data frame with x,y locations to plot.</span>
<span class="sd">    feature: str</span>
<span class="sd">        Expression value or color column to plot.</span>
<span class="sd">    slab_list: List[str]</span>
<span class="sd">        List of slab_plane ids to plot.</span>
<span class="sd">    species: str</span>
<span class="sd">        Species to plot (human, macaque, marmoset).</span>
<span class="sd">    cmap: matplotlib.colors.Colormap</span>
<span class="sd">        Colormap for expression features.</span>
<span class="sd">    fig_width: int</span>
<span class="sd">        Width of the plotted figure.</span>
<span class="sd">    fig_height: int</span>
<span class="sd">        Height of the plotted figure.</span>
<span class="sd">    plot_polygons: bool</span>
<span class="sd">        Overplot geojson polygons. Note that these are note available for human</span>
<span class="sd">        and this should be set to False.</span>
<span class="sd">    point_size: float</span>
<span class="sd">        Size of the points to plot.</span>
<span class="sd">    fix_bounds: bool</span>
<span class="sd">        If True, fix the x and y bounds across all slabs to be the same.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">slab_list</span><span class="p">,</span> <span class="nb">str</span><span class="p">):</span>
        <span class="n">slab_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">slab_list</span><span class="p">]</span>

    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">slab_list</span><span class="p">))</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">slab_list</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
        <span class="n">ax</span> <span class="o">=</span> <span class="p">[</span><span class="n">ax</span><span class="p">]</span>

    <span class="k">if</span> <span class="s1">&#39;x&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">x_col</span> <span class="o">=</span> <span class="s1">&#39;x&#39;</span>
    <span class="k">elif</span> <span class="s1">&#39;x_slab_mm&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">x_col</span> <span class="o">=</span> <span class="s1">&#39;x_slab_mm&#39;</span>
    <span class="k">if</span> <span class="s1">&#39;y&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">y_col</span> <span class="o">=</span> <span class="s1">&#39;y&#39;</span>
    <span class="k">elif</span> <span class="s1">&#39;y_slab_mm&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">y_col</span> <span class="o">=</span> <span class="s1">&#39;y_slab_mm&#39;</span>

    <span class="k">if</span> <span class="n">fix_bounds</span><span class="p">:</span>
        <span class="n">x_min</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span> <span class="o">*</span> <span class="mf">0.8</span>
        <span class="n">x_max</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span> <span class="o">*</span> <span class="mf">1.2</span>
        <span class="n">y_min</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span> <span class="o">*</span> <span class="mf">0.8</span>
        <span class="n">y_max</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span> <span class="o">*</span> <span class="mf">1.2</span>

    <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">slab</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">slab_list</span><span class="p">):</span>

        <span class="k">if</span> <span class="s1">&#39;slab_plane_label&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="n">filtered</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;slab_plane_label&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">slab</span><span class="p">]</span>
        <span class="k">elif</span> <span class="s1">&#39;brain_section_label&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="n">filtered</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">slab</span><span class="p">]</span>
        
        <span class="n">xx</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span>
        <span class="n">yy</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span>
        <span class="n">vv</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">feature</span><span class="p">]</span>
        
        <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">point_size</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">vv</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">point_size</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">vv</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s2">&quot;.&quot;</span><span class="p">)</span>

        <span class="k">if</span> <span class="n">plot_polygons</span><span class="p">:</span>
            <span class="n">plot_geo_data</span><span class="p">(</span><span class="n">get_polygons</span><span class="p">(</span><span class="n">slab</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="n">species</span><span class="p">),</span> <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">])</span>
            
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s1">&#39;equal&#39;</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">fix_bounds</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="n">y_min</span><span class="p">,</span> <span class="n">y_max</span><span class="p">)</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">x_min</span><span class="p">,</span> <span class="n">x_max</span><span class="p">)</span>
        <span class="n">ylim</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="n">ylim</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">ylim</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
        
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">slab</span><span class="p">)</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        
    <span class="n">plt</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.01</span><span class="p">,</span> <span class="n">hspace</span><span class="o">=</span><span class="mf">0.01</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<section id="displaying-the-manual-anatomical-annotations">
<h3>Displaying the manual anatomical annotations.<a class="headerlink" href="#displaying-the-manual-anatomical-annotations" title="Link to this heading">#</a></h3>
<p>We now have a set of cells in x,y slab coordinates and metadata on the anatomical regions these cells have been manually annotated as members of. Below we plot the cells using the colors as defined in our term set, displaying the cells by manually annotated regions of the Basal Ganglia.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_slab_list</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;H22.30.001.CX.16.Plane.07&#39;</span><span class="p">,</span> <span class="s1">&#39;H22.30.001.CX.17.Plane.07&#39;</span><span class="p">,</span>
    <span class="s1">&#39;H22.30.001.CX.19.Plane.07&#39;</span><span class="p">,</span> <span class="s1">&#39;H22.30.001.CX.20.Plane.03&#39;</span><span class="p">,</span> <span class="s1">&#39;H22.30.001.CX.20.Plane.05&#39;</span>
<span class="p">]</span>
<span class="n">color_type</span> <span class="o">=</span> <span class="s1">&#39;parcellation_term_symbol_color&#39;</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
    <span class="n">human_cell_metadata</span><span class="p">,</span>
    <span class="n">color_type</span><span class="p">,</span>
    <span class="n">human_slab_list</span><span class="p">,</span>
    <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span>
    <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Human: Manual Anatomical Annotations&#39;</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>H22.30.001.CX.16.Plane.07.geojson: 100%|██████████| 5.47k/5.47k [00:00&lt;00:00, 54.8kMB/s]
H22.30.001.CX.17.Plane.07.geojson: 100%|██████████| 8.08k/8.08k [00:00&lt;00:00, 103kMB/s]
H22.30.001.CX.19.Plane.07.geojson: 100%|██████████| 5.97k/5.97k [00:00&lt;00:00, 81.5kMB/s]
H22.30.001.CX.20.Plane.03.geojson: 100%|██████████| 3.00k/3.00k [00:00&lt;00:00, 44.1kMB/s]
H22.30.001.CX.20.Plane.05.geojson: 100%|██████████| 2.35k/2.35k [00:00&lt;00:00, 29.1kMB/s]
</pre></div>
</div>
<img alt="../_images/5c07117cfe47c72e46c77478da53c43dc06c61976207118cbd7888bd0eea5422.png" src="../_images/5c07117cfe47c72e46c77478da53c43dc06c61976207118cbd7888bd0eea5422.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_slab_list</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;QM23.50.001.CX.42.Plane.03&#39;</span><span class="p">,</span> <span class="s1">&#39;QM23.50.001.CX.43.Plane.07&#39;</span><span class="p">,</span> <span class="s1">&#39;QM23.50.001.CX.44.Plane.05&#39;</span><span class="p">,</span>
    <span class="s1">&#39;QM23.50.001.CX.45.Plane.05&#39;</span><span class="p">,</span> <span class="s1">&#39;QM23.50.001.CX.46.Plane.05&#39;</span>
<span class="p">]</span>
<span class="n">color_type</span> <span class="o">=</span> <span class="s1">&#39;parcellation_term_symbol_color&#39;</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
    <span class="n">macaque_cell_metadata</span><span class="p">,</span>
    <span class="n">color_type</span><span class="p">,</span>
    <span class="n">macaque_slab_list</span><span class="p">,</span>
    <span class="n">species</span><span class="o">=</span><span class="s1">&#39;macaque&#39;</span><span class="p">,</span>
    <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Macaque: Manual Anatomical Annotations&#39;</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>QM23.50.001.CX.42.Plane.03.geojson: 100%|██████████| 2.33k/2.33k [00:00&lt;00:00, 25.6kMB/s]
QM23.50.001.CX.43.Plane.07.geojson: 100%|██████████| 2.99k/2.99k [00:00&lt;00:00, 47.4kMB/s]
QM23.50.001.CX.44.Plane.05.geojson: 100%|██████████| 5.53k/5.53k [00:00&lt;00:00, 86.0kMB/s]
QM23.50.001.CX.45.Plane.05.geojson: 100%|██████████| 6.83k/6.83k [00:00&lt;00:00, 90.1kMB/s]
QM23.50.001.CX.46.Plane.05.geojson: 100%|██████████| 5.00k/5.00k [00:00&lt;00:00, 54.7kMB/s]
</pre></div>
</div>
<img alt="../_images/c2bbf34196e83305ba102d6a2e5b8f7bbe56521ee8892dc18d26bb991d6c3f96.png" src="../_images/c2bbf34196e83305ba102d6a2e5b8f7bbe56521ee8892dc18d26bb991d6c3f96.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_slab_list</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;CJ23.56.004.CX.42.Plane.05&#39;</span><span class="p">,</span> <span class="s1">&#39;CJ23.56.004.CX.42.Plane.13&#39;</span><span class="p">,</span> <span class="s1">&#39;CJ23.56.004.CX.43.Plane.05&#39;</span><span class="p">,</span>
    <span class="s1">&#39;CJ23.56.004.CX.43.Plane.15&#39;</span><span class="p">,</span> <span class="s1">&#39;CJ23.56.004.CX.44.Plane.03&#39;</span>
<span class="p">]</span>
<span class="n">color_type</span> <span class="o">=</span> <span class="s1">&#39;parcellation_term_symbol_color&#39;</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
    <span class="n">marmoset_cell_metadata</span><span class="p">,</span>
    <span class="n">color_type</span><span class="p">,</span>
    <span class="n">marmoset_slab_list</span><span class="p">,</span>
    <span class="n">species</span><span class="o">=</span><span class="s1">&#39;marmoset&#39;</span><span class="p">,</span>
    <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Marmoset: Manual Anatomical Annotations&#39;</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>CJ23.56.004.CX.42.Plane.05.geojson: 100%|██████████| 6.43k/6.43k [00:00&lt;00:00, 54.2kMB/s]
CJ23.56.004.CX.42.Plane.13.geojson: 100%|██████████| 6.54k/6.54k [00:00&lt;00:00, 55.5kMB/s]
CJ23.56.004.CX.43.Plane.05.geojson: 100%|██████████| 12.2k/12.2k [00:00&lt;00:00, 157kMB/s]
CJ23.56.004.CX.43.Plane.15.geojson: 100%|██████████| 9.98k/9.98k [00:00&lt;00:00, 139kMB/s]
CJ23.56.004.CX.44.Plane.03.geojson: 100%|██████████| 5.77k/5.77k [00:00&lt;00:00, 75.0kMB/s]
</pre></div>
</div>
<img alt="../_images/2e5b0f2803e75264004373a5bb71533fce8646b0287a4a04dea3b8dfec18a694.png" src="../_images/2e5b0f2803e75264004373a5bb71533fce8646b0287a4a04dea3b8dfec18a694.png" />
</div>
</div>
</section>
</section>
<section id="taxonomy-mapping">
<h2>Taxonomy mapping<a class="headerlink" href="#taxonomy-mapping" title="Link to this heading">#</a></h2>
<p>Now that we have our base functions for plotting set up and we’ve exercised them using the parcellation colors, let’s load the taxonomy information so that we can plot the spatial locations of various cell types. Below, we load and merge all the information we need to plot cell colors and taxonomy as done in the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_clustering_analysis_and_annotation.html">previous HMBA-BG tutorials</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">:</span> <span class="s1">&#39;Int64&#39;</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">cluster_annotation_term_set</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term_set&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span>

<span class="n">cluster_annotation_term</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>

<span class="n">cluster_to_cluster_annotation_membership</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_to_cluster_annotation_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">membership_with_cluster_info</span> <span class="o">=</span> <span class="n">cluster_to_cluster_annotation_membership</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">cluster</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)[[</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">]],</span>
    <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span>
<span class="p">)</span>
<span class="n">membership_with_cluster_info</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_annotation_term</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_anno_term&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">membership_groupby</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span>
<span class="p">)</span>

<span class="n">cluster_annotation_term_set</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term_set&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span>

<span class="c1"># term_sets = abc_cache.get_metadata_dataframe(directory=&#39;WHB-taxonomy&#39;, file_name=&#39;cluster_annotation_term_set&#39;).set_index(&#39;label&#39;)</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_order</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;term_order&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_order</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">,</span> <span class="s1">&#39;Class&#39;</span><span class="p">,</span> <span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;Group&#39;</span><span class="p">,</span> <span class="s1">&#39;Cluster&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_order</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span>
    <span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">:</span> <span class="s1">&#39;Neighborhood_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;Class&#39;</span><span class="p">:</span> <span class="s1">&#39;Class_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;Subclass&#39;</span><span class="p">:</span> <span class="s1">&#39;Subclass_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;Group&#39;</span><span class="p">:</span> <span class="s1">&#39;Group_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;Cluster&#39;</span><span class="p">:</span> <span class="s1">&#39;Cluster_order&#39;</span><span class="p">},</span>
    <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>

<span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">cluster_colors</span><span class="p">[</span><span class="n">cluster_annotation_term_set</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span>
<span class="n">cluster_colors</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">,</span> <span class="s1">&#39;Class&#39;</span><span class="p">,</span> <span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;Group&#39;</span><span class="p">,</span> <span class="s1">&#39;Cluster&#39;</span><span class="p">],</span>
    <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Each of the three species has been mapped to the HMBA 10X taxonomy using MapMyCells. Below, we load the mapping of each cell to a taxonomy cluster. This mapping is the entry point into the taxonomy.</p>
<p>Again, we load each species.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_to_cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">macaque_cell_to_cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">marmoset_cell_to_cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>cell_to_cluster_membership.csv: 100%|██████████| 310M/310M [00:52&lt;00:00, 5.87MMB/s]    
cell_to_cluster_membership.csv: 100%|██████████| 195M/195M [00:34&lt;00:00, 5.61MMB/s]    
cell_to_cluster_membership.csv: 100%|██████████| 88.1M/88.1M [00:14&lt;00:00, 6.11MMB/s]  
</pre></div>
</div>
</div>
</div>
<p>The spatial data are registered to an extended version of the taxonomy that contains clusters that are considered Adjacent to the Basal Ganglia. These are cell types that are present in the data but are not considered part of the basal ganglia. These clusters are mapped into a Group, Subclass, Class, and Neighborhood in a single path of the taxonomy called Adjacent. The release consensus taxonomy does not explicitly contain this taxon. Any cell in a cluster that would be annotated into Adjacent, will end up being NaN in the joins between the clusters and the taxonomy. We thus fill in these NaNs, assigning any cell that was not mapped into to a group in the taxonomy an Adjacent cell type at all levels with a grey color. We give these Adjacent taxons an ordering at the very end of the set of taxons at each level.”</p>
<p>Although the BG Consensus Taxonomy is intended to be specific to BG cell types, when the data were sampled for the snRNA-Seq reference, some cells from adjacent non-BG regions were included. Because these cells originated from non-BG regions, we did not assign them a cell type identity (i.e. a label at the Group, Subclass, etc. levels). However, because 1) our spatial transcriptomics datasets also contain cells from adjacent non-BG regions, and 2) MapMyCells requires every cell to be assigned a label from the reference, including Adjacent cells in the reference decreases spurious mapping results in BG-adjacent regions. Adjacent cells are included in the release data for completeness, where they are labeled as “Adjacent” at all taxonomy levels. When visualizing cells in BG regions, we recommend filtering them out.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">fill_adjacent_data</span><span class="p">(</span><span class="n">input_df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">:</span>
    <span class="n">output_df</span> <span class="o">=</span> <span class="n">input_df</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span>
        <span class="p">{</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;cluster_label&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Class&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Cluster&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Group&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Neighborhood&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Subclass&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Neighborhood_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Class_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Subclass_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Group_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Cluster_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Neighborhood_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Neighborhood_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
         <span class="s1">&#39;Class_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Class_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
         <span class="s1">&#39;Subclass_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Subclass_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
         <span class="s1">&#39;Group_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Group_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
         <span class="s1">&#39;Cluster_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Cluster_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
        <span class="p">}</span>
    <span class="p">)</span>
    <span class="k">return</span> <span class="n">output_df</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span>
        <span class="p">{</span>
            <span class="s1">&#39;Neighborhood_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Class_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Subclass_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Group_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Cluster_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
        <span class="p">}</span>
    <span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Now we merge the cell type classification colors and order into our metadata for each species.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">human_cell_to_cluster</span><span class="p">)</span>
<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_color&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_order</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">fill_adjacent_data</span><span class="p">(</span><span class="n">human_cell_metadata</span><span class="p">)</span>

<span class="k">del</span> <span class="n">human_cell_to_cluster</span>

<span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>Neighborhood_color</th>
      <th>Class_color</th>
      <th>Subclass_color</th>
      <th>Group_color</th>
      <th>Cluster_color</th>
      <th>Class_order</th>
      <th>Cluster_order</th>
      <th>Group_order</th>
      <th>Neighborhood_order</th>
      <th>Subclass_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1384732733_104390-20250227</th>
      <td>H22.30.001.CX.20.02.03.01</td>
      <td>1384732733</td>
      <td>sis_cellpose_v1</td>
      <td>9122.90000</td>
      <td>2747.79320</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>1bb3cc09-224a-412b-8a09-59438370c64f</td>
      <td>1384732733</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#a8afa5</td>
      <td>#B97300</td>
      <td>#B97300</td>
      <td>#8b3406</td>
      <td>4</td>
      <td>261</td>
      <td>15</td>
      <td>1</td>
      <td>11</td>
    </tr>
    <tr>
      <th>1326494670_42591-20250227</th>
      <td>H22.30.001.CX.17.07.07.02</td>
      <td>1326494670</td>
      <td>sis_cellpose_v1</td>
      <td>623.67487</td>
      <td>903.07245</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>3d211210-71d6-47f9-a88f-946f56c9ba8c</td>
      <td>1326494670</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#66391F</td>
      <td>#a9de1a</td>
      <td>3</td>
      <td>142</td>
      <td>9</td>
      <td>1</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1326494670_42590-20250227</th>
      <td>H22.30.001.CX.17.07.07.02</td>
      <td>1326494670</td>
      <td>sis_cellpose_v1</td>
      <td>498.75037</td>
      <td>902.13605</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>e059d8e7-29ce-48f9-9105-df1c9ba966c1</td>
      <td>1326494670</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#66391F</td>
      <td>#7db606</td>
      <td>3</td>
      <td>144</td>
      <td>9</td>
      <td>1</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1326494670_42589-20250227</th>
      <td>H22.30.001.CX.17.07.07.02</td>
      <td>1326494670</td>
      <td>sis_cellpose_v1</td>
      <td>421.28458</td>
      <td>900.51600</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>c017a97b-0c6a-41b0-b4b5-6f99e4a5759d</td>
      <td>1326494670</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#f5e689</td>
      <td>#f5e689</td>
      <td>#8c2187</td>
      <td>3</td>
      <td>210</td>
      <td>12</td>
      <td>1</td>
      <td>8</td>
    </tr>
    <tr>
      <th>1326494670_42588-20250227</th>
      <td>H22.30.001.CX.17.07.07.02</td>
      <td>1326494670</td>
      <td>sis_cellpose_v1</td>
      <td>406.79843</td>
      <td>899.12110</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>59579c65-2336-44e4-ae2e-284dd9766ed4</td>
      <td>1326494670</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#6af4cf</td>
      <td>3</td>
      <td>161</td>
      <td>10</td>
      <td>1</td>
      <td>7</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 51 columns</p>
</div></div></div>
</div>
<p>Load the Macaque and Marmoset data in the same way.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_cell_to_cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">macaque_cell_to_cluster</span><span class="p">)</span>
<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_color&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_order</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">fill_adjacent_data</span><span class="p">(</span><span class="n">macaque_cell_metadata</span><span class="p">)</span>

<span class="k">del</span> <span class="n">macaque_cell_to_cluster</span>

<span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>Neighborhood_color</th>
      <th>Class_color</th>
      <th>Subclass_color</th>
      <th>Group_color</th>
      <th>Cluster_color</th>
      <th>Class_order</th>
      <th>Cluster_order</th>
      <th>Group_order</th>
      <th>Neighborhood_order</th>
      <th>Subclass_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1291813781_1-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6328.8564</td>
      <td>271.7173</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>#19613b</td>
      <td>#d0b83c</td>
      <td>#253c8c</td>
      <td>#ff9896</td>
      <td>#681364</td>
      <td>10</td>
      <td>1289</td>
      <td>55</td>
      <td>3</td>
      <td>32</td>
    </tr>
    <tr>
      <th>1291812338_13027-20250227</th>
      <td>QM23.50.001.CX.43.01.11.03</td>
      <td>1291812338</td>
      <td>sis_cellpose_v1</td>
      <td>8314.5330</td>
      <td>2671.2260</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>6de019e9-4c63-470b-b474-3b8d23947d84</td>
      <td>1291812338</td>
      <td>...</td>
      <td>#19613b</td>
      <td>#d0b83c</td>
      <td>#79bdf4</td>
      <td>#e377c2</td>
      <td>#8f8307</td>
      <td>10</td>
      <td>1335</td>
      <td>57</td>
      <td>3</td>
      <td>33</td>
    </tr>
    <tr>
      <th>1291812338_13026-20250227</th>
      <td>QM23.50.001.CX.43.01.11.03</td>
      <td>1291812338</td>
      <td>sis_cellpose_v1</td>
      <td>8180.6080</td>
      <td>2670.6003</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>288bf294-b18b-42e9-b0fc-84234c008a8a</td>
      <td>1291812338</td>
      <td>...</td>
      <td>#19613b</td>
      <td>#d0b83c</td>
      <td>#1655f2</td>
      <td>#d62728</td>
      <td>#307e06</td>
      <td>10</td>
      <td>1166</td>
      <td>52</td>
      <td>3</td>
      <td>31</td>
    </tr>
    <tr>
      <th>1291812338_13025-20250227</th>
      <td>QM23.50.001.CX.43.01.11.03</td>
      <td>1291812338</td>
      <td>sis_cellpose_v1</td>
      <td>8336.7420</td>
      <td>2667.5422</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>37da2f12-be8f-42a3-93ab-0a31ef0e4382</td>
      <td>1291812338</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#66391F</td>
      <td>#62620e</td>
      <td>3</td>
      <td>146</td>
      <td>9</td>
      <td>1</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1291812338_13024-20250227</th>
      <td>QM23.50.001.CX.43.01.11.03</td>
      <td>1291812338</td>
      <td>sis_cellpose_v1</td>
      <td>8315.9370</td>
      <td>2664.6123</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>dca478a5-0c68-451d-a832-80e977e075cb</td>
      <td>1291812338</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#401e66</td>
      <td>#401e66</td>
      <td>#195f8d</td>
      <td>#21e5fe</td>
      <td>1</td>
      <td>25</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 51 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_cell_to_cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">marmoset_cell_to_cluster</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_color&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_order</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">fill_adjacent_data</span><span class="p">(</span><span class="n">marmoset_cell_metadata</span><span class="p">)</span>

<span class="k">del</span> <span class="n">marmoset_cell_to_cluster</span>

<span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>Neighborhood_color</th>
      <th>Class_color</th>
      <th>Subclass_color</th>
      <th>Group_color</th>
      <th>Cluster_color</th>
      <th>Class_order</th>
      <th>Cluster_order</th>
      <th>Group_order</th>
      <th>Neighborhood_order</th>
      <th>Subclass_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1382042951_abmkfncc-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5923.492676</td>
      <td>6160.654297</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>c8c08d8c-5147-4d03-b6f4-ca7bf4d0b8d1</td>
      <td>1382042951</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#66391F</td>
      <td>#5a9ad6</td>
      <td>3</td>
      <td>158</td>
      <td>9</td>
      <td>1</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1386591529_oglkpmal-1-20250227</th>
      <td>CJ23.56.004.CX.43.01.07</td>
      <td>1386591529</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>10066.064453</td>
      <td>9964.978516</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>899a0853-cb22-4b1f-a34c-6e2fad41acd2</td>
      <td>1386591529</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#995C60</td>
      <td>#995C60</td>
      <td>#995C60</td>
      <td>#54ca78</td>
      <td>2</td>
      <td>113</td>
      <td>4</td>
      <td>1</td>
      <td>3</td>
    </tr>
    <tr>
      <th>1386591529_oglkphgf-1-20250227</th>
      <td>CJ23.56.004.CX.43.01.07</td>
      <td>1386591529</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>10032.336914</td>
      <td>9951.043945</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>1f87149d-55fe-4081-b6da-c2bf68189e48</td>
      <td>1386591529</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#f5e689</td>
      <td>#f5e689</td>
      <td>#f222e1</td>
      <td>3</td>
      <td>219</td>
      <td>12</td>
      <td>1</td>
      <td>8</td>
    </tr>
    <tr>
      <th>1386591529_oglkpgff-1-20250227</th>
      <td>CJ23.56.004.CX.43.01.07</td>
      <td>1386591529</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>10078.684570</td>
      <td>9936.422852</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>37a9464c-9dff-4823-a36a-42bb7ca37fef</td>
      <td>1386591529</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#a8afa5</td>
      <td>#66493D</td>
      <td>#66493D</td>
      <td>#1bf6d7</td>
      <td>4</td>
      <td>253</td>
      <td>14</td>
      <td>1</td>
      <td>10</td>
    </tr>
    <tr>
      <th>1386591529_oglikmic-1-20250227</th>
      <td>CJ23.56.004.CX.43.01.07</td>
      <td>1386591529</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>9677.964844</td>
      <td>11089.391602</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>228d9fa6-bea9-4813-8c82-7285f1e5754b</td>
      <td>1386591529</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#995C60</td>
      <td>#CC887A</td>
      <td>#CC887A</td>
      <td>#80f3ea</td>
      <td>2</td>
      <td>130</td>
      <td>5</td>
      <td>1</td>
      <td>4</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 51 columns</p>
</div></div></div>
</div>
<p>Merging in the taxonomy allows one to select cells by the specific taxons they are associated with, as we will see in a function we create below.</p>
<section id="plotting-the-taxonomy">
<h3>Plotting the taxonomy<a class="headerlink" href="#plotting-the-taxonomy" title="Link to this heading">#</a></h3>
<p>Now that the taxonomy is loaded, we plot the the cells showing the spatial locations of the cell types in the Basal Ganglia. Below we plot each species in the dataset for each of the 5 levels in the taxonomy selection using the same set of slabs we used for the above to plot the manual anatomical annotations. As we step through each slab plane, we can see hints of the various regions of the basal ganglia, helped by the manual annotation polygons that we plot of the BG regions.</p>
<p>For clarity, we’ll filter out the Adjacent taxonomy classes as well as the Noneuronal cells. We’ll also leave an example for filtering out just the Adjacent data as well.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Filter out Adjacent</span>
<span class="c1"># filtered_human_metadata = human_cell_metadata[~(human_cell_metadata[&#39;Neighborhood&#39;] == &#39;Adjacent&#39;)]</span>
<span class="c1"># Filter out both adjacent and non-neuronal cells.</span>
<span class="n">filtered_human_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="p">[</span>
    <span class="n">np</span><span class="o">.</span><span class="n">logical_and</span><span class="p">(</span><span class="o">~</span><span class="p">(</span><span class="n">human_cell_metadata</span><span class="p">[</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">),</span>
                   <span class="n">human_cell_metadata</span><span class="p">[</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">&#39;Nonneuron&#39;</span><span class="p">)</span>
<span class="p">]</span>

<span class="k">for</span> <span class="n">color_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;Neighborhood_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Class_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Subclass_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Group_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Cluster_color&#39;</span><span class="p">]:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
        <span class="n">filtered_human_metadata</span><span class="p">,</span>
        <span class="n">color_type</span><span class="p">,</span>
        <span class="n">human_slab_list</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span>
    <span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Human: </span><span class="si">{</span><span class="n">color_type</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;_&quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/6ab3ebf02193ebd6844a9c73bae1599d30cf2b0d1607910f17d92ed8e3f443b6.png" src="../_images/6ab3ebf02193ebd6844a9c73bae1599d30cf2b0d1607910f17d92ed8e3f443b6.png" />
<img alt="../_images/3dd9424d55a7611c353c3e8d10e521d3d1e3364732171ea90a0a87700c94e796.png" src="../_images/3dd9424d55a7611c353c3e8d10e521d3d1e3364732171ea90a0a87700c94e796.png" />
<img alt="../_images/bc5fad164d42fe3b9dbfc45a0e7db2fdc8a150c9536de5440acbd5864a9ad0d9.png" src="../_images/bc5fad164d42fe3b9dbfc45a0e7db2fdc8a150c9536de5440acbd5864a9ad0d9.png" />
<img alt="../_images/2027e057abb4e583152bd9797177fa16e00bf4d742c32af63d2e311a4c7e0a97.png" src="../_images/2027e057abb4e583152bd9797177fa16e00bf4d742c32af63d2e311a4c7e0a97.png" />
<img alt="../_images/5be6bb56013b74c3a86d8b110b1be490e270f762972cf91c6bf4b7a37a07317e.png" src="../_images/5be6bb56013b74c3a86d8b110b1be490e270f762972cf91c6bf4b7a37a07317e.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Filter out Adjacent</span>
<span class="c1"># filtered_macaque_metadata = macaque_cell_metadata[~(macaque_cell_metadata[&#39;Neighborhood&#39;] == &#39;Adjacent&#39;)]</span>
<span class="c1"># Filter out both adjacent and non-neuronal cells.</span>
<span class="n">filtered_macaque_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="p">[</span>
    <span class="n">np</span><span class="o">.</span><span class="n">logical_and</span><span class="p">(</span><span class="o">~</span><span class="p">(</span><span class="n">macaque_cell_metadata</span><span class="p">[</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">),</span>
                   <span class="n">macaque_cell_metadata</span><span class="p">[</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">&#39;Nonneuron&#39;</span><span class="p">)</span>
<span class="p">]</span>
<span class="k">for</span> <span class="n">color_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;Neighborhood_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Class_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Subclass_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Group_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Cluster_color&#39;</span><span class="p">]:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
        <span class="n">filtered_macaque_metadata</span><span class="p">,</span>
        <span class="n">color_type</span><span class="p">,</span>
        <span class="n">macaque_slab_list</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;macaque&#39;</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span>
    <span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Macaque: </span><span class="si">{</span><span class="n">color_type</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;_&quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/eb88a2e1fe197eabe1874efac4b0f67ec7a0e35d56253839ce30f26b40cae315.png" src="../_images/eb88a2e1fe197eabe1874efac4b0f67ec7a0e35d56253839ce30f26b40cae315.png" />
<img alt="../_images/03e89d7a82b48592618bc634b2bf6ae479c31a443aed1eedcb29f63c7a89cfa3.png" src="../_images/03e89d7a82b48592618bc634b2bf6ae479c31a443aed1eedcb29f63c7a89cfa3.png" />
<img alt="../_images/9f5901cc3e4f23ece8a9504534b734fecb595016a7e1fca5b0a299a6ecea0dc7.png" src="../_images/9f5901cc3e4f23ece8a9504534b734fecb595016a7e1fca5b0a299a6ecea0dc7.png" />
<img alt="../_images/70fafc1634e93b96b6c797bd51a035c06ee7fa21d7486adb02807bd141cc29d1.png" src="../_images/70fafc1634e93b96b6c797bd51a035c06ee7fa21d7486adb02807bd141cc29d1.png" />
<img alt="../_images/24a11948d7766f4607a762933858f31fb3b118975650245ef810121846f6ffad.png" src="../_images/24a11948d7766f4607a762933858f31fb3b118975650245ef810121846f6ffad.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Filter out Adjacent</span>
<span class="c1"># filtered_marmoset_metadata = marmoset_cell_metadata[~(marmoset_cell_metadata[&#39;Neighborhood&#39;] == &#39;Adjacent&#39;)]</span>
<span class="c1"># Filter out both adjacent and non-neuronal cells.</span>
<span class="n">filtered_marmoset_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="p">[</span>
    <span class="n">np</span><span class="o">.</span><span class="n">logical_and</span><span class="p">(</span><span class="o">~</span><span class="p">(</span><span class="n">marmoset_cell_metadata</span><span class="p">[</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">),</span>
                   <span class="n">marmoset_cell_metadata</span><span class="p">[</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">&#39;Nonneuron&#39;</span><span class="p">)</span>
<span class="p">]</span>
<span class="k">for</span> <span class="n">color_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;Neighborhood_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Class_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Subclass_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Group_color&#39;</span><span class="p">,</span> <span class="s1">&#39;Cluster_color&#39;</span><span class="p">]:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
        <span class="n">filtered_marmoset_metadata</span><span class="p">,</span>
        <span class="n">color_type</span><span class="p">,</span>
        <span class="n">marmoset_slab_list</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;marmoset&#39;</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span>
    <span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Marmoset: </span><span class="si">{</span><span class="n">color_type</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;_&quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/0482cd104259754db54906041dcd48c6df2db9bd75ae32459414c77da7a0eaae.png" src="../_images/0482cd104259754db54906041dcd48c6df2db9bd75ae32459414c77da7a0eaae.png" />
<img alt="../_images/9c402c4f93a3a028242220298ce88430456031d91df2bb4a3d691401a04089ad.png" src="../_images/9c402c4f93a3a028242220298ce88430456031d91df2bb4a3d691401a04089ad.png" />
<img alt="../_images/a45e8fa8052b6b85d2cde10146e0af6bc98f5987f23e6349561f18a70582d4cc.png" src="../_images/a45e8fa8052b6b85d2cde10146e0af6bc98f5987f23e6349561f18a70582d4cc.png" />
<img alt="../_images/44860b0d0a11d59b538fd6940f3a1d8f9ea6135babade57f8dc150e793b7305a.png" src="../_images/44860b0d0a11d59b538fd6940f3a1d8f9ea6135babade57f8dc150e793b7305a.png" />
<img alt="../_images/1cd56b6cc387ade21694af6f524d6d83c4e8ef192b398c2c558d92ae6f6d30ba.png" src="../_images/1cd56b6cc387ade21694af6f524d6d83c4e8ef192b398c2c558d92ae6f6d30ba.png" />
</div>
</div>
</section>
<section id="selecting-cell-types">
<h3>Selecting cell types<a class="headerlink" href="#selecting-cell-types" title="Link to this heading">#</a></h3>
<p>We define a slightly modified version of our slab plotting function that allows us to plot specific cell types, coloring by cell type level that is nominally lower in the taxonomy. This allows for us to investigate the spatial distribution of cell types more clearly by, for instance, plotting the spatial distributions of Groups within the a Medium Spiny Neuron subclass.</p>
<p>In general, cells associated with individual cell types and taxons can be selected by masking the DataFrame using the desired column and value in that column, as we showed in our examples above. For instance to select the Class, CN LGE GABA, we can create a mask thusly <code class="docutils literal notranslate"><span class="pre">human_cell_metadata['Class']=='CN</span> <span class="pre">LGE</span> <span class="pre">GABA'</span></code> and similarly for the other species. Again, this function can be used to plot individual slab planes and selections as well.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_selected_cells</span><span class="p">(</span>
    <span class="n">df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
    <span class="n">feature</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">selection_list</span><span class="p">:</span> <span class="n">List</span><span class="p">[</span><span class="n">Tuple</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="nb">str</span><span class="p">]],</span>
    <span class="n">slab</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">species</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">cmap</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Colormap</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">40</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
    <span class="n">plot_polygons</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
    <span class="n">point_size</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mf">1.0</span><span class="p">,</span>
    <span class="n">fix_bounds</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>
<span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Plot spatial transcriptomics data selected a cell type and colored by</span>
<span class="sd">    a feature of interest.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    df: pandas.DataFrame</span>
<span class="sd">        cell metadata data frame with x,y locations to plot.</span>
<span class="sd">    feature: str</span>
<span class="sd">        Expression value or color column to plot.</span>
<span class="sd">    selection_list: List[str]</span>
<span class="sd">        Set of cell types to select by. Specify as (&#39;taxonomy_level_name&#39;, &#39;cell type&#39;)</span>
<span class="sd">    slab: str</span>
<span class="sd">        Specific slab to plot.</span>
<span class="sd">    species: str</span>
<span class="sd">        Species to plot (human, macaque, marmoset).</span>
<span class="sd">    cmap: matplotlib.colors.Colormap</span>
<span class="sd">        Colormap for expression features.</span>
<span class="sd">    fid_width: int</span>
<span class="sd">        Width of the plotted figure.</span>
<span class="sd">    fig_height: iont</span>
<span class="sd">        Height of the plotted figure.</span>
<span class="sd">    plot_polygons: bool</span>
<span class="sd">        Overplot geojson polygons. Note that these are note available for human</span>
<span class="sd">        and this should be set to False.</span>
<span class="sd">    point_size: float</span>
<span class="sd">        Size of the points to plot.</span>
<span class="sd">    fix_bounds: bool</span>
<span class="sd">        If True, fix the x and y bounds across all slabs to be the same.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">selection_list</span><span class="p">))</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">selection_list</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
        <span class="n">ax</span> <span class="o">=</span> <span class="p">[</span><span class="n">ax</span><span class="p">]</span>

    <span class="k">if</span> <span class="s1">&#39;slab_plane_label&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">slab_filtered</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;slab_plane_label&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">slab</span><span class="p">]</span>
    <span class="k">elif</span> <span class="s1">&#39;brain_section_label&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">slab_filtered</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">slab</span><span class="p">]</span>

    <span class="k">if</span> <span class="s1">&#39;x&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">x_col</span> <span class="o">=</span> <span class="s1">&#39;x&#39;</span>
    <span class="k">elif</span> <span class="s1">&#39;x_slab_mm&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">x_col</span> <span class="o">=</span> <span class="s1">&#39;x_slab_mm&#39;</span>
    <span class="k">if</span> <span class="s1">&#39;y&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">y_col</span> <span class="o">=</span> <span class="s1">&#39;y&#39;</span>
    <span class="k">elif</span> <span class="s1">&#39;y_slab_mm&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">y_col</span> <span class="o">=</span> <span class="s1">&#39;y_slab_mm&#39;</span>

    <span class="k">if</span> <span class="n">fix_bounds</span><span class="p">:</span>
        <span class="n">x_min</span> <span class="o">=</span> <span class="n">slab_filtered</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>
        <span class="n">x_max</span> <span class="o">=</span> <span class="n">slab_filtered</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span>
        <span class="n">y_min</span> <span class="o">=</span> <span class="n">slab_filtered</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>
        <span class="n">y_max</span> <span class="o">=</span> <span class="n">slab_filtered</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span>
    
    <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="p">(</span><span class="n">col</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">selection_list</span><span class="p">):</span>

        <span class="k">if</span> <span class="s1">&#39;slab_plane_label&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="n">filtered</span> <span class="o">=</span> <span class="n">slab_filtered</span><span class="p">[</span><span class="n">slab_filtered</span><span class="p">[</span><span class="n">col</span><span class="p">]</span><span class="o">==</span><span class="n">name</span><span class="p">]</span>
        <span class="k">elif</span> <span class="s1">&#39;brain_section_label&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="n">filtered</span> <span class="o">=</span> <span class="n">slab_filtered</span><span class="p">[</span><span class="n">slab_filtered</span><span class="p">[</span><span class="n">col</span><span class="p">]</span><span class="o">==</span><span class="n">name</span><span class="p">]</span>
        <span class="n">xx</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span>
        <span class="n">yy</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span>
        <span class="n">vv</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">feature</span><span class="p">]</span>
        
        <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">point_size</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">vv</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">)</span>
        <span class="k">else</span> <span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">point_size</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">vv</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s2">&quot;.&quot;</span><span class="p">)</span>

        <span class="k">if</span> <span class="n">plot_polygons</span><span class="p">:</span>
            <span class="n">plot_geo_data</span><span class="p">(</span><span class="n">get_polygons</span><span class="p">(</span><span class="n">slab</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="n">species</span><span class="p">),</span> <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">])</span>
            
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s1">&#39;equal&#39;</span><span class="p">)</span>
        <span class="n">ylim</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()</span>
        <span class="k">if</span> <span class="n">fix_bounds</span><span class="p">:</span>
            <span class="n">ylim</span> <span class="o">=</span> <span class="p">(</span><span class="n">y_min</span><span class="p">,</span> <span class="n">y_max</span><span class="p">)</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">x_min</span><span class="p">,</span> <span class="n">x_max</span><span class="p">)</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="n">ylim</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">ylim</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
        
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        
    <span class="n">plt</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.01</span><span class="p">,</span> <span class="n">hspace</span><span class="o">=</span><span class="mf">0.01</span><span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">species</span><span class="si">}</span><span class="s1">, </span><span class="si">{</span><span class="n">slab</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<p>Using the above function, we plot the cells colored by their Group color and select by specific classes and subclasses. We can see commonalities in the distribution across the different species and sections,of the cell types exceptionally clearly.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">slab</span> <span class="ow">in</span> <span class="n">human_slab_list</span><span class="p">:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_selected_cells</span><span class="p">(</span>
        <span class="n">human_cell_metadata</span><span class="p">,</span>
        <span class="n">feature</span><span class="o">=</span><span class="s1">&#39;Group_color&#39;</span><span class="p">,</span>
        <span class="n">selection_list</span><span class="o">=</span><span class="p">[</span>
            <span class="p">(</span><span class="s1">&#39;Class&#39;</span><span class="p">,</span> <span class="s1">&#39;CN LGE GABA&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR D1 MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR D2 MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR Hybrid MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;OT Granular GABA&#39;</span><span class="p">)</span>
        <span class="p">],</span>
        <span class="n">slab</span><span class="o">=</span><span class="n">slab</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span>
        <span class="n">fig_width</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span>
        <span class="n">fig_height</span><span class="o">=</span><span class="mi">6</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/00b99fe9328db383feda795212176591d6ccd24ed9a137a3bd62c026f5ccfe03.png" src="../_images/00b99fe9328db383feda795212176591d6ccd24ed9a137a3bd62c026f5ccfe03.png" />
<img alt="../_images/39819bd368bf41ed023747e596cb73e0abab733867348996f1d9b8db8c6b676c.png" src="../_images/39819bd368bf41ed023747e596cb73e0abab733867348996f1d9b8db8c6b676c.png" />
<img alt="../_images/8978c03dc8044e1fa3e8ca3e84c3a3a3f0cf072d0f740f8bfeb146d3f3e887eb.png" src="../_images/8978c03dc8044e1fa3e8ca3e84c3a3a3f0cf072d0f740f8bfeb146d3f3e887eb.png" />
<img alt="../_images/c6d8c9d0b68b281eddb1cdce11f4f01fdb3edd20993b5e7cc81325eee2b4c891.png" src="../_images/c6d8c9d0b68b281eddb1cdce11f4f01fdb3edd20993b5e7cc81325eee2b4c891.png" />
<img alt="../_images/8d3cbea5f7d6574868ea0002aabb90b1348269b7bfc63df819744632b518d55f.png" src="../_images/8d3cbea5f7d6574868ea0002aabb90b1348269b7bfc63df819744632b518d55f.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">slab</span> <span class="ow">in</span> <span class="n">macaque_slab_list</span><span class="p">:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_selected_cells</span><span class="p">(</span>
        <span class="n">macaque_cell_metadata</span><span class="p">,</span>
        <span class="n">feature</span><span class="o">=</span><span class="s1">&#39;Group_color&#39;</span><span class="p">,</span>
        <span class="n">selection_list</span><span class="o">=</span><span class="p">[</span>
            <span class="p">(</span><span class="s1">&#39;Class&#39;</span><span class="p">,</span> <span class="s1">&#39;CN LGE GABA&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR D1 MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR D2 MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR Hybrid MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;OT Granular GABA&#39;</span><span class="p">)</span>
        <span class="p">],</span>
        <span class="n">slab</span><span class="o">=</span><span class="n">slab</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;macaque&#39;</span><span class="p">,</span>
        <span class="n">fig_width</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span>
        <span class="n">fig_height</span><span class="o">=</span><span class="mi">6</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/d050dceb29b69bffcb6bebc97f4089c72bc0bdc8468f4dd5a3758aa4c39606d6.png" src="../_images/d050dceb29b69bffcb6bebc97f4089c72bc0bdc8468f4dd5a3758aa4c39606d6.png" />
<img alt="../_images/926c1f0ac9baeefebdd60d218f728bd832abf7b45264b2fe4467769857a7b399.png" src="../_images/926c1f0ac9baeefebdd60d218f728bd832abf7b45264b2fe4467769857a7b399.png" />
<img alt="../_images/519bad487a5274cfa7ef8fcf3fae2cd11a8f2a40985730236b7fca04960fc08c.png" src="../_images/519bad487a5274cfa7ef8fcf3fae2cd11a8f2a40985730236b7fca04960fc08c.png" />
<img alt="../_images/a7af84b66f340f6992811d17d3ce101bde7ae234aa3ed495842ac635be125837.png" src="../_images/a7af84b66f340f6992811d17d3ce101bde7ae234aa3ed495842ac635be125837.png" />
<img alt="../_images/2f0ad310733546f50811dbd6d03830aa60793b68b14a94c00cf833996c75ad50.png" src="../_images/2f0ad310733546f50811dbd6d03830aa60793b68b14a94c00cf833996c75ad50.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">slab</span> <span class="ow">in</span> <span class="n">marmoset_slab_list</span><span class="p">:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_selected_cells</span><span class="p">(</span>
        <span class="n">marmoset_cell_metadata</span><span class="p">,</span>
        <span class="n">feature</span><span class="o">=</span><span class="s1">&#39;Group_color&#39;</span><span class="p">,</span>
        <span class="n">selection_list</span><span class="o">=</span><span class="p">[</span>
            <span class="p">(</span><span class="s1">&#39;Class&#39;</span><span class="p">,</span> <span class="s1">&#39;CN LGE GABA&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR D1 MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR D2 MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;STR Hybrid MSN&#39;</span><span class="p">),</span>
            <span class="p">(</span><span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;OT Granular GABA&#39;</span><span class="p">)</span>
        <span class="p">],</span>
        <span class="n">slab</span><span class="o">=</span><span class="n">slab</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;marmoset&#39;</span><span class="p">,</span>
        <span class="n">fig_width</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span>
        <span class="n">fig_height</span><span class="o">=</span><span class="mi">6</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/998ae291c892cbff78e8e12c868a7b4318fa2bfef64561865e090763cfcbc4a2.png" src="../_images/998ae291c892cbff78e8e12c868a7b4318fa2bfef64561865e090763cfcbc4a2.png" />
<img alt="../_images/0bea78530499c174ca692a376fb7fa79db1c4d5e89532246f53910001018cc0f.png" src="../_images/0bea78530499c174ca692a376fb7fa79db1c4d5e89532246f53910001018cc0f.png" />
<img alt="../_images/bb9d81e0e258cf0c32790dfe172c24b71b250ad2af13bf1d3b374dff810d6bf1.png" src="../_images/bb9d81e0e258cf0c32790dfe172c24b71b250ad2af13bf1d3b374dff810d6bf1.png" />
<img alt="../_images/dc8bd4f749b59df45309f7a8694e7c5143df83a2df595917294cdeba320f505a.png" src="../_images/dc8bd4f749b59df45309f7a8694e7c5143df83a2df595917294cdeba320f505a.png" />
<img alt="../_images/137b2cf8ec30d0fb66e2efcd2d0fe1f735ff577860fff261b75108993a9b08bc.png" src="../_images/137b2cf8ec30d0fb66e2efcd2d0fe1f735ff577860fff261b75108993a9b08bc.png" />
</div>
</div>
<p>In the next tutorial, we will explore <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_spatial_genes.html">gene expression patterns across these spatial transcriptomics datasets</a>.</p>
<p>To learn more about the taxonomy used in this notebook and the data it is derived from, see the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_clustering_analysis_and_annotation.html">HMBA-BG Clustering Analysis and Taxonomy notebook</a>.</p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="hmba_bg_10X_snRNASeq_tutorial.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression</p>
      </div>
    </a>
    <a class="right-next"
       href="hmba_bg_spatial_genes.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data Overview</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#specimen-metadata">Specimen metadata</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#spatial-slab-plane-coordinates">Spatial, slab-plane coordinates</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#geojson-polygons">Geojson polygons</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#manual-anatomical-annotations">Manual anatomical annotations</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-slabs">Plotting the Slabs</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#displaying-the-manual-anatomical-annotations">Displaying the manual anatomical annotations.</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxonomy-mapping">Taxonomy mapping</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-taxonomy">Plotting the taxonomy</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#selecting-cell-types">Selecting cell types</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>