
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression &#8212; Allen Brain Cell Atlas - Data Access</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=9c3e77be" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/simple_style.css?v=3aaed5dc" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=1ae7504c"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/hmba_bg_spatial_genes';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Resource Pages" href="../docs/README.html" />
    <link rel="prev" title="Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation" href="hmba_bg_spatial_slabs_and_taxonomy.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/allen_logo.jpeg" class="logo__image only-light" alt="Allen Brain Cell Atlas - Data Access - Home"/>
    <script>document.write(`<img src="../_static/allen_logo.jpeg" class="logo__image only-dark" alt="Allen Brain Cell Atlas - Data Access - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Allen Brain Cell Atlas - Data Access
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">INTRODUCTION</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="general_accessing_10x_snRNASeq_tutorial.html">Accessing 10x RNA-seq gene expression data</a></li>
<li class="toctree-l1"><a class="reference internal" href="abc_atlas_selection_example.html">Using cells selected in the ABC Atlas with Gene Expression data.</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">DATA</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WMB_dataset.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv2.html">Whole mouse brain 10Xv2 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv3.html">Whole mouse brain 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10XMulti.html">Whole mouse brain 10XMulti single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10X.html">Clustering analysis of whole adult mouse brain 10X single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-taxonomy.html">Whole adult mouse brain taxonomy of cell types</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-imputed.html">Imputed MERFISH spatial transcriptomics of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-neighborhoods.html">Cluster groups and neighborhood specific UMAP embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Allen-CCF-2020.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-CCF.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_dataset.html">Aging Mouse dataset (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_10Xv3.html">Zeng Aging Mouse 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_taxonomy.html">Zeng Aging Mouse clustering and mapping to the WMB taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zhuang_dataset.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-1.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-2.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-3.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-3)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-4.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-4)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-dataset.html">Consensus Whole Mouse Brain dataset (Macosko, Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-10X.html">Consensus Whole Mouse Brain single cell/nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Consensus-WMB-taxonomy.html">Consensus Whole Mouse Brain integrated taxonomy</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_dataset.html">Human whole-brain transcriptomic cell type atlas (Kimberly Siletti)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10Xv3.html">Human brain 10Xv3 single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-10X-clustering.html">Clustering analysis of whole human brain 10X single nucleus transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WHB-taxonomy.html">Human brain taxonomy of cell types</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_dataset.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-10X.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/ASAP-PMDBS-MapMyCells.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain (WHB) and Seattle Alzheimer’s Disease Brain Cell Atlas (SEA-AD) Taxonomies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/HMBA-BG_dataset.html">Human-Mammalian Brain - Basal Ganglia: Data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-BG-taxonomy-CCN20250428.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq data: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG-Aligned.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Aligned gene expression across species</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-10XMultiome-BG.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq data: Individual species gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/HMBA-Spatial-BG.html">Human-Mammalian Brain - Basal Ganglia: spatial transcriptomics cells for each species</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">NOTEBOOKS</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cluster_annotation_tutorial.html">10x RNA-seq clustering analysis and annotation (CCN20230722)</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle2.html">10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_1.html">10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2a.html">10x RNA-seq gene expression data (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2b.html">10x RNA-seq gene expression data (part 2b)</a></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle3.html">MERFISH whole brain spatial transcriptomics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_1.html">MERFISH whole brain spatial transcriptomics (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2a.html">MERFISH whole brain spatial transcriptomics (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2b.html">MERFISH whole brain spatial transcriptomics (part 2b)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_imputed_genes_example.html">MERFISH whole brain spatial transcriptomics cells with imputed genes</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_groups_and_embeddings_tutorial.html">Cluster neighborhoods and embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_neighborhood_gallery.html">Cell type neighborhood gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="ccf_and_parcellation_annotation_tutorial.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="merfish_ccf_registration_tutorial.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Consensus-WMB-notebooks.html">Consensus Mouse notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_clustering_analysis_and_annotation.html">Consensus Whole Mouse Brain #1: Clustering and Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="consensus_mouse_10X_snRNASeq.html">Consensus Whole Mouse Brain #2: 10X gene expression</a></li>

</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/Zeng_Aging_Mouse_notebooks.html">Aging Mouse Notebooks (Hongkui Zeng)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_clustering_analysis_and_annotation.html">Mouse Aging clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="Zeng_Aging_Mouse_10x_snRNASeq_tutorial.html">Mouse Aging 10X RNA-seq gene expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle4.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="zhuang_merfish_tutorial.html">MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/WHB_notebooks.html">Whole Human Brain 10x scRNA-seq gene expression data</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="WHB_cluster_annotation_tutorial.html">Whole Human Brain 10x RNA-seq clustering analysis and annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_1.html">Whole Human Brain (WHB) 10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="WHB-10x_snRNASeq_tutorial_part_2.html">Whole Human Brain 10x RNA-seq gene expression data (part 2)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/ASAP-PMDBS_notebooks.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (ASAP-PMDBS) Notebooks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_data_and_metadata.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Data Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_siletti_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the Whole Human Brain Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_seaad_taxonomy.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Mapping to the SEA-AD Taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="asap_pmdbs_gene_expression.html">ASAP Human Postmortem-Derived Brain Sequencing Collection (PMDBS): Gene Expression</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../descriptions/HMBA-BG_notebooks.html">Human-Mammalian Brain - Basal Ganglia: Notebooks</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_clustering_analysis_and_annotation.html">Human-Mammalian Brain - Basal Ganglia 10X snRNASeq analysis: clustering and annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_10X_snRNASeq_tutorial.html">Human-Mammalian Brain - Basal Ganglia 10X snRANSeq analysis: gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmba_bg_spatial_slabs_and_taxonomy.html">Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">ADDITIONAL RESOURCES</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../docs/README.html">Resource Pages</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../docs/abbreviation_terms/README.html">Abbreviation lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/cluster_annotations/README.html">Cluster annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/gene_lists/README.html">Gene lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/parcellation_annotations/README.html">Parcellation annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/supplemental_metadata_descriptions/README.html">Supplemental Metadata Descriptions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/neuroglancer_hmba/README.html">HMBA Basal Ganglia: Neuroglancer Views</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access/issues/new?title=Issue%20on%20page%20%2Fnotebooks/hmba_bg_spatial_genes.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/hmba_bg_spatial_genes.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#geojson-polygons">Geojson polygons</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-expression-data">Gene expression data</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-slabs">Plotting the Slabs</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#manual-anatomical-annotations">Manual anatomical annotations</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-heatmap-expression-data">Plotting heatmap expression data.</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualize-gene-expression-for-selected-genes-by-manual-annotation-region">Visualize gene expression for selected genes by manual annotation region.</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxonomy-mapping">Taxonomy mapping</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualize-gene-expression-for-selected-genes-by-the-hmba-bg-taxonomy">Visualize gene expression for selected genes by the HMBA-BG taxonomy.</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="human-mammalian-brain-basal-ganglia-spatial-transcriptomics-gene-expression">
<h1>Human-Mammalian Brain - Basal Ganglia spatial transcriptomics gene expression<a class="headerlink" href="#human-mammalian-brain-basal-ganglia-spatial-transcriptomics-gene-expression" title="Link to this heading">#</a></h1>
<p>In this notebook, we build on the our <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_spatial_slabs_and_taxonomy.html">previous investigation</a> into the cell level metadata, combining the metadata gene expressions for select genes that we extract from the released h5ad files.</p>
<p>You need to be connected to the internet to run this notebook and have run through the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html">getting started notebook</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">geojson</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">typing</span><span class="w"> </span><span class="kn">import</span> <span class="n">List</span><span class="p">,</span> <span class="n">Tuple</span><span class="p">,</span> <span class="n">Optional</span>

<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpl</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">abc_atlas_access.abc_atlas_cache.abc_project_cache</span><span class="w"> </span><span class="kn">import</span> <span class="n">AbcProjectCache</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">abc_atlas_access.abc_atlas_cache.anndata_utils</span><span class="w"> </span><span class="kn">import</span> <span class="n">get_gene_data</span>
</pre></div>
</div>
</div>
</div>
<p>We will interact with the data using the <strong>AbcProjectCache</strong>. This cache object tracks which data has been downloaded and serves the path to the requested data on disk. For metadata, the cache can also directly serve a up a Pandas DataFrame. See the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/getting_started.html"><code class="docutils literal notranslate"><span class="pre">getting_started</span></code></a> notebook for more details on using the cache including installing the cache if it has not already been.</p>
<p><strong>Change the download_base variable to where you would like to download the data in your system.</strong></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">download_base</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">&#39;../../data/abc_atlas&#39;</span><span class="p">)</span>
<span class="n">abc_cache</span> <span class="o">=</span> <span class="n">AbcProjectCache</span><span class="o">.</span><span class="n">from_cache_dir</span><span class="p">(</span>
    <span class="n">download_base</span><span class="p">,</span>
<span class="p">)</span>

<span class="n">abc_cache</span><span class="o">.</span><span class="n">current_manifest</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;releases/20251031/manifest.json&#39;
</pre></div>
</div>
</div>
</div>
<section id="cell-metadata">
<h2>Cell metadata<a class="headerlink" href="#cell-metadata" title="Link to this heading">#</a></h2>
<p>Each species in the HMBA spatial dataset has it’s own h5ad files and associated cell metadata tables. We load each of these in turn starting from Human to Macaque to Marmoset. In the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_spatial_slabs_and_taxonomy.html">previous tutorial</a>, we merged the cell metadata tables with their associated specimen metadatadata and then with the slab_plane coordinates we use to plot the cell locations.</p>
<p>Below we load each of the three species of the HMBA-BG data. We load the metadata tables we need for each species and join them in one cell this time.</p>
<p>Load the Human cells.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load the human cell metadata and specimen metadata</span>
<span class="c1"># and join them together with the slab plane coordinates.</span>

<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">human_specimen</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;specimen_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">)</span>

<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">human_specimen</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_specimen&#39;</span><span class="p">)</span>

<span class="n">human_cell_slab_plane_coordinates</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;slab_plane_coordinates&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">human_cell_slab_plane_coordinates</span><span class="p">[[</span><span class="s1">&#39;x_slab_mm&#39;</span><span class="p">,</span> <span class="s1">&#39;y_slab_mm&#39;</span><span class="p">]],</span>
    <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cell_label&#39;</span>
<span class="p">)</span>

<span class="k">del</span> <span class="n">human_specimen</span>
<span class="k">del</span> <span class="n">human_cell_slab_plane_coordinates</span>

<span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>brain_slab_thickness</th>
      <th>hemisphere</th>
      <th>brain_division_label</th>
      <th>brain_division_ordinal</th>
      <th>donor_label</th>
      <th>slab_plane_ordinal</th>
      <th>slab_plane_label</th>
      <th>slab_plane_order</th>
      <th>x_slab_mm</th>
      <th>y_slab_mm</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1324989690_1-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7564.9175</td>
      <td>1985.1162</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>ddc7b0c5-1bb0-4f02-a1e4-ddacac84c9aa</td>
      <td>1324989690</td>
      <td>...</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.CX</td>
      <td>1</td>
      <td>H22.30.001</td>
      <td>3</td>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>11703</td>
      <td>28.232720</td>
      <td>79.516699</td>
    </tr>
    <tr>
      <th>1324989690_2-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7553.6514</td>
      <td>1986.4812</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>52d3e787-0140-42d2-b7cf-026f398cb597</td>
      <td>1324989690</td>
      <td>...</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.CX</td>
      <td>1</td>
      <td>H22.30.001</td>
      <td>3</td>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>11703</td>
      <td>28.242950</td>
      <td>79.520359</td>
    </tr>
    <tr>
      <th>1324989690_3-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7687.5170</td>
      <td>1987.5452</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>4af7a7a0-823a-47bd-bbe5-344ddd70cc9e</td>
      <td>1324989690</td>
      <td>...</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.CX</td>
      <td>1</td>
      <td>H22.30.001</td>
      <td>3</td>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>11703</td>
      <td>28.129210</td>
      <td>79.461518</td>
    </tr>
    <tr>
      <th>1324989690_4-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7574.2090</td>
      <td>1999.2940</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>a4455148-9d49-41b7-8750-00d7a3fb225e</td>
      <td>1324989690</td>
      <td>...</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.CX</td>
      <td>1</td>
      <td>H22.30.001</td>
      <td>3</td>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>11703</td>
      <td>28.231203</td>
      <td>79.500081</td>
    </tr>
    <tr>
      <th>1324989690_5-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7432.2330</td>
      <td>1999.0550</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>41946bab-3c5a-4890-ae7b-cbffcab06c3e</td>
      <td>1324989690</td>
      <td>...</td>
      <td>4.0</td>
      <td>right</td>
      <td>H22.30.001.CX</td>
      <td>1</td>
      <td>H22.30.001</td>
      <td>3</td>
      <td>H22.30.001.CX.17.Plane.03</td>
      <td>11703</td>
      <td>28.352236</td>
      <td>79.561696</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 31 columns</p>
</div></div></div>
</div>
<p>Load the Macaque cells.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">macaque_specimen</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;specimen_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">)</span>

<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">macaque_specimen</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_specimen&#39;</span><span class="p">)</span>

<span class="n">macaque_cell_slab_plane_coordinates</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;slab_plane_coordinates&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">macaque_cell_slab_plane_coordinates</span><span class="p">[[</span><span class="s1">&#39;x_slab_mm&#39;</span><span class="p">,</span> <span class="s1">&#39;y_slab_mm&#39;</span><span class="p">]],</span>
    <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cell_label&#39;</span>
<span class="p">)</span>

<span class="k">del</span> <span class="n">macaque_specimen</span>
<span class="k">del</span> <span class="n">macaque_cell_slab_plane_coordinates</span>

<span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>brain_slab_thickness</th>
      <th>hemisphere</th>
      <th>brain_division_label</th>
      <th>brain_division_ordinal</th>
      <th>donor_label</th>
      <th>slab_plane_ordinal</th>
      <th>slab_plane_label</th>
      <th>slab_plane_order</th>
      <th>x_slab_mm</th>
      <th>y_slab_mm</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1291813781_1-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6328.8564</td>
      <td>271.71730</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>7</td>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>14407</td>
      <td>49.982522</td>
      <td>35.779509</td>
    </tr>
    <tr>
      <th>1291813781_2-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6313.3560</td>
      <td>297.27830</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>7</td>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>14407</td>
      <td>49.986300</td>
      <td>35.748134</td>
    </tr>
    <tr>
      <th>1291813781_3-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6323.7240</td>
      <td>297.38340</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>7</td>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>14407</td>
      <td>49.976306</td>
      <td>35.751906</td>
    </tr>
    <tr>
      <th>1291813781_4-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6370.7470</td>
      <td>377.92710</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>7</td>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>14407</td>
      <td>49.896221</td>
      <td>35.688888</td>
    </tr>
    <tr>
      <th>1291813781_5-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6375.0240</td>
      <td>386.81006</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>QM23.50.001.CX</td>
      <td>1</td>
      <td>QM23.50.001</td>
      <td>7</td>
      <td>QM23.50.001.CX.44.Plane.07</td>
      <td>14407</td>
      <td>49.888261</td>
      <td>35.681598</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 31 columns</p>
</div></div></div>
</div>
<p>Load the Marmoset cells.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">marmoset_specimen</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;specimen_metadata&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">)</span>

<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">marmoset_specimen</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_specimen&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_slab_plane_coordinates</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;slab_plane_coordinates&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span> <span class="nb">str</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">marmoset_cell_slab_plane_coordinates</span><span class="p">[[</span><span class="s1">&#39;x_slab_mm&#39;</span><span class="p">,</span> <span class="s1">&#39;y_slab_mm&#39;</span><span class="p">]],</span>
    <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cell_label&#39;</span>
<span class="p">)</span>

<span class="k">del</span> <span class="n">marmoset_specimen</span>
<span class="k">del</span> <span class="n">marmoset_cell_slab_plane_coordinates</span>

<span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>brain_slab_thickness</th>
      <th>hemisphere</th>
      <th>brain_division_label</th>
      <th>brain_division_ordinal</th>
      <th>donor_label</th>
      <th>slab_plane_ordinal</th>
      <th>slab_plane_label</th>
      <th>slab_plane_order</th>
      <th>x_slab_mm</th>
      <th>y_slab_mm</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1382042951_abmkfncc-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5923.492676</td>
      <td>6160.654297</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>c8c08d8c-5147-4d03-b6f4-ca7bf4d0b8d1</td>
      <td>1382042951</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>1</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>14201</td>
      <td>22.351041</td>
      <td>13.197057</td>
    </tr>
    <tr>
      <th>1382042951_abmmdgea-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>6137.572266</td>
      <td>6217.075684</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>f10bad2c-9cc7-44bf-bd7e-4e61d28121bd</td>
      <td>1382042951</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>1</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>14201</td>
      <td>22.161344</td>
      <td>13.311199</td>
    </tr>
    <tr>
      <th>1382042951_abmojfce-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5802.674805</td>
      <td>5614.007812</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>080466ff-3dea-44f1-b6e2-7892e8c12c95</td>
      <td>1382042951</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>1</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>14201</td>
      <td>22.313860</td>
      <td>12.638454</td>
    </tr>
    <tr>
      <th>1382042951_abmomnim-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5805.942383</td>
      <td>5599.325684</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>cee3bf8b-a159-42d1-983a-ac149b9edda9</td>
      <td>1382042951</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>1</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>14201</td>
      <td>22.306610</td>
      <td>12.625276</td>
    </tr>
    <tr>
      <th>1382042951_aeiamgga-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>6291.268066</td>
      <td>6068.720215</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>3f24c46e-9ab1-4a54-a6fc-591f752edb48</td>
      <td>1382042951</td>
      <td>...</td>
      <td>5.0</td>
      <td>right</td>
      <td>CJ23.56.004.CX</td>
      <td>1</td>
      <td>CJ23.56.004</td>
      <td>1</td>
      <td>CJ23.56.004.CX.42.Plane.01</td>
      <td>14201</td>
      <td>21.972238</td>
      <td>13.211848</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 31 columns</p>
</div></div></div>
</div>
<section id="geojson-polygons">
<h3>Geojson polygons<a class="headerlink" href="#geojson-polygons" title="Link to this heading">#</a></h3>
<p>The dataset includes polygons outlining select regions of the Basal Ganglia, which can be plotted over slab coordinates. These polygon annotations were manually drawn using a combination of taxonomy cell types, select gene expression from the spatial transcriptomics gene panel, and cell segmentation stains with best attempts made to be consistent across species and planes. Manual annotations include the striatum (STR; encompasses the caudate, putamen, nucleus accumbens, and olfactory tubercle), the globus pallidus (GP) and it’s nested subregions the external (GPe) and internal (GPi) segments, the ventral pallidum (VeP), the substantia nigra (SN), and the subthalamic nucleus (STH). However, please note that these polygons are not derived from registration to the Harmonized Ontology of Mammalian Brain Anatomy (HOMBA). They should instead be used as cellular groupings to orient the user in the anatomical context of the spatial transcriptomics data and subset the data into more manageable, anatomically-related chunks. These polygons are provided in geojson format, and here we use the associated geojson python package to load the jsons for use in our visualization. First though, we load the term set for these data. This provides colors and plotting order for the annotations that we can use in our figures.These are provided in <a class="reference external" href="https://geojson.org/">geojson</a> format. We use the associated python package to load the jsons for use in our analysis.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">value_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;value_sets&#39;</span><span class="p">,</span>
<span class="p">)</span>
<span class="n">pred</span> <span class="o">=</span> <span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span>
<span class="n">parcellation_terms</span> <span class="o">=</span> <span class="n">value_sets</span><span class="p">[</span><span class="n">pred</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">parcellation_terms</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">parcellation_terms</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>value_sets.csv: 100%|██████████| 3.79k/3.79k [00:00&lt;00:00, 45.4kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>table</th>
      <th>field</th>
      <th>description</th>
      <th>order</th>
      <th>external_identifier</th>
      <th>parent_label</th>
      <th>color_hex_triplet</th>
    </tr>
    <tr>
      <th>label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Basal ganglia and adjacent</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>Basal ganglia and adjacent</td>
      <td>1</td>
      <td>basal_ganglia_and_adjacent</td>
      <td>NaN</td>
      <td>#BEBEBE</td>
    </tr>
    <tr>
      <th>STR</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>striatum</td>
      <td>2</td>
      <td>DHBA:10333</td>
      <td>Basal ganglia and adjacent</td>
      <td>#cd86cc</td>
    </tr>
    <tr>
      <th>GP</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>globus pallidus</td>
      <td>3</td>
      <td>DHBA:10342</td>
      <td>Basal ganglia and adjacent</td>
      <td>#7AE361</td>
    </tr>
    <tr>
      <th>GPe</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>external segment of globus pallidus</td>
      <td>4</td>
      <td>DHBA:10343</td>
      <td>Basal ganglia and adjacent</td>
      <td>#53DB33</td>
    </tr>
    <tr>
      <th>GPi</th>
      <td>path</td>
      <td>parcellation_term_symbol</td>
      <td>internal segment of globus pallidus</td>
      <td>5</td>
      <td>DHBA:10344</td>
      <td>Basal ganglia and adjacent</td>
      <td>#A7DB33</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Below we define two functions that will be used in our plotting of the geojsons. The first takes in the data and a matplotlib axis object and plots the geojson data on the provided axis, extracting color and name information from the value/term set. The second loads the data based on the appropriate slab and species provided.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_geo_data</span><span class="p">(</span><span class="n">geo_data</span><span class="p">:</span> <span class="nb">dict</span><span class="p">,</span> <span class="n">ax</span><span class="p">:</span> <span class="n">plt</span><span class="o">.</span><span class="n">Axes</span><span class="p">,</span> <span class="n">terms</span><span class="o">=</span><span class="n">parcellation_terms</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Plot a geojson polygon over the input axis.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    geo_data: dict</span>
<span class="sd">        Geojson formatted dictionary containing Polygons to plot.</span>
<span class="sd">    ax: matplotlib.pyplot.axis</span>
<span class="sd">        Axis object to plot the geojson data over.</span>
<span class="sd">    terms: pandas.DataFrame</span>
<span class="sd">        Term list containing the colors and orders for the parcellation, brain region terms. </span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">geo_data</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="k">return</span> <span class="kc">None</span>
    <span class="c1"># If the geo_data is a FeatureCollection, we need to iterate over its</span>
    <span class="c1"># features and plot each one.</span>
    <span class="k">if</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;FeatureCollection&#39;</span><span class="p">:</span>
        <span class="k">for</span> <span class="n">sub_data</span> <span class="ow">in</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;features&#39;</span><span class="p">]:</span>
            <span class="n">plot_geo_data</span><span class="p">(</span><span class="n">sub_data</span><span class="p">,</span> <span class="n">ax</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Feature&#39;</span><span class="p">:</span>
        <span class="n">geometry_data</span> <span class="o">=</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;geometry&#39;</span><span class="p">]</span>
        <span class="n">name</span> <span class="o">=</span> <span class="n">geo_data</span><span class="p">[</span><span class="s1">&#39;properties&#39;</span><span class="p">][</span><span class="s1">&#39;label&#39;</span><span class="p">]</span>
        <span class="n">color</span> <span class="o">=</span> <span class="n">terms</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">name</span><span class="p">,</span> <span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span> <span class="c1"># Get the color for this object from the term list.</span>
        <span class="c1"># If the geometry is a GeometryCollection, we need to iterate over its</span>
        <span class="c1"># geometries that make up the complete feature.</span>
        <span class="k">if</span> <span class="n">geometry_data</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;GeometryCollection&#39;</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">geometry</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">geometry_data</span><span class="p">[</span><span class="s1">&#39;geometries&#39;</span><span class="p">]):</span>
                <span class="n">coords</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">geojson</span><span class="o">.</span><span class="n">coords</span><span class="p">(</span><span class="n">geometry</span><span class="p">)))</span>
                <span class="c1"># Only label the first geometry in the collection. Others</span>
                <span class="c1"># will be plotted without a label to avoid duplicate legend entries.</span>
                <span class="k">if</span> <span class="n">idx</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
                    <span class="n">ax</span><span class="o">.</span><span class="n">fill</span><span class="p">(</span><span class="n">coords</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">],</span>
                            <span class="n">coords</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">],</span>
                            <span class="n">edgecolor</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
                            <span class="n">fill</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                            <span class="n">linewidth</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
                            <span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">)</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">ax</span><span class="o">.</span><span class="n">fill</span><span class="p">(</span><span class="n">coords</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">],</span>
                            <span class="n">coords</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">],</span>
                            <span class="n">edgecolor</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
                            <span class="n">fill</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                            <span class="n">linewidth</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">coords</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">geojson</span><span class="o">.</span><span class="n">coords</span><span class="p">(</span><span class="n">geometry_data</span><span class="p">)))</span>
            <span class="k">try</span><span class="p">:</span>
                <span class="n">ax</span><span class="o">.</span><span class="n">fill</span><span class="p">(</span><span class="n">coords</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">],</span>
                        <span class="n">coords</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">],</span>
                        <span class="n">edgecolor</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
                        <span class="n">fill</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">linewidth</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
                        <span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">)</span>
            <span class="k">except</span> <span class="ne">ValueError</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
                <span class="kn">import</span><span class="w"> </span><span class="nn">pdb</span><span class="p">;</span> <span class="n">pdb</span><span class="o">.</span><span class="n">set_trace</span><span class="p">()</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;unknown data&quot;</span><span class="p">,</span> <span class="n">geo_data</span><span class="p">)</span>

<span class="k">def</span><span class="w"> </span><span class="nf">get_polygons</span><span class="p">(</span><span class="n">slab_name</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">species</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">dict</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Load the geojson polygons for the specified slab and species.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    slab_name: str</span>
<span class="sd">        Name of the slab to load.</span>
<span class="sd">    species: str</span>
<span class="sd">        Species of the slab, one of &#39;macaque&#39;, &#39;marmoset&#39;, or &#39;human&#39;.</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    dict</span>
<span class="sd">        Geojson formatted dictionary containing the polygons for the specified slab and species.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">if</span> <span class="n">species</span> <span class="o">==</span> <span class="s1">&#39;macaque&#39;</span><span class="p">:</span>
        <span class="n">base_dir</span> <span class="o">=</span> <span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span>
    <span class="k">elif</span> <span class="n">species</span> <span class="o">==</span> <span class="s1">&#39;marmoset&#39;</span><span class="p">:</span>
        <span class="n">base_dir</span> <span class="o">=</span> <span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span>
    <span class="k">elif</span> <span class="n">species</span> <span class="o">==</span> <span class="s1">&#39;human&#39;</span><span class="p">:</span>
        <span class="n">base_dir</span> <span class="o">=</span> <span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s1">&#39;Invalid species&#39;</span><span class="p">)</span>

    <span class="n">file_path</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_file_path</span><span class="p">(</span><span class="n">directory</span><span class="o">=</span><span class="n">base_dir</span><span class="p">,</span> <span class="n">file_name</span><span class="o">=</span><span class="n">slab_name</span><span class="p">)</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">file_path</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="k">return</span> <span class="kc">None</span>

    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">file_path</span><span class="p">)</span> <span class="k">as</span> <span class="n">jfile</span><span class="p">:</span>
        <span class="n">data</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">jfile</span><span class="p">)</span>
    
    <span class="k">return</span> <span class="n">data</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="gene-expression-data">
<h2>Gene expression data<a class="headerlink" href="#gene-expression-data" title="Link to this heading">#</a></h2>
<p>Now we can extract expression for specific genes from the h5ad files and pair it with our gene metadata. In general, we can use the function <code class="docutils literal notranslate"><span class="pre">get_gene_data</span></code> to extract the expression of specific genes for all the cells in a given dataset or a subset of cells.. For more details on using this convenience function, see the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/general_accessing_10x_snRNASeq_tutorial.html">Accessing 10x RNA-seq gene expression data</a> tutorial notebook.</p>
<p>For this tutorial we will use precomputed tables of the expression for specific genes to compare expression both spatially and by the taxonomy.</p>
<p>To use the gene expression data, we first load the set of genes for each of the 3 species in HMBA-BG.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_genes</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;gene&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;gene_identifier&#39;</span><span class="p">)</span>
<span class="n">human_genes</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>gene.csv: 100%|██████████| 27.4k/27.4k [00:00&lt;00:00, 209kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene_symbol</th>
      <th>description</th>
      <th>transcript_identifier</th>
      <th>chromosome</th>
      <th>molecular_type</th>
    </tr>
    <tr>
      <th>gene_identifier</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>ENSG00000118777</th>
      <td>ABCG2</td>
      <td>ATP binding cassette subfamily G member 2 (Jun...</td>
      <td>ENST00000237612</td>
      <td>chr4</td>
      <td>protein_coding</td>
    </tr>
    <tr>
      <th>ENSG00000103740</th>
      <td>ACSBG1</td>
      <td>acyl-CoA synthetase bubblegum family member 1</td>
      <td>ENST00000258873</td>
      <td>chr15</td>
      <td>protein_coding</td>
    </tr>
    <tr>
      <th>ENSG00000042980</th>
      <td>ADAM28</td>
      <td>ADAM metallopeptidase domain 28</td>
      <td>ENST00000521629</td>
      <td>chr8</td>
      <td>protein_coding</td>
    </tr>
    <tr>
      <th>ENSG00000140873</th>
      <td>ADAMTS18</td>
      <td>ADAM metallopeptidase with thrombospondin type...</td>
      <td>ENST00000282849</td>
      <td>chr16</td>
      <td>protein_coding</td>
    </tr>
    <tr>
      <th>ENSG00000173157</th>
      <td>ADAMTS20</td>
      <td>ADAM metallopeptidase with thrombospondin type...</td>
      <td>ENST00000549670</td>
      <td>chr12</td>
      <td>protein_coding</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_genes</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;gene&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;gene_identifier&#39;</span><span class="p">)</span>
<span class="n">macaque_genes</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>gene.csv: 100%|██████████| 51.0k/51.0k [00:00&lt;00:00, 207kMB/s] 
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene_symbol</th>
      <th>description</th>
      <th>molecular_type</th>
      <th>ensembl_gene_identifier</th>
      <th>ncbi_gene_identifier</th>
      <th>ncbi_gene_symbol</th>
      <th>ncbi_description</th>
      <th>match_method</th>
    </tr>
    <tr>
      <th>gene_identifier</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>ENSMMUG00000015100</th>
      <td>ABO</td>
      <td>ABO, alpha 1-3-N-acetylgalactosaminyltransfera...</td>
      <td>protein_coding</td>
      <td>ENSEMBL:ENSMMUG00000015100</td>
      <td>NCBIGene:722252</td>
      <td>ABO</td>
      <td>ABO, alpha 1-3-N-acetylgalactosaminyltransfera...</td>
      <td>ensembl_biomart</td>
    </tr>
    <tr>
      <th>ENSMMUG00000009661</th>
      <td>ADAM12</td>
      <td>ADAM metallopeptidase domain 12</td>
      <td>protein_coding</td>
      <td>ENSEMBL:ENSMMUG00000009661</td>
      <td>NCBIGene:696405</td>
      <td>ADAM12</td>
      <td>ADAM metallopeptidase domain 12</td>
      <td>ensembl_biomart</td>
    </tr>
    <tr>
      <th>ENSMMUG00000000824</th>
      <td>ADAMTS18</td>
      <td>ADAM metallopeptidase with thrombospondin type...</td>
      <td>protein_coding</td>
      <td>ENSEMBL:ENSMMUG00000000824</td>
      <td>NCBIGene:712710</td>
      <td>ADAMTS18</td>
      <td>ADAM metallopeptidase with thrombospondin type...</td>
      <td>ensembl_biomart</td>
    </tr>
    <tr>
      <th>ENSMMUG00000044658</th>
      <td>ADAMTSL1</td>
      <td>ADAMTS like 1</td>
      <td>protein_coding</td>
      <td>ENSEMBL:ENSMMUG00000044658</td>
      <td>NCBIGene:712844</td>
      <td>ADAMTSL1</td>
      <td>ADAMTS like 1</td>
      <td>ensembl_biomart</td>
    </tr>
    <tr>
      <th>ENSMMUG00000017120</th>
      <td>ADARB2</td>
      <td>adenosine deaminase RNA specific B2 (inactive)</td>
      <td>protein_coding</td>
      <td>ENSEMBL:ENSMMUG00000017120</td>
      <td>NCBIGene:722075</td>
      <td>ADARB2</td>
      <td>adenosine deaminase RNA specific B2 (inactive)</td>
      <td>ensembl_biomart</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_genes</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;gene&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;gene_identifier&#39;</span><span class="p">)</span>
<span class="n">marmoset_genes</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>gene.csv: 100%|██████████| 22.4k/22.4k [00:00&lt;00:00, 247kMB/s]
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene_symbol</th>
      <th>description</th>
      <th>molecular_type</th>
    </tr>
    <tr>
      <th>gene_identifier</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>NCBIGene:100393956</th>
      <td>ABI3BP</td>
      <td>ABI family member 3 binding protein</td>
      <td>protein_coding</td>
    </tr>
    <tr>
      <th>NCBIGene:100388523</th>
      <td>ABL2</td>
      <td>ABL proto-oncogene 2, non-receptor tyrosine ki...</td>
      <td>protein_coding</td>
    </tr>
    <tr>
      <th>NCBIGene:100389219</th>
      <td>ABLIM1</td>
      <td>actin binding LIM protein 1</td>
      <td>protein_coding</td>
    </tr>
    <tr>
      <th>NCBIGene:100400228</th>
      <td>ACAP2</td>
      <td>ArfGAP with coiled-coil, ankyrin repeat and PH...</td>
      <td>protein_coding</td>
    </tr>
    <tr>
      <th>NCBIGene:100392257</th>
      <td>ACHE</td>
      <td>acetylcholinesterase (Cartwright blood group)</td>
      <td>protein_coding</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We define a set of genes to load via their symbol. We reuse the smaller overlapping subset of genes we used in the previous <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_10X_snRNASeq_tutorial.html">10X gene expression notebook</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gene_names</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;AQP4&#39;</span><span class="p">,</span> <span class="s1">&#39;DRD1&#39;</span><span class="p">,</span> <span class="s1">&#39;DRD2&#39;</span><span class="p">,</span> <span class="s1">&#39;SLC17A6&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<p>In the cell below, we show how to use the <code class="docutils literal notranslate"><span class="pre">get_gene_data</span></code> function to extract the expression of specific genes from the h5ad metadata. In the commented code, we combine our abc_cache object, cell metadata, gene metadata, and gene symbols we would like select in our <code class="docutils literal notranslate"><span class="pre">get_gene_data</span></code> function that downloads and extracts gene expression data from the h5ad file. Note that this function is an example and should not be used on small systems to extract a large number of genes for all cells at once. For a small subset of cells, it can be used to pull expression data from the h5ad file for all genes. The methods below will pull data from the files normalized with log2cpm, however one can also request the <code class="docutils literal notranslate"><span class="pre">raw</span></code> counts data as well.</p>
<p>For this tutorial we will skip using these functions and instead use a pre-computed gene expression to save time. For specifics on how to use the below functions see <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/general_accessing_10x_snRNASeq_tutorial.html">this tutorial</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">human_gene_data = get_gene_data(</span>
<span class="sd">    abc_atlas_cache=abc_cache,</span>
<span class="sd">    all_cells=human_cell_metadata,</span>
<span class="sd">    all_genes=human_genes,</span>
<span class="sd">    selected_genes=gene_names,</span>
<span class="sd">    data_type=&#39;log2&#39;</span>
<span class="sd">)</span>
<span class="sd">macaque_gene_data = get_gene_data(</span>
<span class="sd">    abc_atlas_cache=abc_cache,</span>
<span class="sd">    all_cells=macaque_cell_metadata,</span>
<span class="sd">    all_genes=macaque_genes,</span>
<span class="sd">    selected_genes=gene_names,</span>
<span class="sd">    data_type=&#39;log2&#39;</span>
<span class="sd">)</span>
<span class="sd">marmoset_gene_data = get_gene_data(</span>
<span class="sd">    abc_atlas_cache=abc_cache,</span>
<span class="sd">    all_cells=marmoset_cell_metadata,</span>
<span class="sd">    all_genes=marmoset_genes,</span>
<span class="sd">    selected_genes=gene_names,</span>
<span class="sd">    data_type=&#39;raw&#39;</span>
<span class="sd">)&quot;&quot;&quot;</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&quot;\nhuman_gene_data = get_gene_data(\n    abc_atlas_cache=abc_cache,\n    all_cells=human_cell_metadata,\n    all_genes=human_genes,\n    selected_genes=gene_names,\n    data_type=&#39;log2&#39;\n)\nmacaque_gene_data = get_gene_data(\n    abc_atlas_cache=abc_cache,\n    all_cells=macaque_cell_metadata,\n    all_genes=macaque_genes,\n    selected_genes=gene_names,\n    data_type=&#39;log2&#39;\n)\nmarmoset_gene_data = get_gene_data(\n    abc_atlas_cache=abc_cache,\n    all_cells=marmoset_cell_metadata,\n    all_genes=marmoset_genes,\n    selected_genes=gene_names,\n    data_type=&#39;raw&#39;\n)&quot;
</pre></div>
</div>
</div>
</div>
<p>We load the precomputed gene expressions for each species below instead of running the processing. Note that the Human and Marmoset data contain all 4 of the genes we requested, however AQP4 is not in the Macaque data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_gene_data</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;example_gene_expression&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">human_gene_data</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>example_gene_expression.csv: 100%|██████████| 236M/236M [00:38&lt;00:00, 6.16MMB/s]    
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>AQP4</th>
      <th>DRD1</th>
      <th>SLC17A6</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1321115602_1-20250227</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>1321115602_2-20250227</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>1321115602_3-20250227</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>1321115602_4-20250227</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>1321115602_5-20250227</th>
      <td>0.0</td>
      <td>0.0</td>
      <td>3.823535</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_gene_data</span> <span class="o">=</span> <span class="n">macaque_genes</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;example_gene_expression&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">macaque_gene_data</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>example_gene_expression.csv: 100%|██████████| 160M/160M [00:27&lt;00:00, 5.81MMB/s]   
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>DRD1</th>
      <th>DRD2</th>
      <th>SLC17A6</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1291813781_1-20250227</th>
      <td>0.0</td>
      <td>9.967226</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1291813781_2-20250227</th>
      <td>0.0</td>
      <td>0.000000</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1291813781_3-20250227</th>
      <td>0.0</td>
      <td>0.000000</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1291813781_4-20250227</th>
      <td>0.0</td>
      <td>0.000000</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1291813781_5-20250227</th>
      <td>0.0</td>
      <td>0.000000</td>
      <td>0.0</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_gene_data</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;example_gene_expression&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">marmoset_gene_data</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>example_gene_expression.csv: 100%|██████████| 64.2M/64.2M [00:11&lt;00:00, 5.74MMB/s]  
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>AQP4</th>
      <th>DRD1</th>
      <th>DRD2</th>
      <th>SLC17A6</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1382042951_abmkfncc-1-20250227</th>
      <td>3.0</td>
      <td>1.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1382042951_abmmdgea-1-20250227</th>
      <td>11.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1382042951_abmojfce-1-20250227</th>
      <td>11.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1382042951_abmomnim-1-20250227</th>
      <td>5.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1382042951_aeiamgga-1-20250227</th>
      <td>0.0</td>
      <td>2.0</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Now that we’ve loaded our expression matrix tables, indexed by cell_label, we can merge them into our cell metadata tables.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">human_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">human_gene_data</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">macaque_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">macaque_gene_data</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">marmoset_gene_data</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="c1"># clean up memory</span>
<span class="k">del</span> <span class="n">human_gene_data</span>
<span class="k">del</span> <span class="n">macaque_gene_data</span>
<span class="k">del</span> <span class="n">marmoset_gene_data</span>
</pre></div>
</div>
</div>
</div>
<section id="plotting-the-slabs">
<h3>Plotting the Slabs<a class="headerlink" href="#plotting-the-slabs" title="Link to this heading">#</a></h3>
<p>We define a helper function <em>plot_slabs</em> to visualize the cells for a specified set of brain sections either by colorized metadata or gene expression. This function can be used to plot individual brain sections as well.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">plot_slabs</span><span class="p">(</span>
    <span class="n">df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
    <span class="n">feature</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">slab_list</span><span class="p">:</span> <span class="n">List</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
    <span class="n">species</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">cmap</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Colormap</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">vmin</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="nb">float</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">vmax</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="nb">float</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">fig_width</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">40</span><span class="p">,</span>
    <span class="n">fig_height</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
    <span class="n">plot_polygons</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
    <span class="n">point_size</span><span class="p">:</span> <span class="nb">float</span> <span class="o">=</span> <span class="mf">1.0</span><span class="p">,</span>
    <span class="n">unit</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;[log2(CPT + 1)]&#39;</span><span class="p">,</span>
    <span class="n">fix_bounds</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Tuple</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Figure</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Plot spatial transcriptomics data.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    df: pandas.DataFrame</span>
<span class="sd">        cell metadata data frame with x,y locations to plot.</span>
<span class="sd">    feature: str</span>
<span class="sd">        Expression value or color column to plot.</span>
<span class="sd">    slab_list: List[str]</span>
<span class="sd">        List of slab_plane ids to plot.</span>
<span class="sd">    species: str</span>
<span class="sd">        Species to plot (human, macaque, marmoset).</span>
<span class="sd">    cmap: matplotlib.colors.Colormap</span>
<span class="sd">        Colormap for expression features.</span>
<span class="sd">    vmin: float</span>
<span class="sd">        Minimum value for color mapping. Defaults to min of values in the feature column. Values lower than</span>
<span class="sd">        this will be plotted as grey. </span>
<span class="sd">    vmax: float</span>
<span class="sd">        Maximum value for color mapping. Defaults to max of values in the feature column. Values higher than</span>
<span class="sd">        this will be plotted as grey.</span>
<span class="sd">    fig_width: int</span>
<span class="sd">        Width of the plotted figure.</span>
<span class="sd">    fig_height: int</span>
<span class="sd">        Height of the plotted figure.</span>
<span class="sd">    plot_polygons: bool</span>
<span class="sd">        Overplot geojson polygons. Note that these are note available for human</span>
<span class="sd">        and this should be set to False.</span>
<span class="sd">    point_size: float</span>
<span class="sd">        Size of the points to plot.</span>
<span class="sd">    fix_bounds: bool</span>
<span class="sd">        If True, fix the x and y bounds across all slabs to be the same.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">slab_list</span><span class="p">,</span> <span class="nb">str</span><span class="p">):</span>
        <span class="n">slab_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">slab_list</span><span class="p">]</span>
    
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">slab_list</span><span class="p">))</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">slab_list</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
        <span class="n">ax</span> <span class="o">=</span> <span class="p">[</span><span class="n">ax</span><span class="p">]</span>

    <span class="k">if</span> <span class="s1">&#39;x&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">x_col</span> <span class="o">=</span> <span class="s1">&#39;x&#39;</span>
    <span class="k">elif</span> <span class="s1">&#39;x_slab_mm&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">x_col</span> <span class="o">=</span> <span class="s1">&#39;x_slab_mm&#39;</span>
    <span class="k">if</span> <span class="s1">&#39;y&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">y_col</span> <span class="o">=</span> <span class="s1">&#39;y&#39;</span>
    <span class="k">elif</span> <span class="s1">&#39;y_slab_mm&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">y_col</span> <span class="o">=</span> <span class="s1">&#39;y_slab_mm&#39;</span>

    <span class="k">if</span> <span class="n">fix_bounds</span><span class="p">:</span>
        <span class="n">x_min</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>
        <span class="n">x_max</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span>
        <span class="n">y_min</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>
        <span class="n">y_max</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span>

    <span class="n">vv</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">feature</span><span class="p">]</span>
    <span class="k">if</span> <span class="n">vmin</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">vmin</span> <span class="o">=</span> <span class="n">vv</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>

    <span class="k">if</span> <span class="n">vmax</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">vmax</span> <span class="o">=</span> <span class="n">vv</span><span class="o">.</span><span class="n">max</span><span class="p">()</span>

    <span class="n">norm</span> <span class="o">=</span> <span class="n">mpl</span><span class="o">.</span><span class="n">colors</span><span class="o">.</span><span class="n">Normalize</span><span class="p">(</span><span class="n">vmin</span><span class="o">=</span><span class="n">vmin</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="n">vmax</span><span class="p">)</span>
    <span class="n">sm</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">ScalarMappable</span><span class="p">(</span><span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span> <span class="n">norm</span><span class="o">=</span><span class="n">norm</span><span class="p">)</span>
    <span class="n">cmap</span> <span class="o">=</span> <span class="n">sm</span><span class="o">.</span><span class="n">get_cmap</span><span class="p">()</span>
    
    <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">slab</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">slab_list</span><span class="p">):</span>

        <span class="k">if</span> <span class="s1">&#39;slab_plane_label&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="n">filtered</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;slab_plane_label&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">slab</span><span class="p">]</span>
        <span class="k">elif</span> <span class="s1">&#39;brain_section_label&#39;</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="n">filtered</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">slab</span><span class="p">]</span>
        
        <span class="n">xx</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">x_col</span><span class="p">]</span>
        <span class="n">yy</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">y_col</span><span class="p">]</span>
        <span class="n">vv</span> <span class="o">=</span> <span class="n">filtered</span><span class="p">[</span><span class="n">feature</span><span class="p">]</span>

        <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">mask</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">logical_and</span><span class="p">(</span><span class="n">vv</span> <span class="o">&gt;=</span> <span class="n">vmin</span><span class="p">,</span> <span class="n">vv</span> <span class="o">&lt;=</span> <span class="n">vmax</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">mask</span><span class="o">.</span><span class="n">any</span><span class="p">():</span>
                <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">[</span><span class="o">~</span><span class="n">mask</span><span class="p">],</span> <span class="n">yy</span><span class="p">[</span><span class="o">~</span><span class="n">mask</span><span class="p">],</span> <span class="n">s</span><span class="o">=</span><span class="mf">1.0</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="s1">&#39;#D3D3D3&#39;</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">)</span>
                <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">[</span><span class="n">mask</span><span class="p">],</span> <span class="n">yy</span><span class="p">[</span><span class="n">mask</span><span class="p">],</span> <span class="n">s</span><span class="o">=</span><span class="mf">1.0</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">vv</span><span class="p">[</span><span class="n">mask</span><span class="p">],</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">)</span>
            <span class="k">else</span> <span class="p">:</span>
                <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mf">1.0</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">vv</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s2">&quot;.&quot;</span><span class="p">)</span>
        <span class="k">else</span> <span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">point_size</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="n">vv</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s2">&quot;.&quot;</span><span class="p">)</span>

        <span class="k">if</span> <span class="n">plot_polygons</span><span class="p">:</span>
            <span class="n">plot_geo_data</span><span class="p">(</span><span class="n">get_polygons</span><span class="p">(</span><span class="n">slab</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="n">species</span><span class="p">),</span> <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">])</span>
            
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s1">&#39;equal&#39;</span><span class="p">)</span>
        <span class="n">ylim</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()</span>
        <span class="k">if</span> <span class="n">fix_bounds</span><span class="p">:</span>
            <span class="n">ylim</span> <span class="o">=</span> <span class="p">(</span><span class="n">y_min</span><span class="p">,</span> <span class="n">y_max</span><span class="p">)</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">x_min</span><span class="p">,</span> <span class="n">x_max</span><span class="p">)</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="n">ylim</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">ylim</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
        
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">slab</span><span class="p">)</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">markerscale</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>


    <span class="n">cbar</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">colorbar</span><span class="p">(</span><span class="n">sm</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">orientation</span><span class="o">=</span><span class="s1">&#39;vertical&#39;</span><span class="p">,</span> <span class="n">fraction</span><span class="o">=</span><span class="mf">0.01</span><span class="p">,</span> <span class="n">pad</span><span class="o">=</span><span class="mf">0.01</span><span class="p">)</span>
    <span class="n">cbar</span><span class="o">.</span><span class="n">set_label</span><span class="p">(</span><span class="n">unit</span><span class="p">)</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.01</span><span class="p">,</span> <span class="n">hspace</span><span class="o">=</span><span class="mf">0.01</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<p>Below we plot the spatial expression of our genes across the different species. We select a minium expression value of which effectively masks out zero expression cells from the plots, showing a bit more contrast. Feel free to experiment with different vmin and vmax values in investigate the gene expressions more in depth.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_gene_names</span> <span class="o">=</span> <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">human_slab_list</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;H22.30.001.CX.16.Plane.07&#39;</span><span class="p">,</span> <span class="s1">&#39;H22.30.001.CX.17.Plane.07&#39;</span><span class="p">,</span>
    <span class="s1">&#39;H22.30.001.CX.19.Plane.07&#39;</span><span class="p">,</span> <span class="s1">&#39;H22.30.001.CX.20.Plane.03&#39;</span><span class="p">,</span> <span class="s1">&#39;H22.30.001.CX.20.Plane.05&#39;</span>
<span class="p">]</span>
<span class="k">for</span> <span class="n">gene_symbol</span> <span class="ow">in</span> <span class="n">human_gene_names</span><span class="p">:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
        <span class="n">human_cell_metadata_with_genes</span><span class="p">,</span>
        <span class="n">gene_symbol</span><span class="p">,</span>
        <span class="n">human_slab_list</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span>
        <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma</span><span class="p">,</span>
        <span class="n">vmin</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
        <span class="n">vmax</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Gene Expression: </span><span class="si">{</span><span class="n">gene_symbol</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/95afe8dc522014273b2809a6f20f6a10b5539f729a8e63978fd1083b7a4663a5.png" src="../_images/95afe8dc522014273b2809a6f20f6a10b5539f729a8e63978fd1083b7a4663a5.png" />
<img alt="../_images/a1d4d4f05b998dab7253ccd884d234049e5ee7a05f9bf831abbddc21d4a26eaa.png" src="../_images/a1d4d4f05b998dab7253ccd884d234049e5ee7a05f9bf831abbddc21d4a26eaa.png" />
<img alt="../_images/aa02326982a42f3b3033d4cd88c7b14739c043087257a8d53c72c288903cf457.png" src="../_images/aa02326982a42f3b3033d4cd88c7b14739c043087257a8d53c72c288903cf457.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_gene_names</span> <span class="o">=</span> <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">macaque_slab_list</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;QM23.50.001.CX.42.Plane.05&#39;</span><span class="p">,</span> <span class="s1">&#39;QM23.50.001.CX.43.Plane.07&#39;</span><span class="p">,</span> <span class="s1">&#39;QM23.50.001.CX.44.Plane.05&#39;</span><span class="p">,</span>
    <span class="s1">&#39;QM23.50.001.CX.45.Plane.05&#39;</span><span class="p">,</span> <span class="s1">&#39;QM23.50.001.CX.46.Plane.05&#39;</span>
<span class="p">]</span>
<span class="k">for</span> <span class="n">gene_symbol</span> <span class="ow">in</span> <span class="n">macaque_gene_names</span><span class="p">:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
        <span class="n">macaque_cell_metadata_with_genes</span><span class="p">,</span>
        <span class="n">gene_symbol</span><span class="p">,</span>
        <span class="n">macaque_slab_list</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;macaque&#39;</span><span class="p">,</span>
        <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma</span><span class="p">,</span>
        <span class="n">vmin</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
        <span class="n">vmax</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Gene Expression: </span><span class="si">{</span><span class="n">gene_symbol</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>QM23.50.001.CX.42.Plane.05.geojson: 100%|██████████| 1.61k/1.61k [00:00&lt;00:00, 16.1kMB/s]
</pre></div>
</div>
<img alt="../_images/d36661178b0394318fcabc269e5f3787a34a335dcbe10f1a5fc1305921f25418.png" src="../_images/d36661178b0394318fcabc269e5f3787a34a335dcbe10f1a5fc1305921f25418.png" />
<img alt="../_images/1d76eb9cb24a4d6802eadd54c1a3d8d151110f40cf75efb4c2268362d7bc6d6d.png" src="../_images/1d76eb9cb24a4d6802eadd54c1a3d8d151110f40cf75efb4c2268362d7bc6d6d.png" />
<img alt="../_images/fa9651bc2e33a0cfc86eabe7e7fd0b0e5fdbc7ee4ba33f0c07fe795d42b906c7.png" src="../_images/fa9651bc2e33a0cfc86eabe7e7fd0b0e5fdbc7ee4ba33f0c07fe795d42b906c7.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_gene_names</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">marmoset_slab_list</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;CJ23.56.004.CX.42.Plane.05&#39;</span><span class="p">,</span> <span class="s1">&#39;CJ23.56.004.CX.42.Plane.13&#39;</span><span class="p">,</span> <span class="s1">&#39;CJ23.56.004.CX.43.Plane.05&#39;</span><span class="p">,</span>
    <span class="s1">&#39;CJ23.56.004.CX.43.Plane.15&#39;</span><span class="p">,</span> <span class="s1">&#39;CJ23.56.004.CX.44.Plane.03&#39;</span>
<span class="p">]</span>
<span class="k">for</span> <span class="n">gene_symbol</span> <span class="ow">in</span> <span class="n">marmoset_gene_names</span><span class="p">:</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_slabs</span><span class="p">(</span>
        <span class="n">marmoset_cell_metadata_with_genes</span><span class="p">,</span>
        <span class="n">gene_symbol</span><span class="p">,</span>
        <span class="n">marmoset_slab_list</span><span class="p">,</span>
        <span class="n">species</span><span class="o">=</span><span class="s1">&#39;marmoset&#39;</span><span class="p">,</span>
        <span class="n">vmin</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
        <span class="n">vmax</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;Raw Counts&#39;</span><span class="p">,</span>
        <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma</span><span class="p">,</span>
        <span class="n">plot_polygons</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Gene Expression: </span><span class="si">{</span><span class="n">gene_symbol</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/5298be010c02e46cd03b1ccfe101899273043ed3ae99bce654175193ea4f46b6.png" src="../_images/5298be010c02e46cd03b1ccfe101899273043ed3ae99bce654175193ea4f46b6.png" />
<img alt="../_images/447900005e688317b1241e9f618de355ea1275dab0857aaa40ed8d651bfd16d1.png" src="../_images/447900005e688317b1241e9f618de355ea1275dab0857aaa40ed8d651bfd16d1.png" />
<img alt="../_images/5f7d935cb21fa5b82b4b6e5dee01d61c46de1fef89a51348216f1e3ee46f9a44.png" src="../_images/5f7d935cb21fa5b82b4b6e5dee01d61c46de1fef89a51348216f1e3ee46f9a44.png" />
<img alt="../_images/490e48e68a195a4f3ef57e618c9065e47b11fcb24aa08093f6d920c02c3fe3e2.png" src="../_images/490e48e68a195a4f3ef57e618c9065e47b11fcb24aa08093f6d920c02c3fe3e2.png" />
</div>
</div>
</section>
</section>
<section id="manual-anatomical-annotations">
<h2>Manual anatomical annotations<a class="headerlink" href="#manual-anatomical-annotations" title="Link to this heading">#</a></h2>
<p>We load tables that allow us to assign cells to the manual annotations plotted using the geojsons. These tables contain boolean assignments to a given manual annotation and column specifying a single unique assignment.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_anatomical_annotations</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_anatomical_annotations&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">macaque_cell_anatomical_annotations</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_anatomical_annotations&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">marmoset_cell_anatomical_annotations</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_anatomical_annotations&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">value_sets</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;value_sets&#39;</span><span class="p">,</span>
<span class="p">)</span>
<span class="n">pred</span> <span class="o">=</span> <span class="n">value_sets</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span>
<span class="n">parcellation_terms</span> <span class="o">=</span> <span class="n">value_sets</span><span class="p">[</span><span class="n">pred</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">parcellation_terms</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;label&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">parcellation_terms</span>

<span class="k">def</span><span class="w"> </span><span class="nf">extract_value_set</span><span class="p">(</span><span class="n">cell_metadata_df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">input_value_set</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">input_value_set_label</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Add color and order columns to the cell metadata dataframe based on the input</span>
<span class="sd">    value set.</span>

<span class="sd">    Columns are added as {input_value_set_label}_color and {input_value_set_label}_order.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    cell_metadata_df : pd.DataFrame</span>
<span class="sd">        DataFrame containing cell metadata.</span>
<span class="sd">    input_value_set : pd.DataFrame</span>
<span class="sd">        DataFrame containing the value set information.</span>
<span class="sd">    input_value_set_label : str</span>
<span class="sd">        The the column name to extract color and order information for. will be added to the cell metadata.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">cell_metadata_df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">input_value_set_label</span><span class="si">}</span><span class="s1">_color&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_value_set</span><span class="p">[</span>
        <span class="n">input_value_set</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">input_value_set_label</span>
    <span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cell_metadata_df</span><span class="p">[</span><span class="n">input_value_set_label</span><span class="p">]][</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">cell_metadata_df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">input_value_set_label</span><span class="si">}</span><span class="s1">_order&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_value_set</span><span class="p">[</span>
        <span class="n">input_value_set</span><span class="p">[</span><span class="s1">&#39;field&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">input_value_set_label</span>
    <span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">cell_metadata_df</span><span class="p">[</span><span class="n">input_value_set_label</span><span class="p">]][</span><span class="s1">&#39;order&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">human_cell_anatomical_annotations</span><span class="p">[[</span><span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">]]</span>
<span class="p">)</span>
<span class="n">macaque_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">macaque_cell_anatomical_annotations</span><span class="p">[[</span><span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">]]</span>
<span class="p">)</span>
<span class="n">marmoset_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">marmoset_cell_anatomical_annotations</span><span class="p">[[</span><span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">]]</span>
<span class="p">)</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">human_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">parcellation_terms</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">)</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">macaque_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">parcellation_terms</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">)</span>
<span class="n">extract_value_set</span><span class="p">(</span><span class="n">marmoset_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">parcellation_terms</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">)</span>

<span class="c1"># clean up memory</span>
<span class="k">del</span> <span class="n">human_cell_anatomical_annotations</span>
<span class="k">del</span> <span class="n">macaque_cell_anatomical_annotations</span>
<span class="k">del</span> <span class="n">marmoset_cell_anatomical_annotations</span>
</pre></div>
</div>
</div>
</div>
<section id="plotting-heatmap-expression-data">
<h3>Plotting heatmap expression data.<a class="headerlink" href="#plotting-heatmap-expression-data" title="Link to this heading">#</a></h3>
<p>We define a helper functions <em>aggregate_by_metadata</em> to compute the average expression for a given category. Note that this is a simple wrapper of the Pandas GroupBy functionality. Other summary statistics beyond just the <code class="docutils literal notranslate"><span class="pre">mean</span></code> are listed here: <a class="reference external" href="https://pandas.pydata.org/docs/reference/groupby.html#dataframegroupby-computations-descriptive-stats">https://pandas.pydata.org/docs/reference/groupby.html#dataframegroupby-computations-descriptive-stats</a></p>
<p>We also define a function to plot the resultant averaged data in a heatmap.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">aggregate_by_metadata</span><span class="p">(</span><span class="n">df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span> <span class="n">gnames</span><span class="p">:</span> <span class="n">List</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span> <span class="n">value</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">sort</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Aggregate gene expression data by metadata.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    df: pandas.DataFrame</span>
<span class="sd">        DataFrame containing gene expression and metadata.</span>
<span class="sd">    gnames: List[str]</span>
<span class="sd">        List of gene names to aggregate.</span>
<span class="sd">    value: str</span>
<span class="sd">    sort: bool</span>
<span class="sd">        Metadata column to group by.</span>
<span class="sd">        If True, sort the output by the first gene in gnames.</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    pandas.DataFrame</span>
<span class="sd">        DataFrame with aggregated gene expression values.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">grouped</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">value</span><span class="p">)[</span><span class="n">gnames</span><span class="p">]</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span>
    <span class="k">if</span> <span class="n">sort</span><span class="p">:</span>
        <span class="n">grouped</span> <span class="o">=</span> <span class="n">grouped</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="n">gnames</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">ascending</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">grouped</span>


<span class="k">def</span><span class="w"> </span><span class="nf">plot_heatmap</span><span class="p">(</span>
        <span class="n">df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">,</span>
        <span class="n">fig_width</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
        <span class="n">fig_height</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span>
        <span class="n">cmap</span><span class="p">:</span> <span class="n">mpl</span><span class="o">.</span><span class="n">colormaps</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma</span><span class="p">,</span>
        <span class="n">vmax</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="nb">float</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
        <span class="n">unit</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;[log2(CPT + 1)]&#39;</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Tuple</span><span class="p">[</span><span class="n">plt</span><span class="o">.</span><span class="n">Figure</span><span class="p">,</span> <span class="n">plt</span><span class="o">.</span><span class="n">Axes</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Plot a heatmap from a DataFrame.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    df: pandas.DataFrame</span>
<span class="sd">        DataFrame to plot as a heatmap.</span>
<span class="sd">    fig_width: int</span>
<span class="sd">        Width of the figure.</span>
<span class="sd">    fig_height: int</span>
<span class="sd">        Height of the figure.</span>
<span class="sd">    cmap: matplotlib.colors.Colormap</span>
<span class="sd">        Colormap to use for the heatmap.</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    fig: matplotlib.pyplot.Figure</span>
<span class="sd">        Figure object containing the heatmap.</span>
<span class="sd">    ax: matplotlib.pyplot.Axes</span>
<span class="sd">        Axes object containing the heatmap.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">arr</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="s1">&#39;float&#39;</span><span class="p">)</span>

    <span class="n">vmin</span> <span class="o">=</span> <span class="n">arr</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>
    <span class="n">vmax</span> <span class="o">=</span> <span class="n">arr</span><span class="o">.</span><span class="n">max</span><span class="p">()</span>
    <span class="n">norm</span> <span class="o">=</span> <span class="n">mpl</span><span class="o">.</span><span class="n">colors</span><span class="o">.</span><span class="n">Normalize</span><span class="p">(</span><span class="n">vmin</span><span class="o">=</span><span class="n">vmin</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="n">vmax</span><span class="p">)</span>
    <span class="n">sm</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">ScalarMappable</span><span class="p">(</span><span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span> <span class="n">norm</span><span class="o">=</span><span class="n">norm</span><span class="p">)</span>
    <span class="n">cmap</span> <span class="o">=</span> <span class="n">sm</span><span class="o">.</span><span class="n">get_cmap</span><span class="p">()</span>

    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">arr</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span> <span class="n">aspect</span><span class="o">=</span><span class="s1">&#39;auto&#39;</span><span class="p">)</span>
    <span class="n">xlabs</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span>
    <span class="n">ylabs</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">values</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">xlabs</span><span class="p">)))</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">xlabs</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">ylabs</span><span class="p">)))</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">(</span><span class="n">ylabs</span><span class="p">)</span>

    <span class="n">cbar</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">colorbar</span><span class="p">(</span><span class="n">sm</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">orientation</span><span class="o">=</span><span class="s1">&#39;vertical&#39;</span><span class="p">,</span> <span class="n">fraction</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span> <span class="n">pad</span><span class="o">=</span><span class="mf">0.01</span><span class="p">)</span>
    <span class="n">cbar</span><span class="o">.</span><span class="n">set_label</span><span class="p">(</span><span class="n">unit</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="visualize-gene-expression-for-selected-genes-by-manual-annotation-region">
<h3>Visualize gene expression for selected genes by manual annotation region.<a class="headerlink" href="#visualize-gene-expression-for-selected-genes-by-manual-annotation-region" title="Link to this heading">#</a></h3>
<p>Below we average the expression of our selected genes across the different manually annotated regions that each cell is uniquely assigned to. We compute these expressions for each species individually. For a side by side comparison of gene expression heatmaps for each species, see the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_10X_snRNASeq_tutorial.html">HMBA 10X gene expression tutorial</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_gene_names</span> <span class="o">=</span> <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">aggregate_by_metadata</span><span class="p">(</span><span class="n">human_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">human_gene_names</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">,</span> <span class="n">sort</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">agg</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">parcellation_terms</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">parcellation_terms</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">agg</span><span class="o">.</span><span class="n">index</span><span class="p">)]]</span>
<span class="n">plot_heatmap</span><span class="p">(</span><span class="n">agg</span><span class="p">,</span> <span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Human Anatomical Annotations&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/0f1bfcab6a00e5a145f384d84667b2e7dafacebc7ba2078568bda28cd4b54136.png" src="../_images/0f1bfcab6a00e5a145f384d84667b2e7dafacebc7ba2078568bda28cd4b54136.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_gene_names</span> <span class="o">=</span> <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">aggregate_by_metadata</span><span class="p">(</span><span class="n">macaque_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">macaque_gene_names</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">,</span> <span class="n">sort</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">agg</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">parcellation_terms</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">parcellation_terms</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">agg</span><span class="o">.</span><span class="n">index</span><span class="p">)]]</span>
<span class="n">plot_heatmap</span><span class="p">(</span><span class="n">agg</span><span class="p">,</span> <span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Macaque Anatomical Annotations&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/d3be216768243ff09c2d818a668224d9e38abd828dadd9f2a338a9cac82ef0b6.png" src="../_images/d3be216768243ff09c2d818a668224d9e38abd828dadd9f2a338a9cac82ef0b6.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_gene_names</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">aggregate_by_metadata</span><span class="p">(</span><span class="n">marmoset_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">marmoset_gene_names</span><span class="p">,</span> <span class="s1">&#39;parcellation_term_symbol&#39;</span><span class="p">,</span> <span class="n">sort</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">agg</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">parcellation_terms</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">parcellation_terms</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">agg</span><span class="o">.</span><span class="n">index</span><span class="p">)]]</span>
<span class="n">plot_heatmap</span><span class="p">(</span><span class="n">agg</span><span class="p">,</span> <span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;[Ave Raw Counts]&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Marmoset Anatomical Annotations&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/c3cad212c054f52e6d5da9a623dc8ec37938d6202546d5d64cdfe0a6677cf86b.png" src="../_images/c3cad212c054f52e6d5da9a623dc8ec37938d6202546d5d64cdfe0a6677cf86b.png" />
</div>
</div>
</section>
</section>
<section id="taxonomy-mapping">
<h2>Taxonomy mapping<a class="headerlink" href="#taxonomy-mapping" title="Link to this heading">#</a></h2>
<p>Finally, let’s load the taxonomy information so that we can plot gene expression versus various taxons in a heatmap for each species. Below, we load and merge all the information we need to plot cell colors and taxonomy as done in previous <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_clustering_analysis_and_annotation.html">HMBA tutorial</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster&#39;</span><span class="p">,</span>
    <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">:</span> <span class="s1">&#39;Int64&#39;</span><span class="p">}</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">cluster_annotation_term_set</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term_set&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span>

<span class="n">cluster_annotation_term</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>

<span class="n">cluster_to_cluster_annotation_membership</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_to_cluster_annotation_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">)</span>
<span class="n">membership_with_cluster_info</span> <span class="o">=</span> <span class="n">cluster_to_cluster_annotation_membership</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
    <span class="n">cluster</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)[[</span><span class="s1">&#39;number_of_cells&#39;</span><span class="p">]],</span>
    <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span>
<span class="p">)</span>
<span class="n">membership_with_cluster_info</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_annotation_term</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_anno_term&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">membership_groupby</span> <span class="o">=</span> <span class="n">membership_with_cluster_info</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="s1">&#39;cluster_annotation_term_set_name&#39;</span><span class="p">]</span>
<span class="p">)</span>

<span class="n">cluster_annotation_term_set</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-BG-taxonomy-CCN20250428&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cluster_annotation_term_set&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;label&#39;</span><span class="p">:</span> <span class="s1">&#39;cluster_annotation_term_label&#39;</span><span class="p">})</span>

<span class="c1"># term_sets = abc_cache.get_metadata_dataframe(directory=&#39;WHB-taxonomy&#39;, file_name=&#39;cluster_annotation_term_set&#39;).set_index(&#39;label&#39;)</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;cluster_annotation_term_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_order</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;term_order&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_order</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">,</span> <span class="s1">&#39;Class&#39;</span><span class="p">,</span> <span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;Group&#39;</span><span class="p">,</span> <span class="s1">&#39;Cluster&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">cluster_order</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span>
    <span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">:</span> <span class="s1">&#39;Neighborhood_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;Class&#39;</span><span class="p">:</span> <span class="s1">&#39;Class_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;Subclass&#39;</span><span class="p">:</span> <span class="s1">&#39;Subclass_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;Group&#39;</span><span class="p">:</span> <span class="s1">&#39;Group_order&#39;</span><span class="p">,</span>
             <span class="s1">&#39;Cluster&#39;</span><span class="p">:</span> <span class="s1">&#39;Cluster_order&#39;</span><span class="p">},</span>
    <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>

<span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">membership_groupby</span><span class="p">[</span><span class="s1">&#39;color_hex_triplet&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">first</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">()</span>
<span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">cluster_colors</span><span class="p">[</span><span class="n">cluster_annotation_term_set</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span>
<span class="n">cluster_colors</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;Neighborhood&#39;</span><span class="p">,</span> <span class="s1">&#39;Class&#39;</span><span class="p">,</span> <span class="s1">&#39;Subclass&#39;</span><span class="p">,</span> <span class="s1">&#39;Group&#39;</span><span class="p">,</span> <span class="s1">&#39;Cluster&#39;</span><span class="p">],</span>
    <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Each of the three species gene expression has been mapped to the HMBA 10X taxonomy using MapMyCells. Below we load the mapping of each cell to the cluster it was assigned data for each species in the HMBA-BG dataset.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_to_cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-H22.30.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">macaque_cell_to_cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-MERSCOPE-QM23.50.001-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>

<span class="n">marmoset_cell_to_cluster</span> <span class="o">=</span> <span class="n">abc_cache</span><span class="o">.</span><span class="n">get_metadata_dataframe</span><span class="p">(</span>
    <span class="n">directory</span><span class="o">=</span><span class="s1">&#39;HMBA-Xenium-CJ23.56.004-BG&#39;</span><span class="p">,</span>
    <span class="n">file_name</span><span class="o">=</span><span class="s1">&#39;cell_to_cluster_membership&#39;</span><span class="p">,</span>
<span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>As in the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_spatial_slabs_and_taxonomy.html">previous spatial tutorial</a>, the spatial data of the 3 HMBA-BG species in this release are registered to a extended version of the the taxonomy that contains clusters that are considered Adjacent to the Basal Ganglia. These are cell types that are present in the data but are not considered part of the basal ganglia. These clusters are mapped into a Group, Subclass, Class, and Neighborhood in a single path of the taxonomy called Adjacent. The release consensus taxonomy does not explicitly contain this taxon. We thus add them below, assigning any cell that was not mapped into to a group in the taxonomy an Adjacent cell type at all levels with a grey color. We give these Adjacent taxons a odering at the very end of the set of taxons at each level.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span><span class="w"> </span><span class="nf">fill_adjacent_data</span><span class="p">(</span><span class="n">input_df</span><span class="p">):</span>
    <span class="n">output_df</span> <span class="o">=</span> <span class="n">input_df</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span>
        <span class="p">{</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;cluster_label&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Class&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Cluster&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Group&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Neighborhood&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Subclass&#39;</span><span class="p">:</span> <span class="s1">&#39;Adjacent&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Neighborhood_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Class_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Subclass_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Group_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Cluster_color&#39;</span><span class="p">:</span> <span class="s1">&#39;#808080&#39;</span><span class="p">,</span>
         <span class="s1">&#39;Neighborhood_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Neighborhood_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
         <span class="s1">&#39;Class_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Class_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
         <span class="s1">&#39;Subclass_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Subclass_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
         <span class="s1">&#39;Group_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Group_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
         <span class="s1">&#39;Cluster_order&#39;</span><span class="p">:</span> <span class="n">cluster_order</span><span class="o">.</span><span class="n">max</span><span class="p">()[</span><span class="s1">&#39;Cluster_order&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span>
        <span class="p">}</span>
    <span class="p">)</span>
    <span class="k">return</span> <span class="n">output_df</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span>
        <span class="p">{</span>
            <span class="s1">&#39;Neighborhood_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Class_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Subclass_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Group_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Cluster_order&#39;</span><span class="p">:</span> <span class="s1">&#39;int&#39;</span><span class="p">,</span>
        <span class="p">}</span>
    <span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">human_cell_to_cluster</span><span class="p">)</span>
<span class="n">human_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_color&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_order</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">human_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">fill_adjacent_data</span><span class="p">(</span><span class="n">human_cell_metadata_with_genes</span><span class="p">)</span>

<span class="k">del</span> <span class="n">human_cell_to_cluster</span>

<span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>Neighborhood_color</th>
      <th>Class_color</th>
      <th>Subclass_color</th>
      <th>Group_color</th>
      <th>Cluster_color</th>
      <th>Class_order</th>
      <th>Cluster_order</th>
      <th>Group_order</th>
      <th>Neighborhood_order</th>
      <th>Subclass_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1324989690_1-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7564.9175</td>
      <td>1985.1162</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>ddc7b0c5-1bb0-4f02-a1e4-ddacac84c9aa</td>
      <td>1324989690</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#88772d</td>
      <td>3</td>
      <td>160</td>
      <td>10</td>
      <td>1</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1324989690_2-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7553.6514</td>
      <td>1986.4812</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>52d3e787-0140-42d2-b7cf-026f398cb597</td>
      <td>1324989690</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#401e66</td>
      <td>#401e66</td>
      <td>#195f8d</td>
      <td>#238c7a</td>
      <td>1</td>
      <td>12</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1324989690_3-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7687.5170</td>
      <td>1987.5452</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>4af7a7a0-823a-47bd-bbe5-344ddd70cc9e</td>
      <td>1324989690</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#401e66</td>
      <td>#401e66</td>
      <td>#195f8d</td>
      <td>#238c7a</td>
      <td>1</td>
      <td>12</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1324989690_4-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7574.2090</td>
      <td>1999.2940</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>a4455148-9d49-41b7-8750-00d7a3fb225e</td>
      <td>1324989690</td>
      <td>...</td>
      <td>#19613b</td>
      <td>#d0b83c</td>
      <td>#253c8c</td>
      <td>#ff9896</td>
      <td>#7892e2</td>
      <td>10</td>
      <td>1277</td>
      <td>55</td>
      <td>3</td>
      <td>32</td>
    </tr>
    <tr>
      <th>1324989690_5-20250227</th>
      <td>H22.30.001.CX.17.03.03.03</td>
      <td>1324989690</td>
      <td>sis_cellpose_v1</td>
      <td>7432.2330</td>
      <td>1999.0550</td>
      <td>True</td>
      <td>HMBA-MERSCOPE-H22.30.001-BG</td>
      <td>MERSCOPE-H22.30.001-BG</td>
      <td>41946bab-3c5a-4890-ae7b-cbffcab06c3e</td>
      <td>1324989690</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#66391F</td>
      <td>#a9de1a</td>
      <td>3</td>
      <td>142</td>
      <td>9</td>
      <td>1</td>
      <td>7</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 54 columns</p>
</div></div></div>
</div>
<p>Load the Macaque and Marmoset data in the same way.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">macaque_cell_to_cluster</span><span class="p">)</span>
<span class="n">macaque_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_color&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_order</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">macaque_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">fill_adjacent_data</span><span class="p">(</span><span class="n">macaque_cell_metadata_with_genes</span><span class="p">)</span>

<span class="k">del</span> <span class="n">macaque_cell_to_cluster</span>

<span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>Neighborhood_color</th>
      <th>Class_color</th>
      <th>Subclass_color</th>
      <th>Group_color</th>
      <th>Cluster_color</th>
      <th>Class_order</th>
      <th>Cluster_order</th>
      <th>Group_order</th>
      <th>Neighborhood_order</th>
      <th>Subclass_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1291813781_1-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6328.8564</td>
      <td>271.71730</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>#19613b</td>
      <td>#d0b83c</td>
      <td>#253c8c</td>
      <td>#ff9896</td>
      <td>#681364</td>
      <td>10</td>
      <td>1289</td>
      <td>55</td>
      <td>3</td>
      <td>32</td>
    </tr>
    <tr>
      <th>1291813781_2-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6313.3560</td>
      <td>297.27830</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#a8afa5</td>
      <td>#66493D</td>
      <td>#66493D</td>
      <td>#e86aa0</td>
      <td>4</td>
      <td>246</td>
      <td>14</td>
      <td>1</td>
      <td>10</td>
    </tr>
    <tr>
      <th>1291813781_3-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6323.7240</td>
      <td>297.38340</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#995C60</td>
      <td>#CC887A</td>
      <td>#CC887A</td>
      <td>#c79107</td>
      <td>2</td>
      <td>126</td>
      <td>5</td>
      <td>1</td>
      <td>4</td>
    </tr>
    <tr>
      <th>1291813781_4-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6370.7470</td>
      <td>377.92710</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#995C60</td>
      <td>#5A662E</td>
      <td>#5A662E</td>
      <td>#eb2f3c</td>
      <td>2</td>
      <td>137</td>
      <td>6</td>
      <td>1</td>
      <td>5</td>
    </tr>
    <tr>
      <th>1291813781_5-20250227</th>
      <td>QM23.50.001.CX.44.03.07.02</td>
      <td>1291813781</td>
      <td>sis_cellpose_v1</td>
      <td>6375.0240</td>
      <td>386.81006</td>
      <td>False</td>
      <td>HMBA-MERSCOPE-QM23.50.001-BG</td>
      <td>MERSCOPE-QM23.50.001-BG</td>
      <td>NaN</td>
      <td>1291813781</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#a8afa5</td>
      <td>#a8afa5</td>
      <td>#a8afa5</td>
      <td>#f16ebe</td>
      <td>4</td>
      <td>288</td>
      <td>17</td>
      <td>1</td>
      <td>13</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 54 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">marmoset_cell_to_cluster</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_color&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_order</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
<span class="n">marmoset_cell_metadata_with_genes</span> <span class="o">=</span> <span class="n">fill_adjacent_data</span><span class="p">(</span><span class="n">marmoset_cell_metadata_with_genes</span><span class="p">)</span>

<span class="k">del</span> <span class="n">marmoset_cell_to_cluster</span>

<span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>brain_section_barcode</th>
      <th>segmentation_job_id</th>
      <th>x_experiment</th>
      <th>y_experiment</th>
      <th>qc_pass</th>
      <th>dataset_label</th>
      <th>feature_matrix_label</th>
      <th>abc_sample_id</th>
      <th>brain_section_barcode_specimen</th>
      <th>...</th>
      <th>Neighborhood_color</th>
      <th>Class_color</th>
      <th>Subclass_color</th>
      <th>Group_color</th>
      <th>Cluster_color</th>
      <th>Class_order</th>
      <th>Cluster_order</th>
      <th>Group_order</th>
      <th>Neighborhood_order</th>
      <th>Subclass_order</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1382042951_abmkfncc-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5923.492676</td>
      <td>6160.654297</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>c8c08d8c-5147-4d03-b6f4-ca7bf4d0b8d1</td>
      <td>1382042951</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#66391F</td>
      <td>#5a9ad6</td>
      <td>3</td>
      <td>158</td>
      <td>9</td>
      <td>1</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1382042951_abmmdgea-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>6137.572266</td>
      <td>6217.075684</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>f10bad2c-9cc7-44bf-bd7e-4e61d28121bd</td>
      <td>1382042951</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#401e66</td>
      <td>#401e66</td>
      <td>#195f8d</td>
      <td>#9c3de0</td>
      <td>1</td>
      <td>36</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1382042951_abmojfce-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5802.674805</td>
      <td>5614.007812</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>080466ff-3dea-44f1-b6e2-7892e8c12c95</td>
      <td>1382042951</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#401e66</td>
      <td>#564860</td>
      <td>#564860</td>
      <td>#f12367</td>
      <td>1</td>
      <td>81</td>
      <td>3</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <th>1382042951_abmomnim-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>5805.942383</td>
      <td>5599.325684</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>cee3bf8b-a159-42d1-983a-ac149b9edda9</td>
      <td>1382042951</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#401e66</td>
      <td>#564860</td>
      <td>#564860</td>
      <td>#82d054</td>
      <td>1</td>
      <td>84</td>
      <td>3</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <th>1382042951_aeiamgga-1-20250227</th>
      <td>CJ23.56.004.CX.42.01.06</td>
      <td>1382042951</td>
      <td>xenium_cell_segmentation_kit_v1</td>
      <td>6291.268066</td>
      <td>6068.720215</td>
      <td>True</td>
      <td>HMBA-Xenium-CJ23.56.004-BG</td>
      <td>Xenium-CJ23.56.004-BG</td>
      <td>3f24c46e-9ab1-4a54-a6fc-591f752edb48</td>
      <td>1382042951</td>
      <td>...</td>
      <td>#a8afa5</td>
      <td>#594a26</td>
      <td>#594a26</td>
      <td>#66391F</td>
      <td>#5a9ad6</td>
      <td>3</td>
      <td>158</td>
      <td>9</td>
      <td>1</td>
      <td>7</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 55 columns</p>
</div></div></div>
</div>
<section id="visualize-gene-expression-for-selected-genes-by-the-hmba-bg-taxonomy">
<h3>Visualize gene expression for selected genes by the HMBA-BG taxonomy.<a class="headerlink" href="#visualize-gene-expression-for-selected-genes-by-the-hmba-bg-taxonomy" title="Link to this heading">#</a></h3>
<p>Below we average the expression of our selected genes across taxons from the HMBA-BG taxonomy. We show the expression for taxons at the class level for each species. For a side by side comparison of gene expression heatmaps for each species, see the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_10X_snRNASeq_tutorial.html">HMBA 10X gene expression tutorial</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_gene_names</span> <span class="o">=</span> <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">human_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">aggregate_by_metadata</span><span class="p">(</span><span class="n">human_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">human_gene_names</span><span class="p">,</span> <span class="s1">&#39;Class&#39;</span><span class="p">)</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">agg</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="nb">list</span><span class="p">(</span><span class="nb">reversed</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">agg</span><span class="o">.</span><span class="n">index</span><span class="p">)))]</span>
<span class="n">plot_heatmap</span><span class="p">(</span><span class="n">agg</span><span class="p">,</span> <span class="mi">8</span><span class="p">,</span> <span class="mi">11</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="mi">15</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Human - Class&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/cd884a9974fba0b2945d779d148a789effc0c14f435e567c50b2fba6cd70dee8.png" src="../_images/cd884a9974fba0b2945d779d148a789effc0c14f435e567c50b2fba6cd70dee8.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">macaque_gene_names</span> <span class="o">=</span> <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">macaque_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">aggregate_by_metadata</span><span class="p">(</span><span class="n">macaque_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">macaque_gene_names</span><span class="p">,</span> <span class="s1">&#39;Class&#39;</span><span class="p">)</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">agg</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="nb">list</span><span class="p">(</span><span class="nb">reversed</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">agg</span><span class="o">.</span><span class="n">index</span><span class="p">)))]</span>
<span class="n">plot_heatmap</span><span class="p">(</span><span class="n">agg</span><span class="p">,</span> <span class="mi">8</span><span class="p">,</span> <span class="mi">11</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="mi">15</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Macaque - Class&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/c1f8157491ae08d16139cdccc31316cc62e701e16ab88ee999de8c618f440731.png" src="../_images/c1f8157491ae08d16139cdccc31316cc62e701e16ab88ee999de8c618f440731.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">marmoset_gene_names</span> <span class="o">=</span> <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span>
    <span class="n">marmoset_cell_metadata_with_genes</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">gene_names</span><span class="p">)</span>
<span class="p">]</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">aggregate_by_metadata</span><span class="p">(</span><span class="n">marmoset_cell_metadata_with_genes</span><span class="p">,</span> <span class="n">marmoset_gene_names</span><span class="p">,</span> <span class="s1">&#39;Class&#39;</span><span class="p">)</span>
<span class="n">agg</span> <span class="o">=</span> <span class="n">agg</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="nb">list</span><span class="p">(</span><span class="nb">reversed</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">agg</span><span class="o">.</span><span class="n">index</span><span class="p">)))]</span>
<span class="n">plot_heatmap</span><span class="p">(</span><span class="n">agg</span><span class="p">,</span> <span class="mi">8</span><span class="p">,</span> <span class="mi">11</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="mi">15</span><span class="p">,</span> <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;[Ave Raw Counts]&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Marmoset - Class&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/5b47632ec03feca46d4eb43181524702b08820dab1c44336f99c80eb8d55f74e.png" src="../_images/5b47632ec03feca46d4eb43181524702b08820dab1c44336f99c80eb8d55f74e.png" />
</div>
</div>
<p>To learn more about the spatial data see the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_spatial_slabs_and_taxonomy.html">HBMA-BG Spatial Slabs and Taxonomy notebook</a>.</p>
<p>To learn more about the taxonomy used in this notebook and the data it is derived from, see the <a class="reference external" href="https://alleninstitute.github.io/abc_atlas_access/notebooks/hmba_bg_clustering_analysis_and_annotation.html">HMBA-BG Clustering Analysis and Taxonomy notebook</a>.</p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="hmba_bg_spatial_slabs_and_taxonomy.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Human-Mammalian Brain - Basal Ganglia spatial transcriptomic slab coordinates and annotation</p>
      </div>
    </a>
    <a class="right-next"
       href="../docs/README.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Resource Pages</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#geojson-polygons">Geojson polygons</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-expression-data">Gene expression data</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-the-slabs">Plotting the Slabs</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#manual-anatomical-annotations">Manual anatomical annotations</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plotting-heatmap-expression-data">Plotting heatmap expression data.</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualize-gene-expression-for-selected-genes-by-manual-annotation-region">Visualize gene expression for selected genes by manual annotation region.</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxonomy-mapping">Taxonomy mapping</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualize-gene-expression-for-selected-genes-by-the-hmba-bg-taxonomy">Visualize gene expression for selected genes by the HMBA-BG taxonomy.</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>