{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MERFISH whole brain spatial transcriptomics (part 1)\n",
    "\n",
    "The MERFISH data consists of 59 coronal sections (10 micron thick) covering the whole anterior to posterior extent of the brain (200 micron apart) from a single adult male mouse. Each section is assayed using the MERSCOPE platform with a 500 gene panel designed to enable us to map cells to the whole mouse brain taxonomy.\n",
    "\n",
    "The MERSCOPE platform uses combinatorial labeling, sequential imaging and error robust barcodes to obtain quantitative (counts) RNA-expression of 100s-1000s of genes. Each probe binds to a specific transcripts resulting in “dots” in the image. Cells are segmented using CellPose software based on DAPI/PolyT stains.\n",
    "\n",
    "Gene expression data for each section is stored as individual h5ad files. The puprose of this set of notebooks is to provide and overview of the data, the file organization and how to combine the data and metadata through example use cases."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import anndata\n",
    "import time\n",
    "import json\n",
    "import matplotlib.pyplot as plt\n",
    "import requests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "user_home = os.getenv('HOME')\n",
    "download_base = os.path.join(user_home,'abc_download_root')\n",
    "\n",
    "url = 'https://allen-brain-cell-atlas.s3-us-west-2.amazonaws.com/releases/20230630/manifest.json'\n",
    "manifest = json.loads(requests.get(url).text)\n",
    "    \n",
    "metadata = manifest['file_listing']['MERFISH-C57BL6J-638850']['metadata']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "view_directory = os.path.join( download_base, \n",
    "                               manifest['directory_listing']['MERFISH-C57BL6J-638850']['directories']['metadata']['relative_path'], \n",
    "                              'views')\n",
    "cache_views = False\n",
    "if cache_views :\n",
    "    os.makedirs( view_directory, exist_ok=True )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data overview\n",
    "\n",
    "### Cell metadata\n",
    "\n",
    "Essential cell metadata is stored as a dataframe. Each row represents one cell indexed by a *cell label*.\n",
    "\n",
    "Each cell is associated with a *brain section label*, *donor label*, *donor genotype*, *donor sex* and *matrix_prefix* identifying which data package this cell is part of. Each cell also has a set of x, y, z coordinates generated by rotating each section so that it is upright with midline approximatedly in the middle of the frame. The z coordinate takes into account the phyiscal spacing between each section.\n",
    "\n",
    "Each cell is mapped to the whole mouse brain using a hierachical correlation method, resulting in the assignment of a *cluster alias\" and *average correlation score*. Score of less than 0.5 is considered a poor mapping result and should not be used in downstream analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of cells =  4330907\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>brain_section_label</th>\n",
       "      <th>cluster_alias</th>\n",
       "      <th>average_correlation_score</th>\n",
       "      <th>matrix_label</th>\n",
       "      <th>donor_label</th>\n",
       "      <th>low_quality_mapping</th>\n",
       "      <th>donor_genotype</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>x</th>\n",
       "      <th>y</th>\n",
       "      <th>z</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_label</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1015221640100570419</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.582276</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>False</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>5.171280</td>\n",
       "      <td>6.489884</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015221640100590598</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.410491</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>True</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015221640100820600</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.568130</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>False</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>6.547826</td>\n",
       "      <td>5.712030</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015221640100580476</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.417244</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>True</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015221640100580189</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.453351</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>True</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    brain_section_label  cluster_alias  \\\n",
       "cell_label                                               \n",
       "1015221640100570419   C57BL6J-638850.01           4090   \n",
       "1015221640100590598   C57BL6J-638850.01           4090   \n",
       "1015221640100820600   C57BL6J-638850.01           4090   \n",
       "1015221640100580476   C57BL6J-638850.01           4090   \n",
       "1015221640100580189   C57BL6J-638850.01           4090   \n",
       "\n",
       "                     average_correlation_score       matrix_label  \\\n",
       "cell_label                                                          \n",
       "1015221640100570419                   0.582276  C57BL6J-638850.01   \n",
       "1015221640100590598                   0.410491  C57BL6J-638850.01   \n",
       "1015221640100820600                   0.568130  C57BL6J-638850.01   \n",
       "1015221640100580476                   0.417244  C57BL6J-638850.01   \n",
       "1015221640100580189                   0.453351  C57BL6J-638850.01   \n",
       "\n",
       "                        donor_label  low_quality_mapping donor_genotype  \\\n",
       "cell_label                                                                \n",
       "1015221640100570419  C57BL6J-638850                False          wt/wt   \n",
       "1015221640100590598  C57BL6J-638850                 True          wt/wt   \n",
       "1015221640100820600  C57BL6J-638850                False          wt/wt   \n",
       "1015221640100580476  C57BL6J-638850                 True          wt/wt   \n",
       "1015221640100580189  C57BL6J-638850                 True          wt/wt   \n",
       "\n",
       "                    donor_sex         x         y    z  \n",
       "cell_label                                              \n",
       "1015221640100570419         M  5.171280  6.489884  0.0  \n",
       "1015221640100590598         M       NaN       NaN  NaN  \n",
       "1015221640100820600         M  6.547826  5.712030  0.0  \n",
       "1015221640100580476         M       NaN       NaN  NaN  \n",
       "1015221640100580189         M       NaN       NaN  NaN  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rpath = metadata['cell_metadata']['files']['csv']['relative_path']\n",
    "file = os.path.join( download_base, rpath)\n",
    "cell = pd.read_csv(file, dtype={\"cell_label\":str})\n",
    "cell.set_index('cell_label',inplace=True)\n",
    "print(\"Number of cells = \", len(cell))\n",
    "cell.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We define a small helper fuction that uses pandas groupby function to see how many unique items are associated for each field and list them out if the number of items is small."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def print_column_info(df) :\n",
    "    \n",
    "    for c in df.columns :\n",
    "        grouped = df[[c]].groupby(c).count()\n",
    "        members = ''\n",
    "        if len(grouped) < 30 :\n",
    "            members = str(list(grouped.index))\n",
    "        print(\"Number of unique %s = %d %s\" % (c,len(grouped),members))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unique brain_section_label = 59 \n",
      "Number of unique cluster_alias = 5194 \n",
      "Number of unique average_correlation_score = 4328111 \n",
      "Number of unique matrix_label = 59 \n",
      "Number of unique donor_label = 1 ['C57BL6J-638850']\n",
      "Number of unique low_quality_mapping = 2 [False, True]\n",
      "Number of unique donor_genotype = 1 ['wt/wt']\n",
      "Number of unique donor_sex = 1 ['M']\n",
      "Number of unique x = 3940980 \n",
      "Number of unique y = 3940980 \n",
      "Number of unique z = 59 \n"
     ]
    }
   ],
   "source": [
    "print_column_info(cell)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also bring in the pivot table from the \"cluster annotation tutorial\" to associate each cell with terms at each cell type classification level and the corresponding color."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "taxonomy_metadata = manifest['file_listing']['WMB-taxonomy']['metadata']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neurotransmitter</th>\n",
       "      <th>division</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>supertype</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_alias</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Glut</td>\n",
       "      <td>1 Pallium glutamatergic</td>\n",
       "      <td>01 IT-ET Glut</td>\n",
       "      <td>018 L2 IT PPP-APr Glut</td>\n",
       "      <td>0081 L2 IT PPP-APr Glut_3</td>\n",
       "      <td>0327 L2 IT PPP-APr Glut_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Glut</td>\n",
       "      <td>1 Pallium glutamatergic</td>\n",
       "      <td>01 IT-ET Glut</td>\n",
       "      <td>018 L2 IT PPP-APr Glut</td>\n",
       "      <td>0081 L2 IT PPP-APr Glut_3</td>\n",
       "      <td>0328 L2 IT PPP-APr Glut_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Glut</td>\n",
       "      <td>1 Pallium glutamatergic</td>\n",
       "      <td>01 IT-ET Glut</td>\n",
       "      <td>018 L2 IT PPP-APr Glut</td>\n",
       "      <td>0080 L2 IT PPP-APr Glut_2</td>\n",
       "      <td>0323 L2 IT PPP-APr Glut_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Glut</td>\n",
       "      <td>1 Pallium glutamatergic</td>\n",
       "      <td>01 IT-ET Glut</td>\n",
       "      <td>018 L2 IT PPP-APr Glut</td>\n",
       "      <td>0080 L2 IT PPP-APr Glut_2</td>\n",
       "      <td>0324 L2 IT PPP-APr Glut_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Glut</td>\n",
       "      <td>1 Pallium glutamatergic</td>\n",
       "      <td>01 IT-ET Glut</td>\n",
       "      <td>018 L2 IT PPP-APr Glut</td>\n",
       "      <td>0080 L2 IT PPP-APr Glut_2</td>\n",
       "      <td>0326 L2 IT PPP-APr Glut_2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              neurotransmitter                 division          class  \\\n",
       "cluster_alias                                                            \n",
       "1                         Glut  1 Pallium glutamatergic  01 IT-ET Glut   \n",
       "2                         Glut  1 Pallium glutamatergic  01 IT-ET Glut   \n",
       "3                         Glut  1 Pallium glutamatergic  01 IT-ET Glut   \n",
       "4                         Glut  1 Pallium glutamatergic  01 IT-ET Glut   \n",
       "5                         Glut  1 Pallium glutamatergic  01 IT-ET Glut   \n",
       "\n",
       "                             subclass                  supertype  \\\n",
       "cluster_alias                                                      \n",
       "1              018 L2 IT PPP-APr Glut  0081 L2 IT PPP-APr Glut_3   \n",
       "2              018 L2 IT PPP-APr Glut  0081 L2 IT PPP-APr Glut_3   \n",
       "3              018 L2 IT PPP-APr Glut  0080 L2 IT PPP-APr Glut_2   \n",
       "4              018 L2 IT PPP-APr Glut  0080 L2 IT PPP-APr Glut_2   \n",
       "5              018 L2 IT PPP-APr Glut  0080 L2 IT PPP-APr Glut_2   \n",
       "\n",
       "                                 cluster  \n",
       "cluster_alias                             \n",
       "1              0327 L2 IT PPP-APr Glut_3  \n",
       "2              0328 L2 IT PPP-APr Glut_3  \n",
       "3              0323 L2 IT PPP-APr Glut_2  \n",
       "4              0324 L2 IT PPP-APr Glut_2  \n",
       "5              0326 L2 IT PPP-APr Glut_2  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rpath = taxonomy_metadata['cluster_to_cluster_annotation_membership_pivoted']['files']['csv']['relative_path']\n",
    "file = os.path.join( download_base, rpath)\n",
    "cluster_details = pd.read_csv(file)\n",
    "cluster_details.set_index('cluster_alias', inplace=True)\n",
    "cluster_details.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neurotransmitter_color</th>\n",
       "      <th>division_color</th>\n",
       "      <th>class_color</th>\n",
       "      <th>subclass_color</th>\n",
       "      <th>supertype_color</th>\n",
       "      <th>cluster_color</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_alias</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>#2B93DF</td>\n",
       "      <td>#4D58FF</td>\n",
       "      <td>#f74294</td>\n",
       "      <td>#0F6632</td>\n",
       "      <td>#266DFF</td>\n",
       "      <td>#26FF8C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>#2B93DF</td>\n",
       "      <td>#4D58FF</td>\n",
       "      <td>#f74294</td>\n",
       "      <td>#0F6632</td>\n",
       "      <td>#266DFF</td>\n",
       "      <td>#280099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>#2B93DF</td>\n",
       "      <td>#4D58FF</td>\n",
       "      <td>#f74294</td>\n",
       "      <td>#0F6632</td>\n",
       "      <td>#002BCC</td>\n",
       "      <td>#5CCCA0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>#2B93DF</td>\n",
       "      <td>#4D58FF</td>\n",
       "      <td>#f74294</td>\n",
       "      <td>#0F6632</td>\n",
       "      <td>#002BCC</td>\n",
       "      <td>#CC6D1F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>#2B93DF</td>\n",
       "      <td>#4D58FF</td>\n",
       "      <td>#f74294</td>\n",
       "      <td>#0F6632</td>\n",
       "      <td>#002BCC</td>\n",
       "      <td>#FF8E26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              neurotransmitter_color division_color class_color  \\\n",
       "cluster_alias                                                     \n",
       "1                            #2B93DF        #4D58FF     #f74294   \n",
       "2                            #2B93DF        #4D58FF     #f74294   \n",
       "3                            #2B93DF        #4D58FF     #f74294   \n",
       "4                            #2B93DF        #4D58FF     #f74294   \n",
       "5                            #2B93DF        #4D58FF     #f74294   \n",
       "\n",
       "              subclass_color supertype_color cluster_color  \n",
       "cluster_alias                                               \n",
       "1                    #0F6632         #266DFF       #26FF8C  \n",
       "2                    #0F6632         #266DFF       #280099  \n",
       "3                    #0F6632         #002BCC       #5CCCA0  \n",
       "4                    #0F6632         #002BCC       #CC6D1F  \n",
       "5                    #0F6632         #002BCC       #FF8E26  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rpath = taxonomy_metadata['cluster_to_cluster_annotation_membership_color']['files']['csv']['relative_path']\n",
    "file = os.path.join( download_base, rpath)\n",
    "cluster_colors = pd.read_csv(file)\n",
    "cluster_colors.set_index('cluster_alias', inplace=True)\n",
    "cluster_colors.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>brain_section_label</th>\n",
       "      <th>cluster_alias</th>\n",
       "      <th>average_correlation_score</th>\n",
       "      <th>matrix_label</th>\n",
       "      <th>donor_label</th>\n",
       "      <th>low_quality_mapping</th>\n",
       "      <th>donor_genotype</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>x</th>\n",
       "      <th>y</th>\n",
       "      <th>...</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>supertype</th>\n",
       "      <th>cluster</th>\n",
       "      <th>neurotransmitter_color</th>\n",
       "      <th>division_color</th>\n",
       "      <th>class_color</th>\n",
       "      <th>subclass_color</th>\n",
       "      <th>supertype_color</th>\n",
       "      <th>cluster_color</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_label</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1015221640100570419</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.582276</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>False</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>5.171280</td>\n",
       "      <td>6.489884</td>\n",
       "      <td>...</td>\n",
       "      <td>22 MY GABA</td>\n",
       "      <td>282 MY Calcb Chol</td>\n",
       "      <td>0999 MY Calcb Chol</td>\n",
       "      <td>5100 MY Calcb Chol</td>\n",
       "      <td>#73E785</td>\n",
       "      <td>#9F3DCC</td>\n",
       "      <td>#E5C865</td>\n",
       "      <td>#FF4D81</td>\n",
       "      <td>#2E5499</td>\n",
       "      <td>#91FF4D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015221640100590598</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.410491</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>True</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>22 MY GABA</td>\n",
       "      <td>282 MY Calcb Chol</td>\n",
       "      <td>0999 MY Calcb Chol</td>\n",
       "      <td>5100 MY Calcb Chol</td>\n",
       "      <td>#73E785</td>\n",
       "      <td>#9F3DCC</td>\n",
       "      <td>#E5C865</td>\n",
       "      <td>#FF4D81</td>\n",
       "      <td>#2E5499</td>\n",
       "      <td>#91FF4D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015221640100820600</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.568130</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>False</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>6.547826</td>\n",
       "      <td>5.712030</td>\n",
       "      <td>...</td>\n",
       "      <td>22 MY GABA</td>\n",
       "      <td>282 MY Calcb Chol</td>\n",
       "      <td>0999 MY Calcb Chol</td>\n",
       "      <td>5100 MY Calcb Chol</td>\n",
       "      <td>#73E785</td>\n",
       "      <td>#9F3DCC</td>\n",
       "      <td>#E5C865</td>\n",
       "      <td>#FF4D81</td>\n",
       "      <td>#2E5499</td>\n",
       "      <td>#91FF4D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015221640100580476</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.417244</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>True</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>22 MY GABA</td>\n",
       "      <td>282 MY Calcb Chol</td>\n",
       "      <td>0999 MY Calcb Chol</td>\n",
       "      <td>5100 MY Calcb Chol</td>\n",
       "      <td>#73E785</td>\n",
       "      <td>#9F3DCC</td>\n",
       "      <td>#E5C865</td>\n",
       "      <td>#FF4D81</td>\n",
       "      <td>#2E5499</td>\n",
       "      <td>#91FF4D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015221640100580189</th>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>4090</td>\n",
       "      <td>0.453351</td>\n",
       "      <td>C57BL6J-638850.01</td>\n",
       "      <td>C57BL6J-638850</td>\n",
       "      <td>True</td>\n",
       "      <td>wt/wt</td>\n",
       "      <td>M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>22 MY GABA</td>\n",
       "      <td>282 MY Calcb Chol</td>\n",
       "      <td>0999 MY Calcb Chol</td>\n",
       "      <td>5100 MY Calcb Chol</td>\n",
       "      <td>#73E785</td>\n",
       "      <td>#9F3DCC</td>\n",
       "      <td>#E5C865</td>\n",
       "      <td>#FF4D81</td>\n",
       "      <td>#2E5499</td>\n",
       "      <td>#91FF4D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    brain_section_label  cluster_alias  \\\n",
       "cell_label                                               \n",
       "1015221640100570419   C57BL6J-638850.01           4090   \n",
       "1015221640100590598   C57BL6J-638850.01           4090   \n",
       "1015221640100820600   C57BL6J-638850.01           4090   \n",
       "1015221640100580476   C57BL6J-638850.01           4090   \n",
       "1015221640100580189   C57BL6J-638850.01           4090   \n",
       "\n",
       "                     average_correlation_score       matrix_label  \\\n",
       "cell_label                                                          \n",
       "1015221640100570419                   0.582276  C57BL6J-638850.01   \n",
       "1015221640100590598                   0.410491  C57BL6J-638850.01   \n",
       "1015221640100820600                   0.568130  C57BL6J-638850.01   \n",
       "1015221640100580476                   0.417244  C57BL6J-638850.01   \n",
       "1015221640100580189                   0.453351  C57BL6J-638850.01   \n",
       "\n",
       "                        donor_label  low_quality_mapping donor_genotype  \\\n",
       "cell_label                                                                \n",
       "1015221640100570419  C57BL6J-638850                False          wt/wt   \n",
       "1015221640100590598  C57BL6J-638850                 True          wt/wt   \n",
       "1015221640100820600  C57BL6J-638850                False          wt/wt   \n",
       "1015221640100580476  C57BL6J-638850                 True          wt/wt   \n",
       "1015221640100580189  C57BL6J-638850                 True          wt/wt   \n",
       "\n",
       "                    donor_sex         x         y  ...       class  \\\n",
       "cell_label                                         ...               \n",
       "1015221640100570419         M  5.171280  6.489884  ...  22 MY GABA   \n",
       "1015221640100590598         M       NaN       NaN  ...  22 MY GABA   \n",
       "1015221640100820600         M  6.547826  5.712030  ...  22 MY GABA   \n",
       "1015221640100580476         M       NaN       NaN  ...  22 MY GABA   \n",
       "1015221640100580189         M       NaN       NaN  ...  22 MY GABA   \n",
       "\n",
       "                              subclass           supertype  \\\n",
       "cell_label                                                   \n",
       "1015221640100570419  282 MY Calcb Chol  0999 MY Calcb Chol   \n",
       "1015221640100590598  282 MY Calcb Chol  0999 MY Calcb Chol   \n",
       "1015221640100820600  282 MY Calcb Chol  0999 MY Calcb Chol   \n",
       "1015221640100580476  282 MY Calcb Chol  0999 MY Calcb Chol   \n",
       "1015221640100580189  282 MY Calcb Chol  0999 MY Calcb Chol   \n",
       "\n",
       "                                cluster neurotransmitter_color division_color  \\\n",
       "cell_label                                                                      \n",
       "1015221640100570419  5100 MY Calcb Chol                #73E785        #9F3DCC   \n",
       "1015221640100590598  5100 MY Calcb Chol                #73E785        #9F3DCC   \n",
       "1015221640100820600  5100 MY Calcb Chol                #73E785        #9F3DCC   \n",
       "1015221640100580476  5100 MY Calcb Chol                #73E785        #9F3DCC   \n",
       "1015221640100580189  5100 MY Calcb Chol                #73E785        #9F3DCC   \n",
       "\n",
       "                    class_color subclass_color supertype_color cluster_color  \n",
       "cell_label                                                                    \n",
       "1015221640100570419     #E5C865        #FF4D81         #2E5499       #91FF4D  \n",
       "1015221640100590598     #E5C865        #FF4D81         #2E5499       #91FF4D  \n",
       "1015221640100820600     #E5C865        #FF4D81         #2E5499       #91FF4D  \n",
       "1015221640100580476     #E5C865        #FF4D81         #2E5499       #91FF4D  \n",
       "1015221640100580189     #E5C865        #FF4D81         #2E5499       #91FF4D  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_extended = cell.join(cluster_details,on='cluster_alias')\n",
    "cell_extended = cell_extended.join(cluster_colors,on='cluster_alias')\n",
    "cell_extended.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unique brain_section_label = 59 \n",
      "Number of unique cluster_alias = 5194 \n",
      "Number of unique average_correlation_score = 4328111 \n",
      "Number of unique matrix_label = 59 \n",
      "Number of unique donor_label = 1 ['C57BL6J-638850']\n",
      "Number of unique low_quality_mapping = 2 [False, True]\n",
      "Number of unique donor_genotype = 1 ['wt/wt']\n",
      "Number of unique donor_sex = 1 ['M']\n",
      "Number of unique x = 3940980 \n",
      "Number of unique y = 3940980 \n",
      "Number of unique z = 59 \n",
      "Number of unique neurotransmitter = 9 ['Chol', 'Dopa', 'GABA', 'GABA-Glyc', 'Glut', 'Glut-GABA', 'Hist', 'Nora', 'Sero']\n",
      "Number of unique division = 7 ['1 Pallium glutamatergic', '2 Subpallium GABAergic', '3 PAL-sAMY-TH-HY-MB-HB neuronal', '4 CBX-MOB-other neuronal', '5 Neuroglial', '6 Vascular', '7 Immune']\n",
      "Number of unique class = 32 \n",
      "Number of unique subclass = 307 \n",
      "Number of unique supertype = 1114 \n",
      "Number of unique cluster = 5194 \n"
     ]
    }
   ],
   "source": [
    "print_column_info(cell_extended)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For convenience, we can cache this view for later reuse."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if cache_views :\n",
    "    file = os.path.join( view_directory, 'cell_metadata_with_cluster_annotation.csv')\n",
    "    cell_extended.to_csv( file )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene expression matrices\n",
    "\n",
    "The expression of each section is stored as an individual anndata h5ad file with minimal metadata. In this next section, we provide example code on how to open the file and connect with the rich cell level metadata discussed above.\n",
    "\n",
    "For each section, there are two h5ad files one storing the raw counts and the other log normalization of it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_matrices = manifest['file_listing']['MERFISH-C57BL6J-638850']['expression_matrices']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_matrices['C57BL6J-638850.38']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rpath = expression_matrices['C57BL6J-638850.38']['log2']['files']['h5ad']['relative_path']\n",
    "file = os.path.join( download_base, rpath)\n",
    "print(file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We use the anndata's *read_h5ad* function to open the package for the log2 normalization file. The \"backed='r'\" makes use of the lazy loading functionality to only load required data. By default, anndata will load the entire expression matrix in memory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = anndata.read_h5ad(file,backed='r')\n",
    "print(adata)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Genes are represented as \"variables\". For this data, the var dataframe is indexed by the Ensembl gene identifier with one metadata column gene symbol."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"Number of genes = \", len(adata.var))\n",
    "adata.var.index[0:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cells are represented as \"observations\". We can easily connect the cells in each section to the extended cell metadata and filter out poorly mapped cells with simple pandas functions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_filtered = cell_extended.loc[adata.obs.index]\n",
    "pred = cell_filtered['low_quality_mapping'] == False\n",
    "cell_filtered = cell_filtered[pred]\n",
    "\n",
    "print(\"Number of cells = \", len(cell_filtered))\n",
    "print_column_info( cell_filtered )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We define a small helper function *plot section* to visualize the cells in anatomical context colorized by: neurotransmitter identity, cell types division, class and subclass."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_section( xx, yy, cc=None, val=None, fig_width = 8, fig_height = 8, cmap=None ) :\n",
    "    fig, ax = plt.subplots()\n",
    "    fig.set_size_inches(fig_width, fig_height)\n",
    "    if cmap is not None :\n",
    "        plt.scatter(xx,yy,s=0.5,c=val,marker='.',cmap=cmap)\n",
    "    elif cc is not None :\n",
    "        plt.scatter(xx,yy,s=0.5,color=cc,marker='.')\n",
    "    ax.set_ylim(11,0)\n",
    "    ax.set_xlim(0,11)\n",
    "    ax.axis('equal')\n",
    "    ax.set_xticks([])\n",
    "    ax.set_yticks([])\n",
    "    return fig, ax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax = plot_section(cell_filtered['x'], cell_filtered['y'],cell_filtered['neurotransmitter_color'])\n",
    "res = ax.set_title(\"Neuortransmitter Identity\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax = plot_section(cell_filtered['x'], cell_filtered['y'],cell_filtered['division_color'])\n",
    "res = ax.set_title(\"Cell Types: Division\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax = plot_section(cell_filtered['x'], cell_filtered['y'],cell_filtered['class_color'])\n",
    "res = ax.set_title(\"Cell Types: Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax = plot_section(cell_filtered['x'], cell_filtered['y'],cell_filtered['subclass_color'])\n",
    "res = ax.set_title(\"Cell Types: Subclass\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example use cases\n",
    "\n",
    "In this section, we explore the same two use cases as with the 10x data. The first example looks at the expression of nine canonical neurotransmitter transporter genes and the second the expression of gene Tac2. To support these use cases, we will create a smaller submatrix (all cells and 10 genes) that read it into memory. *Note this operation takes around 10 seconds*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ntgenes = ['Slc17a7','Slc17a6','Slc17a8','Slc32a1','Slc6a5','Slc6a3','Slc6a4']\n",
    "exgenes = ['Tac2']\n",
    "gnames = ntgenes + exgenes\n",
    "pred = [x in gnames for x in adata.var.gene_symbol]\n",
    "gene_filtered = adata.var[pred]\n",
    "gene_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "start = time.process_time()\n",
    "asubset = adata[cell_filtered.index,gene_filtered.index].to_memory()\n",
    "print(\"time taken: \", time.process_time() - start)\n",
    "print(asubset)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We define two helper functions to (1) *create_expression_dataframe*: create joined gene expression and cell metadata dataframe based for a set of input genes and (2) *aggregate_by_metadata* which computes the average expression for each term in a given category."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_expression_dataframe( ad, gf ) :\n",
    "    gdata = ad[:,gf.index].to_df()\n",
    "    gdata.columns = gf.gene_symbol\n",
    "    joined = cell_filtered.join( gdata )\n",
    "    return joined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def aggregate_by_metadata( df, gnames, value, sort = False ) :\n",
    "    grouped = df.groupby(value)[gnames].mean()\n",
    "    if sort :\n",
    "        grouped = grouped.sort_values(by=gnames[0],ascending=False)\n",
    "    return grouped"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Expression of canonical neurotransmitter transporter genes\n",
    "\n",
    "During analysis, clusters were assigned neurotransmitter identities based on the expression of of canonical neurotransmitter transporter genes. In this example, we create a dataframe comprising of cells in the 'C57BL6J-638850.40' section and expression of the 9 solute carrier family genes. We then group the cells by the assigned neurotransmitter class and compute the mean expression for each group and visualized as a colorized table. The results are similar of the 10x data restricted to cells from the thalamus."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pred = [x in ntgenes for x in asubset.var.gene_symbol]\n",
    "gf = asubset.var[pred]\n",
    "\n",
    "ntexp = create_expression_dataframe( asubset, gf )\n",
    "agg = aggregate_by_metadata(ntexp, gf.gene_symbol, 'neurotransmitter')\n",
    "agg = agg[ntgenes]\n",
    "agg.style.background_gradient(cmap='Reds')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The expression data can also be visualized as scatter point in anatomical context. Gene Slc17a7 is highly enriched in the cortex , hippocampus, medial habenula with some expression in the thalamus. On the other hand gene Slc32a1 is enriched in the striatum with sparse expression on the cortical plate."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax = plot_section(ntexp['x'], ntexp['y'],val=ntexp['Slc17a7'],cmap='Reds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax = plot_section(ntexp['x'], ntexp['y'],val=ntexp['Slc32a1'],cmap='Reds')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Expression of Tachykinin 2 (Tac2) in the whole brain\n",
    "\n",
    "In mice, the tachykinin 2 (Tac2) gene encodes neuropeptide called neurokinin B (NkB). Tac2 is produced by neurons in specific regions of the brain know to be invovled in emotion and social behavior. Based on [ISH data](https://mouse.brain-map.org/experiment/siv?id=77279001&imageId=77284584&initImage=ish&coordSystem=pixel&x=5384.5&y=3832.5&z=3) from the Allen Mouse Brain Atlas, Tac 2 is sparsely expressed in the mouse isocortex and densely enriched is specific subcortical regions such the medial habenula (MH), the amygdala and hypothalamus.\n",
    "\n",
    "<br>\n",
    "<div>\n",
    "<img src=\"Tac2_ISH_Coronal_Section.JPG\" width=\"600\"/>\n",
    "</div>\n",
    "\n",
    "\n",
    "In this example, we create a dataframe comprising of cells in section 'C57BL6J-638850.40' and expression values of Tac2 in those cells."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gf = asubset.var[asubset.var.gene_symbol == 'Tac2']\n",
    "tac2 = create_expression_dataframe( asubset, gf )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Grouping cells by neurotransmitter identites and computing the mean expression in each group, we can observed that Tac2 gene is highly enriched in cholinergic cell types with some expression in Glut-GABA types."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "agg = aggregate_by_metadata(tac2, gf.gene_symbol, 'neurotransmitter', True).head(8)\n",
    "agg.style.background_gradient(cmap='Reds')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Grouping cells by cell types class, shows that Tac2 is enriched in class \"08 MH-LH Glut\" with cells restricted to the medial (MH) and lateral (LH) habenula and a mixture of glutamatergic and cholinergic types."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "agg = aggregate_by_metadata(tac2, gf.gene_symbol, 'class', True).head(8)\n",
    "agg.style.background_gradient(cmap='Reds')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "At the next level, grouping by subclass, shows Tac2 is enriched in sublcasses with specificity to the medial habenula (MH), bed muclei of the stria terminalis (BST) and main olfactory blub (MOB)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "agg = aggregate_by_metadata(tac2, gf.gene_symbol, 'subclass', True).head(8)\n",
    "agg.style.background_gradient(cmap='Reds')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Plotting Tac2 expression in anatomical coordinates results in a image that is comparable to the ISH image data showing enriched expression in the medial habenula, amydala and hypothalamus and sparse expression in the cortex."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig,ax = plot_section(tac2['x'], tac2['y'],val=tac2['Tac2'],cmap='Reds')\n",
    "res = ax.set_title('Gene Tac2')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Close h5ad file and clean up"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.file.close()\n",
    "del adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
