

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang) &#8212; Allen Brain Cell Atlas - Data Access</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/zhuang_merfish_tutorial';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Resource Pages" href="../docs/README.html" />
    <link rel="prev" title="A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)" href="../descriptions/notebook_subtitle4.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
    
    
      
    
    
    <img src="../_static/allen_logo.jpeg" class="logo__image only-light" alt="Logo image"/>
    <script>document.write(`<img src="../_static/allen_logo.jpeg" class="logo__image only-dark" alt="Logo image"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Allen Brain Cell Atlas - Data Access
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">INTRODUCTION</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">DATA</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/data_release_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv2.html">Whole mouse brain 10Xv2 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10Xv3.html">Whole mouse brain 10Xv3 single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10XMulti.html">Whole mouse brain 10XMulti single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-10X.html">Clustering analysis of whole adult mouse brain 10X single cell transcriptomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-taxonomy.html">Whole adult mouse brain taxonomy of cell types</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/WMB-neighborhoods.html">Cluster groups and neighborhood specific UMAP embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Allen-CCF-2020.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/MERFISH-C57BL6J-638850-CCF.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/data_release_subtitle2.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-1.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-2.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-3.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-3)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../descriptions/Zhuang-ABCA-4.html">MERFISH spatial transcriptomics dataset of a single adult mouse brain (Zhuang-ABCA-4)</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">NOTEBOOKS</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle1.html">Mouse whole-brain transcriptomic cell type atlas (Hongkui Zeng)</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-3"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="cluster_annotation_tutorial.html">10x RNA-seq clustering analysis and annotation (CCN20230722)</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle2.html">10x scRNA-seq gene expression data</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-4"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_1.html">10x RNA-seq gene expression data (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2a.html">10x RNA-seq gene expression data (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="10x_snRNASeq_tutorial_part_2b.html">10x RNA-seq gene expression data (part 2b)</a></li>
</ul>
</li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../descriptions/notebook_subtitle3.html">MERFISH whole brain spatial transcriptomics</a><input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-5"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_1.html">MERFISH whole brain spatial transcriptomics (part 1)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2a.html">MERFISH whole brain spatial transcriptomics (part 2a)</a></li>
<li class="toctree-l3"><a class="reference internal" href="merfish_tutorial_part_2b.html">MERFISH whole brain spatial transcriptomics (part 2b)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="cluster_groups_and_embeddings_tutorial.html">Cluster neighborhoods and embeddings</a></li>
<li class="toctree-l2"><a class="reference internal" href="cluster_neighborhood_gallery.html">Cell type neighborhood gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="ccf_and_parcellation_annotation_tutorial.html">Allen Mouse Common Coordinate Framework (2020 version)</a></li>
<li class="toctree-l2"><a class="reference internal" href="merfish_ccf_registration_tutorial.html">MERFISH CCF mapped coordinates</a></li>
</ul>
</li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../descriptions/notebook_subtitle4.html">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-6"><i class="fa-solid fa-chevron-down"></i></label><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">ADDITIONAL RESOURCES</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../docs/README.html">Resource Pages</a><input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-7"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../docs/gene_lists/README.html">Gene lists</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/cluster_annotations/README.html">Cluster annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../docs/parcellation_annotations/README.html">Parcellation annotations</a></li>
</ul>
</li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/AllenInstitute/abc_atlas_access/issues/new?title=Issue%20on%20page%20%2Fnotebooks/zhuang_merfish_tutorial.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/zhuang_merfish_tutorial.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cluster-annotation">Cluster annotation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-panel">Gene panel</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-expression-matrix">Gene expression matrix</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#example-use-case">Example use case</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ccf-registration-and-parcellation-annotation">CCF registration and parcellation annotation</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="merfish-whole-mouse-brain-spatial-transcriptomics-xiaowei-zhuang">
<h1>MERFISH whole mouse brain spatial transcriptomics (Xiaowei Zhuang)<a class="headerlink" href="#merfish-whole-mouse-brain-spatial-transcriptomics-xiaowei-zhuang" title="Permalink to this heading">#</a></h1>
<p>A collection of in situ, spatially resolved transcriptomic profiles of individual cells in the whole mouse brain by multiplexed error-robust fluorescence in situ hybridization (MERFISH) consisting of ~9 million cells using a 1122 gene panel. We performed MERFISH imaging on 245 coronal and sagittal sections from four animal, obtained 9.3 million segmented cells that passed quality control, and integrated the MERFISH data from the four animals with the scRNA-seq data from the Allen Institute to classify cells. We applied a series of filters to select a subset of cells to be visualized on the ABC atlas. We first removed six fractured tissue slices and 9.1 million cells remained after this step. Then we aligned the spatial coordinates of the cells to the Allen-CCF-2020. For coronal slices that can be registered to the CCF, we used the CCF coordinates to define the coordinates of the center point of the midline and removed cells that substantially passed the midline in the other hemisphere (which has not been registered to the CCF). For the sagittal slices that can be registered to the CCF, we used the CCF coordinates to define the coordinates of the center point of the tissue and removed cells that substantially passed the posterior edge (which has not been registered to the CCF). For the 31 anterior and posterior coronal slices and 3 lateral sagittal slices that cannot be registered to the CCF, we manually aligned and oriented the slices. The x, y coordinates are experimentally measured coordinates after rotating and aligning the tissue slices to the CCF, and the z coordinates are estimated position of each tissue slice in the 3D Allen-CCF 2020 space along the slicing axis based on either the registration results (for slices that can be registered to CCF) or positions of the slices measured during tissue sectioning (for the slices that cannot be registered). The z position is set to zero when the estimated position becomes zero or negative. 8.4 million cells remained after this step. The cell-by-gene matrix of the 8.4 millions cells can be downloaded from the AWS bucket of this animal. We then filtered the cells by cell-classification (label transfer) confidence scores calculated during MERFISH-scRNAseq data integration. 7.0 million cells passed the confidence score threshold for cell subclass label transfer and 5.8 million cells further passed the confidence score threshold for cell cluster label transfer. These 5.8 million cells are included in the cell metadata file that can be downloaded from the the AWS bucket and are displayed on the ABC Atlas. The CCF coordinates of the 5.4 million cells that were registered to the 3D Allen-CCF can be downloaded from the CCF coordinate files in the AWB bucket. The collection spans four mouse specimens (2 coronal sets and 2 sagittal sets). Cells are mapped to the whole mouse brain taxonomy (WMB-taxonomy) and Allen Common Coordinate Framework (Allen-CCF-2020). Refer to <a class="reference external" href="https://doi.org/10.1101/2023.03.06.531348">Zhang et al, 2023</a> for more details.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">anndata</span>
<span class="kn">import</span> <span class="nn">time</span>
<span class="kn">import</span> <span class="nn">json</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">requests</span>
</pre></div>
</div>
</div>
</div>
<p>The prerequisite for running this notebook is that the data have been downloaded to local directory maintaining the organization from the manifest.json. <strong>Change the download_base variable to where you have downloaded the data in your system.</strong></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">version</span> <span class="o">=</span> <span class="s1">&#39;20230830&#39;</span>
<span class="n">download_base</span> <span class="o">=</span> <span class="s1">&#39;../../abc_download_root&#39;</span>

<span class="n">use_local_cache</span> <span class="o">=</span> <span class="kc">False</span>
<span class="n">manifest_path</span> <span class="o">=</span> <span class="s1">&#39;releases/</span><span class="si">%s</span><span class="s1">/manifest.json&#39;</span> <span class="o">%</span> <span class="n">version</span>

<span class="k">if</span> <span class="ow">not</span> <span class="n">use_local_cache</span> <span class="p">:</span>
    <span class="n">url</span> <span class="o">=</span> <span class="s1">&#39;https://allen-brain-cell-atlas.s3-us-west-2.amazonaws.com/&#39;</span> <span class="o">+</span> <span class="n">manifest_path</span>
    <span class="n">manifest</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="n">requests</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">url</span><span class="p">)</span><span class="o">.</span><span class="n">text</span><span class="p">)</span>
<span class="k">else</span> <span class="p">:</span>
    <span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">download_base</span><span class="p">,</span><span class="n">manifest_path</span><span class="p">)</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="s1">&#39;rb&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">manifest</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">datasets</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Zhuang-ABCA-1&#39;</span><span class="p">,</span><span class="s1">&#39;Zhuang-ABCA-2&#39;</span><span class="p">,</span><span class="s1">&#39;Zhuang-ABCA-3&#39;</span><span class="p">,</span><span class="s1">&#39;Zhuang-ABCA-4&#39;</span><span class="p">]</span>
<span class="n">example_section</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;Zhuang-ABCA-1&#39;</span><span class="p">:</span><span class="s1">&#39;Zhuang-ABCA-1.079&#39;</span><span class="p">,</span>
                   <span class="s1">&#39;Zhuang-ABCA-2&#39;</span><span class="p">:</span><span class="s1">&#39;Zhuang-ABCA-2.037&#39;</span><span class="p">,</span>
                   <span class="s1">&#39;Zhuang-ABCA-3&#39;</span><span class="p">:</span><span class="s1">&#39;Zhuang-ABCA-3.010&#39;</span><span class="p">,</span>
                   <span class="s1">&#39;Zhuang-ABCA-4&#39;</span><span class="p">:</span><span class="s1">&#39;Zhuang-ABCA-4.002&#39;</span><span class="p">}</span>

<span class="n">metadata</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
    <span class="n">metadata</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">manifest</span><span class="p">[</span><span class="s1">&#39;file_listing&#39;</span><span class="p">][</span><span class="n">d</span><span class="p">][</span><span class="s1">&#39;metadata&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cache_views</span> <span class="o">=</span> <span class="kc">False</span>

<span class="n">view_directory</span> <span class="o">=</span> <span class="p">{}</span> 

<span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
    
    <span class="n">view_directory</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> 
                                     <span class="n">manifest</span><span class="p">[</span><span class="s1">&#39;directory_listing&#39;</span><span class="p">][</span><span class="n">d</span><span class="p">][</span><span class="s1">&#39;directories&#39;</span><span class="p">][</span><span class="s1">&#39;metadata&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">],</span> 
                                     <span class="s1">&#39;views&#39;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">cache_views</span> <span class="p">:</span>
        <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span> <span class="n">view_directory</span><span class="p">[</span><span class="n">d</span><span class="p">],</span> <span class="n">exist_ok</span><span class="o">=</span><span class="kc">True</span> <span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="data-overview">
<h2>Data overview<a class="headerlink" href="#data-overview" title="Permalink to this heading">#</a></h2>
<section id="cell-metadata">
<h3>Cell metadata<a class="headerlink" href="#cell-metadata" title="Permalink to this heading">#</a></h3>
<p>Essential cell metadata is stored as a dataframe. Each row represents one cell indexed by a <em>cell label</em>.</p>
<p>Each cell is associated with a <em>brain section label</em>, <em>donor label</em>, <em>donor genotype</em>, <em>donor sex</em> and <em>matrix_prefix</em> identifying which data package this cell is part of. Each cell also has a set of x, y, z coordinates generated by rotating each section so that it is upright with midline approximatedly in the middle of the frame.</p>
<p>Each cell is mapped to the whole mouse brain taxonomy resulting in the assignment of a <em>cluster alias</em> and <em>confidence scores</em>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell</span> <span class="o">=</span> <span class="p">{}</span>

<span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
    
    <span class="n">rpath</span> <span class="o">=</span> <span class="n">metadata</span><span class="p">[</span><span class="n">d</span><span class="p">][</span><span class="s1">&#39;cell_metadata&#39;</span><span class="p">][</span><span class="s1">&#39;files&#39;</span><span class="p">][</span><span class="s1">&#39;csv&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">]</span>
    <span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> <span class="n">rpath</span><span class="p">)</span>
    <span class="n">cell</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;cell_label&quot;</span><span class="p">:</span><span class="nb">str</span><span class="p">})</span>
    <span class="n">cell</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="n">sdf</span> <span class="o">=</span> <span class="n">cell</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">)</span>
    
    <span class="nb">print</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="s2">&quot;:&quot;</span><span class="p">,</span><span class="s2">&quot;Number of cells = &quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">cell</span><span class="p">[</span><span class="n">d</span><span class="p">]),</span> <span class="s2">&quot;, &quot;</span><span class="p">,</span> <span class="s2">&quot;Number of sections =&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">sdf</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Zhuang-ABCA-1 : Number of cells =  2846908 ,  Number of sections = 147
Zhuang-ABCA-2 : Number of cells =  1227408 ,  Number of sections = 66
Zhuang-ABCA-3 : Number of cells =  1585843 ,  Number of sections = 23
Zhuang-ABCA-4 : Number of cells =  162578 ,  Number of sections = 3
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell</span><span class="p">[</span><span class="n">datasets</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>feature_matrix_label</th>
      <th>donor_label</th>
      <th>donor_genotype</th>
      <th>donor_sex</th>
      <th>cluster_alias</th>
      <th>x</th>
      <th>y</th>
      <th>z</th>
      <th>subclass_confidence_score</th>
      <th>cluster_confidence_score</th>
      <th>high_quality_transfer</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>182941331246012878296807398333956011710</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>704</td>
      <td>0.682522</td>
      <td>3.366483</td>
      <td>7.829530</td>
      <td>0.969933</td>
      <td>0.718088</td>
      <td>True</td>
    </tr>
    <tr>
      <th>221260934538535633595532020856387724686</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5243</td>
      <td>0.667690</td>
      <td>3.442241</td>
      <td>7.829530</td>
      <td>0.850554</td>
      <td>0.850554</td>
      <td>True</td>
    </tr>
    <tr>
      <th>22228792606814781533240955623030943708</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>14939</td>
      <td>0.638731</td>
      <td>3.474328</td>
      <td>7.829530</td>
      <td>0.888285</td>
      <td>0.649581</td>
      <td>True</td>
    </tr>
    <tr>
      <th>272043042552227961220474294517855477150</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>14939</td>
      <td>0.653425</td>
      <td>3.433218</td>
      <td>7.829530</td>
      <td>0.900000</td>
      <td>0.607080</td>
      <td>True</td>
    </tr>
    <tr>
      <th>110116287883089187971185374239350249328</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5254</td>
      <td>0.623896</td>
      <td>3.513574</td>
      <td>7.829530</td>
      <td>0.999978</td>
      <td>0.689511</td>
      <td>True</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>94310525370042131911495836073267655162</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5255</td>
      <td>0.997247</td>
      <td>3.823236</td>
      <td>9.717769</td>
      <td>0.971385</td>
      <td>0.943456</td>
      <td>True</td>
    </tr>
    <tr>
      <th>298798481479578578007190103666214714353</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>14939</td>
      <td>1.043871</td>
      <td>3.706231</td>
      <td>9.717769</td>
      <td>0.956377</td>
      <td>0.905493</td>
      <td>True</td>
    </tr>
    <tr>
      <th>330756942354980576352210203729462562749</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5261</td>
      <td>1.037680</td>
      <td>3.759890</td>
      <td>9.717769</td>
      <td>0.811520</td>
      <td>0.788901</td>
      <td>True</td>
    </tr>
    <tr>
      <th>47305871059582831548494138048361484565</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5252</td>
      <td>1.044169</td>
      <td>3.758463</td>
      <td>9.717769</td>
      <td>0.991924</td>
      <td>0.634152</td>
      <td>True</td>
    </tr>
    <tr>
      <th>64578198410898899234789748167671783948</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5263</td>
      <td>1.042301</td>
      <td>3.589421</td>
      <td>9.717769</td>
      <td>0.889413</td>
      <td>0.645770</td>
      <td>True</td>
    </tr>
  </tbody>
</table>
<p>2846908 rows × 12 columns</p>
</div></div></div>
</div>
</section>
<section id="cluster-annotation">
<h3>Cluster annotation<a class="headerlink" href="#cluster-annotation" title="Permalink to this heading">#</a></h3>
<p>Read in the pivot table from the “cluster annotation tutorial” to associate each cell with terms at each cell type classification level and the corresponding color.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">taxonomy_metadata</span> <span class="o">=</span> <span class="n">manifest</span><span class="p">[</span><span class="s1">&#39;file_listing&#39;</span><span class="p">][</span><span class="s1">&#39;WMB-taxonomy&#39;</span><span class="p">][</span><span class="s1">&#39;metadata&#39;</span><span class="p">]</span>

<span class="n">rpath</span> <span class="o">=</span> <span class="n">taxonomy_metadata</span><span class="p">[</span><span class="s1">&#39;cluster_to_cluster_annotation_membership_pivoted&#39;</span><span class="p">][</span><span class="s1">&#39;files&#39;</span><span class="p">][</span><span class="s1">&#39;csv&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">]</span>
<span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> <span class="n">rpath</span><span class="p">)</span>
<span class="n">cluster_details</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="n">keep_default_na</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">cluster_details</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">rpath</span> <span class="o">=</span> <span class="n">taxonomy_metadata</span><span class="p">[</span><span class="s1">&#39;cluster_to_cluster_annotation_membership_color&#39;</span><span class="p">][</span><span class="s1">&#39;files&#39;</span><span class="p">][</span><span class="s1">&#39;csv&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">]</span>
<span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> <span class="n">rpath</span><span class="p">)</span>
<span class="n">cluster_colors</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
<span class="n">cluster_colors</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span> <span class="o">=</span> <span class="p">{}</span>

<span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">cell</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_details</span><span class="p">,</span><span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cluster_colors</span><span class="p">,</span><span class="n">on</span><span class="o">=</span><span class="s1">&#39;cluster_alias&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_extended</span><span class="p">[</span><span class="n">datasets</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>brain_section_label</th>
      <th>feature_matrix_label</th>
      <th>donor_label</th>
      <th>donor_genotype</th>
      <th>donor_sex</th>
      <th>cluster_alias</th>
      <th>x</th>
      <th>y</th>
      <th>z</th>
      <th>subclass_confidence_score</th>
      <th>...</th>
      <th>neurotransmitter</th>
      <th>class</th>
      <th>subclass</th>
      <th>supertype</th>
      <th>cluster</th>
      <th>neurotransmitter_color</th>
      <th>class_color</th>
      <th>subclass_color</th>
      <th>supertype_color</th>
      <th>cluster_color</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>182941331246012878296807398333956011710</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>704</td>
      <td>0.682522</td>
      <td>3.366483</td>
      <td>7.829530</td>
      <td>0.969933</td>
      <td>...</td>
      <td>GABA</td>
      <td>06 CTX-CGE GABA</td>
      <td>049 Lamp5 Gaba</td>
      <td>0199 Lamp5 Gaba_1</td>
      <td>0709 Lamp5 Gaba_1</td>
      <td>#FF3358</td>
      <td>#CCFF33</td>
      <td>#FF764D</td>
      <td>#DC00FF</td>
      <td>#998900</td>
    </tr>
    <tr>
      <th>221260934538535633595532020856387724686</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5243</td>
      <td>0.667690</td>
      <td>3.442241</td>
      <td>7.829530</td>
      <td>0.850554</td>
      <td>...</td>
      <td></td>
      <td>33 Vascular</td>
      <td>331 Peri NN</td>
      <td>1191 Peri NN_1</td>
      <td>5304 Peri NN_1</td>
      <td>#666666</td>
      <td>#858881</td>
      <td>#82992E</td>
      <td>#2F00CC</td>
      <td>#BB1FCC</td>
    </tr>
    <tr>
      <th>22228792606814781533240955623030943708</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>14939</td>
      <td>0.638731</td>
      <td>3.474328</td>
      <td>7.829530</td>
      <td>0.888285</td>
      <td>...</td>
      <td></td>
      <td>30 Astro-Epen</td>
      <td>319 Astro-TE NN</td>
      <td>1163 Astro-TE NN_3</td>
      <td>5225 Astro-TE NN_3</td>
      <td>#666666</td>
      <td>#594a26</td>
      <td>#3DCCB1</td>
      <td>#a8afa5</td>
      <td>#551799</td>
    </tr>
    <tr>
      <th>272043042552227961220474294517855477150</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>14939</td>
      <td>0.653425</td>
      <td>3.433218</td>
      <td>7.829530</td>
      <td>0.900000</td>
      <td>...</td>
      <td></td>
      <td>30 Astro-Epen</td>
      <td>319 Astro-TE NN</td>
      <td>1163 Astro-TE NN_3</td>
      <td>5225 Astro-TE NN_3</td>
      <td>#666666</td>
      <td>#594a26</td>
      <td>#3DCCB1</td>
      <td>#a8afa5</td>
      <td>#551799</td>
    </tr>
    <tr>
      <th>110116287883089187971185374239350249328</th>
      <td>Zhuang-ABCA-1.089</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5254</td>
      <td>0.623896</td>
      <td>3.513574</td>
      <td>7.829530</td>
      <td>0.999978</td>
      <td>...</td>
      <td></td>
      <td>33 Vascular</td>
      <td>333 Endo NN</td>
      <td>1193 Endo NN_1</td>
      <td>5310 Endo NN_1</td>
      <td>#666666</td>
      <td>#858881</td>
      <td>#994567</td>
      <td>#00992A</td>
      <td>#FFB473</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>94310525370042131911495836073267655162</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5255</td>
      <td>0.997247</td>
      <td>3.823236</td>
      <td>9.717769</td>
      <td>0.971385</td>
      <td>...</td>
      <td></td>
      <td>33 Vascular</td>
      <td>333 Endo NN</td>
      <td>1193 Endo NN_1</td>
      <td>5311 Endo NN_1</td>
      <td>#666666</td>
      <td>#858881</td>
      <td>#994567</td>
      <td>#00992A</td>
      <td>#CC3D76</td>
    </tr>
    <tr>
      <th>298798481479578578007190103666214714353</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>14939</td>
      <td>1.043871</td>
      <td>3.706231</td>
      <td>9.717769</td>
      <td>0.956377</td>
      <td>...</td>
      <td></td>
      <td>30 Astro-Epen</td>
      <td>319 Astro-TE NN</td>
      <td>1163 Astro-TE NN_3</td>
      <td>5225 Astro-TE NN_3</td>
      <td>#666666</td>
      <td>#594a26</td>
      <td>#3DCCB1</td>
      <td>#a8afa5</td>
      <td>#551799</td>
    </tr>
    <tr>
      <th>330756942354980576352210203729462562749</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5261</td>
      <td>1.037680</td>
      <td>3.759890</td>
      <td>9.717769</td>
      <td>0.811520</td>
      <td>...</td>
      <td></td>
      <td>33 Vascular</td>
      <td>330 VLMC NN</td>
      <td>1188 VLMC NN_2</td>
      <td>5301 VLMC NN_2</td>
      <td>#666666</td>
      <td>#858881</td>
      <td>#653D66</td>
      <td>#4D5CFF</td>
      <td>#79CC5C</td>
    </tr>
    <tr>
      <th>47305871059582831548494138048361484565</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5252</td>
      <td>1.044169</td>
      <td>3.758463</td>
      <td>9.717769</td>
      <td>0.991924</td>
      <td>...</td>
      <td></td>
      <td>33 Vascular</td>
      <td>333 Endo NN</td>
      <td>1193 Endo NN_1</td>
      <td>5309 Endo NN_1</td>
      <td>#666666</td>
      <td>#858881</td>
      <td>#994567</td>
      <td>#00992A</td>
      <td>#5C9994</td>
    </tr>
    <tr>
      <th>64578198410898899234789748167671783948</th>
      <td>Zhuang-ABCA-1.110</td>
      <td>Zhuang-ABCA-1</td>
      <td>Zhuang-ABCA-1</td>
      <td>wt/wt</td>
      <td>F</td>
      <td>5263</td>
      <td>1.042301</td>
      <td>3.589421</td>
      <td>9.717769</td>
      <td>0.889413</td>
      <td>...</td>
      <td></td>
      <td>33 Vascular</td>
      <td>330 VLMC NN</td>
      <td>1187 VLMC NN_1</td>
      <td>5298 VLMC NN_1</td>
      <td>#666666</td>
      <td>#858881</td>
      <td>#653D66</td>
      <td>#66391F</td>
      <td>#3B9900</td>
    </tr>
  </tbody>
</table>
<p>2846908 rows × 22 columns</p>
</div></div></div>
</div>
<p>For convenience, we can cache this view for later reuse.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">if</span> <span class="n">cache_views</span> <span class="p">:</span>
    
    <span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
        <span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">view_directory</span><span class="p">[</span><span class="n">d</span><span class="p">],</span> <span class="s1">&#39;cell_metadata_with_cluster_annotation.csv&#39;</span><span class="p">)</span>
        <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span> <span class="n">file</span> <span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="gene-panel">
<h3>Gene panel<a class="headerlink" href="#gene-panel" title="Permalink to this heading">#</a></h3>
<p>All 4 datasets shares the same 1122 gene panel selected to enable faciliate the mapping to transcriptomically defined cell types taxonomies. Each gene is uniquely identifier by an Ensembl ID. It is best practice to gene identifier to for tracking and data interchange as gene symbols are not unique and can change over time.</p>
<p>Each row of the gene dataframe has Ensembl gene identifier, a gene symbol and name.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">rpath</span> <span class="o">=</span> <span class="n">metadata</span><span class="p">[</span><span class="n">datasets</span><span class="p">[</span><span class="mi">0</span><span class="p">]][</span><span class="s1">&#39;gene&#39;</span><span class="p">][</span><span class="s1">&#39;files&#39;</span><span class="p">][</span><span class="s1">&#39;csv&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">]</span>
<span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> <span class="n">rpath</span><span class="p">)</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
<span class="n">gene</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;gene_identifier&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Number of genes = &quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">gene</span><span class="p">))</span>
<span class="n">gene</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of genes =  1122
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene_symbol</th>
      <th>name</th>
      <th>mapped_ncbi_identifier</th>
    </tr>
    <tr>
      <th>gene_identifier</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>ENSMUSG00000024798</th>
      <td>Htr7</td>
      <td>5-hydroxytryptamine (serotonin) receptor 7</td>
      <td>NCBIGene:15566</td>
    </tr>
    <tr>
      <th>ENSMUSG00000042385</th>
      <td>Gzmk</td>
      <td>granzyme K</td>
      <td>NCBIGene:14945</td>
    </tr>
    <tr>
      <th>ENSMUSG00000036198</th>
      <td>Arhgap36</td>
      <td>Rho GTPase activating protein 36</td>
      <td>NCBIGene:75404</td>
    </tr>
    <tr>
      <th>ENSMUSG00000028780</th>
      <td>Sema3c</td>
      <td>sema domain, immunoglobulin domain (Ig), short...</td>
      <td>NCBIGene:20348</td>
    </tr>
    <tr>
      <th>ENSMUSG00000015843</th>
      <td>Rxrg</td>
      <td>retinoid X receptor gamma</td>
      <td>NCBIGene:20183</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="gene-expression-matrix">
<h3>Gene expression matrix<a class="headerlink" href="#gene-expression-matrix" title="Permalink to this heading">#</a></h3>
<p>Expression values for all sections are stored in an anndata h5ad format per brain with minimal metadata. There are two h5ad files one storing the raw counts and the other log normalization of it. In this next section, we provide example code on how to open the file and connect with the rich cell level metadata discussed above.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">expression_matrices</span> <span class="o">=</span> <span class="p">{}</span>

<span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
    <span class="n">expression_matrices</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">manifest</span><span class="p">[</span><span class="s1">&#39;file_listing&#39;</span><span class="p">][</span><span class="n">d</span><span class="p">][</span><span class="s1">&#39;expression_matrices&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<p>We define a small helper function <em>plot sections</em> to visualize the cells in anatomical context colorized by: neurotransmitter identity, cell types division, class and subclass.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">subplot_section</span><span class="p">(</span> <span class="n">ax</span><span class="p">,</span> <span class="n">xx</span><span class="p">,</span> <span class="n">yy</span><span class="p">,</span> <span class="n">cc</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">val</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="kc">None</span> <span class="p">)</span> <span class="p">:</span>
    
    <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="p">:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span><span class="n">yy</span><span class="p">,</span><span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span><span class="n">c</span><span class="o">=</span><span class="n">val</span><span class="p">,</span><span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span><span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">cc</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="p">:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span><span class="n">yy</span><span class="p">,</span><span class="n">s</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="n">cc</span><span class="p">,</span><span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">11</span><span class="p">,</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">11</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s1">&#39;equal&#39;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">plot_sections</span><span class="p">(</span> <span class="n">cell_extended</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="n">cc</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">val</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">fig_width</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span> <span class="n">fig_height</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="kc">None</span> <span class="p">)</span> <span class="p">:</span>
    
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="n">fig_width</span><span class="p">,</span> <span class="n">fig_height</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">d</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">cell_extended</span><span class="p">):</span>
        
        <span class="n">pred</span> <span class="o">=</span> <span class="p">(</span><span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">][</span><span class="s1">&#39;brain_section_label&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">example_section</span><span class="p">[</span><span class="n">d</span><span class="p">])</span>
        <span class="n">section</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">][</span><span class="n">pred</span><span class="p">]</span> 
        
        <span class="k">if</span> <span class="n">cmap</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="p">:</span>
            <span class="n">subplot_section</span><span class="p">(</span> <span class="n">ax</span><span class="o">.</span><span class="n">flat</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">section</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span> <span class="n">section</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span> <span class="n">val</span><span class="o">=</span><span class="n">section</span><span class="p">[</span><span class="n">val</span><span class="p">],</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">)</span>
        <span class="k">elif</span> <span class="n">cc</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="p">:</span>
            <span class="n">subplot_section</span><span class="p">(</span> <span class="n">ax</span><span class="o">.</span><span class="n">flat</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">section</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">],</span> <span class="n">section</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">],</span> <span class="n">section</span><span class="p">[</span><span class="n">cc</span><span class="p">])</span>
            
        <span class="n">ax</span><span class="o">.</span><span class="n">flat</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">d</span><span class="p">)</span>
        
    <span class="k">return</span> <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_extended</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="s1">&#39;neurotransmitter_color&#39;</span> <span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Neurotransmitter Identity&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/1312f2b639115762cc5d351828a4544c56a10d0046a255ecb2b517a69fdd06c6.png" src="../_images/1312f2b639115762cc5d351828a4544c56a10d0046a255ecb2b517a69fdd06c6.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_extended</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="s1">&#39;class_color&#39;</span> <span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Cell Type Classes&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/7f92f3d3a30a4ea3aaedb2505e41626d2baa680b794c5139a822db7bc29e313a.png" src="../_images/7f92f3d3a30a4ea3aaedb2505e41626d2baa680b794c5139a822db7bc29e313a.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_extended</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="s1">&#39;subclass_color&#39;</span> <span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Cell Type Subclasses&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/b2807401cd6beb05a21eee4c16dee6cd1b6a76a83ba93ad0b3ab7d71a59741dd.png" src="../_images/b2807401cd6beb05a21eee4c16dee6cd1b6a76a83ba93ad0b3ab7d71a59741dd.png" />
</div>
</div>
</section>
</section>
<section id="example-use-case">
<h2>Example use case<a class="headerlink" href="#example-use-case" title="Permalink to this heading">#</a></h2>
<p>In this section, we visualize the expression of nine canonical neurotransmitter transporter genes. To support these use cases, we will create a smaller submatrix (all cells and 9 genes) that read it into dataframe. <em>Note this operation takes around 2-5 minutes</em>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gnames</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Slc17a7&#39;</span><span class="p">,</span><span class="s1">&#39;Slc17a6&#39;</span><span class="p">,</span><span class="s1">&#39;Slc17a8&#39;</span><span class="p">,</span><span class="s1">&#39;Slc32a1&#39;</span><span class="p">,</span><span class="s1">&#39;Slc6a5&#39;</span><span class="p">,</span><span class="s1">&#39;Slc6a3&#39;</span><span class="p">,</span><span class="s1">&#39;Slc6a4&#39;</span><span class="p">]</span>
<span class="n">pred</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="ow">in</span> <span class="n">gnames</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">gene</span><span class="o">.</span><span class="n">gene_symbol</span><span class="p">]</span>
<span class="n">gene_filtered</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">pred</span><span class="p">]</span>
<span class="n">gene_filtered</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene_symbol</th>
      <th>name</th>
      <th>mapped_ncbi_identifier</th>
    </tr>
    <tr>
      <th>gene_identifier</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>ENSMUSG00000019935</th>
      <td>Slc17a8</td>
      <td>solute carrier family 17 (sodium-dependent ino...</td>
      <td>NCBIGene:216227</td>
    </tr>
    <tr>
      <th>ENSMUSG00000021609</th>
      <td>Slc6a3</td>
      <td>solute carrier family 6 (neurotransmitter tran...</td>
      <td>NCBIGene:13162</td>
    </tr>
    <tr>
      <th>ENSMUSG00000037771</th>
      <td>Slc32a1</td>
      <td>solute carrier family 32 (GABA vesicular trans...</td>
      <td>NCBIGene:22348</td>
    </tr>
    <tr>
      <th>ENSMUSG00000039728</th>
      <td>Slc6a5</td>
      <td>solute carrier family 6 (neurotransmitter tran...</td>
      <td>NCBIGene:104245</td>
    </tr>
    <tr>
      <th>ENSMUSG00000070570</th>
      <td>Slc17a7</td>
      <td>solute carrier family 17 (sodium-dependent ino...</td>
      <td>NCBIGene:72961</td>
    </tr>
    <tr>
      <th>ENSMUSG00000020838</th>
      <td>Slc6a4</td>
      <td>solute carrier family 6 (neurotransmitter tran...</td>
      <td>NCBIGene:15567</td>
    </tr>
    <tr>
      <th>ENSMUSG00000030500</th>
      <td>Slc17a6</td>
      <td>solute carrier family 17 (sodium-dependent ino...</td>
      <td>NCBIGene:140919</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell_expression</span> <span class="o">=</span> <span class="p">{}</span>

<span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
    
    <span class="n">expression_matrices</span><span class="p">[</span><span class="n">d</span><span class="p">]</span>
    <span class="n">rpath</span> <span class="o">=</span> <span class="n">expression_matrices</span><span class="p">[</span><span class="n">d</span><span class="p">][</span><span class="n">d</span><span class="p">][</span><span class="s1">&#39;log2&#39;</span><span class="p">][</span><span class="s1">&#39;files&#39;</span><span class="p">][</span><span class="s1">&#39;h5ad&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">]</span>
    <span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> <span class="n">rpath</span><span class="p">)</span>
    
    <span class="n">adata</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="n">backed</span><span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">)</span>
    
    <span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">process_time</span><span class="p">()</span>
    <span class="n">gdata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[:,</span><span class="n">gene_filtered</span><span class="o">.</span><span class="n">index</span><span class="p">]</span><span class="o">.</span><span class="n">to_df</span><span class="p">()</span>
    <span class="n">gdata</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">gene_filtered</span><span class="o">.</span><span class="n">gene_symbol</span>
    <span class="n">cell_expression</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">gdata</span> <span class="p">)</span>
    
    <span class="nb">print</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="s2">&quot;-&quot;</span><span class="p">,</span><span class="s2">&quot;time taken: &quot;</span><span class="p">,</span> <span class="n">time</span><span class="o">.</span><span class="n">process_time</span><span class="p">()</span> <span class="o">-</span> <span class="n">start</span><span class="p">)</span>
    
    <span class="n">adata</span><span class="o">.</span><span class="n">file</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>
    <span class="k">del</span> <span class="n">adata</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Zhuang-ABCA-1 - time taken:  26.131841844
Zhuang-ABCA-2 - time taken:  10.865160033999999
Zhuang-ABCA-3 - time taken:  12.162818516000002
Zhuang-ABCA-4 - time taken:  1.3441603709999868
</pre></div>
</div>
</div>
</div>
<p>Visualize genes Slc17a7, Slc17a6, Slc32a1 for an example section from each brain</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_expression</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="n">val</span><span class="o">=</span><span class="s1">&#39;Slc17a7&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Gene Slc17a7&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/611e883dae40bd3f29799cd63241ed44b29e45998129ff2750bb49ac78ab8b52.png" src="../_images/611e883dae40bd3f29799cd63241ed44b29e45998129ff2750bb49ac78ab8b52.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_expression</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="n">val</span><span class="o">=</span><span class="s1">&#39;Slc17a6&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Gene Slc17a6&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/4f2c249f6d3ff8cab40a75bf99ae3ec9fdab2cc84d46f3dc3788f8d14f32699b.png" src="../_images/4f2c249f6d3ff8cab40a75bf99ae3ec9fdab2cc84d46f3dc3788f8d14f32699b.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_expression</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="n">val</span><span class="o">=</span><span class="s1">&#39;Slc32a1&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">cm</span><span class="o">.</span><span class="n">magma_r</span><span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Gene Slc32a1&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/627735b273c9e806b22eb634cd80a1b179120dfd5da028347d73aef4aee42f42.png" src="../_images/627735b273c9e806b22eb634cd80a1b179120dfd5da028347d73aef4aee42f42.png" />
</div>
</div>
<section id="ccf-registration-and-parcellation-annotation">
<h3>CCF registration and parcellation annotation<a class="headerlink" href="#ccf-registration-and-parcellation-annotation" title="Permalink to this heading">#</a></h3>
<p>Each brain specimen has been registered to Allen CCFv3 atlas, resulting in an x, y, z coordinates and parcellation_index for each cell.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ccf_coordinates</span> <span class="o">=</span> <span class="p">{}</span>

<span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
    
    <span class="n">rpath</span> <span class="o">=</span> <span class="n">manifest</span><span class="p">[</span><span class="s1">&#39;file_listing&#39;</span><span class="p">][</span><span class="n">d</span><span class="o">+</span><span class="s1">&#39;-CCF&#39;</span><span class="p">][</span><span class="s1">&#39;metadata&#39;</span><span class="p">][</span><span class="s1">&#39;ccf_coordinates&#39;</span><span class="p">][</span><span class="s1">&#39;files&#39;</span><span class="p">][</span><span class="s1">&#39;csv&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">]</span>
    <span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> <span class="n">rpath</span><span class="p">)</span>
    <span class="n">ccf_coordinates</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
    <span class="n">ccf_coordinates</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;cell_label&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="n">ccf_coordinates</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;x&#39;</span><span class="p">:</span><span class="s1">&#39;x_ccf&#39;</span><span class="p">,</span><span class="s1">&#39;y&#39;</span><span class="p">:</span><span class="s1">&#39;y_ccf&#39;</span><span class="p">,</span><span class="s1">&#39;z&#39;</span><span class="p">:</span><span class="s1">&#39;z_ccf&#39;</span><span class="p">},</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">ccf_coordinates</span><span class="p">[</span><span class="n">d</span><span class="p">],</span><span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ccf_coordinates</span><span class="p">[</span><span class="n">datasets</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>x_ccf</th>
      <th>y_ccf</th>
      <th>z_ccf</th>
      <th>parcellation_index</th>
    </tr>
    <tr>
      <th>cell_label</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>182941331246012878296807398333956011710</th>
      <td>7.902190</td>
      <td>3.048426</td>
      <td>0.582962</td>
      <td>0</td>
    </tr>
    <tr>
      <th>221260934538535633595532020856387724686</th>
      <td>7.906513</td>
      <td>3.145200</td>
      <td>0.577602</td>
      <td>0</td>
    </tr>
    <tr>
      <th>22228792606814781533240955623030943708</th>
      <td>7.906110</td>
      <td>3.182761</td>
      <td>0.553731</td>
      <td>0</td>
    </tr>
    <tr>
      <th>272043042552227961220474294517855477150</th>
      <td>7.904627</td>
      <td>3.131808</td>
      <td>0.563525</td>
      <td>0</td>
    </tr>
    <tr>
      <th>110116287883089187971185374239350249328</th>
      <td>7.907236</td>
      <td>3.230647</td>
      <td>0.543048</td>
      <td>0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>94310525370042131911495836073267655162</th>
      <td>9.681244</td>
      <td>4.453979</td>
      <td>0.852027</td>
      <td>0</td>
    </tr>
    <tr>
      <th>298798481479578578007190103666214714353</th>
      <td>9.676999</td>
      <td>4.291647</td>
      <td>0.899531</td>
      <td>1109</td>
    </tr>
    <tr>
      <th>330756942354980576352210203729462562749</th>
      <td>9.678760</td>
      <td>4.363282</td>
      <td>0.894082</td>
      <td>1109</td>
    </tr>
    <tr>
      <th>47305871059582831548494138048361484565</th>
      <td>9.678641</td>
      <td>4.360346</td>
      <td>0.901195</td>
      <td>1109</td>
    </tr>
    <tr>
      <th>64578198410898899234789748167671783948</th>
      <td>9.673530</td>
      <td>4.138034</td>
      <td>0.897311</td>
      <td>1109</td>
    </tr>
  </tbody>
</table>
<p>2616328 rows × 4 columns</p>
</div></div></div>
</div>
<p>Read in the pivot table from the “parcellation annotation tutorial” to associate each cell with terms at each anatomical parcellation level and the corresponding color.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">metadata</span> <span class="o">=</span> <span class="n">manifest</span><span class="p">[</span><span class="s1">&#39;file_listing&#39;</span><span class="p">][</span><span class="s1">&#39;Allen-CCF-2020&#39;</span><span class="p">][</span><span class="s1">&#39;metadata&#39;</span><span class="p">]</span>
<span class="n">rpath</span> <span class="o">=</span> <span class="n">metadata</span><span class="p">[</span><span class="s1">&#39;parcellation_to_parcellation_term_membership_acronym&#39;</span><span class="p">][</span><span class="s1">&#39;files&#39;</span><span class="p">][</span><span class="s1">&#39;csv&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">]</span>
<span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> <span class="n">rpath</span><span class="p">)</span>
<span class="n">parcellation_annotation</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
<span class="n">parcellation_annotation</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;parcellation_index&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">parcellation_annotation</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;parcellation_</span><span class="si">%s</span><span class="s1">&#39;</span><span class="o">%</span> <span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span>  <span class="n">parcellation_annotation</span><span class="o">.</span><span class="n">columns</span><span class="p">]</span>

<span class="n">rpath</span> <span class="o">=</span> <span class="n">metadata</span><span class="p">[</span><span class="s1">&#39;parcellation_to_parcellation_term_membership_color&#39;</span><span class="p">][</span><span class="s1">&#39;files&#39;</span><span class="p">][</span><span class="s1">&#39;csv&#39;</span><span class="p">][</span><span class="s1">&#39;relative_path&#39;</span><span class="p">]</span>
<span class="n">file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">download_base</span><span class="p">,</span> <span class="n">rpath</span><span class="p">)</span>
<span class="n">parcellation_color</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
<span class="n">parcellation_color</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;parcellation_index&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">parcellation_color</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;parcellation_</span><span class="si">%s</span><span class="s1">&#39;</span><span class="o">%</span> <span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span>  <span class="n">parcellation_color</span><span class="o">.</span><span class="n">columns</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">datasets</span> <span class="p">:</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">parcellation_annotation</span><span class="p">,</span><span class="n">on</span><span class="o">=</span><span class="s1">&#39;parcellation_index&#39;</span><span class="p">)</span>
    <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">cell_extended</span><span class="p">[</span><span class="n">d</span><span class="p">]</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">parcellation_color</span><span class="p">,</span><span class="n">on</span><span class="o">=</span><span class="s1">&#39;parcellation_index&#39;</span><span class="p">)</span>   
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_extended</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="s1">&#39;parcellation_division_color&#39;</span> <span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Parcellation - division&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/41c9bc1332689bd3b10f7b42398a8342538948a71295e4ea5f262c103b859bd7.png" src="../_images/41c9bc1332689bd3b10f7b42398a8342538948a71295e4ea5f262c103b859bd7.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_extended</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="s1">&#39;parcellation_structure_color&#39;</span> <span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Parcellation - structure&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/4d09a69f3e556ab867529b6f9ccf4f0a2efe6014cf3044ba52bbbf86486a3171.png" src="../_images/4d09a69f3e556ab867529b6f9ccf4f0a2efe6014cf3044ba52bbbf86486a3171.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plot_sections</span><span class="p">(</span> <span class="n">cell_extended</span><span class="p">,</span> <span class="n">example_section</span><span class="p">,</span> <span class="s1">&#39;parcellation_substructure_color&#39;</span> <span class="p">)</span>
<span class="n">res</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s1">&#39;Parcellation - substructure&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/565f879962f99f62404884759d27aefd906921527d5ed4c9f400dc7838544245.png" src="../_images/565f879962f99f62404884759d27aefd906921527d5ed4c9f400dc7838544245.png" />
</div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="../descriptions/notebook_subtitle4.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">A molecularly defined and spatially resolved cell atlas of the whole mouse brain (Xiaowei Zhuang)</p>
      </div>
    </a>
    <a class="right-next"
       href="../docs/README.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Resource Pages</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-overview">Data overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-metadata">Cell metadata</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cluster-annotation">Cluster annotation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-panel">Gene panel</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-expression-matrix">Gene expression matrix</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#example-use-case">Example use case</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ccf-registration-and-parcellation-annotation">CCF registration and parcellation annotation</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>